<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.7.32">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">


<title>Package Functions Help – EnrichGT</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
/* CSS for syntax highlighting */
html { -webkit-text-size-adjust: 100%; }
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
  }
pre.numberSource { margin-left: 3em;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
</style>


<script src="site_libs/quarto-nav/quarto-nav.js"></script>
<script src="site_libs/quarto-nav/headroom.min.js"></script>
<script src="site_libs/clipboard/clipboard.min.js"></script>
<script src="site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="site_libs/quarto-search/fuse.min.js"></script>
<script src="site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="./">
<script src="site_libs/quarto-html/quarto.js" type="module"></script>
<script src="site_libs/quarto-html/tabsets/tabsets.js" type="module"></script>
<script src="site_libs/quarto-html/popper.min.js"></script>
<script src="site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="site_libs/quarto-html/anchor.min.js"></script>
<link href="site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="site_libs/quarto-html/quarto-syntax-highlighting-37eea08aefeeee20ff55810ff984fec1.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="site_libs/bootstrap/bootstrap.min.js"></script>
<link href="site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="site_libs/bootstrap/bootstrap-960cc9eac5395d1658253b679c30d018.min.css" rel="stylesheet" append-hash="true" id="quarto-bootstrap" data-mode="light">
<script id="quarto-search-options" type="application/json">{
  "location": "navbar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "end",
  "type": "overlay",
  "limit": 50,
  "keyboard-shortcut": [
    "f",
    "/",
    "s"
  ],
  "show-item-context": false,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-text-placeholder": "",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit",
    "search-label": "Search"
  }
}</script>


<link rel="stylesheet" href="styles.css">
</head>

<body class="nav-fixed quarto-light">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top">
    <nav class="navbar navbar-expand-lg " data-bs-theme="dark">
      <div class="navbar-container container-fluid">
      <div class="navbar-brand-container mx-auto">
    <a href="./index.html" class="navbar-brand navbar-brand-logo">
    <img src="./EnrichGTlogo.png" alt="" class="navbar-logo">
    </a>
    <a class="navbar-brand" href="./index.html">
    <span class="navbar-title">EnrichGT</span>
    </a>
  </div>
            <div id="quarto-search" class="" title="Search"></div>
          <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbarCollapse" aria-controls="navbarCollapse" role="menu" aria-expanded="false" aria-label="Toggle navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
  <span class="navbar-toggler-icon"></span>
</button>
          <div class="collapse navbar-collapse" id="navbarCollapse">
            <ul class="navbar-nav navbar-nav-scroll me-auto">
  <li class="nav-item">
    <a class="nav-link" href="./index.html"> 
<span class="menu-text">Home</span></a>
  </li>  
  <li class="nav-item">
    <a class="nav-link active" href="./pkgfuns.html" aria-current="page"> 
<span class="menu-text">Package Functions</span></a>
  </li>  
  <li class="nav-item">
    <a class="nav-link" href="https://github.com/ZhimingYe/EnrichGT/"> 
<span class="menu-text">Github</span></a>
  </li>  
  <li class="nav-item">
    <a class="nav-link" href="./acknowledgement.html"> 
<span class="menu-text">Acknowledgement</span></a>
  </li>  
</ul>
          </div> <!-- /navcollapse -->
            <div class="quarto-navbar-tools">
</div>
      </div> <!-- /container-fluid -->
    </nav>
</header>
<!-- content -->
<div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-article page-navbar">
<!-- sidebar -->
<!-- margin-sidebar -->
    <div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
        <nav id="TOC" role="doc-toc" class="toc-active">
    <h2 id="toc-title">On this page</h2>
   
  <ul>
  <li><a href="#description" id="toc-description" class="nav-link active" data-scroll-target="#description">DESCRIPTION</a></li>
  <li><a href="#comparison_reactor_base-comparison-reactor-base" id="toc-comparison_reactor_base-comparison-reactor-base" class="nav-link" data-scroll-target="#comparison_reactor_base-comparison-reactor-base"><code>comparison_reactor_base</code>: Comparison Reactor Base</a>
  <ul class="collapse">
  <li><a href="#description-1" id="toc-description-1" class="nav-link" data-scroll-target="#description-1">Description</a></li>
  <li><a href="#details" id="toc-details" class="nav-link" data-scroll-target="#details">Details</a></li>
  <li><a href="#methods" id="toc-methods" class="nav-link" data-scroll-target="#methods">Methods</a>
  <ul class="collapse">
  <li><a href="#public-methods" id="toc-public-methods" class="nav-link" data-scroll-target="#public-methods">Public methods</a></li>
  <li><a href="#method-new" id="toc-method-new" class="nav-link" data-scroll-target="#method-new">Method <code>new()</code></a></li>
  <li><a href="#method-append_enriched_result" id="toc-method-append_enriched_result" class="nav-link" data-scroll-target="#method-append_enriched_result">Method <code>append_enriched_result()</code></a></li>
  <li><a href="#method-summarize" id="toc-method-summarize" class="nav-link" data-scroll-target="#method-summarize">Method <code>summarize()</code></a></li>
  <li><a href="#method-prefilter_by_p_adj" id="toc-method-prefilter_by_p_adj" class="nav-link" data-scroll-target="#method-prefilter_by_p_adj">Method <code>prefilter_by_p_adj()</code></a></li>
  <li><a href="#method-find_relationship" id="toc-method-find_relationship" class="nav-link" data-scroll-target="#method-find_relationship">Method <code>find_relationship()</code></a></li>
  <li><a href="#method-fetch_relationship" id="toc-method-fetch_relationship" class="nav-link" data-scroll-target="#method-fetch_relationship">Method <code>fetch_relationship()</code></a></li>
  <li><a href="#method-fetch_biological_theme" id="toc-method-fetch_biological_theme" class="nav-link" data-scroll-target="#method-fetch_biological_theme">Method <code>fetch_biological_theme()</code></a></li>
  <li><a href="#method-split_by_cluster" id="toc-method-split_by_cluster" class="nav-link" data-scroll-target="#method-split_by_cluster">Method <code>split_by_cluster()</code></a></li>
  <li><a href="#method-get_splited_list" id="toc-method-get_splited_list" class="nav-link" data-scroll-target="#method-get_splited_list">Method <code>get_splited_list()</code></a></li>
  <li><a href="#method-do_recluster" id="toc-method-do_recluster" class="nav-link" data-scroll-target="#method-do_recluster">Method <code>do_recluster()</code></a></li>
  <li><a href="#method-get_recluster_result" id="toc-method-get_recluster_result" class="nav-link" data-scroll-target="#method-get_recluster_result">Method <code>get_recluster_result()</code></a></li>
  <li><a href="#method-print" id="toc-method-print" class="nav-link" data-scroll-target="#method-print">Method <code>print()</code></a></li>
  <li><a href="#method-clone" id="toc-method-clone" class="nav-link" data-scroll-target="#method-clone">Method <code>clone()</code></a></li>
  </ul></li>
  <li><a href="#examples-7" id="toc-examples-7" class="nav-link" data-scroll-target="#examples-7">Examples</a></li>
  </ul></li>
  <li><a href="#comparison_reactor_gsea-gsea-comparison-reactor" id="toc-comparison_reactor_gsea-gsea-comparison-reactor" class="nav-link" data-scroll-target="#comparison_reactor_gsea-gsea-comparison-reactor"><code>comparison_reactor_gsea</code>: GSEA Comparison Reactor</a>
  <ul class="collapse">
  <li><a href="#description-2" id="toc-description-2" class="nav-link" data-scroll-target="#description-2">Description</a></li>
  <li><a href="#details-1" id="toc-details-1" class="nav-link" data-scroll-target="#details-1">Details</a></li>
  <li><a href="#super-class" id="toc-super-class" class="nav-link" data-scroll-target="#super-class">Super class</a></li>
  <li><a href="#methods-1" id="toc-methods-1" class="nav-link" data-scroll-target="#methods-1">Methods</a>
  <ul class="collapse">
  <li><a href="#public-methods-1" id="toc-public-methods-1" class="nav-link" data-scroll-target="#public-methods-1">Public methods</a></li>
  <li><a href="#method-new-1" id="toc-method-new-1" class="nav-link" data-scroll-target="#method-new-1">Method <code>new()</code></a></li>
  <li><a href="#method-make_plans" id="toc-method-make_plans" class="nav-link" data-scroll-target="#method-make_plans">Method <code>make_plans()</code></a></li>
  <li><a href="#method-prefilter_by_nes" id="toc-method-prefilter_by_nes" class="nav-link" data-scroll-target="#method-prefilter_by_nes">Method <code>prefilter_by_NES()</code></a></li>
  <li><a href="#method-clone-1" id="toc-method-clone-1" class="nav-link" data-scroll-target="#method-clone-1">Method <code>clone()</code></a></li>
  </ul></li>
  <li><a href="#seealso" id="toc-seealso" class="nav-link" data-scroll-target="#seealso">Seealso</a></li>
  <li><a href="#examples-10" id="toc-examples-10" class="nav-link" data-scroll-target="#examples-10">Examples</a></li>
  </ul></li>
  <li><a href="#comparison_reactor_ora-ora-comparison-reactor" id="toc-comparison_reactor_ora-ora-comparison-reactor" class="nav-link" data-scroll-target="#comparison_reactor_ora-ora-comparison-reactor"><code>comparison_reactor_ora</code>: ORA Comparison Reactor</a>
  <ul class="collapse">
  <li><a href="#description-3" id="toc-description-3" class="nav-link" data-scroll-target="#description-3">Description</a></li>
  <li><a href="#details-2" id="toc-details-2" class="nav-link" data-scroll-target="#details-2">Details</a></li>
  <li><a href="#super-class-1" id="toc-super-class-1" class="nav-link" data-scroll-target="#super-class-1">Super class</a></li>
  <li><a href="#methods-2" id="toc-methods-2" class="nav-link" data-scroll-target="#methods-2">Methods</a>
  <ul class="collapse">
  <li><a href="#public-methods-2" id="toc-public-methods-2" class="nav-link" data-scroll-target="#public-methods-2">Public methods</a></li>
  <li><a href="#method-new-2" id="toc-method-new-2" class="nav-link" data-scroll-target="#method-new-2">Method <code>new()</code></a></li>
  <li><a href="#method-make_plans-1" id="toc-method-make_plans-1" class="nav-link" data-scroll-target="#method-make_plans-1">Method <code>make_plans()</code></a></li>
  <li><a href="#method-clone-2" id="toc-method-clone-2" class="nav-link" data-scroll-target="#method-clone-2">Method <code>clone()</code></a></li>
  </ul></li>
  <li><a href="#seealso-1" id="toc-seealso-1" class="nav-link" data-scroll-target="#seealso-1">Seealso</a></li>
  <li><a href="#examples-12" id="toc-examples-12" class="nav-link" data-scroll-target="#examples-12">Examples</a></li>
  </ul></li>
  <li><a href="#convert_annotations_genes-convert-gene-annotations-from-any-keys-to-any-keys" id="toc-convert_annotations_genes-convert-gene-annotations-from-any-keys-to-any-keys" class="nav-link" data-scroll-target="#convert_annotations_genes-convert-gene-annotations-from-any-keys-to-any-keys"><code>convert_annotations_genes</code>: Convert gene annotations from any keys to any keys</a>
  <ul class="collapse">
  <li><a href="#description-4" id="toc-description-4" class="nav-link" data-scroll-target="#description-4">Description</a></li>
  <li><a href="#usage-20" id="toc-usage-20" class="nav-link" data-scroll-target="#usage-20">Usage</a></li>
  <li><a href="#arguments-13" id="toc-arguments-13" class="nav-link" data-scroll-target="#arguments-13">Arguments</a></li>
  <li><a href="#value" id="toc-value" class="nav-link" data-scroll-target="#value">Value</a></li>
  </ul></li>
  <li><a href="#database_from_gmt-parse-gmt-format-gene-set-files" id="toc-database_from_gmt-parse-gmt-format-gene-set-files" class="nav-link" data-scroll-target="#database_from_gmt-parse-gmt-format-gene-set-files"><code>database_from_gmt</code>: Parse GMT format gene set files</a>
  <ul class="collapse">
  <li><a href="#description-5" id="toc-description-5" class="nav-link" data-scroll-target="#description-5">Description</a></li>
  <li><a href="#usage-21" id="toc-usage-21" class="nav-link" data-scroll-target="#usage-21">Usage</a></li>
  <li><a href="#arguments-14" id="toc-arguments-14" class="nav-link" data-scroll-target="#arguments-14">Arguments</a></li>
  <li><a href="#author" id="toc-author" class="nav-link" data-scroll-target="#author">Author</a></li>
  <li><a href="#value-1" id="toc-value-1" class="nav-link" data-scroll-target="#value-1">Value</a></li>
  <li><a href="#examples-13" id="toc-examples-13" class="nav-link" data-scroll-target="#examples-13">Examples</a></li>
  </ul></li>
  <li><a href="#egt_compare_groups-2-group-comparison-of-enrichment-results-and-further-clustering-and-visualizing" id="toc-egt_compare_groups-2-group-comparison-of-enrichment-results-and-further-clustering-and-visualizing" class="nav-link" data-scroll-target="#egt_compare_groups-2-group-comparison-of-enrichment-results-and-further-clustering-and-visualizing"><code>egt_compare_groups</code>: 2-Group Comparison of enrichment results and further clustering and visualizing</a>
  <ul class="collapse">
  <li><a href="#description-6" id="toc-description-6" class="nav-link" data-scroll-target="#description-6">Description</a></li>
  <li><a href="#usage-22" id="toc-usage-22" class="nav-link" data-scroll-target="#usage-22">Usage</a></li>
  <li><a href="#arguments-15" id="toc-arguments-15" class="nav-link" data-scroll-target="#arguments-15">Arguments</a></li>
  <li><a href="#details-3" id="toc-details-3" class="nav-link" data-scroll-target="#details-3">Details</a></li>
  <li><a href="#author-1" id="toc-author-1" class="nav-link" data-scroll-target="#author-1">Author</a></li>
  <li><a href="#value-2" id="toc-value-2" class="nav-link" data-scroll-target="#value-2">Value</a></li>
  </ul></li>
  <li><a href="#egt_comparison_reactor-framework-for-comparing-and-analyzing-enrichment-results" id="toc-egt_comparison_reactor-framework-for-comparing-and-analyzing-enrichment-results" class="nav-link" data-scroll-target="#egt_comparison_reactor-framework-for-comparing-and-analyzing-enrichment-results"><code>egt_comparison_reactor</code>: Framework for comparing and analyzing enrichment results</a>
  <ul class="collapse">
  <li><a href="#description-7" id="toc-description-7" class="nav-link" data-scroll-target="#description-7">Description</a></li>
  <li><a href="#usage-23" id="toc-usage-23" class="nav-link" data-scroll-target="#usage-23">Usage</a></li>
  <li><a href="#details-4" id="toc-details-4" class="nav-link" data-scroll-target="#details-4">Details</a></li>
  <li><a href="#seealso-2" id="toc-seealso-2" class="nav-link" data-scroll-target="#seealso-2">Seealso</a></li>
  <li><a href="#examples-14" id="toc-examples-14" class="nav-link" data-scroll-target="#examples-14">Examples</a></li>
  </ul></li>
  <li><a href="#egt_enrichment_analysis-perform-over-representation-analysis-ora" id="toc-egt_enrichment_analysis-perform-over-representation-analysis-ora" class="nav-link" data-scroll-target="#egt_enrichment_analysis-perform-over-representation-analysis-ora"><code>egt_enrichment_analysis</code>: Perform Over-Representation Analysis (ORA)</a>
  <ul class="collapse">
  <li><a href="#description-8" id="toc-description-8" class="nav-link" data-scroll-target="#description-8">Description</a></li>
  <li><a href="#usage-24" id="toc-usage-24" class="nav-link" data-scroll-target="#usage-24">Usage</a></li>
  <li><a href="#arguments-16" id="toc-arguments-16" class="nav-link" data-scroll-target="#arguments-16">Arguments</a></li>
  <li><a href="#details-5" id="toc-details-5" class="nav-link" data-scroll-target="#details-5">Details</a></li>
  <li><a href="#author-2" id="toc-author-2" class="nav-link" data-scroll-target="#author-2">Author</a></li>
  <li><a href="#value-3" id="toc-value-3" class="nav-link" data-scroll-target="#value-3">Value</a></li>
  <li><a href="#examples-15" id="toc-examples-15" class="nav-link" data-scroll-target="#examples-15">Examples</a></li>
  </ul></li>
  <li><a href="#egt_fetch_biological_theme-generate-biological-theme-wordcloud-from-enrichment-results" id="toc-egt_fetch_biological_theme-generate-biological-theme-wordcloud-from-enrichment-results" class="nav-link" data-scroll-target="#egt_fetch_biological_theme-generate-biological-theme-wordcloud-from-enrichment-results"><code>egt_fetch_biological_theme</code>: Generate Biological Theme Wordcloud from Enrichment Results</a>
  <ul class="collapse">
  <li><a href="#description-9" id="toc-description-9" class="nav-link" data-scroll-target="#description-9">Description</a></li>
  <li><a href="#usage-25" id="toc-usage-25" class="nav-link" data-scroll-target="#usage-25">Usage</a></li>
  <li><a href="#arguments-17" id="toc-arguments-17" class="nav-link" data-scroll-target="#arguments-17">Arguments</a></li>
  <li><a href="#author-3" id="toc-author-3" class="nav-link" data-scroll-target="#author-3">Author</a></li>
  <li><a href="#value-4" id="toc-value-4" class="nav-link" data-scroll-target="#value-4">Value</a></li>
  </ul></li>
  <li><a href="#egt_fetch_termwise_relationship-fetch-and-visualize-termwise-relationships-in-enrichment-analysis-results" id="toc-egt_fetch_termwise_relationship-fetch-and-visualize-termwise-relationships-in-enrichment-analysis-results" class="nav-link" data-scroll-target="#egt_fetch_termwise_relationship-fetch-and-visualize-termwise-relationships-in-enrichment-analysis-results"><code>egt_fetch_termwise_relationship</code>: Fetch and visualize termwise relationships in enrichment analysis results</a>
  <ul class="collapse">
  <li><a href="#description-10" id="toc-description-10" class="nav-link" data-scroll-target="#description-10">Description</a></li>
  <li><a href="#usage-26" id="toc-usage-26" class="nav-link" data-scroll-target="#usage-26">Usage</a></li>
  <li><a href="#arguments-18" id="toc-arguments-18" class="nav-link" data-scroll-target="#arguments-18">Arguments</a></li>
  <li><a href="#value-5" id="toc-value-5" class="nav-link" data-scroll-target="#value-5">Value</a></li>
  <li><a href="#examples-16" id="toc-examples-16" class="nav-link" data-scroll-target="#examples-16">Examples</a></li>
  </ul></li>
  <li><a href="#egt_generate_quarto_report-export-quarto-report" id="toc-egt_generate_quarto_report-export-quarto-report" class="nav-link" data-scroll-target="#egt_generate_quarto_report-export-quarto-report"><code>egt_generate_quarto_report</code>: Export Quarto Report</a>
  <ul class="collapse">
  <li><a href="#description-11" id="toc-description-11" class="nav-link" data-scroll-target="#description-11">Description</a></li>
  <li><a href="#usage-27" id="toc-usage-27" class="nav-link" data-scroll-target="#usage-27">Usage</a></li>
  <li><a href="#arguments-19" id="toc-arguments-19" class="nav-link" data-scroll-target="#arguments-19">Arguments</a></li>
  <li><a href="#value-6" id="toc-value-6" class="nav-link" data-scroll-target="#value-6">Value</a></li>
  </ul></li>
  <li><a href="#egt_gsea_analysis-perform-gene-set-enrichment-analysis-gsea" id="toc-egt_gsea_analysis-perform-gene-set-enrichment-analysis-gsea" class="nav-link" data-scroll-target="#egt_gsea_analysis-perform-gene-set-enrichment-analysis-gsea"><code>egt_gsea_analysis</code>: Perform Gene Set Enrichment Analysis (GSEA)</a>
  <ul class="collapse">
  <li><a href="#description-12" id="toc-description-12" class="nav-link" data-scroll-target="#description-12">Description</a></li>
  <li><a href="#usage-28" id="toc-usage-28" class="nav-link" data-scroll-target="#usage-28">Usage</a></li>
  <li><a href="#arguments-20" id="toc-arguments-20" class="nav-link" data-scroll-target="#arguments-20">Arguments</a></li>
  <li><a href="#details-6" id="toc-details-6" class="nav-link" data-scroll-target="#details-6">Details</a></li>
  <li><a href="#author-4" id="toc-author-4" class="nav-link" data-scroll-target="#author-4">Author</a></li>
  <li><a href="#value-7" id="toc-value-7" class="nav-link" data-scroll-target="#value-7">Value</a></li>
  <li><a href="#examples-17" id="toc-examples-17" class="nav-link" data-scroll-target="#examples-17">Examples</a></li>
  </ul></li>
  <li><a href="#egt_infer_act-infering-pathway-or-transcript-factors-activity-from-enrichgt-meta-gene-modules" id="toc-egt_infer_act-infering-pathway-or-transcript-factors-activity-from-enrichgt-meta-gene-modules" class="nav-link" data-scroll-target="#egt_infer_act-infering-pathway-or-transcript-factors-activity-from-enrichgt-meta-gene-modules"><code>egt_infer_act</code>: Infering Pathway or Transcript Factors activity from EnrichGT meta-gene modules</a>
  <ul class="collapse">
  <li><a href="#description-13" id="toc-description-13" class="nav-link" data-scroll-target="#description-13">Description</a></li>
  <li><a href="#usage-29" id="toc-usage-29" class="nav-link" data-scroll-target="#usage-29">Usage</a></li>
  <li><a href="#arguments-21" id="toc-arguments-21" class="nav-link" data-scroll-target="#arguments-21">Arguments</a></li>
  <li><a href="#author-5" id="toc-author-5" class="nav-link" data-scroll-target="#author-5">Author</a></li>
  <li><a href="#value-8" id="toc-value-8" class="nav-link" data-scroll-target="#value-8">Value</a></li>
  </ul></li>
  <li><a href="#egt_llm_multi_summary-compare-multiple-llm-summaries" id="toc-egt_llm_multi_summary-compare-multiple-llm-summaries" class="nav-link" data-scroll-target="#egt_llm_multi_summary-compare-multiple-llm-summaries"><code>egt_llm_multi_summary</code>: Compare Multiple LLM Summaries</a>
  <ul class="collapse">
  <li><a href="#description-14" id="toc-description-14" class="nav-link" data-scroll-target="#description-14">Description</a></li>
  <li><a href="#usage-30" id="toc-usage-30" class="nav-link" data-scroll-target="#usage-30">Usage</a></li>
  <li><a href="#arguments-22" id="toc-arguments-22" class="nav-link" data-scroll-target="#arguments-22">Arguments</a></li>
  <li><a href="#value-9" id="toc-value-9" class="nav-link" data-scroll-target="#value-9">Value</a></li>
  </ul></li>
  <li><a href="#egt_llm_summary-summarize-enrichgt-results-using-llm" id="toc-egt_llm_summary-summarize-enrichgt-results-using-llm" class="nav-link" data-scroll-target="#egt_llm_summary-summarize-enrichgt-results-using-llm"><code>egt_llm_summary</code>: Summarize EnrichGT results using LLM</a>
  <ul class="collapse">
  <li><a href="#description-15" id="toc-description-15" class="nav-link" data-scroll-target="#description-15">Description</a></li>
  <li><a href="#usage-31" id="toc-usage-31" class="nav-link" data-scroll-target="#usage-31">Usage</a></li>
  <li><a href="#arguments-23" id="toc-arguments-23" class="nav-link" data-scroll-target="#arguments-23">Arguments</a></li>
  <li><a href="#note" id="toc-note" class="nav-link" data-scroll-target="#note">Note</a></li>
  <li><a href="#references" id="toc-references" class="nav-link" data-scroll-target="#references">References</a></li>
  <li><a href="#seealso-3" id="toc-seealso-3" class="nav-link" data-scroll-target="#seealso-3">Seealso</a></li>
  <li><a href="#value-10" id="toc-value-10" class="nav-link" data-scroll-target="#value-10">Value</a></li>
  <li><a href="#examples-18" id="toc-examples-18" class="nav-link" data-scroll-target="#examples-18">Examples</a></li>
  </ul></li>
  <li><a href="#egt_plot_gsea-generate-gsea-enrichment-plots" id="toc-egt_plot_gsea-generate-gsea-enrichment-plots" class="nav-link" data-scroll-target="#egt_plot_gsea-generate-gsea-enrichment-plots"><code>egt_plot_gsea</code>: Generate GSEA Enrichment Plots</a>
  <ul class="collapse">
  <li><a href="#description-16" id="toc-description-16" class="nav-link" data-scroll-target="#description-16">Description</a></li>
  <li><a href="#usage-32" id="toc-usage-32" class="nav-link" data-scroll-target="#usage-32">Usage</a></li>
  <li><a href="#arguments-24" id="toc-arguments-24" class="nav-link" data-scroll-target="#arguments-24">Arguments</a></li>
  <li><a href="#author-6" id="toc-author-6" class="nav-link" data-scroll-target="#author-6">Author</a></li>
  <li><a href="#value-11" id="toc-value-11" class="nav-link" data-scroll-target="#value-11">Value</a></li>
  </ul></li>
  <li><a href="#egt_plot_results-visualize-enrichment-results-using-simple-plot" id="toc-egt_plot_results-visualize-enrichment-results-using-simple-plot" class="nav-link" data-scroll-target="#egt_plot_results-visualize-enrichment-results-using-simple-plot"><code>egt_plot_results</code>: Visualize enrichment results using simple plot</a>
  <ul class="collapse">
  <li><a href="#description-17" id="toc-description-17" class="nav-link" data-scroll-target="#description-17">Description</a></li>
  <li><a href="#usage-33" id="toc-usage-33" class="nav-link" data-scroll-target="#usage-33">Usage</a></li>
  <li><a href="#arguments-25" id="toc-arguments-25" class="nav-link" data-scroll-target="#arguments-25">Arguments</a></li>
  <li><a href="#author-7" id="toc-author-7" class="nav-link" data-scroll-target="#author-7">Author</a></li>
  <li><a href="#value-12" id="toc-value-12" class="nav-link" data-scroll-target="#value-12">Value</a></li>
  </ul></li>
  <li><a href="#egt_recluster_analysis-cluster-and-re-enrichment-enrichment-results" id="toc-egt_recluster_analysis-cluster-and-re-enrichment-enrichment-results" class="nav-link" data-scroll-target="#egt_recluster_analysis-cluster-and-re-enrichment-enrichment-results"><code>egt_recluster_analysis</code>: Cluster and re-enrichment enrichment results</a>
  <ul class="collapse">
  <li><a href="#description-18" id="toc-description-18" class="nav-link" data-scroll-target="#description-18">Description</a></li>
  <li><a href="#usage-34" id="toc-usage-34" class="nav-link" data-scroll-target="#usage-34">Usage</a></li>
  <li><a href="#arguments-26" id="toc-arguments-26" class="nav-link" data-scroll-target="#arguments-26">Arguments</a></li>
  <li><a href="#details-7" id="toc-details-7" class="nav-link" data-scroll-target="#details-7">Details</a></li>
  <li><a href="#author-8" id="toc-author-8" class="nav-link" data-scroll-target="#author-8">Author</a></li>
  <li><a href="#value-13" id="toc-value-13" class="nav-link" data-scroll-target="#value-13">Value</a></li>
  <li><a href="#examples-19" id="toc-examples-19" class="nav-link" data-scroll-target="#examples-19">Examples</a></li>
  </ul></li>
  <li><a href="#egt_summary-generate-html-summary-for-enrichgt-object" id="toc-egt_summary-generate-html-summary-for-enrichgt-object" class="nav-link" data-scroll-target="#egt_summary-generate-html-summary-for-enrichgt-object"><code>egt_summary</code>: Generate HTML Summary for EnrichGT Object</a>
  <ul class="collapse">
  <li><a href="#description-19" id="toc-description-19" class="nav-link" data-scroll-target="#description-19">Description</a></li>
  <li><a href="#usage-35" id="toc-usage-35" class="nav-link" data-scroll-target="#usage-35">Usage</a></li>
  <li><a href="#arguments-27" id="toc-arguments-27" class="nav-link" data-scroll-target="#arguments-27">Arguments</a></li>
  <li><a href="#value-14" id="toc-value-14" class="nav-link" data-scroll-target="#value-14">Value</a></li>
  <li><a href="#examples-20" id="toc-examples-20" class="nav-link" data-scroll-target="#examples-20">Examples</a></li>
  </ul></li>
  <li><a href="#egt_web_interface-launch-enrichgt-web-interface" id="toc-egt_web_interface-launch-enrichgt-web-interface" class="nav-link" data-scroll-target="#egt_web_interface-launch-enrichgt-web-interface"><code>egt_web_interface</code>: Launch EnrichGT Web Interface</a>
  <ul class="collapse">
  <li><a href="#description-20" id="toc-description-20" class="nav-link" data-scroll-target="#description-20">Description</a></li>
  <li><a href="#usage-36" id="toc-usage-36" class="nav-link" data-scroll-target="#usage-36">Usage</a></li>
  <li><a href="#arguments-28" id="toc-arguments-28" class="nav-link" data-scroll-target="#arguments-28">Arguments</a></li>
  <li><a href="#details-8" id="toc-details-8" class="nav-link" data-scroll-target="#details-8">Details</a></li>
  <li><a href="#author-9" id="toc-author-9" class="nav-link" data-scroll-target="#author-9">Author</a></li>
  <li><a href="#seealso-4" id="toc-seealso-4" class="nav-link" data-scroll-target="#seealso-4">Seealso</a></li>
  <li><a href="#value-15" id="toc-value-15" class="nav-link" data-scroll-target="#value-15">Value</a></li>
  <li><a href="#examples-21" id="toc-examples-21" class="nav-link" data-scroll-target="#examples-21">Examples</a></li>
  </ul></li>
  <li><a href="#genes_with_weights-create-a-ranked-gene-list-for-gsea-analysis" id="toc-genes_with_weights-create-a-ranked-gene-list-for-gsea-analysis" class="nav-link" data-scroll-target="#genes_with_weights-create-a-ranked-gene-list-for-gsea-analysis"><code>genes_with_weights</code>: Create a ranked gene list for GSEA analysis</a>
  <ul class="collapse">
  <li><a href="#description-21" id="toc-description-21" class="nav-link" data-scroll-target="#description-21">Description</a></li>
  <li><a href="#usage-37" id="toc-usage-37" class="nav-link" data-scroll-target="#usage-37">Usage</a></li>
  <li><a href="#arguments-29" id="toc-arguments-29" class="nav-link" data-scroll-target="#arguments-29">Arguments</a></li>
  <li><a href="#author-10" id="toc-author-10" class="nav-link" data-scroll-target="#author-10">Author</a></li>
  <li><a href="#value-16" id="toc-value-16" class="nav-link" data-scroll-target="#value-16">Value</a></li>
  <li><a href="#examples-22" id="toc-examples-22" class="nav-link" data-scroll-target="#examples-22">Examples</a></li>
  </ul></li>
  <li><a href="#database_...-get-database-for-enrichment-analysis" id="toc-database_...-get-database-for-enrichment-analysis" class="nav-link" data-scroll-target="#database_...-get-database-for-enrichment-analysis"><code>database_...</code>: Get database for enrichment analysis</a>
  <ul class="collapse">
  <li><a href="#description-22" id="toc-description-22" class="nav-link" data-scroll-target="#description-22">Description</a></li>
  <li><a href="#usage-38" id="toc-usage-38" class="nav-link" data-scroll-target="#usage-38">Usage</a></li>
  <li><a href="#arguments-30" id="toc-arguments-30" class="nav-link" data-scroll-target="#arguments-30">Arguments</a></li>
  <li><a href="#author-11" id="toc-author-11" class="nav-link" data-scroll-target="#author-11">Author</a></li>
  <li><a href="#value-17" id="toc-value-17" class="nav-link" data-scroll-target="#value-17">Value</a></li>
  </ul></li>
  <li><a href="#database_kegg-get-kegg-database-from-kegg-website" id="toc-database_kegg-get-kegg-database-from-kegg-website" class="nav-link" data-scroll-target="#database_kegg-get-kegg-database-from-kegg-website"><code>database_KEGG</code>: Get KEGG database from KEGG website</a>
  <ul class="collapse">
  <li><a href="#description-23" id="toc-description-23" class="nav-link" data-scroll-target="#description-23">Description</a></li>
  <li><a href="#usage-39" id="toc-usage-39" class="nav-link" data-scroll-target="#usage-39">Usage</a></li>
  <li><a href="#arguments-31" id="toc-arguments-31" class="nav-link" data-scroll-target="#arguments-31">Arguments</a></li>
  <li><a href="#value-18" id="toc-value-18" class="nav-link" data-scroll-target="#value-18">Value</a></li>
  </ul></li>
  <li><a href="#delete--filter-enrichment-results-by-description-pattern" id="toc-delete--filter-enrichment-results-by-description-pattern" class="nav-link" data-scroll-target="#delete--filter-enrichment-results-by-description-pattern"><code>%-delete-&gt;%</code>: Filter Enrichment Results by Description Pattern</a>
  <ul class="collapse">
  <li><a href="#description-24" id="toc-description-24" class="nav-link" data-scroll-target="#description-24">Description</a></li>
  <li><a href="#usage-40" id="toc-usage-40" class="nav-link" data-scroll-target="#usage-40">Usage</a></li>
  <li><a href="#arguments-32" id="toc-arguments-32" class="nav-link" data-scroll-target="#arguments-32">Arguments</a></li>
  <li><a href="#details-9" id="toc-details-9" class="nav-link" data-scroll-target="#details-9">Details</a></li>
  <li><a href="#value-19" id="toc-value-19" class="nav-link" data-scroll-target="#value-19">Value</a></li>
  <li><a href="#examples-23" id="toc-examples-23" class="nav-link" data-scroll-target="#examples-23">Examples</a></li>
  </ul></li>
  </ul>
<div class="toc-actions"><ul><li><a href="https://github.com/ZhimingYe/EnrichGT_website/blob/main/pkgfuns.qmd" class="toc-action"><i class="bi bi-github"></i>View source</a></li><li><a href="https://github.com/ZhimingYe/EnrichGT_website/issues/new" class="toc-action"><i class="bi empty"></i>Report an issue</a></li></ul></div></nav>
    </div>
<!-- main -->
<main class="content" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default">
<div class="quarto-title">
<h1 class="title">Package Functions Help</h1>
</div>



<div class="quarto-title-meta">

    
  
    
  </div>
  


</header>


<section id="description" class="level1">
<h1>DESCRIPTION</h1>
<pre><code>Type: Package
Package: EnrichGT
Title: EnrichGT - all in one enrichment analysis solution
Version: 2.2.0
Authors@R: c(
    person("Zhiming", "Ye", , "garnetcrow@hotmail.com", role = c("aut", "cre")),
    person("Runchen", "Wang", , "runchen_wang@163.com", role = "cph")
  )
Description: Do biological enrichment analysis and parsing and clustering
    enrichment result to insightful results in just ONE package
License: GPL-V3
URL: https://zhimingye.github.io/EnrichGT/
Depends: 
    R (&gt;= 4.1.0)
Imports: 
    AnnotationDbi,
    cli,
    dplyr,
    ellmer,
    fgsea,
    fontawesome,
    forcats,
    ggdendro,
    ggplot2,
    ggwordcloud,
    glue,
    GO.db,
    graphics,
    grDevices,
    gt,
    htmltools,
    Matrix,
    methods,
    proxy,
    qvalue,
    R6,
    RColorBrewer,
    Rcpp,
    reactome.db,
    rlang,
    scales,
    stats,
    stringr,
    text2vec,
    tibble,
    utils,
    xfun
Suggests: 
    BiocManager,
    readr,
    testthat (&gt;= 3.0.0)
LinkingTo: 
    Rcpp
Config/testthat/edition: 3
Encoding: UTF-8
LazyData: true
RoxygenNote: 7.3.2</code></pre>
</section>
<section id="comparison_reactor_base-comparison-reactor-base" class="level1">
<h1><code>comparison_reactor_base</code>: Comparison Reactor Base</h1>
<section id="description-1" class="level2">
<h2 class="anchored" data-anchor-id="description-1">Description</h2>
<p>Typically created via <code>[egt_comparison_reactor](egt_comparison_reactor)()</code>A framework for comparing and analyzing enrichment results across multiple groups. This is the base class that provides core functionality for both ORA and GSEA analysis.</p>
</section>
<section id="details" class="level2">
<h2 class="anchored" data-anchor-id="details">Details</h2>
<p>The comparison reactor allows: - Appending multiple enrichment results from different groups - Filtering results by p-value or NES score - Comparing results between groups - Identifying relationships between enriched terms - Visualizing biological themes - Sub-clustering results for deeper analysis</p>
</section>
<section id="methods" class="level2">
<h2 class="anchored" data-anchor-id="methods">Methods</h2>
<section id="public-methods" class="level3">
<h3 class="anchored" data-anchor-id="public-methods">Public methods</h3>
<ul>
<li><a href="#method-comparison_reactor_base-new"><code>comparison_reactor_base$new()</code></a></li>
<li><a href="#method-comparison_reactor_base-append_enriched_result"><code>comparison_reactor_base$append_enriched_result()</code></a></li>
<li><a href="#method-comparison_reactor_base-summarize"><code>comparison_reactor_base$summarize()</code></a></li>
<li><a href="#method-comparison_reactor_base-prefilter_by_p_adj"><code>comparison_reactor_base$prefilter_by_p_adj()</code></a></li>
<li><a href="#method-comparison_reactor_base-find_relationship"><code>comparison_reactor_base$find_relationship()</code></a></li>
<li><a href="#method-comparison_reactor_base-fetch_relationship"><code>comparison_reactor_base$fetch_relationship()</code></a></li>
<li><a href="#method-comparison_reactor_base-fetch_biological_theme"><code>comparison_reactor_base$fetch_biological_theme()</code></a></li>
<li><a href="#method-comparison_reactor_base-split_by_cluster"><code>comparison_reactor_base$split_by_cluster()</code></a></li>
<li><a href="#method-comparison_reactor_base-get_splited_list"><code>comparison_reactor_base$get_splited_list()</code></a></li>
<li><a href="#method-comparison_reactor_base-do_recluster"><code>comparison_reactor_base$do_recluster()</code></a></li>
<li><a href="#method-comparison_reactor_base-get_recluster_result"><code>comparison_reactor_base$get_recluster_result()</code></a></li>
<li><a href="#method-comparison_reactor_base-print"><code>comparison_reactor_base$print()</code></a></li>
<li><a href="#method-comparison_reactor_base-clone"><code>comparison_reactor_base$clone()</code></a></li>
</ul>
</section>
<section id="method-new" class="level3">
<h3 class="anchored" data-anchor-id="method-new">Method <code>new()</code></h3>
<p>Create a new comparison reactor base object</p>
<section id="usage" class="level4">
<h4 class="anchored" data-anchor-id="usage">Usage</h4>
<pre><code>comparison_reactor_base$new(Type = NULL)</code></pre>
</section>
<section id="arguments" class="level4">
<h4 class="anchored" data-anchor-id="arguments">Arguments</h4>
<ul>
<li><code>Type</code>: The analysis type (“ORA” or “GSEA”)</li>
</ul>
</section>
</section>
<section id="method-append_enriched_result" class="level3">
<h3 class="anchored" data-anchor-id="method-append_enriched_result">Method <code>append_enriched_result()</code></h3>
<p>Add enrichment results to the reactor</p>
<section id="usage-1" class="level4">
<h4 class="anchored" data-anchor-id="usage-1">Usage</h4>
<pre><code>comparison_reactor_base$append_enriched_result(x, group)</code></pre>
</section>
<section id="arguments-1" class="level4">
<h4 class="anchored" data-anchor-id="arguments-1">Arguments</h4>
<ul>
<li><code>x</code>: Data.frame of enrichment results (must contain p-value/adjusted p-value columns)</li>
<li><code>group</code>: Character string naming the group for these results</li>
</ul>
</section>
<section id="returns" class="level4">
<h4 class="anchored" data-anchor-id="returns">Returns</h4>
<p>The reactor object (invisible) for method chaining</p>
</section>
<section id="examples" class="level4">
<h4 class="anchored" data-anchor-id="examples">Examples</h4>
<pre><code>\dontrun{
reactor &lt;- egt_comparison_reactor("ORA")
reactor$append_enriched_result(ora_result1, "group1")
}</code></pre>
</section>
</section>
<section id="method-summarize" class="level3">
<h3 class="anchored" data-anchor-id="method-summarize">Method <code>summarize()</code></h3>
<p>Print summary of groups in reactor</p>
<section id="usage-2" class="level4">
<h4 class="anchored" data-anchor-id="usage-2">Usage</h4>
<pre><code>comparison_reactor_base$summarize()</code></pre>
</section>
</section>
<section id="method-prefilter_by_p_adj" class="level3">
<h3 class="anchored" data-anchor-id="method-prefilter_by_p_adj">Method <code>prefilter_by_p_adj()</code></h3>
<p>Filter enrichment results by adjusted p-value cutoff</p>
<section id="usage-3" class="level4">
<h4 class="anchored" data-anchor-id="usage-3">Usage</h4>
<pre><code>comparison_reactor_base$prefilter_by_p_adj(x = 0.05)</code></pre>
</section>
<section id="arguments-2" class="level4">
<h4 class="anchored" data-anchor-id="arguments-2">Arguments</h4>
<ul>
<li><code>x</code>: Numeric cutoff for adjusted p-values (default 0.05)</li>
</ul>
</section>
<section id="returns-1" class="level4">
<h4 class="anchored" data-anchor-id="returns-1">Returns</h4>
<p>The reactor object (invisible) for method chaining</p>
</section>
<section id="examples-1" class="level4">
<h4 class="anchored" data-anchor-id="examples-1">Examples</h4>
<pre><code>\dontrun{
reactor$prefilter_by_p_adj(0.01) # Use stricter cutoff
}</code></pre>
</section>
</section>
<section id="method-find_relationship" class="level3">
<h3 class="anchored" data-anchor-id="method-find_relationship">Method <code>find_relationship()</code></h3>
<p>Identify relationships between enriched terms across groups</p>
<section id="usage-4" class="level4">
<h4 class="anchored" data-anchor-id="usage-4">Usage</h4>
<pre><code>comparison_reactor_base$find_relationship(
  Num = NULL,
  dist_method = "euclidean",
  hclust_method = "ward.D2",
  ...
)</code></pre>
</section>
<section id="arguments-3" class="level4">
<h4 class="anchored" data-anchor-id="arguments-3">Arguments</h4>
<ul>
<li><code>Num</code>: Number of top terms to consider from each group</li>
<li><code>dist_method</code>: Distance method for clustering (default “euclidean”)</li>
<li><code>hclust_method</code>: Hierarchical clustering method (default “ward.D2”)</li>
<li><code>...</code>: Additional parameters passed to heatmap function</li>
</ul>
</section>
<section id="returns-2" class="level4">
<h4 class="anchored" data-anchor-id="returns-2">Returns</h4>
<p>The reactor object (invisible) for method chaining</p>
</section>
<section id="examples-2" class="level4">
<h4 class="anchored" data-anchor-id="examples-2">Examples</h4>
<pre><code>\dontrun{
reactor$find_relationship(Num = 5, dist_method = "manhattan")
}</code></pre>
</section>
</section>
<section id="method-fetch_relationship" class="level3">
<h3 class="anchored" data-anchor-id="method-fetch_relationship">Method <code>fetch_relationship()</code></h3>
<p>Retrieve relationship data between terms</p>
<section id="usage-5" class="level4">
<h4 class="anchored" data-anchor-id="usage-5">Usage</h4>
<pre><code>comparison_reactor_base$fetch_relationship()</code></pre>
</section>
<section id="returns-3" class="level4">
<h4 class="anchored" data-anchor-id="returns-3">Returns</h4>
<p>Data.frame containing term relationships and cluster assignments</p>
</section>
<section id="examples-3" class="level4">
<h4 class="anchored" data-anchor-id="examples-3">Examples</h4>
<pre><code>\dontrun{
relation_df &lt;- reactor$fetch_relationship()
head(relation_df)
}</code></pre>
</section>
</section>
<section id="method-fetch_biological_theme" class="level3">
<h3 class="anchored" data-anchor-id="method-fetch_biological_theme">Method <code>fetch_biological_theme()</code></h3>
<p>Generate wordcloud visualization of biological themes</p>
<section id="usage-6" class="level4">
<h4 class="anchored" data-anchor-id="usage-6">Usage</h4>
<pre><code>comparison_reactor_base$fetch_biological_theme(...)</code></pre>
</section>
<section id="arguments-4" class="level4">
<h4 class="anchored" data-anchor-id="arguments-4">Arguments</h4>
<ul>
<li><code>...</code>: Additional parameters passed to wordcloud generator</li>
</ul>
</section>
<section id="returns-4" class="level4">
<h4 class="anchored" data-anchor-id="returns-4">Returns</h4>
<p>List containing ggplot2 wordcloud objects</p>
</section>
<section id="examples-4" class="level4">
<h4 class="anchored" data-anchor-id="examples-4">Examples</h4>
<pre><code>\dontrun{
wordclouds &lt;- reactor$fetch_biological_theme()
wordclouds[[1]] # View first wordcloud
}</code></pre>
</section>
</section>
<section id="method-split_by_cluster" class="level3">
<h3 class="anchored" data-anchor-id="method-split_by_cluster">Method <code>split_by_cluster()</code></h3>
<p>Split results by identified clusters for further analysis</p>
<section id="usage-7" class="level4">
<h4 class="anchored" data-anchor-id="usage-7">Usage</h4>
<pre><code>comparison_reactor_base$split_by_cluster(...)</code></pre>
</section>
<section id="arguments-5" class="level4">
<h4 class="anchored" data-anchor-id="arguments-5">Arguments</h4>
<ul>
<li><code>...</code>: Additional parameters</li>
</ul>
</section>
<section id="returns-5" class="level4">
<h4 class="anchored" data-anchor-id="returns-5">Returns</h4>
<p>The reactor object (invisible) for method chaining</p>
</section>
</section>
<section id="method-get_splited_list" class="level3">
<h3 class="anchored" data-anchor-id="method-get_splited_list">Method <code>get_splited_list()</code></h3>
<p>Get list of results split by cluster</p>
<section id="usage-8" class="level4">
<h4 class="anchored" data-anchor-id="usage-8">Usage</h4>
<pre><code>comparison_reactor_base$get_splited_list()</code></pre>
</section>
<section id="returns-6" class="level4">
<h4 class="anchored" data-anchor-id="returns-6">Returns</h4>
<p>List of data frames split by cluster</p>
</section>
</section>
<section id="method-do_recluster" class="level3">
<h3 class="anchored" data-anchor-id="method-do_recluster">Method <code>do_recluster()</code></h3>
<p>Perform sub-clustering within existing clusters</p>
<section id="usage-9" class="level4">
<h4 class="anchored" data-anchor-id="usage-9">Usage</h4>
<pre><code>comparison_reactor_base$do_recluster(
  ClusterNum = 10,
  P.adj = 0.05,
  force = F,
  nTop = 10,
  method = "ward.D2",
  ...
)</code></pre>
</section>
<section id="arguments-6" class="level4">
<h4 class="anchored" data-anchor-id="arguments-6">Arguments</h4>
<ul>
<li><code>ClusterNum</code>: Number of sub-clusters to generate</li>
<li><code>P.adj</code>: Adjusted p-value cutoff (default 0.05)</li>
<li><code>force</code>: Whether to force reclustering (default FALSE)</li>
<li><code>nTop</code>: Number of top terms to consider (default 10)</li>
<li><code>method</code>: Clustering method (default “ward.D2”)</li>
<li><code>...</code>: Additional parameters passed to clustering functions</li>
</ul>
</section>
<section id="returns-7" class="level4">
<h4 class="anchored" data-anchor-id="returns-7">Returns</h4>
<p>The reactor object (invisible) for method chaining</p>
</section>
<section id="examples-5" class="level4">
<h4 class="anchored" data-anchor-id="examples-5">Examples</h4>
<pre><code>\dontrun{
reactor$do_recluster(ClusterNum = 5, method = "complete")
}</code></pre>
</section>
</section>
<section id="method-get_recluster_result" class="level3">
<h3 class="anchored" data-anchor-id="method-get_recluster_result">Method <code>get_recluster_result()</code></h3>
<p>Retrieve sub-clustering results</p>
<section id="usage-10" class="level4">
<h4 class="anchored" data-anchor-id="usage-10">Usage</h4>
<pre><code>comparison_reactor_base$get_recluster_result()</code></pre>
</section>
<section id="returns-8" class="level4">
<h4 class="anchored" data-anchor-id="returns-8">Returns</h4>
<p>List containing reclustering results</p>
</section>
<section id="examples-6" class="level4">
<h4 class="anchored" data-anchor-id="examples-6">Examples</h4>
<pre><code>\dontrun{
recluster_results &lt;- reactor$get_recluster_result()
names(recluster_results)
}</code></pre>
</section>
</section>
<section id="method-print" class="level3">
<h3 class="anchored" data-anchor-id="method-print">Method <code>print()</code></h3>
<section id="usage-11" class="level4">
<h4 class="anchored" data-anchor-id="usage-11">Usage</h4>
<pre><code>comparison_reactor_base$print(...)</code></pre>
</section>
</section>
<section id="method-clone" class="level3">
<h3 class="anchored" data-anchor-id="method-clone">Method <code>clone()</code></h3>
<p>The objects of this class are cloneable with this method.</p>
<section id="usage-12" class="level4">
<h4 class="anchored" data-anchor-id="usage-12">Usage</h4>
<pre><code>comparison_reactor_base$clone(deep = FALSE)</code></pre>
</section>
<section id="arguments-7" class="level4">
<h4 class="anchored" data-anchor-id="arguments-7">Arguments</h4>
<ul>
<li><code>deep</code>: Whether to make a deep clone.</li>
</ul>
</section>
</section>
</section>
<section id="examples-7" class="level2">
<h2 class="anchored" data-anchor-id="examples-7">Examples</h2>
<div class="sourceCode" id="cb22"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb22-1"><a href="#cb22-1" aria-hidden="true" tabindex="-1"></a><span class="do">## ------------------------------------------------</span></span>
<span id="cb22-2"><a href="#cb22-2" aria-hidden="true" tabindex="-1"></a><span class="do">## Method `comparison_reactor_base$append_enriched_result`</span></span>
<span id="cb22-3"><a href="#cb22-3" aria-hidden="true" tabindex="-1"></a><span class="do">## ------------------------------------------------</span></span>
<span id="cb22-4"><a href="#cb22-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb22-5"><a href="#cb22-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb22-6"><a href="#cb22-6" aria-hidden="true" tabindex="-1"></a>reactor <span class="ot">&lt;-</span> <span class="fu">egt_comparison_reactor</span>(<span class="st">"ORA"</span>)</span>
<span id="cb22-7"><a href="#cb22-7" aria-hidden="true" tabindex="-1"></a>reactor<span class="sc">$</span><span class="fu">append_enriched_result</span>(ora_result1, <span class="st">"group1"</span>)</span>
<span id="cb22-8"><a href="#cb22-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb22-9"><a href="#cb22-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb22-10"><a href="#cb22-10" aria-hidden="true" tabindex="-1"></a><span class="do">## ------------------------------------------------</span></span>
<span id="cb22-11"><a href="#cb22-11" aria-hidden="true" tabindex="-1"></a><span class="do">## Method `comparison_reactor_base$prefilter_by_p_adj`</span></span>
<span id="cb22-12"><a href="#cb22-12" aria-hidden="true" tabindex="-1"></a><span class="do">## ------------------------------------------------</span></span>
<span id="cb22-13"><a href="#cb22-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb22-14"><a href="#cb22-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb22-15"><a href="#cb22-15" aria-hidden="true" tabindex="-1"></a>reactor<span class="sc">$</span><span class="fu">prefilter_by_p_adj</span>(<span class="fl">0.01</span>) <span class="co"># Use stricter cutoff</span></span>
<span id="cb22-16"><a href="#cb22-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb22-17"><a href="#cb22-17" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb22-18"><a href="#cb22-18" aria-hidden="true" tabindex="-1"></a><span class="do">## ------------------------------------------------</span></span>
<span id="cb22-19"><a href="#cb22-19" aria-hidden="true" tabindex="-1"></a><span class="do">## Method `comparison_reactor_base$find_relationship`</span></span>
<span id="cb22-20"><a href="#cb22-20" aria-hidden="true" tabindex="-1"></a><span class="do">## ------------------------------------------------</span></span>
<span id="cb22-21"><a href="#cb22-21" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb22-22"><a href="#cb22-22" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb22-23"><a href="#cb22-23" aria-hidden="true" tabindex="-1"></a>reactor<span class="sc">$</span><span class="fu">find_relationship</span>(<span class="at">Num =</span> <span class="dv">5</span>, <span class="at">dist_method =</span> <span class="st">"manhattan"</span>)</span>
<span id="cb22-24"><a href="#cb22-24" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb22-25"><a href="#cb22-25" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb22-26"><a href="#cb22-26" aria-hidden="true" tabindex="-1"></a><span class="do">## ------------------------------------------------</span></span>
<span id="cb22-27"><a href="#cb22-27" aria-hidden="true" tabindex="-1"></a><span class="do">## Method `comparison_reactor_base$fetch_relationship`</span></span>
<span id="cb22-28"><a href="#cb22-28" aria-hidden="true" tabindex="-1"></a><span class="do">## ------------------------------------------------</span></span>
<span id="cb22-29"><a href="#cb22-29" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb22-30"><a href="#cb22-30" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb22-31"><a href="#cb22-31" aria-hidden="true" tabindex="-1"></a>relation_df <span class="ot">&lt;-</span> reactor<span class="sc">$</span><span class="fu">fetch_relationship</span>()</span>
<span id="cb22-32"><a href="#cb22-32" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(relation_df)</span>
<span id="cb22-33"><a href="#cb22-33" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb22-34"><a href="#cb22-34" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb22-35"><a href="#cb22-35" aria-hidden="true" tabindex="-1"></a><span class="do">## ------------------------------------------------</span></span>
<span id="cb22-36"><a href="#cb22-36" aria-hidden="true" tabindex="-1"></a><span class="do">## Method `comparison_reactor_base$fetch_biological_theme`</span></span>
<span id="cb22-37"><a href="#cb22-37" aria-hidden="true" tabindex="-1"></a><span class="do">## ------------------------------------------------</span></span>
<span id="cb22-38"><a href="#cb22-38" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb22-39"><a href="#cb22-39" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb22-40"><a href="#cb22-40" aria-hidden="true" tabindex="-1"></a>wordclouds <span class="ot">&lt;-</span> reactor<span class="sc">$</span><span class="fu">fetch_biological_theme</span>()</span>
<span id="cb22-41"><a href="#cb22-41" aria-hidden="true" tabindex="-1"></a>wordclouds[[<span class="dv">1</span>]] <span class="co"># View first wordcloud</span></span>
<span id="cb22-42"><a href="#cb22-42" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb22-43"><a href="#cb22-43" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb22-44"><a href="#cb22-44" aria-hidden="true" tabindex="-1"></a><span class="do">## ------------------------------------------------</span></span>
<span id="cb22-45"><a href="#cb22-45" aria-hidden="true" tabindex="-1"></a><span class="do">## Method `comparison_reactor_base$do_recluster`</span></span>
<span id="cb22-46"><a href="#cb22-46" aria-hidden="true" tabindex="-1"></a><span class="do">## ------------------------------------------------</span></span>
<span id="cb22-47"><a href="#cb22-47" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb22-48"><a href="#cb22-48" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb22-49"><a href="#cb22-49" aria-hidden="true" tabindex="-1"></a>reactor<span class="sc">$</span><span class="fu">do_recluster</span>(<span class="at">ClusterNum =</span> <span class="dv">5</span>, <span class="at">method =</span> <span class="st">"complete"</span>)</span>
<span id="cb22-50"><a href="#cb22-50" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb22-51"><a href="#cb22-51" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb22-52"><a href="#cb22-52" aria-hidden="true" tabindex="-1"></a><span class="do">## ------------------------------------------------</span></span>
<span id="cb22-53"><a href="#cb22-53" aria-hidden="true" tabindex="-1"></a><span class="do">## Method `comparison_reactor_base$get_recluster_result`</span></span>
<span id="cb22-54"><a href="#cb22-54" aria-hidden="true" tabindex="-1"></a><span class="do">## ------------------------------------------------</span></span>
<span id="cb22-55"><a href="#cb22-55" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb22-56"><a href="#cb22-56" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb22-57"><a href="#cb22-57" aria-hidden="true" tabindex="-1"></a>recluster_results <span class="ot">&lt;-</span> reactor<span class="sc">$</span><span class="fu">get_recluster_result</span>()</span>
<span id="cb22-58"><a href="#cb22-58" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(recluster_results)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</section>
</section>
<section id="comparison_reactor_gsea-gsea-comparison-reactor" class="level1">
<h1><code>comparison_reactor_gsea</code>: GSEA Comparison Reactor</h1>
<section id="description-2" class="level2">
<h2 class="anchored" data-anchor-id="description-2">Description</h2>
<p>Typically created via <code>[egt_comparison_reactor](egt_comparison_reactor)("GSEA")</code></p>
</section>
<section id="details-1" class="level2">
<h2 class="anchored" data-anchor-id="details-1">Details</h2>
<p>This reactor is optimized for comparing GSEA results across multiple groups, with methods tailored for NES (Normalized Enrichment Score) based comparisons.</p>
</section>
<section id="super-class" class="level2">
<h2 class="anchored" data-anchor-id="super-class">Super class</h2>
<p><code>[EnrichGT::comparison_reactor_base](https://rdrr.io/pkg/EnrichGT/man/comparison_reactor_base.html)</code> -&gt; <code>comparison_reactor_gsea</code></p>
</section>
<section id="methods-1" class="level2">
<h2 class="anchored" data-anchor-id="methods-1">Methods</h2>
<section id="public-methods-1" class="level3">
<h3 class="anchored" data-anchor-id="public-methods-1">Public methods</h3>
<ul>
<li><a href="#method-comparison_reactor_gsea-new"><code>comparison_reactor_gsea$new()</code></a></li>
<li><a href="#method-comparison_reactor_gsea-make_plans"><code>comparison_reactor_gsea$make_plans()</code></a></li>
<li><a href="#method-comparison_reactor_gsea-prefilter_by_NES"><code>comparison_reactor_gsea$prefilter_by_NES()</code></a></li>
<li><a href="#method-comparison_reactor_gsea-clone"><code>comparison_reactor_gsea$clone()</code></a></li>
</ul>
</section>
<section id="method-new-1" class="level3">
<h3 class="anchored" data-anchor-id="method-new-1">Method <code>new()</code></h3>
<p>Create a new GSEA comparison reactor</p>
<section id="usage-13" class="level4">
<h4 class="anchored" data-anchor-id="usage-13">Usage</h4>
<pre><code>comparison_reactor_gsea$new()</code></pre>
</section>
</section>
<section id="method-make_plans" class="level3">
<h3 class="anchored" data-anchor-id="method-make_plans">Method <code>make_plans()</code></h3>
<p>Create comparison plans between specified GSEA groups</p>
<section id="usage-14" class="level4">
<h4 class="anchored" data-anchor-id="usage-14">Usage</h4>
<pre><code>comparison_reactor_gsea$make_plans(group = "auto", use_value = "NES")</code></pre>
</section>
<section id="arguments-8" class="level4">
<h4 class="anchored" data-anchor-id="arguments-8">Arguments</h4>
<ul>
<li><code>group</code>: Character vector of group names to compare or “auto” for all groups</li>
<li><code>use_value</code>: Which value to use for comparison (“NES” for normalized enrichment score)</li>
</ul>
</section>
<section id="returns-9" class="level4">
<h4 class="anchored" data-anchor-id="returns-9">Returns</h4>
<p>The reactor object (invisible) for method chaining</p>
</section>
<section id="examples-8" class="level4">
<h4 class="anchored" data-anchor-id="examples-8">Examples</h4>
<pre><code>\dontrun{
gsea_reactor$make_plans(group = c("group1", "group2"))
}</code></pre>
</section>
</section>
<section id="method-prefilter_by_nes" class="level3">
<h3 class="anchored" data-anchor-id="method-prefilter_by_nes">Method <code>prefilter_by_NES()</code></h3>
<p>Filter GSEA results by normalized enrichment score cutoff</p>
<section id="usage-15" class="level4">
<h4 class="anchored" data-anchor-id="usage-15">Usage</h4>
<pre><code>comparison_reactor_gsea$prefilter_by_NES(x = 1)</code></pre>
</section>
<section id="arguments-9" class="level4">
<h4 class="anchored" data-anchor-id="arguments-9">Arguments</h4>
<ul>
<li><code>x</code>: Numeric cutoff for absolute NES values (default 1)</li>
</ul>
</section>
<section id="returns-10" class="level4">
<h4 class="anchored" data-anchor-id="returns-10">Returns</h4>
<p>The reactor object (invisible) for method chaining</p>
</section>
<section id="examples-9" class="level4">
<h4 class="anchored" data-anchor-id="examples-9">Examples</h4>
<pre><code>\dontrun{
gsea_reactor$prefilter_by_NES(1.5) # Filter for stronger effects
}</code></pre>
</section>
</section>
<section id="method-clone-1" class="level3">
<h3 class="anchored" data-anchor-id="method-clone-1">Method <code>clone()</code></h3>
<p>The objects of this class are cloneable with this method.</p>
<section id="usage-16" class="level4">
<h4 class="anchored" data-anchor-id="usage-16">Usage</h4>
<pre><code>comparison_reactor_gsea$clone(deep = FALSE)</code></pre>
</section>
<section id="arguments-10" class="level4">
<h4 class="anchored" data-anchor-id="arguments-10">Arguments</h4>
<ul>
<li><code>deep</code>: Whether to make a deep clone.</li>
</ul>
</section>
</section>
</section>
<section id="seealso" class="level2">
<h2 class="anchored" data-anchor-id="seealso">Seealso</h2>
<p><code>[comparison_reactor_base](comparison_reactor_base)</code> for inherited methods A specialized reactor for comparing Gene Set Enrichment Analysis (GSEA) results. Inherits from comparison_reactor_base and provides GSEA-specific functionality.</p>
</section>
<section id="examples-10" class="level2">
<h2 class="anchored" data-anchor-id="examples-10">Examples</h2>
<div class="sourceCode" id="cb29"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb29-1"><a href="#cb29-1" aria-hidden="true" tabindex="-1"></a><span class="do">## ------------------------------------------------</span></span>
<span id="cb29-2"><a href="#cb29-2" aria-hidden="true" tabindex="-1"></a><span class="do">## Method `comparison_reactor_gsea$make_plans`</span></span>
<span id="cb29-3"><a href="#cb29-3" aria-hidden="true" tabindex="-1"></a><span class="do">## ------------------------------------------------</span></span>
<span id="cb29-4"><a href="#cb29-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-5"><a href="#cb29-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-6"><a href="#cb29-6" aria-hidden="true" tabindex="-1"></a>gsea_reactor<span class="sc">$</span><span class="fu">make_plans</span>(<span class="at">group =</span> <span class="fu">c</span>(<span class="st">"group1"</span>, <span class="st">"group2"</span>))</span>
<span id="cb29-7"><a href="#cb29-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-8"><a href="#cb29-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-9"><a href="#cb29-9" aria-hidden="true" tabindex="-1"></a><span class="do">## ------------------------------------------------</span></span>
<span id="cb29-10"><a href="#cb29-10" aria-hidden="true" tabindex="-1"></a><span class="do">## Method `comparison_reactor_gsea$prefilter_by_NES`</span></span>
<span id="cb29-11"><a href="#cb29-11" aria-hidden="true" tabindex="-1"></a><span class="do">## ------------------------------------------------</span></span>
<span id="cb29-12"><a href="#cb29-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-13"><a href="#cb29-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-14"><a href="#cb29-14" aria-hidden="true" tabindex="-1"></a>gsea_reactor<span class="sc">$</span><span class="fu">prefilter_by_NES</span>(<span class="fl">1.5</span>) <span class="co"># Filter for stronger effects</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</section>
</section>
<section id="comparison_reactor_ora-ora-comparison-reactor" class="level1">
<h1><code>comparison_reactor_ora</code>: ORA Comparison Reactor</h1>
<section id="description-3" class="level2">
<h2 class="anchored" data-anchor-id="description-3">Description</h2>
<p>Typically created via <code>[egt_comparison_reactor](egt_comparison_reactor)("ORA")</code></p>
</section>
<section id="details-2" class="level2">
<h2 class="anchored" data-anchor-id="details-2">Details</h2>
<p>This reactor is optimized for comparing ORA results across multiple groups, with methods tailored for p-value based comparisons.</p>
</section>
<section id="super-class-1" class="level2">
<h2 class="anchored" data-anchor-id="super-class-1">Super class</h2>
<p><code>[EnrichGT::comparison_reactor_base](https://rdrr.io/pkg/EnrichGT/man/comparison_reactor_base.html)</code> -&gt; <code>comparison_reactor_ora</code></p>
</section>
<section id="methods-2" class="level2">
<h2 class="anchored" data-anchor-id="methods-2">Methods</h2>
<section id="public-methods-2" class="level3">
<h3 class="anchored" data-anchor-id="public-methods-2">Public methods</h3>
<ul>
<li><a href="#method-comparison_reactor_ora-new"><code>comparison_reactor_ora$new()</code></a></li>
<li><a href="#method-comparison_reactor_ora-make_plans"><code>comparison_reactor_ora$make_plans()</code></a></li>
<li><a href="#method-comparison_reactor_ora-clone"><code>comparison_reactor_ora$clone()</code></a></li>
</ul>
</section>
<section id="method-new-2" class="level3">
<h3 class="anchored" data-anchor-id="method-new-2">Method <code>new()</code></h3>
<p>Create a new ORA comparison reactor</p>
<section id="usage-17" class="level4">
<h4 class="anchored" data-anchor-id="usage-17">Usage</h4>
<pre><code>comparison_reactor_ora$new()</code></pre>
</section>
</section>
<section id="method-make_plans-1" class="level3">
<h3 class="anchored" data-anchor-id="method-make_plans-1">Method <code>make_plans()</code></h3>
<p>Create comparison plans between specified ORA groups</p>
<section id="usage-18" class="level4">
<h4 class="anchored" data-anchor-id="usage-18">Usage</h4>
<pre><code>comparison_reactor_ora$make_plans(group = "auto", use_value = "p")</code></pre>
</section>
<section id="arguments-11" class="level4">
<h4 class="anchored" data-anchor-id="arguments-11">Arguments</h4>
<ul>
<li><code>group</code>: Character vector of group names to compare or “auto” for all groups</li>
<li><code>use_value</code>: Which value to use for comparison (“p” for p-value or “padj” for adjusted p-value)</li>
</ul>
</section>
<section id="returns-11" class="level4">
<h4 class="anchored" data-anchor-id="returns-11">Returns</h4>
<p>The reactor object (invisible) for method chaining</p>
</section>
<section id="examples-11" class="level4">
<h4 class="anchored" data-anchor-id="examples-11">Examples</h4>
<pre><code>\dontrun{
ora_reactor$make_plans(group = c("group1", "group2"), use_value = "padj")
}</code></pre>
</section>
</section>
<section id="method-clone-2" class="level3">
<h3 class="anchored" data-anchor-id="method-clone-2">Method <code>clone()</code></h3>
<p>The objects of this class are cloneable with this method.</p>
<section id="usage-19" class="level4">
<h4 class="anchored" data-anchor-id="usage-19">Usage</h4>
<pre><code>comparison_reactor_ora$clone(deep = FALSE)</code></pre>
</section>
<section id="arguments-12" class="level4">
<h4 class="anchored" data-anchor-id="arguments-12">Arguments</h4>
<ul>
<li><code>deep</code>: Whether to make a deep clone.</li>
</ul>
</section>
</section>
</section>
<section id="seealso-1" class="level2">
<h2 class="anchored" data-anchor-id="seealso-1">Seealso</h2>
<p><code>[comparison_reactor_base](comparison_reactor_base)</code> for inherited methods A specialized reactor for comparing Over-Representation Analysis (ORA) results. Inherits from comparison_reactor_base and provides ORA-specific functionality.</p>
</section>
<section id="examples-12" class="level2">
<h2 class="anchored" data-anchor-id="examples-12">Examples</h2>
<div class="sourceCode" id="cb34"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb34-1"><a href="#cb34-1" aria-hidden="true" tabindex="-1"></a><span class="do">## ------------------------------------------------</span></span>
<span id="cb34-2"><a href="#cb34-2" aria-hidden="true" tabindex="-1"></a><span class="do">## Method `comparison_reactor_ora$make_plans`</span></span>
<span id="cb34-3"><a href="#cb34-3" aria-hidden="true" tabindex="-1"></a><span class="do">## ------------------------------------------------</span></span>
<span id="cb34-4"><a href="#cb34-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-5"><a href="#cb34-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-6"><a href="#cb34-6" aria-hidden="true" tabindex="-1"></a>ora_reactor<span class="sc">$</span><span class="fu">make_plans</span>(<span class="at">group =</span> <span class="fu">c</span>(<span class="st">"group1"</span>, <span class="st">"group2"</span>), <span class="at">use_value =</span> <span class="st">"padj"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</section>
</section>
<section id="convert_annotations_genes-convert-gene-annotations-from-any-keys-to-any-keys" class="level1">
<h1><code>convert_annotations_genes</code>: Convert gene annotations from any keys to any keys</h1>
<section id="description-4" class="level2">
<h2 class="anchored" data-anchor-id="description-4">Description</h2>
<p>Convert gene annotations from any keys to any keys</p>
</section>
<section id="usage-20" class="level2">
<h2 class="anchored" data-anchor-id="usage-20">Usage</h2>
<div class="sourceCode" id="cb35"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb35-1"><a href="#cb35-1" aria-hidden="true" tabindex="-1"></a><span class="fu">convert_annotations_genes</span>(genes, from_what, to_what, OrgDB)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</section>
<section id="arguments-13" class="level2">
<h2 class="anchored" data-anchor-id="arguments-13">Arguments</h2>
<ul>
<li><code>genes</code>: gene vector</li>
<li><code>from_what</code>: input type (like “SYMBOL”,“ENTREZID”,“ENSEMBL”,“GENENAME”,…), keys should be supported by AnnotationDbi. Search for the help page of AnnotationDbi for further help.</li>
<li><code>to_what</code>: output type (like “SYMBOL”,“ENTREZID”,“ENSEMBL”,“GENENAME”,…), keys should be supported by AnnotationDbi. Search for the help page of AnnotationDbi for further help. Can be multiple items E.g. <code>c("ENTREZID","ENSEMBL","GENENAME")</code></li>
<li><code>OrgDB</code>: human = org.Hs.eg.db, mouse = org.Mm.eg.db, search BioConductor website for further help</li>
</ul>
</section>
<section id="value" class="level2">
<h2 class="anchored" data-anchor-id="value">Value</h2>
<p>a data.frame</p>
</section>
</section>
<section id="database_from_gmt-parse-gmt-format-gene-set-files" class="level1">
<h1><code>database_from_gmt</code>: Parse GMT format gene set files</h1>
<section id="description-5" class="level2">
<h2 class="anchored" data-anchor-id="description-5">Description</h2>
<p>Reads gene set files in GMT format (e.g., from MSigDB or WikiPathways) and converts them to a data frame suitable for enrichment analysis. Can optionally convert ENTREZ IDs to gene symbols.</p>
</section>
<section id="usage-21" class="level2">
<h2 class="anchored" data-anchor-id="usage-21">Usage</h2>
<div class="sourceCode" id="cb36"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb36-1"><a href="#cb36-1" aria-hidden="true" tabindex="-1"></a><span class="fu">database_from_gmt</span>(gmtfile, <span class="at">OrgDB =</span> <span class="cn">NULL</span>, <span class="at">convert_2_symbols =</span> T)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</section>
<section id="arguments-14" class="level2">
<h2 class="anchored" data-anchor-id="arguments-14">Arguments</h2>
<ul>
<li><code>gmtfile</code>: Path to GMT format file</li>
<li><code>OrgDB</code>: Annotation database for ID conversion (e.g., org.Hs.eg.db for human). Required if convert_2_symbols=TRUE.</li>
<li><code>convert_2_symbols</code>: Logical indicating whether to convert ENTREZ IDs to gene symbols. Default is TRUE.</li>
</ul>
</section>
<section id="author" class="level2">
<h2 class="anchored" data-anchor-id="author">Author</h2>
<p>Original GMT parser by Guangchuang Yu (https://github.com/YuLab-SMU/gson). Cache system and enhancements by Zhiming Ye.</p>
</section>
<section id="value-1" class="level2">
<h2 class="anchored" data-anchor-id="value-1">Value</h2>
<p>A data frame with columns:</p>
<ul>
<li>term: Gene set name</li>
<li>gene: Gene identifiers (symbols or ENTREZ IDs)</li>
</ul>
<p>If input has 3 columns, includes an additional ID column.</p>
</section>
<section id="examples-13" class="level2">
<h2 class="anchored" data-anchor-id="examples-13">Examples</h2>
<div class="sourceCode" id="cb37"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb37-1"><a href="#cb37-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Read MSigDB hallmark gene sets</span></span>
<span id="cb37-2"><a href="#cb37-2" aria-hidden="true" tabindex="-1"></a>gmt_file <span class="ot">&lt;-</span> <span class="fu">system.file</span>(<span class="st">"extdata"</span>, <span class="st">"h.all.v7.4.symbols.gmt"</span>, <span class="at">package =</span> <span class="st">"EnrichGT"</span>)</span>
<span id="cb37-3"><a href="#cb37-3" aria-hidden="true" tabindex="-1"></a>gene_sets <span class="ot">&lt;-</span> <span class="fu">database_from_gmt</span>(gmt_file)</span>
<span id="cb37-4"><a href="#cb37-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb37-5"><a href="#cb37-5" aria-hidden="true" tabindex="-1"></a><span class="co"># Read WikiPathways with ENTREZ to symbol conversion</span></span>
<span id="cb37-6"><a href="#cb37-6" aria-hidden="true" tabindex="-1"></a>gmt_file <span class="ot">&lt;-</span> <span class="st">"wikipathways-20220310-gmt-Homo_sapiens.gmt"</span></span>
<span id="cb37-7"><a href="#cb37-7" aria-hidden="true" tabindex="-1"></a>gene_sets <span class="ot">&lt;-</span> <span class="fu">database_from_gmt</span>(gmt_file, <span class="at">OrgDB =</span> org.Hs.eg.db)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</section>
</section>
<section id="egt_compare_groups-2-group-comparison-of-enrichment-results-and-further-clustering-and-visualizing" class="level1">
<h1><code>egt_compare_groups</code>: 2-Group Comparison of enrichment results and further clustering and visualizing</h1>
<section id="description-6" class="level2">
<h2 class="anchored" data-anchor-id="description-6">Description</h2>
<p>See <code>?egt_enrichment_analysis()</code></p>
</section>
<section id="usage-22" class="level2">
<h2 class="anchored" data-anchor-id="usage-22">Usage</h2>
<div class="sourceCode" id="cb38"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb38-1"><a href="#cb38-1" aria-hidden="true" tabindex="-1"></a><span class="fu">egt_compare_groups</span>(</span>
<span id="cb38-2"><a href="#cb38-2" aria-hidden="true" tabindex="-1"></a>  obj.test,</span>
<span id="cb38-3"><a href="#cb38-3" aria-hidden="true" tabindex="-1"></a>  obj.ctrl,</span>
<span id="cb38-4"><a href="#cb38-4" aria-hidden="true" tabindex="-1"></a>  <span class="at">name.test =</span> <span class="cn">NULL</span>,</span>
<span id="cb38-5"><a href="#cb38-5" aria-hidden="true" tabindex="-1"></a>  <span class="at">name.ctrl =</span> <span class="cn">NULL</span>,</span>
<span id="cb38-6"><a href="#cb38-6" aria-hidden="true" tabindex="-1"></a>  <span class="at">ClusterNum =</span> <span class="dv">15</span>,</span>
<span id="cb38-7"><a href="#cb38-7" aria-hidden="true" tabindex="-1"></a>  <span class="at">P.adj =</span> <span class="fl">0.05</span>,</span>
<span id="cb38-8"><a href="#cb38-8" aria-hidden="true" tabindex="-1"></a>  <span class="at">force =</span> F,</span>
<span id="cb38-9"><a href="#cb38-9" aria-hidden="true" tabindex="-1"></a>  <span class="at">nTop =</span> <span class="dv">10</span>,</span>
<span id="cb38-10"><a href="#cb38-10" aria-hidden="true" tabindex="-1"></a>  <span class="at">method =</span> <span class="st">"ward.D2"</span>,</span>
<span id="cb38-11"><a href="#cb38-11" aria-hidden="true" tabindex="-1"></a>  ...</span>
<span id="cb38-12"><a href="#cb38-12" aria-hidden="true" tabindex="-1"></a>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</section>
<section id="arguments-15" class="level2">
<h2 class="anchored" data-anchor-id="arguments-15">Arguments</h2>
<ul>
<li><code>obj.test</code>: the enriched object from tested group. WARNING: <code>obj.test</code> and <code>obj.ctrl</code> should come from same database (e.g.&nbsp;GO Biological Process(GOBP)).</li>
<li><code>obj.ctrl</code>: the enriched object from control group. WARNING: <code>obj.test</code> and <code>obj.ctrl</code> should come from same database (e.g.&nbsp;GO Biological Process(GOBP)).</li>
<li><code>name.test</code>: optional, the name of the testing group. If is <code>NULL</code>, the object name of <code>obj.test</code> will be used.</li>
<li><code>name.ctrl</code>: optional, the name of the control group. If is <code>NULL</code>, the object name of <code>obj.ctrl</code> will be used.</li>
<li><code>ClusterNum</code>: how many cluster will be clustered</li>
<li><code>P.adj</code>: p.adjust cut-off. To avoid slow visualization, you can make stricter p-cut off.</li>
<li><code>force</code>: ignore all auto-self-checks, which is useful</li>
<li><code>nTop</code>: keep n top items according to p-adj in each cluster.</li>
<li><code>method</code>: the agglomeration method to be used. This should be (an unambiguous abbreviation of) one of “ward.D”, “ward.D2”, “single”, “complete”, “average” (= UPGMA), “mcquitty” (= WPGMA), “median” (= WPGMC) or “centroid” (= UPGMC).</li>
<li><code>...</code>: Others options.</li>
</ul>
</section>
<section id="details-3" class="level2">
<h2 class="anchored" data-anchor-id="details-3">Details</h2>
<p>Execute <code>obj.test</code> VS <code>obj.ctrl</code> tests, showing pathway overlaps (or differences) and meta-gene modules of test group and control group. Supports ORA and GSEA results (enriched object or data.frame). !WARNING!: <code>obj.test</code> and <code>obj.ctrl</code> should come from same database (e.g.&nbsp;GO Biological Process(GOBP)).</p>
</section>
<section id="author-1" class="level2">
<h2 class="anchored" data-anchor-id="author-1">Author</h2>
<p>Zhiming Ye</p>
</section>
<section id="value-2" class="level2">
<h2 class="anchored" data-anchor-id="value-2">Value</h2>
<p><code>List</code> containing multiple <code>EnrichGT_obj</code> objects. The <code>List</code> contains objects with overlapped enriched terms, unique enrich terms.</p>
</section>
</section>
<section id="egt_comparison_reactor-framework-for-comparing-and-analyzing-enrichment-results" class="level1">
<h1><code>egt_comparison_reactor</code>: Framework for comparing and analyzing enrichment results</h1>
<p>across multiple groups</p>
<section id="description-7" class="level2">
<h2 class="anchored" data-anchor-id="description-7">Description</h2>
<p>Framework for comparing and analyzing enrichment results across multiple groups</p>
</section>
<section id="usage-23" class="level2">
<h2 class="anchored" data-anchor-id="usage-23">Usage</h2>
<div class="sourceCode" id="cb39"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb39-1"><a href="#cb39-1" aria-hidden="true" tabindex="-1"></a><span class="fu">egt_comparison_reactor</span>(<span class="at">Type =</span> <span class="cn">NULL</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</section>
<section id="details-4" class="level2">
<h2 class="anchored" data-anchor-id="details-4">Details</h2>
<p>Type should be one of “ORA” or “GSEA”. For functions inside reactor, please see below ‘See also’ (in the bottom of this doc)</p>
</section>
<section id="seealso-2" class="level2">
<h2 class="anchored" data-anchor-id="seealso-2">Seealso</h2>
<p><code>[comparison_reactor_base](comparison_reactor_base)</code> for the base class documentation <code>[comparison_reactor_ora](comparison_reactor_ora)</code> for ORA-specific functionality <code>[comparison_reactor_gsea](comparison_reactor_gsea)</code> for GSEA-specific functionality</p>
</section>
<section id="examples-14" class="level2">
<h2 class="anchored" data-anchor-id="examples-14">Examples</h2>
<div class="sourceCode" id="cb40"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb40-1"><a href="#cb40-1" aria-hidden="true" tabindex="-1"></a><span class="co"># ORA example</span></span>
<span id="cb40-2"><a href="#cb40-2" aria-hidden="true" tabindex="-1"></a>reactor <span class="ot">&lt;-</span> <span class="fu">egt_comparison_reactor</span>(<span class="st">"ORA"</span>)</span>
<span id="cb40-3"><a href="#cb40-3" aria-hidden="true" tabindex="-1"></a>reactor<span class="sc">$</span><span class="fu">append_enriched_result</span>(ora_result1, <span class="st">"group_1_go"</span>)</span>
<span id="cb40-4"><a href="#cb40-4" aria-hidden="true" tabindex="-1"></a>reactor<span class="sc">$</span><span class="fu">append_enriched_result</span>(ora_result3, <span class="st">"group_3_go"</span>)</span>
<span id="cb40-5"><a href="#cb40-5" aria-hidden="true" tabindex="-1"></a>reactor<span class="sc">$</span><span class="fu">prefilter_by_p_adj</span>(<span class="fl">0.05</span>)</span>
<span id="cb40-6"><a href="#cb40-6" aria-hidden="true" tabindex="-1"></a>reactor<span class="sc">$</span><span class="fu">make_plans</span>(<span class="at">group =</span> <span class="fu">c</span>(<span class="st">"group_1_go"</span>,<span class="st">"group_3_go"</span>), <span class="at">use_value =</span> <span class="st">"p"</span>)</span>
<span id="cb40-7"><a href="#cb40-7" aria-hidden="true" tabindex="-1"></a>reactor<span class="sc">$</span><span class="fu">find_relationship</span>(<span class="at">Num =</span> <span class="dv">3</span>)</span>
<span id="cb40-8"><a href="#cb40-8" aria-hidden="true" tabindex="-1"></a>wordcloudFigure <span class="ot">&lt;-</span> reactor<span class="sc">$</span><span class="fu">fetch_biological_theme</span>()</span>
<span id="cb40-9"><a href="#cb40-9" aria-hidden="true" tabindex="-1"></a>wordcloudFigure[[<span class="dv">1</span>]]</span>
<span id="cb40-10"><a href="#cb40-10" aria-hidden="true" tabindex="-1"></a>reactor<span class="sc">$</span><span class="fu">split_by_cluster</span>()</span>
<span id="cb40-11"><a href="#cb40-11" aria-hidden="true" tabindex="-1"></a>reactor<span class="sc">$</span><span class="fu">do_recluster</span>(<span class="at">ClusterNum =</span> <span class="dv">10</span>)</span>
<span id="cb40-12"><a href="#cb40-12" aria-hidden="true" tabindex="-1"></a>cls_res <span class="ot">&lt;-</span> reactor<span class="sc">$</span><span class="fu">get_recluster_result</span>()</span>
<span id="cb40-13"><a href="#cb40-13" aria-hidden="true" tabindex="-1"></a>relation_df <span class="ot">&lt;-</span> reactor<span class="sc">$</span><span class="fu">fetch_relationship</span>()</span>
<span id="cb40-14"><a href="#cb40-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-15"><a href="#cb40-15" aria-hidden="true" tabindex="-1"></a><span class="co"># GSEA can do more with:</span></span>
<span id="cb40-16"><a href="#cb40-16" aria-hidden="true" tabindex="-1"></a>gsea_reactor <span class="ot">&lt;-</span> <span class="fu">egt_comparison_reactor</span>(<span class="st">"GSEA"</span>)</span>
<span id="cb40-17"><a href="#cb40-17" aria-hidden="true" tabindex="-1"></a>gsea_reactor<span class="sc">$</span><span class="fu">prefilter_by_NES</span>(<span class="fl">1.5</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</section>
</section>
<section id="egt_enrichment_analysis-perform-over-representation-analysis-ora" class="level1">
<h1><code>egt_enrichment_analysis</code>: Perform Over-Representation Analysis (ORA)</h1>
<section id="description-8" class="level2">
<h2 class="anchored" data-anchor-id="description-8">Description</h2>
<p>ORA compares the proportion of genes in your target list that belong to specific categories (pathways, GO terms etc.) against the expected proportion in a background set. This implementation uses hash tables for efficient gene counting and supports parallel processing for analyzing multiple gene lists.</p>
</section>
<section id="usage-24" class="level2">
<h2 class="anchored" data-anchor-id="usage-24">Usage</h2>
<div class="sourceCode" id="cb41"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb41-1"><a href="#cb41-1" aria-hidden="true" tabindex="-1"></a><span class="fu">egt_enrichment_analysis</span>(</span>
<span id="cb41-2"><a href="#cb41-2" aria-hidden="true" tabindex="-1"></a>  genes,</span>
<span id="cb41-3"><a href="#cb41-3" aria-hidden="true" tabindex="-1"></a>  database,</span>
<span id="cb41-4"><a href="#cb41-4" aria-hidden="true" tabindex="-1"></a>  <span class="at">p_adj_methods =</span> <span class="st">"BH"</span>,</span>
<span id="cb41-5"><a href="#cb41-5" aria-hidden="true" tabindex="-1"></a>  <span class="at">p_val_cut_off =</span> <span class="fl">0.5</span>,</span>
<span id="cb41-6"><a href="#cb41-6" aria-hidden="true" tabindex="-1"></a>  <span class="at">background_genes =</span> <span class="cn">NULL</span>,</span>
<span id="cb41-7"><a href="#cb41-7" aria-hidden="true" tabindex="-1"></a>  <span class="at">min_geneset_size =</span> <span class="dv">10</span>,</span>
<span id="cb41-8"><a href="#cb41-8" aria-hidden="true" tabindex="-1"></a>  <span class="at">max_geneset_size =</span> <span class="dv">500</span>,</span>
<span id="cb41-9"><a href="#cb41-9" aria-hidden="true" tabindex="-1"></a>  <span class="at">multi_cores =</span> <span class="dv">0</span></span>
<span id="cb41-10"><a href="#cb41-10" aria-hidden="true" tabindex="-1"></a>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</section>
<section id="arguments-16" class="level2">
<h2 class="anchored" data-anchor-id="arguments-16">Arguments</h2>
<ul>
<li><code>genes</code>: Input genes, either:</li>
<li>Character vector of gene IDs (e.g., <code>c("TP53","BRCA1")</code>)</li>
<li>Named numeric vector from <code>genes_with_weights()</code> (will split by expression direction)</li>
<li>List of gene vectors for multiple comparisons (e.g., by cell type)</li>
<li><code>database</code>: Gene set database, either:</li>
<li>Built-in database from <code>database_GO_BP()</code>, <code>database_KEGG()</code> etc.</li>
<li>Custom data frame with columns: (ID, Pathway_Name, Genes) or (Pathway_Name, Genes)</li>
<li>GMT file loaded via <code>database_from_gmt()</code></li>
<li><code>p_val_cut_off</code>: Adjusted p-value cutoff (default 0.5)</li>
<li><code>background_genes</code>: Custom background genes (default: all genes in database)</li>
<li><code>min_geneset_size</code>: Minimum genes per set (default 10)</li>
<li><code>max_geneset_size</code>: Maximum genes per set (default 500)</li>
<li><code>multi_cores</code>: (Please don’t use this since it has several known bugs) Number of cores for parallel processing (default 0 = serial)</li>
<li><code>p_adj_method</code>: Multiple testing correction method (default “BH” for Benjamini-Hochberg)</li>
</ul>
</section>
<section id="details-5" class="level2">
<h2 class="anchored" data-anchor-id="details-5">Details</h2>
<p>Identifies enriched biological pathways or gene sets in a gene list using high-performance C++ implementation with parallel processing support.</p>
</section>
<section id="author-2" class="level2">
<h2 class="anchored" data-anchor-id="author-2">Author</h2>
<p>Zhiming Ye</p>
</section>
<section id="value-3" class="level2">
<h2 class="anchored" data-anchor-id="value-3">Value</h2>
<p>A data frame with columns:</p>
<ul>
<li>ID: Gene set identifier</li>
<li>Description: Gene set name</li>
<li>GeneRatio: Enriched genes / input genes</li>
<li>BgRatio: Set genes / background genes</li>
<li>pvalue: Raw p-value</li>
<li>p.adjust: Adjusted p-value</li>
<li>geneID: Enriched genes</li>
<li>Count: Number of enriched genes</li>
</ul>
<p>For weighted input, additional columns show up/down regulated genes.</p>
</section>
<section id="examples-15" class="level2">
<h2 class="anchored" data-anchor-id="examples-15">Examples</h2>
<div class="sourceCode" id="cb42"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb42-1"><a href="#cb42-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Basic ORA with GO Biological Processes</span></span>
<span id="cb42-2"><a href="#cb42-2" aria-hidden="true" tabindex="-1"></a>genes <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"TP53"</span>, <span class="st">"BRCA1"</span>, <span class="st">"EGFR"</span>, <span class="st">"CDK2"</span>)</span>
<span id="cb42-3"><a href="#cb42-3" aria-hidden="true" tabindex="-1"></a>res <span class="ot">&lt;-</span> <span class="fu">egt_enrichment_analysis</span>(genes, <span class="fu">database_GO_BP</span>())</span>
<span id="cb42-4"><a href="#cb42-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb42-5"><a href="#cb42-5" aria-hidden="true" tabindex="-1"></a><span class="co"># ORA with DEG results (split by direction)</span></span>
<span id="cb42-6"><a href="#cb42-6" aria-hidden="true" tabindex="-1"></a>deg_genes <span class="ot">&lt;-</span> <span class="fu">genes_with_weights</span>(DEG<span class="sc">$</span>gene, DEG<span class="sc">$</span>log2FC)</span>
<span id="cb42-7"><a href="#cb42-7" aria-hidden="true" tabindex="-1"></a>res <span class="ot">&lt;-</span> <span class="fu">egt_enrichment_analysis</span>(deg_genes, <span class="fu">database_KEGG</span>())</span>
<span id="cb42-8"><a href="#cb42-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb42-9"><a href="#cb42-9" aria-hidden="true" tabindex="-1"></a><span class="co"># Multi-group ORA with parallel processing</span></span>
<span id="cb42-10"><a href="#cb42-10" aria-hidden="true" tabindex="-1"></a>gene_lists <span class="ot">&lt;-</span> <span class="fu">list</span>(</span>
<span id="cb42-11"><a href="#cb42-11" aria-hidden="true" tabindex="-1"></a>  <span class="at">Macrophages =</span> <span class="fu">c</span>(<span class="st">"CD68"</span>, <span class="st">"CD163"</span>, <span class="st">"CD169"</span>),</span>
<span id="cb42-12"><a href="#cb42-12" aria-hidden="true" tabindex="-1"></a>  <span class="at">Fibroblasts =</span> <span class="fu">c</span>(<span class="st">"COL1A1"</span>, <span class="st">"COL1A2"</span>, <span class="st">"ACTA2"</span>)</span>
<span id="cb42-13"><a href="#cb42-13" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb42-14"><a href="#cb42-14" aria-hidden="true" tabindex="-1"></a>res <span class="ot">&lt;-</span> <span class="fu">egt_enrichment_analysis</span>(gene_lists, <span class="fu">database_Reactome</span>(), <span class="at">multi_cores=</span><span class="dv">0</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</section>
</section>
<section id="egt_fetch_biological_theme-generate-biological-theme-wordcloud-from-enrichment-results" class="level1">
<h1><code>egt_fetch_biological_theme</code>: Generate Biological Theme Wordcloud from Enrichment Results</h1>
<section id="description-9" class="level2">
<h2 class="anchored" data-anchor-id="description-9">Description</h2>
<p>Creates a wordcloud visualization from enrichment results, either from a data.frame or an EnrichGT_obj. For data.frames, filters by p.adjust &lt; 0.05 by default.</p>
</section>
<section id="usage-25" class="level2">
<h2 class="anchored" data-anchor-id="usage-25">Usage</h2>
<div class="sourceCode" id="cb43"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb43-1"><a href="#cb43-1" aria-hidden="true" tabindex="-1"></a><span class="fu">egt_fetch_biological_theme</span>(x, <span class="at">cluster =</span> <span class="cn">NULL</span>, <span class="at">skip_filtering =</span> F, ...)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</section>
<section id="arguments-17" class="level2">
<h2 class="anchored" data-anchor-id="arguments-17">Arguments</h2>
<ul>
<li><code>x</code>: Input object - either a data.frame with enrichment results or an EnrichGT_obj</li>
<li><code>cluster</code>: Cluster name/number (required if x is EnrichGT_obj)</li>
<li><code>skip_filtering</code>: Logical, if TRUE skips filtering of data.frame by p.adjust</li>
<li><code>...</code>: Additional arguments passed to wordcloud_generator2</li>
</ul>
</section>
<section id="author-3" class="level2">
<h2 class="anchored" data-anchor-id="author-3">Author</h2>
<p>Zhiming Ye</p>
</section>
<section id="value-4" class="level2">
<h2 class="anchored" data-anchor-id="value-4">Value</h2>
<p>ggwordcloud object</p>
</section>
</section>
<section id="egt_fetch_termwise_relationship-fetch-and-visualize-termwise-relationships-in-enrichment-analysis-results" class="level1">
<h1><code>egt_fetch_termwise_relationship</code>: Fetch and visualize termwise relationships in enrichment analysis results</h1>
<section id="description-10" class="level2">
<h2 class="anchored" data-anchor-id="description-10">Description</h2>
<p>This function analyzes the relationships between enriched terms and creates a hierarchical clustering visualization. Please avoiding querying too large dataset.</p>
</section>
<section id="usage-26" class="level2">
<h2 class="anchored" data-anchor-id="usage-26">Usage</h2>
<div class="sourceCode" id="cb44"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb44-1"><a href="#cb44-1" aria-hidden="true" tabindex="-1"></a><span class="fu">egt_fetch_termwise_relationship</span>(</span>
<span id="cb44-2"><a href="#cb44-2" aria-hidden="true" tabindex="-1"></a>  x,</span>
<span id="cb44-3"><a href="#cb44-3" aria-hidden="true" tabindex="-1"></a>  database,</span>
<span id="cb44-4"><a href="#cb44-4" aria-hidden="true" tabindex="-1"></a>  <span class="at">according_to =</span> <span class="st">"Description"</span>,</span>
<span id="cb44-5"><a href="#cb44-5" aria-hidden="true" tabindex="-1"></a>  <span class="at">ClusterNum =</span> <span class="dv">4</span>,</span>
<span id="cb44-6"><a href="#cb44-6" aria-hidden="true" tabindex="-1"></a>  <span class="at">method =</span> <span class="st">"ward.D2"</span>,</span>
<span id="cb44-7"><a href="#cb44-7" aria-hidden="true" tabindex="-1"></a>  <span class="at">force =</span> F,</span>
<span id="cb44-8"><a href="#cb44-8" aria-hidden="true" tabindex="-1"></a>  <span class="at">maxLength =</span> <span class="dv">40</span></span>
<span id="cb44-9"><a href="#cb44-9" aria-hidden="true" tabindex="-1"></a>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</section>
<section id="arguments-18" class="level2">
<h2 class="anchored" data-anchor-id="arguments-18">Arguments</h2>
<ul>
<li><code>x</code>: A character vector of terms to analyze</li>
<li><code>database</code>: Query database. For example, <code>database_GO_BP(org.Hs.eg.db)</code>. If you are quering fused result, you can input a list like <code>list(database_GO_BP(org.Hs.eg.db), database_Reactome(org.Hs.eg.db))</code>.</li>
<li><code>according_to</code>: Character string specifying which column to use for filtering, default is “Description”</li>
<li><code>ClusterNum</code>: Integer specifying the number of clusters, default is 4</li>
<li><code>method</code>: Character string specifying the hierarchical clustering method, default is “ward.D2”</li>
<li><code>force</code>: Avoid checking the length of x. Large query will cause long calculation time and poor results.</li>
<li><code>maxLength</code>: Warp length max than what terms</li>
</ul>
</section>
<section id="value-5" class="level2">
<h2 class="anchored" data-anchor-id="value-5">Value</h2>
<p>A list containing:</p>
</section>
<section id="examples-16" class="level2">
<h2 class="anchored" data-anchor-id="examples-16">Examples</h2>
<div class="sourceCode" id="cb45"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb45-1"><a href="#cb45-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Example usage:</span></span>
<span id="cb45-2"><a href="#cb45-2" aria-hidden="true" tabindex="-1"></a>result <span class="ot">&lt;-</span> <span class="fu">egt_fetch_termwise_relationship</span>(</span>
<span id="cb45-3"><a href="#cb45-3" aria-hidden="true" tabindex="-1"></a>  <span class="at">x =</span> <span class="fu">c</span>(<span class="st">"pathway1"</span>, <span class="st">"pathway2"</span>),</span>
<span id="cb45-4"><a href="#cb45-4" aria-hidden="true" tabindex="-1"></a>  <span class="at">database =</span> pathway_db,</span>
<span id="cb45-5"><a href="#cb45-5" aria-hidden="true" tabindex="-1"></a>  <span class="at">according_to =</span> <span class="st">"Description"</span></span>
<span id="cb45-6"><a href="#cb45-6" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb45-7"><a href="#cb45-7" aria-hidden="true" tabindex="-1"></a>result<span class="sc">$</span>Figure</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</section>
</section>
<section id="egt_generate_quarto_report-export-quarto-report" class="level1">
<h1><code>egt_generate_quarto_report</code>: Export Quarto Report</h1>
<section id="description-11" class="level2">
<h2 class="anchored" data-anchor-id="description-11">Description</h2>
<p>Export Quarto Report</p>
</section>
<section id="usage-27" class="level2">
<h2 class="anchored" data-anchor-id="usage-27">Usage</h2>
<div class="sourceCode" id="cb46"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb46-1"><a href="#cb46-1" aria-hidden="true" tabindex="-1"></a><span class="fu">egt_generate_quarto_report</span>(</span>
<span id="cb46-2"><a href="#cb46-2" aria-hidden="true" tabindex="-1"></a>  re_enrichment_results,</span>
<span id="cb46-3"><a href="#cb46-3" aria-hidden="true" tabindex="-1"></a>  <span class="at">output_path =</span> <span class="st">"Report.qmd"</span>,</span>
<span id="cb46-4"><a href="#cb46-4" aria-hidden="true" tabindex="-1"></a>  <span class="at">type =</span> <span class="st">"html"</span></span>
<span id="cb46-5"><a href="#cb46-5" aria-hidden="true" tabindex="-1"></a>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</section>
<section id="arguments-19" class="level2">
<h2 class="anchored" data-anchor-id="arguments-19">Arguments</h2>
<ul>
<li><code>re_enrichment_results</code>: The <code>EnrichGT_obj</code>, AI summarized result is more recommanded.</li>
<li><code>output_path</code>: Path of the output qmd file (e.g., <code>test.qmd</code>)</li>
<li><code>type</code>: export pdf or html. In default output is HTML format.</li>
</ul>
</section>
<section id="value-6" class="level2">
<h2 class="anchored" data-anchor-id="value-6">Value</h2>
<p>A quarto document</p>
</section>
</section>
<section id="egt_gsea_analysis-perform-gene-set-enrichment-analysis-gsea" class="level1">
<h1><code>egt_gsea_analysis</code>: Perform Gene Set Enrichment Analysis (GSEA)</h1>
<section id="description-12" class="level2">
<h2 class="anchored" data-anchor-id="description-12">Description</h2>
<p>GSEA analyzes whether predefined gene sets show statistically significant enrichment at the top or bottom of a ranked gene list. This implementation uses the fast fgsea algorithm from the fgsea package.</p>
</section>
<section id="usage-28" class="level2">
<h2 class="anchored" data-anchor-id="usage-28">Usage</h2>
<div class="sourceCode" id="cb47"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb47-1"><a href="#cb47-1" aria-hidden="true" tabindex="-1"></a><span class="fu">egt_gsea_analysis</span>(</span>
<span id="cb47-2"><a href="#cb47-2" aria-hidden="true" tabindex="-1"></a>  genes,</span>
<span id="cb47-3"><a href="#cb47-3" aria-hidden="true" tabindex="-1"></a>  database,</span>
<span id="cb47-4"><a href="#cb47-4" aria-hidden="true" tabindex="-1"></a>  <span class="at">p_val_cut_off =</span> <span class="fl">0.5</span>,</span>
<span id="cb47-5"><a href="#cb47-5" aria-hidden="true" tabindex="-1"></a>  <span class="at">min_geneset_size =</span> <span class="dv">10</span>,</span>
<span id="cb47-6"><a href="#cb47-6" aria-hidden="true" tabindex="-1"></a>  <span class="at">max_geneset_size =</span> <span class="dv">500</span>,</span>
<span id="cb47-7"><a href="#cb47-7" aria-hidden="true" tabindex="-1"></a>  <span class="at">gseaParam =</span> <span class="dv">1</span></span>
<span id="cb47-8"><a href="#cb47-8" aria-hidden="true" tabindex="-1"></a>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</section>
<section id="arguments-20" class="level2">
<h2 class="anchored" data-anchor-id="arguments-20">Arguments</h2>
<ul>
<li><code>genes</code>: A named numeric vector of ranked genes where:</li>
<li>Names are gene identifiers</li>
<li>Values are ranking metric (e.g., log2 fold change, PCA loading)Must be sorted in descending order (use <code>genes_with_weights()</code> to prepare)</li>
<li><code>database</code>: Gene set database, either:</li>
<li>Built-in database from <code>database_GO_BP()</code>, <code>database_KEGG()</code> etc.</li>
<li>Custom data frame with columns: (ID, Pathway_Name, Genes) or (Pathway_Name, Genes)</li>
<li>GMT file loaded via <code>database_from_gmt()</code></li>
<li><code>p_val_cut_off</code>: Adjusted p-value cutoff (default 0.5)</li>
<li><code>min_geneset_size</code>: Minimum genes per set (default 10)</li>
<li><code>max_geneset_size</code>: Maximum genes per set (default 500)</li>
<li><code>gseaParam</code>: GSEA parameter controlling weight of ranking (default 1)</li>
<li><code>p_adj_method</code>: Multiple testing correction method (default “BH” for Benjamini-Hochberg)</li>
</ul>
</section>
<section id="details-6" class="level2">
<h2 class="anchored" data-anchor-id="details-6">Details</h2>
<p>Identifies enriched biological pathways in a ranked gene list using the fgsea algorithm.</p>
</section>
<section id="author-4" class="level2">
<h2 class="anchored" data-anchor-id="author-4">Author</h2>
<p>Based on fgsea package by Alexey Sergushichev et al.</p>
</section>
<section id="value-7" class="level2">
<h2 class="anchored" data-anchor-id="value-7">Value</h2>
<p>A data frame with columns:</p>
<ul>
<li>ID: Gene set identifier</li>
<li>Description: Gene set name</li>
<li>ES: Enrichment score</li>
<li>NES: Normalized enrichment score</li>
<li>pvalue: Raw p-value</li>
<li>p.adjust: Adjusted p-value</li>
<li>core_enrichment: Leading edge genes</li>
</ul>
</section>
<section id="examples-17" class="level2">
<h2 class="anchored" data-anchor-id="examples-17">Examples</h2>
<div class="sourceCode" id="cb48"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb48-1"><a href="#cb48-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Using differential expression results</span></span>
<span id="cb48-2"><a href="#cb48-2" aria-hidden="true" tabindex="-1"></a>ranked_genes <span class="ot">&lt;-</span> <span class="fu">genes_with_weights</span>(DEG<span class="sc">$</span>gene, DEG<span class="sc">$</span>log2FC)</span>
<span id="cb48-3"><a href="#cb48-3" aria-hidden="true" tabindex="-1"></a>res <span class="ot">&lt;-</span> <span class="fu">egt_gsea_analysis</span>(ranked_genes, <span class="fu">database_GO_BP</span>())</span>
<span id="cb48-4"><a href="#cb48-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb48-5"><a href="#cb48-5" aria-hidden="true" tabindex="-1"></a><span class="co"># Using PCA loadings</span></span>
<span id="cb48-6"><a href="#cb48-6" aria-hidden="true" tabindex="-1"></a>ranked_genes <span class="ot">&lt;-</span> <span class="fu">genes_with_weights</span>(<span class="fu">rownames</span>(pca<span class="sc">$</span>rotation), pca<span class="sc">$</span>rotation[,<span class="dv">1</span>])</span>
<span id="cb48-7"><a href="#cb48-7" aria-hidden="true" tabindex="-1"></a>res <span class="ot">&lt;-</span> <span class="fu">egt_gsea_analysis</span>(ranked_genes, <span class="fu">database_KEGG</span>())</span>
<span id="cb48-8"><a href="#cb48-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb48-9"><a href="#cb48-9" aria-hidden="true" tabindex="-1"></a><span class="co"># Custom gene sets from GMT file</span></span>
<span id="cb48-10"><a href="#cb48-10" aria-hidden="true" tabindex="-1"></a>ranked_genes <span class="ot">&lt;-</span> <span class="fu">genes_with_weights</span>(genes, weights)</span>
<span id="cb48-11"><a href="#cb48-11" aria-hidden="true" tabindex="-1"></a>res <span class="ot">&lt;-</span> <span class="fu">egt_gsea_analysis</span>(ranked_genes, <span class="fu">database_from_gmt</span>(<span class="st">"pathways.gmt"</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</section>
</section>
<section id="egt_infer_act-infering-pathway-or-transcript-factors-activity-from-enrichgt-meta-gene-modules" class="level1">
<h1><code>egt_infer_act</code>: Infering Pathway or Transcript Factors activity from EnrichGT meta-gene modules</h1>
<section id="description-13" class="level2">
<h2 class="anchored" data-anchor-id="description-13">Description</h2>
<p>Only supports gene symbols. <a href="https://saezlab.github.io/progeny/">PROGENy</a> is a comprehensive resource containing a curated collection of pathways and their target genes, with weights for each interaction. <a href="https://github.com/saezlab/CollecTRI">CollecTRI</a> is a comprehensive resource containing a curated collection of TFs and their transcriptional targets compiled from 12 different resources. This collection provides an increased coverage of transcription factors and a superior performance in identifying perturbed TFs compared to our previous. If when doing re-enrichment, you select a high number of clusters, that may cause low gene number in each meta-gene module, and then can’t be infered sucessfully. So if result is empty, please increase the number of re-clustering when doing it.</p>
</section>
<section id="usage-29" class="level2">
<h2 class="anchored" data-anchor-id="usage-29">Usage</h2>
<div class="sourceCode" id="cb49"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb49-1"><a href="#cb49-1" aria-hidden="true" tabindex="-1"></a><span class="fu">egt_infer_act</span>(x, <span class="at">DB =</span> <span class="st">"collectri"</span>, <span class="at">species =</span> <span class="st">"human"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</section>
<section id="arguments-21" class="level2">
<h2 class="anchored" data-anchor-id="arguments-21">Arguments</h2>
<ul>
<li><code>x</code>: an EnrichGT_obj object.</li>
<li><code>DB</code>: can be “progeny” (the Pathway activity database), or “collectri” (TF activity database)</li>
<li><code>species</code>: can be “human” or “mouse”</li>
</ul>
</section>
<section id="author-5" class="level2">
<h2 class="anchored" data-anchor-id="author-5">Author</h2>
<p>Zhiming Ye, Saez-Rodriguez Lab (The decoupleR package, https://saezlab.github.io/decoupleR/)</p>
</section>
<section id="value-8" class="level2">
<h2 class="anchored" data-anchor-id="value-8">Value</h2>
<p>an ORA result list</p>
</section>
</section>
<section id="egt_llm_multi_summary-compare-multiple-llm-summaries" class="level1">
<h1><code>egt_llm_multi_summary</code>: Compare Multiple LLM Summaries</h1>
<section id="description-14" class="level2">
<h2 class="anchored" data-anchor-id="description-14">Description</h2>
<p>Generate summaries using multiple LLM models and create a comparison object.</p>
</section>
<section id="usage-30" class="level2">
<h2 class="anchored" data-anchor-id="usage-30">Usage</h2>
<div class="sourceCode" id="cb50"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb50-1"><a href="#cb50-1" aria-hidden="true" tabindex="-1"></a><span class="fu">egt_llm_multi_summary</span>(</span>
<span id="cb50-2"><a href="#cb50-2" aria-hidden="true" tabindex="-1"></a>  x,</span>
<span id="cb50-3"><a href="#cb50-3" aria-hidden="true" tabindex="-1"></a>  chat_list,</span>
<span id="cb50-4"><a href="#cb50-4" aria-hidden="true" tabindex="-1"></a>  <span class="at">lang =</span> <span class="st">"English"</span>,</span>
<span id="cb50-5"><a href="#cb50-5" aria-hidden="true" tabindex="-1"></a>  <span class="at">background_knowledges =</span> <span class="cn">NULL</span>,</span>
<span id="cb50-6"><a href="#cb50-6" aria-hidden="true" tabindex="-1"></a>  <span class="at">comparison_prompt =</span> <span class="cn">NULL</span></span>
<span id="cb50-7"><a href="#cb50-7" aria-hidden="true" tabindex="-1"></a>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</section>
<section id="arguments-22" class="level2">
<h2 class="anchored" data-anchor-id="arguments-22">Arguments</h2>
<ul>
<li><code>x</code>: An EnrichGT_obj object from egt_recluster_analysis()</li>
<li><code>chat_list</code>: A named list of LLM chat objects</li>
<li><code>lang</code>: Language for summaries (“English” or “Chinese”)</li>
<li><code>background_knowledges</code>: Additional Reference (e.g, Papers or guide) for LLM for reference</li>
<li><code>comparison_prompt</code>: Custom prompt for generating comparison summary</li>
</ul>
</section>
<section id="value-9" class="level2">
<h2 class="anchored" data-anchor-id="value-9">Value</h2>
<p>EnrichGT_obj with LLM_Comparison slot filled</p>
</section>
</section>
<section id="egt_llm_summary-summarize-enrichgt-results-using-llm" class="level1">
<h1><code>egt_llm_summary</code>: Summarize EnrichGT results using LLM</h1>
<section id="description-15" class="level2">
<h2 class="anchored" data-anchor-id="description-15">Description</h2>
<p>This function uses a Large Language Model (LLM) to generate summaries for pathway clusters and gene modules in an EnrichGT_obj object.</p>
</section>
<section id="usage-31" class="level2">
<h2 class="anchored" data-anchor-id="usage-31">Usage</h2>
<div class="sourceCode" id="cb51"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb51-1"><a href="#cb51-1" aria-hidden="true" tabindex="-1"></a><span class="fu">egt_llm_summary</span>(</span>
<span id="cb51-2"><a href="#cb51-2" aria-hidden="true" tabindex="-1"></a>  x,</span>
<span id="cb51-3"><a href="#cb51-3" aria-hidden="true" tabindex="-1"></a>  chat,</span>
<span id="cb51-4"><a href="#cb51-4" aria-hidden="true" tabindex="-1"></a>  <span class="at">lang =</span> <span class="st">"English"</span>,</span>
<span id="cb51-5"><a href="#cb51-5" aria-hidden="true" tabindex="-1"></a>  <span class="at">model_name =</span> <span class="cn">NULL</span>,</span>
<span id="cb51-6"><a href="#cb51-6" aria-hidden="true" tabindex="-1"></a>  <span class="at">background_knowledges =</span> <span class="cn">NULL</span></span>
<span id="cb51-7"><a href="#cb51-7" aria-hidden="true" tabindex="-1"></a>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</section>
<section id="arguments-23" class="level2">
<h2 class="anchored" data-anchor-id="arguments-23">Arguments</h2>
<ul>
<li><code>x</code>: An EnrichGT_obj object created by <code>[egt_recluster_analysis](egt_recluster_analysis)</code>.</li>
<li><code>chat</code>: An LLM chat object created by the <code>ellmer</code> package.</li>
<li><code>lang</code>: Language pass to LLM. Can be <code>English</code> or <code>Chinese</code>.</li>
<li><code>background_knowledges</code>: Additional Reference (e.g, Papers or guide) for LLM for reference. A Single Character String should be provided.</li>
</ul>
</section>
<section id="note" class="level2">
<h2 class="anchored" data-anchor-id="note">Note</h2>
<p>It is recommended not to add system prompts when creating the chat object. The function provides its own carefully crafted prompts for biological analysis.</p>
</section>
<section id="references" class="level2">
<h2 class="anchored" data-anchor-id="references">References</h2>
<p>For more information about creating chat objects, see the <a href="https://ellmer.tidyverse.org/index.html">ellmer package documentation</a>.</p>
</section>
<section id="seealso-3" class="level2">
<h2 class="anchored" data-anchor-id="seealso-3">Seealso</h2>
<p><code>[egt_recluster_analysis](egt_recluster_analysis)</code> to create the input object.</p>
</section>
<section id="value-10" class="level2">
<h2 class="anchored" data-anchor-id="value-10">Value</h2>
<p>Returns the input EnrichGT_obj object with added LLM annotations in the <code>LLM_Annotation</code> slot. The annotations include:</p>
<ul>
<li><code>pathways</code>: Summaries of pathway clusters</li>
<li><code>genes_and_title</code>: Summaries of gene modules and their titles</li>
</ul>
</section>
<section id="examples-18" class="level2">
<h2 class="anchored" data-anchor-id="examples-18">Examples</h2>
<div class="sourceCode" id="cb52"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb52-1"><a href="#cb52-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Create LLM chat object</span></span>
<span id="cb52-2"><a href="#cb52-2" aria-hidden="true" tabindex="-1"></a>chat <span class="ot">&lt;-</span> <span class="fu">chat_deepseek</span>(<span class="at">api_key =</span> YOUR_API_KEY, <span class="at">model =</span> <span class="st">"deepseek-chat"</span>, <span class="at">system_prompt =</span> <span class="st">""</span>)</span>
<span id="cb52-3"><a href="#cb52-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb52-4"><a href="#cb52-4" aria-hidden="true" tabindex="-1"></a><span class="co"># Run enrichment analysis and get EnrichGT_obj</span></span>
<span id="cb52-5"><a href="#cb52-5" aria-hidden="true" tabindex="-1"></a>re_enrichment_results <span class="ot">&lt;-</span> <span class="fu">egt_recluster_analysis</span>(...)</span>
<span id="cb52-6"><a href="#cb52-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb52-7"><a href="#cb52-7" aria-hidden="true" tabindex="-1"></a><span class="co"># Get LLM summaries</span></span>
<span id="cb52-8"><a href="#cb52-8" aria-hidden="true" tabindex="-1"></a>re_enrichment_results <span class="ot">&lt;-</span> <span class="fu">egt_llm_summary</span>(re_enrichment_results, chat)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</section>
</section>
<section id="egt_plot_gsea-generate-gsea-enrichment-plots" class="level1">
<h1><code>egt_plot_gsea</code>: Generate GSEA Enrichment Plots</h1>
<section id="description-16" class="level2">
<h2 class="anchored" data-anchor-id="description-16">Description</h2>
<p>This function creates graphical representations of Gene Set Enrichment Analysis (GSEA) results, including either a multi-panel GSEA table plot for multiple pathways or a single pathway enrichment plot. The visualization leverages the <code>fgsea</code> package’s plotting functions.</p>
</section>
<section id="usage-32" class="level2">
<h2 class="anchored" data-anchor-id="usage-32">Usage</h2>
<div class="sourceCode" id="cb53"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb53-1"><a href="#cb53-1" aria-hidden="true" tabindex="-1"></a><span class="fu">egt_plot_gsea</span>(resGSEA<span class="sc">$</span>Description[<span class="dv">1</span>],<span class="at">genes =</span> <span class="fu">genes_with_weights</span>(<span class="at">genes =</span> DEGexample<span class="sc">$</span>...<span class="dv">1</span>, <span class="at">weights =</span> DEGexample<span class="sc">$</span>log2FoldChange),<span class="at">database =</span> <span class="fu">database_GO_BP</span>(org.Hs.eg.db))</span>
<span id="cb53-2"><a href="#cb53-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb53-3"><a href="#cb53-3" aria-hidden="true" tabindex="-1"></a><span class="fu">egt_plot_gsea</span>(resGSEA[<span class="dv">1</span><span class="sc">:</span><span class="dv">8</span>,],<span class="at">genes =</span> <span class="fu">genes_with_weights</span>(<span class="at">genes =</span> DEGexample<span class="sc">$</span>...<span class="dv">1</span>, <span class="at">weights =</span> DEGexample<span class="sc">$</span>log2FoldChange),<span class="at">database =</span> <span class="fu">database_GO_BP</span>(org.Hs.eg.db))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</section>
<section id="arguments-24" class="level2">
<h2 class="anchored" data-anchor-id="arguments-24">Arguments</h2>
<ul>
<li><code>x</code>: A GSEA result object. Can be either:</li>
<li>A data frame containing GSEA results (requires columns: pvalue, p.adjust, Description)</li>
<li>A character string specifying a single pathway name</li>
<li><code>genes</code>: A named numeric vector from <code>genes_with_weights()</code>. These should match the gene identifiers used in the GSEA analysis.</li>
<li><code>database</code>: A database <code>data.frame</code>, You can obtain it from <code>database_xxx()</code> functions. This should correspond to the database used in the original GSEA analysis.</li>
</ul>
</section>
<section id="author-6" class="level2">
<h2 class="anchored" data-anchor-id="author-6">Author</h2>
<p>Zhiming Ye, warpped from <code>fgsea</code></p>
</section>
<section id="value-11" class="level2">
<h2 class="anchored" data-anchor-id="value-11">Value</h2>
<p>A ggplot object:</p>
<ul>
<li>When <code>x</code> is a data frame: Returns a multi-panel plot showing normalized enrichment scores (NES), p-values, and leading edge plots for top pathways</li>
<li>When <code>x</code> is a pathway name: Returns an enrichment plot showing the running enrichment score for the specified pathway</li>
</ul>
</section>
</section>
<section id="egt_plot_results-visualize-enrichment-results-using-simple-plot" class="level1">
<h1><code>egt_plot_results</code>: Visualize enrichment results using simple plot</h1>
<section id="description-17" class="level2">
<h2 class="anchored" data-anchor-id="description-17">Description</h2>
<p>This plot is the most widely like <code>enrichplot::dotplot()</code>used method to visualize enriched terms. It shows the enrichment scores (e.g.&nbsp;p values) and gene ratio or NES as dot size and color / or bar height. Users can specify the number of terms using <code>ntop</code> or selected terms to color via the <code>low.col</code> and <code>hi.col</code>.</p>
</section>
<section id="usage-33" class="level2">
<h2 class="anchored" data-anchor-id="usage-33">Usage</h2>
<div class="sourceCode" id="cb54"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb54-1"><a href="#cb54-1" aria-hidden="true" tabindex="-1"></a><span class="fu">egt_plot_results</span>(</span>
<span id="cb54-2"><a href="#cb54-2" aria-hidden="true" tabindex="-1"></a>  x,</span>
<span id="cb54-3"><a href="#cb54-3" aria-hidden="true" tabindex="-1"></a>  <span class="at">ntop =</span> <span class="cn">NULL</span>,</span>
<span id="cb54-4"><a href="#cb54-4" aria-hidden="true" tabindex="-1"></a>  <span class="at">showIDs =</span> F,</span>
<span id="cb54-5"><a href="#cb54-5" aria-hidden="true" tabindex="-1"></a>  <span class="at">max_len_descript =</span> <span class="dv">40</span>,</span>
<span id="cb54-6"><a href="#cb54-6" aria-hidden="true" tabindex="-1"></a>  <span class="at">keepAll =</span> F,</span>
<span id="cb54-7"><a href="#cb54-7" aria-hidden="true" tabindex="-1"></a>  <span class="at">maskNoise =</span> <span class="dv">3</span>,</span>
<span id="cb54-8"><a href="#cb54-8" aria-hidden="true" tabindex="-1"></a>  ...,</span>
<span id="cb54-9"><a href="#cb54-9" aria-hidden="true" tabindex="-1"></a>  <span class="at">P.adj =</span> <span class="cn">NULL</span></span>
<span id="cb54-10"><a href="#cb54-10" aria-hidden="true" tabindex="-1"></a>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</section>
<section id="arguments-25" class="level2">
<h2 class="anchored" data-anchor-id="arguments-25">Arguments</h2>
<ul>
<li><code>x</code>: a data frame form enriched result like <code>egt_enrichment_analysis()</code> or <code>egt_gsea_analysis()</code>, or an re-clustered <code>EnrichGT</code> object</li>
<li><code>ntop</code>: Show top N in each cluster. In default, for origin enriched result, showing top 15; for re-clustered object, showing top 5 in each cluster.</li>
<li><code>showIDs</code>: bool, show pathway IDs or not. Default is FALSE</li>
<li><code>max_len_descript</code>: the label format length, default as 40.</li>
<li><code>keepAll</code>: Do filtering to avoid overlap of same genes or not</li>
<li><code>maskNoise</code>: (Only works with re-enriched object) Cut-off value to mask rare population in cluster tree. Less than its value in a specifc child tree will be ignore (because it may hit only by coincidence). Set maskNoise = 0 to ignore this.</li>
<li><code>...</code>: Other param</li>
<li><code>P.adj</code>: (Only works with origin data.frame) If pass an origin data.frame from original enriched result, you can specify the P-adjust value cut off. If is null, default is 0.05. When passing <code>EnrichGT_obj</code>, this filter is previously done by <code>egt_recluster_analysis</code>.</li>
<li><code>low.col</code>: the color for the lowest</li>
<li><code>hi.col</code>: the color for the highest</li>
</ul>
</section>
<section id="author-7" class="level2">
<h2 class="anchored" data-anchor-id="author-7">Author</h2>
<p>Zhiming Ye</p>
</section>
<section id="value-12" class="level2">
<h2 class="anchored" data-anchor-id="value-12">Value</h2>
<p>a ggplot2 object</p>
</section>
</section>
<section id="egt_recluster_analysis-cluster-and-re-enrichment-enrichment-results" class="level1">
<h1><code>egt_recluster_analysis</code>: Cluster and re-enrichment enrichment results</h1>
<section id="description-18" class="level2">
<h2 class="anchored" data-anchor-id="description-18">Description</h2>
<p>Performs hierarchical clustering on enrichment results (ORA or GSEA) based on gene-term associations to reduce redundancy and improve biological interpretation. The function helps identify coherent groups of related terms while preserving important but less significant findings.</p>
</section>
<section id="usage-34" class="level2">
<h2 class="anchored" data-anchor-id="usage-34">Usage</h2>
<div class="sourceCode" id="cb55"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb55-1"><a href="#cb55-1" aria-hidden="true" tabindex="-1"></a><span class="fu">egt_recluster_analysis</span>(</span>
<span id="cb55-2"><a href="#cb55-2" aria-hidden="true" tabindex="-1"></a>  x,</span>
<span id="cb55-3"><a href="#cb55-3" aria-hidden="true" tabindex="-1"></a>  <span class="at">ClusterNum =</span> <span class="dv">10</span>,</span>
<span id="cb55-4"><a href="#cb55-4" aria-hidden="true" tabindex="-1"></a>  <span class="at">P.adj =</span> <span class="fl">0.05</span>,</span>
<span id="cb55-5"><a href="#cb55-5" aria-hidden="true" tabindex="-1"></a>  <span class="at">force =</span> F,</span>
<span id="cb55-6"><a href="#cb55-6" aria-hidden="true" tabindex="-1"></a>  <span class="at">nTop =</span> <span class="dv">10</span>,</span>
<span id="cb55-7"><a href="#cb55-7" aria-hidden="true" tabindex="-1"></a>  <span class="at">method =</span> <span class="st">"ward.D2"</span>,</span>
<span id="cb55-8"><a href="#cb55-8" aria-hidden="true" tabindex="-1"></a>  ...</span>
<span id="cb55-9"><a href="#cb55-9" aria-hidden="true" tabindex="-1"></a>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</section>
<section id="arguments-26" class="level2">
<h2 class="anchored" data-anchor-id="arguments-26">Arguments</h2>
<ul>
<li><code>x</code>: Enrichment result from <code>EnrichGT</code> or <code>clusterProfiler</code>. For multi-database results, provide a <code>list</code>.</li>
<li><code>ClusterNum</code>: Number of clusters to create (default: 10).</li>
<li><code>P.adj</code>: Adjusted p-value cutoff (default: 0.05). Stricter values improve performance.</li>
<li><code>force</code>: Logical to bypass validation checks (default: FALSE).</li>
<li><code>nTop</code>: Number of top terms to keep per cluster by p-value (default: 10).</li>
<li><code>method</code>: Hierarchical clustering method (default: “ward.D2”). One of: “ward.D”, “ward.D2”, “single”, “complete”, “average” (UPGMA), “mcquitty” (WPGMA), “median” (WPGMC), or “centroid” (UPGMC).</li>
<li><code>...</code>: Additional arguments passed to clustering functions.</li>
</ul>
</section>
<section id="details-7" class="level2">
<h2 class="anchored" data-anchor-id="details-7">Details</h2>
<p>Input requirements by analysis type: ORA results: Required columns: “ID”, “Description”, “GeneRatio”, “pvalue”, “p.adjust”, “geneID”, “Count” GSEA results: Required columns: “ID”, “Description”, “NES”, “pvalue”, “p.adjust”, “core_enrichment” compareClusterResult: Either the compareClusterResult object or a data frame with: - All ORA columns listed above - Additional “Cluster” column Multi-database: Provide as a named list of the above result types</p>
</section>
<section id="author-8" class="level2">
<h2 class="anchored" data-anchor-id="author-8">Author</h2>
<p>Zhiming Ye</p>
</section>
<section id="value-13" class="level2">
<h2 class="anchored" data-anchor-id="value-13">Value</h2>
<p>An <code>EnrichGT_obj</code> containing:</p>
<ul>
<li>enriched_result: Filtered results data frame</li>
<li>gt_object: Formatted <code>gt_tbl</code> table object</li>
<li>gene_modules: List of gene modules per cluster</li>
<li>pathway_clusters: Pathway names by cluster</li>
<li>clustering_tree: <code>hclust</code> object for visualization</li>
<li>raw_enriched_result: Unfiltered results table</li>
</ul>
</section>
<section id="examples-19" class="level2">
<h2 class="anchored" data-anchor-id="examples-19">Examples</h2>
<div class="sourceCode" id="cb56"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb56-1"><a href="#cb56-1" aria-hidden="true" tabindex="-1"></a><span class="co"># ORA example</span></span>
<span id="cb56-2"><a href="#cb56-2" aria-hidden="true" tabindex="-1"></a>res <span class="ot">&lt;-</span> <span class="fu">egt_recluster_analysis</span>(ora_result, <span class="at">ClusterNum=</span><span class="dv">8</span>)</span>
<span id="cb56-3"><a href="#cb56-3" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(res<span class="sc">@</span>clustering_tree)</span>
<span id="cb56-4"><a href="#cb56-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb56-5"><a href="#cb56-5" aria-hidden="true" tabindex="-1"></a><span class="co"># GSEA example</span></span>
<span id="cb56-6"><a href="#cb56-6" aria-hidden="true" tabindex="-1"></a>gsea_res <span class="ot">&lt;-</span> <span class="fu">egt_recluster_analysis</span>(gsea_result, <span class="at">method=</span><span class="st">"average"</span>)</span>
<span id="cb56-7"><a href="#cb56-7" aria-hidden="true" tabindex="-1"></a>gsea_res</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</section>
</section>
<section id="egt_summary-generate-html-summary-for-enrichgt-object" class="level1">
<h1><code>egt_summary</code>: Generate HTML Summary for EnrichGT Object</h1>
<section id="description-19" class="level2">
<h2 class="anchored" data-anchor-id="description-19">Description</h2>
<p>This function generates an HTML-formatted summary of enrichment results that can be displayed in RStudio viewer or web browsers.</p>
</section>
<section id="usage-35" class="level2">
<h2 class="anchored" data-anchor-id="usage-35">Usage</h2>
<div class="sourceCode" id="cb57"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb57-1"><a href="#cb57-1" aria-hidden="true" tabindex="-1"></a><span class="fu">egt_summary</span>(x, name)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</section>
<section id="arguments-27" class="level2">
<h2 class="anchored" data-anchor-id="arguments-27">Arguments</h2>
<ul>
<li><code>x</code>: An EnrichGT_obj object</li>
<li><code>name</code>: The cluster name to summarize</li>
</ul>
</section>
<section id="value-14" class="level2">
<h2 class="anchored" data-anchor-id="value-14">Value</h2>
<p>HTML formatted summary that can be viewed with htmltools::html_print() or similar</p>
</section>
<section id="examples-20" class="level2">
<h2 class="anchored" data-anchor-id="examples-20">Examples</h2>
<div class="sourceCode" id="cb58"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb58-1"><a href="#cb58-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Generate HTML summary for cluster 1</span></span>
<span id="cb58-2"><a href="#cb58-2" aria-hidden="true" tabindex="-1"></a>html_summary <span class="ot">&lt;-</span> <span class="fu">egt_summary</span>(obj, <span class="st">"Cluster_1"</span>)</span>
<span id="cb58-3"><a href="#cb58-3" aria-hidden="true" tabindex="-1"></a>htmltools<span class="sc">::</span><span class="fu">html_print</span>(html_summary)</span>
<span id="cb58-4"><a href="#cb58-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb58-5"><a href="#cb58-5" aria-hidden="true" tabindex="-1"></a><span class="co"># Or view in RStudio viewer</span></span>
<span id="cb58-6"><a href="#cb58-6" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span> (<span class="fu">interactive</span>()) {</span>
<span id="cb58-7"><a href="#cb58-7" aria-hidden="true" tabindex="-1"></a>  html_summary <span class="ot">&lt;-</span> <span class="fu">egt_summary</span>(obj, <span class="st">"Cluster_1"</span>)</span>
<span id="cb58-8"><a href="#cb58-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">print</span>(html_summary)</span>
<span id="cb58-9"><a href="#cb58-9" aria-hidden="true" tabindex="-1"></a>}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</section>
</section>
<section id="egt_web_interface-launch-enrichgt-web-interface" class="level1">
<h1><code>egt_web_interface</code>: Launch EnrichGT Web Interface</h1>
<section id="description-20" class="level2">
<h2 class="anchored" data-anchor-id="description-20">Description</h2>
<p>Launch an interactive Shiny web application that provides a user-friendly interface for performing enrichment analysis using EnrichGT package functions.</p>
</section>
<section id="usage-36" class="level2">
<h2 class="anchored" data-anchor-id="usage-36">Usage</h2>
<div class="sourceCode" id="cb59"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb59-1"><a href="#cb59-1" aria-hidden="true" tabindex="-1"></a><span class="fu">egt_web_interface</span>(</span>
<span id="cb59-2"><a href="#cb59-2" aria-hidden="true" tabindex="-1"></a>  <span class="at">LLM =</span> <span class="cn">NULL</span>,</span>
<span id="cb59-3"><a href="#cb59-3" aria-hidden="true" tabindex="-1"></a>  <span class="at">port =</span> <span class="cn">NULL</span>,</span>
<span id="cb59-4"><a href="#cb59-4" aria-hidden="true" tabindex="-1"></a>  <span class="at">host =</span> <span class="st">"127.0.0.1"</span>,</span>
<span id="cb59-5"><a href="#cb59-5" aria-hidden="true" tabindex="-1"></a>  <span class="at">launch.browser =</span> <span class="cn">TRUE</span></span>
<span id="cb59-6"><a href="#cb59-6" aria-hidden="true" tabindex="-1"></a>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</section>
<section id="arguments-28" class="level2">
<h2 class="anchored" data-anchor-id="arguments-28">Arguments</h2>
<ul>
<li><code>LLM</code>: An optional LLM chat object created by the <code>ellmer</code> package. If provided, enables LLM-powered summarization of recluster analysis results. If NULL (default), LLM features are disabled.</li>
<li><code>port</code>: Port number for the application. If NULL (default), a random available port will be used.</li>
<li><code>host</code>: IP address that the application should listen on. Default is “127.0.0.1” (localhost).</li>
<li><code>launch.browser</code>: Logical. If TRUE (default), the web browser will be launched automatically.</li>
</ul>
</section>
<section id="details-8" class="level2">
<h2 class="anchored" data-anchor-id="details-8">Details</h2>
<p>The web interface includes major analysis modules in EnrichGT.</p>
</section>
<section id="author-9" class="level2">
<h2 class="anchored" data-anchor-id="author-9">Author</h2>
<p>Zhiming Ye</p>
</section>
<section id="seealso-4" class="level2">
<h2 class="anchored" data-anchor-id="seealso-4">Seealso</h2>
<p><code>[egt_enrichment_analysis](egt_enrichment_analysis)</code>, <code>[egt_gsea_analysis](egt_gsea_analysis)</code>, <code>[egt_recluster_analysis](egt_recluster_analysis)</code>, <code>[egt_plot_results](egt_plot_results)</code>, <code>[egt_llm_summary](egt_llm_summary)</code></p>
</section>
<section id="value-15" class="level2">
<h2 class="anchored" data-anchor-id="value-15">Value</h2>
<p>Starts a Shiny application</p>
</section>
<section id="examples-21" class="level2">
<h2 class="anchored" data-anchor-id="examples-21">Examples</h2>
<div class="sourceCode" id="cb60"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb60-1"><a href="#cb60-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Launch the web interface with default settings</span></span>
<span id="cb60-2"><a href="#cb60-2" aria-hidden="true" tabindex="-1"></a><span class="fu">egt_web_interface</span>()</span>
<span id="cb60-3"><a href="#cb60-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb60-4"><a href="#cb60-4" aria-hidden="true" tabindex="-1"></a><span class="co"># Launch with LLM support</span></span>
<span id="cb60-5"><a href="#cb60-5" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ellmer)</span>
<span id="cb60-6"><a href="#cb60-6" aria-hidden="true" tabindex="-1"></a>chat <span class="ot">&lt;-</span> <span class="fu">chat_deepseek</span>(<span class="at">api_key =</span> <span class="st">"your_api_key"</span>, <span class="at">model =</span> <span class="st">"deepseek-chat"</span>)</span>
<span id="cb60-7"><a href="#cb60-7" aria-hidden="true" tabindex="-1"></a><span class="fu">egt_web_interface</span>(<span class="at">LLM =</span> chat)</span>
<span id="cb60-8"><a href="#cb60-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb60-9"><a href="#cb60-9" aria-hidden="true" tabindex="-1"></a><span class="co"># Launch on a specific port</span></span>
<span id="cb60-10"><a href="#cb60-10" aria-hidden="true" tabindex="-1"></a><span class="fu">egt_web_interface</span>(<span class="at">port =</span> <span class="dv">3838</span>)</span>
<span id="cb60-11"><a href="#cb60-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb60-12"><a href="#cb60-12" aria-hidden="true" tabindex="-1"></a><span class="co"># Launch without opening browser automatically</span></span>
<span id="cb60-13"><a href="#cb60-13" aria-hidden="true" tabindex="-1"></a><span class="fu">egt_web_interface</span>(<span class="at">launch.browser =</span> <span class="cn">FALSE</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</section>
</section>
<section id="genes_with_weights-create-a-ranked-gene-list-for-gsea-analysis" class="level1">
<h1><code>genes_with_weights</code>: Create a ranked gene list for GSEA analysis</h1>
<section id="description-21" class="level2">
<h2 class="anchored" data-anchor-id="description-21">Description</h2>
<p>Takes gene identifiers and corresponding weights (like log2 fold changes) and returns a ranked vector suitable for Gene Set Enrichment Analysis (GSEA).</p>
</section>
<section id="usage-37" class="level2">
<h2 class="anchored" data-anchor-id="usage-37">Usage</h2>
<div class="sourceCode" id="cb61"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb61-1"><a href="#cb61-1" aria-hidden="true" tabindex="-1"></a><span class="fu">genes_with_weights</span>(genes, weights)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</section>
<section id="arguments-29" class="level2">
<h2 class="anchored" data-anchor-id="arguments-29">Arguments</h2>
<ul>
<li><code>genes</code>: Character vector of gene identifiers (e.g., gene symbols or ENTREZ IDs)</li>
<li><code>weights</code>: Numeric vector of weights for each gene (typically log2 fold changes)</li>
</ul>
</section>
<section id="author-10" class="level2">
<h2 class="anchored" data-anchor-id="author-10">Author</h2>
<p>Zhiming Ye</p>
</section>
<section id="value-16" class="level2">
<h2 class="anchored" data-anchor-id="value-16">Value</h2>
<p>A named numeric vector sorted in descending order by weight, where: - Names are gene identifiers - Values are the corresponding weights</p>
</section>
<section id="examples-22" class="level2">
<h2 class="anchored" data-anchor-id="examples-22">Examples</h2>
<div class="sourceCode" id="cb62"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb62-1"><a href="#cb62-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Example using differential expression results</span></span>
<span id="cb62-2"><a href="#cb62-2" aria-hidden="true" tabindex="-1"></a>genes <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"TP53"</span>, <span class="st">"BRCA1"</span>, <span class="st">"EGFR"</span>)</span>
<span id="cb62-3"><a href="#cb62-3" aria-hidden="true" tabindex="-1"></a>log2fc <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="fl">1.5</span>, <span class="sc">-</span><span class="fl">2.1</span>, <span class="fl">0.8</span>)</span>
<span id="cb62-4"><a href="#cb62-4" aria-hidden="true" tabindex="-1"></a>ranked_genes <span class="ot">&lt;-</span> <span class="fu">genes_with_weights</span>(genes, log2fc)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</section>
</section>
<section id="database_...-get-database-for-enrichment-analysis" class="level1">
<h1><code>database_...</code>: Get database for enrichment analysis</h1>
<section id="description-22" class="level2">
<h2 class="anchored" data-anchor-id="description-22">Description</h2>
<p>Get Gene Ontology (GO), Reactome, and other term-to-gene database, for enrichment analysis</p>
</section>
<section id="usage-38" class="level2">
<h2 class="anchored" data-anchor-id="usage-38">Usage</h2>
<div class="sourceCode" id="cb63"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb63-1"><a href="#cb63-1" aria-hidden="true" tabindex="-1"></a><span class="fu">database_GO_BP</span>(<span class="at">OrgDB =</span> org.Hs.eg.db)</span>
<span id="cb63-2"><a href="#cb63-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb63-3"><a href="#cb63-3" aria-hidden="true" tabindex="-1"></a><span class="fu">database_GO_CC</span>(<span class="at">OrgDB =</span> org.Hs.eg.db)</span>
<span id="cb63-4"><a href="#cb63-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb63-5"><a href="#cb63-5" aria-hidden="true" tabindex="-1"></a><span class="fu">database_GO_MF</span>(<span class="at">OrgDB =</span> org.Hs.eg.db)</span>
<span id="cb63-6"><a href="#cb63-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb63-7"><a href="#cb63-7" aria-hidden="true" tabindex="-1"></a><span class="fu">database_GO_ALL</span>(<span class="at">OrgDB =</span> org.Hs.eg.db)</span>
<span id="cb63-8"><a href="#cb63-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb63-9"><a href="#cb63-9" aria-hidden="true" tabindex="-1"></a><span class="fu">database_Reactome</span>(<span class="at">OrgDB =</span> org.Hs.eg.db)</span>
<span id="cb63-10"><a href="#cb63-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb63-11"><a href="#cb63-11" aria-hidden="true" tabindex="-1"></a><span class="fu">database_progeny_human</span>()</span>
<span id="cb63-12"><a href="#cb63-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb63-13"><a href="#cb63-13" aria-hidden="true" tabindex="-1"></a><span class="fu">database_progeny_mouse</span>()</span>
<span id="cb63-14"><a href="#cb63-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb63-15"><a href="#cb63-15" aria-hidden="true" tabindex="-1"></a><span class="fu">database_CollecTRI_human</span>()</span>
<span id="cb63-16"><a href="#cb63-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb63-17"><a href="#cb63-17" aria-hidden="true" tabindex="-1"></a><span class="fu">database_CollecTRI_mouse</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</section>
<section id="arguments-30" class="level2">
<h2 class="anchored" data-anchor-id="arguments-30">Arguments</h2>
<ul>
<li><code>OrgDB</code>: The AnnotationDbi database to fetch pathway data and convert gene IDs to gene symbols. For human it would be <code>org.Hs.eg.db</code>, for mouse it would be <code>org.Mm.eg.db</code>. In AnnotationDbi there are many species, please search <code>AnnotationDbi</code> for other species annotation database. GO and Reactome should add this, progeny and collectri do not.</li>
</ul>
</section>
<section id="author-11" class="level2">
<h2 class="anchored" data-anchor-id="author-11">Author</h2>
<p>Zhiming Ye. Part of functions were inspired by <code>clusterProfiler</code> but with brand new implement.</p>
</section>
<section id="value-17" class="level2">
<h2 class="anchored" data-anchor-id="value-17">Value</h2>
<p>a data.frame with ID, terms and genes</p>
</section>
</section>
<section id="database_kegg-get-kegg-database-from-kegg-website" class="level1">
<h1><code>database_KEGG</code>: Get KEGG database from KEGG website</h1>
<section id="description-23" class="level2">
<h2 class="anchored" data-anchor-id="description-23">Description</h2>
<p>KEGG is a commercialized database. So EnrichGT can’t pre-cache them locally. You can use this function to fetch KEGG database pathways and modules.</p>
</section>
<section id="usage-39" class="level2">
<h2 class="anchored" data-anchor-id="usage-39">Usage</h2>
<div class="sourceCode" id="cb64"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb64-1"><a href="#cb64-1" aria-hidden="true" tabindex="-1"></a><span class="fu">database_KEGG</span>(<span class="at">kegg_organism=</span><span class="st">"hsa"</span>,<span class="at">OrgDB =</span> org.Hs.eg.db,<span class="at">kegg_modules=</span>F,<span class="at">local_cache=</span>F)</span>
<span id="cb64-2"><a href="#cb64-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb64-3"><a href="#cb64-3" aria-hidden="true" tabindex="-1"></a><span class="fu">database_KEGG_show_organism</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</section>
<section id="arguments-31" class="level2">
<h2 class="anchored" data-anchor-id="arguments-31">Arguments</h2>
<ul>
<li><code>kegg_organism</code>: Determine which species data from KEGG will be fetch. For human, it would be <code>hsa</code>(in default); For mouse, it would be <code>mmu</code>. If you wants other species, see <code>database_kegg_show_organism()</code> for details.</li>
<li><code>OrgDB</code>: The AnnotationDbi database to convert KEGG gene ID to gene symbols. For human it would be <code>org.Hs.eg.db</code>, for mouse it would be <code>org.Mm.eg.db</code>. In AnnotationDbi there are many species, please search <code>AnnotationDbi</code> for other species annotation database.</li>
<li><code>kegg_modules</code>: If TRUE, returns KEGG module; If FALSE returns KEGG pathways. In default, this is setted to FALSE to get mouse commonly used KEGG pathways.</li>
<li><code>local_cache</code>: cache a copy in local working folder. It will be saved as a <code>.enrichgt_cache</code> file in working dictionary. The <code>.enrichgt_cache</code> is just a <code>.rds</code> file, feel free to read it using <code>readRDS()</code>.</li>
</ul>
</section>
<section id="value-18" class="level2">
<h2 class="anchored" data-anchor-id="value-18">Value</h2>
<p>data.frame contains KEGG annotations</p>
</section>
</section>
<section id="delete--filter-enrichment-results-by-description-pattern" class="level1">
<h1><code>%-delete-&gt;%</code>: Filter Enrichment Results by Description Pattern</h1>
<section id="description-24" class="level2">
<h2 class="anchored" data-anchor-id="description-24">Description</h2>
<p>Infix operator to filter enrichment results by matching against Description field. For EnrichGT_obj objects, re-runs clustering analysis after filtering.</p>
</section>
<section id="usage-40" class="level2">
<h2 class="anchored" data-anchor-id="usage-40">Usage</h2>
<div class="sourceCode" id="cb65"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb65-1"><a href="#cb65-1" aria-hidden="true" tabindex="-1"></a>x <span class="sc">%-delete-&gt;%</span> y</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</section>
<section id="arguments-32" class="level2">
<h2 class="anchored" data-anchor-id="arguments-32">Arguments</h2>
<ul>
<li><code>x</code>: Either an EnrichGT_obj object or data.frame containing enrichment results</li>
<li><code>y</code>: Regular expression pattern to match against Description field</li>
</ul>
</section>
<section id="details-9" class="level2">
<h2 class="anchored" data-anchor-id="details-9">Details</h2>
<p>This operator helps refine enrichment results by removing terms matching the given pattern from the Description field. When applied to EnrichGT_obj, it preserves all original parameters and re-runs the clustering analysis on the filtered results.</p>
</section>
<section id="value-19" class="level2">
<h2 class="anchored" data-anchor-id="value-19">Value</h2>
<p>For EnrichGT_obj input: A new EnrichGT_obj with filtered and re-clustered results. For data.frame input: A filtered data.frame.</p>
</section>
<section id="examples-23" class="level2">
<h2 class="anchored" data-anchor-id="examples-23">Examples</h2>
<div class="sourceCode" id="cb66"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb66-1"><a href="#cb66-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Filter out "ribosome" related terms</span></span>
<span id="cb66-2"><a href="#cb66-2" aria-hidden="true" tabindex="-1"></a>filtered_results <span class="ot">&lt;-</span> reenrichment_obj <span class="sc">%-delete-&gt;%</span> <span class="st">"ribosome"</span></span>
<span id="cb66-3"><a href="#cb66-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb66-4"><a href="#cb66-4" aria-hidden="true" tabindex="-1"></a><span class="co"># Filter data.frame directly</span></span>
<span id="cb66-5"><a href="#cb66-5" aria-hidden="true" tabindex="-1"></a>filtered_df <span class="ot">&lt;-</span> df <span class="sc">%-delete-&gt;%</span> <span class="st">"metabolism"</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>


</section>
</section>

<a onclick="window.scrollTo(0, 0); return false;" role="button" id="quarto-back-to-top"><i class="bi bi-arrow-up"></i> Back to top</a></main> <!-- /main -->
<script id="quarto-html-after-body" type="application/javascript">
  window.document.addEventListener("DOMContentLoaded", function (event) {
    const icon = "";
    const anchorJS = new window.AnchorJS();
    anchorJS.options = {
      placement: 'right',
      icon: icon
    };
    anchorJS.add('.anchored');
    const isCodeAnnotation = (el) => {
      for (const clz of el.classList) {
        if (clz.startsWith('code-annotation-')) {                     
          return true;
        }
      }
      return false;
    }
    const onCopySuccess = function(e) {
      // button target
      const button = e.trigger;
      // don't keep focus
      button.blur();
      // flash "checked"
      button.classList.add('code-copy-button-checked');
      var currentTitle = button.getAttribute("title");
      button.setAttribute("title", "Copied!");
      let tooltip;
      if (window.bootstrap) {
        button.setAttribute("data-bs-toggle", "tooltip");
        button.setAttribute("data-bs-placement", "left");
        button.setAttribute("data-bs-title", "Copied!");
        tooltip = new bootstrap.Tooltip(button, 
          { trigger: "manual", 
            customClass: "code-copy-button-tooltip",
            offset: [0, -8]});
        tooltip.show();    
      }
      setTimeout(function() {
        if (tooltip) {
          tooltip.hide();
          button.removeAttribute("data-bs-title");
          button.removeAttribute("data-bs-toggle");
          button.removeAttribute("data-bs-placement");
        }
        button.setAttribute("title", currentTitle);
        button.classList.remove('code-copy-button-checked');
      }, 1000);
      // clear code selection
      e.clearSelection();
    }
    const getTextToCopy = function(trigger) {
        const codeEl = trigger.previousElementSibling.cloneNode(true);
        for (const childEl of codeEl.children) {
          if (isCodeAnnotation(childEl)) {
            childEl.remove();
          }
        }
        return codeEl.innerText;
    }
    const clipboard = new window.ClipboardJS('.code-copy-button:not([data-in-quarto-modal])', {
      text: getTextToCopy
    });
    clipboard.on('success', onCopySuccess);
    if (window.document.getElementById('quarto-embedded-source-code-modal')) {
      const clipboardModal = new window.ClipboardJS('.code-copy-button[data-in-quarto-modal]', {
        text: getTextToCopy,
        container: window.document.getElementById('quarto-embedded-source-code-modal')
      });
      clipboardModal.on('success', onCopySuccess);
    }
      var localhostRegex = new RegExp(/^(?:http|https):\/\/localhost\:?[0-9]*\//);
      var mailtoRegex = new RegExp(/^mailto:/);
        var filterRegex = new RegExp('/' + window.location.host + '/');
      var isInternal = (href) => {
          return filterRegex.test(href) || localhostRegex.test(href) || mailtoRegex.test(href);
      }
      // Inspect non-navigation links and adorn them if external
     var links = window.document.querySelectorAll('a[href]:not(.nav-link):not(.navbar-brand):not(.toc-action):not(.sidebar-link):not(.sidebar-item-toggle):not(.pagination-link):not(.no-external):not([aria-hidden]):not(.dropdown-item):not(.quarto-navigation-tool):not(.about-link)');
      for (var i=0; i<links.length; i++) {
        const link = links[i];
        if (!isInternal(link.href)) {
          // undo the damage that might have been done by quarto-nav.js in the case of
          // links that we want to consider external
          if (link.dataset.originalHref !== undefined) {
            link.href = link.dataset.originalHref;
          }
        }
      }
    function tippyHover(el, contentFn, onTriggerFn, onUntriggerFn) {
      const config = {
        allowHTML: true,
        maxWidth: 500,
        delay: 100,
        arrow: false,
        appendTo: function(el) {
            return el.parentElement;
        },
        interactive: true,
        interactiveBorder: 10,
        theme: 'quarto',
        placement: 'bottom-start',
      };
      if (contentFn) {
        config.content = contentFn;
      }
      if (onTriggerFn) {
        config.onTrigger = onTriggerFn;
      }
      if (onUntriggerFn) {
        config.onUntrigger = onUntriggerFn;
      }
      window.tippy(el, config); 
    }
    const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
    for (var i=0; i<noterefs.length; i++) {
      const ref = noterefs[i];
      tippyHover(ref, function() {
        // use id or data attribute instead here
        let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
        try { href = new URL(href).hash; } catch {}
        const id = href.replace(/^#\/?/, "");
        const note = window.document.getElementById(id);
        if (note) {
          return note.innerHTML;
        } else {
          return "";
        }
      });
    }
    const xrefs = window.document.querySelectorAll('a.quarto-xref');
    const processXRef = (id, note) => {
      // Strip column container classes
      const stripColumnClz = (el) => {
        el.classList.remove("page-full", "page-columns");
        if (el.children) {
          for (const child of el.children) {
            stripColumnClz(child);
          }
        }
      }
      stripColumnClz(note)
      if (id === null || id.startsWith('sec-')) {
        // Special case sections, only their first couple elements
        const container = document.createElement("div");
        if (note.children && note.children.length > 2) {
          container.appendChild(note.children[0].cloneNode(true));
          for (let i = 1; i < note.children.length; i++) {
            const child = note.children[i];
            if (child.tagName === "P" && child.innerText === "") {
              continue;
            } else {
              container.appendChild(child.cloneNode(true));
              break;
            }
          }
          if (window.Quarto?.typesetMath) {
            window.Quarto.typesetMath(container);
          }
          return container.innerHTML
        } else {
          if (window.Quarto?.typesetMath) {
            window.Quarto.typesetMath(note);
          }
          return note.innerHTML;
        }
      } else {
        // Remove any anchor links if they are present
        const anchorLink = note.querySelector('a.anchorjs-link');
        if (anchorLink) {
          anchorLink.remove();
        }
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(note);
        }
        if (note.classList.contains("callout")) {
          return note.outerHTML;
        } else {
          return note.innerHTML;
        }
      }
    }
    for (var i=0; i<xrefs.length; i++) {
      const xref = xrefs[i];
      tippyHover(xref, undefined, function(instance) {
        instance.disable();
        let url = xref.getAttribute('href');
        let hash = undefined; 
        if (url.startsWith('#')) {
          hash = url;
        } else {
          try { hash = new URL(url).hash; } catch {}
        }
        if (hash) {
          const id = hash.replace(/^#\/?/, "");
          const note = window.document.getElementById(id);
          if (note !== null) {
            try {
              const html = processXRef(id, note.cloneNode(true));
              instance.setContent(html);
            } finally {
              instance.enable();
              instance.show();
            }
          } else {
            // See if we can fetch this
            fetch(url.split('#')[0])
            .then(res => res.text())
            .then(html => {
              const parser = new DOMParser();
              const htmlDoc = parser.parseFromString(html, "text/html");
              const note = htmlDoc.getElementById(id);
              if (note !== null) {
                const html = processXRef(id, note);
                instance.setContent(html);
              } 
            }).finally(() => {
              instance.enable();
              instance.show();
            });
          }
        } else {
          // See if we can fetch a full url (with no hash to target)
          // This is a special case and we should probably do some content thinning / targeting
          fetch(url)
          .then(res => res.text())
          .then(html => {
            const parser = new DOMParser();
            const htmlDoc = parser.parseFromString(html, "text/html");
            const note = htmlDoc.querySelector('main.content');
            if (note !== null) {
              // This should only happen for chapter cross references
              // (since there is no id in the URL)
              // remove the first header
              if (note.children.length > 0 && note.children[0].tagName === "HEADER") {
                note.children[0].remove();
              }
              const html = processXRef(null, note);
              instance.setContent(html);
            } 
          }).finally(() => {
            instance.enable();
            instance.show();
          });
        }
      }, function(instance) {
      });
    }
        let selectedAnnoteEl;
        const selectorForAnnotation = ( cell, annotation) => {
          let cellAttr = 'data-code-cell="' + cell + '"';
          let lineAttr = 'data-code-annotation="' +  annotation + '"';
          const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
          return selector;
        }
        const selectCodeLines = (annoteEl) => {
          const doc = window.document;
          const targetCell = annoteEl.getAttribute("data-target-cell");
          const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
          const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
          const lines = annoteSpan.getAttribute("data-code-lines").split(",");
          const lineIds = lines.map((line) => {
            return targetCell + "-" + line;
          })
          let top = null;
          let height = null;
          let parent = null;
          if (lineIds.length > 0) {
              //compute the position of the single el (top and bottom and make a div)
              const el = window.document.getElementById(lineIds[0]);
              top = el.offsetTop;
              height = el.offsetHeight;
              parent = el.parentElement.parentElement;
            if (lineIds.length > 1) {
              const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
              const bottom = lastEl.offsetTop + lastEl.offsetHeight;
              height = bottom - top;
            }
            if (top !== null && height !== null && parent !== null) {
              // cook up a div (if necessary) and position it 
              let div = window.document.getElementById("code-annotation-line-highlight");
              if (div === null) {
                div = window.document.createElement("div");
                div.setAttribute("id", "code-annotation-line-highlight");
                div.style.position = 'absolute';
                parent.appendChild(div);
              }
              div.style.top = top - 2 + "px";
              div.style.height = height + 4 + "px";
              div.style.left = 0;
              let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
              if (gutterDiv === null) {
                gutterDiv = window.document.createElement("div");
                gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
                gutterDiv.style.position = 'absolute';
                const codeCell = window.document.getElementById(targetCell);
                const gutter = codeCell.querySelector('.code-annotation-gutter');
                gutter.appendChild(gutterDiv);
              }
              gutterDiv.style.top = top - 2 + "px";
              gutterDiv.style.height = height + 4 + "px";
            }
            selectedAnnoteEl = annoteEl;
          }
        };
        const unselectCodeLines = () => {
          const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
          elementsIds.forEach((elId) => {
            const div = window.document.getElementById(elId);
            if (div) {
              div.remove();
            }
          });
          selectedAnnoteEl = undefined;
        };
          // Handle positioning of the toggle
      window.addEventListener(
        "resize",
        throttle(() => {
          elRect = undefined;
          if (selectedAnnoteEl) {
            selectCodeLines(selectedAnnoteEl);
          }
        }, 10)
      );
      function throttle(fn, ms) {
      let throttle = false;
      let timer;
        return (...args) => {
          if(!throttle) { // first call gets through
              fn.apply(this, args);
              throttle = true;
          } else { // all the others get throttled
              if(timer) clearTimeout(timer); // cancel #2
              timer = setTimeout(() => {
                fn.apply(this, args);
                timer = throttle = false;
              }, ms);
          }
        };
      }
        // Attach click handler to the DT
        const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
        for (const annoteDlNode of annoteDls) {
          annoteDlNode.addEventListener('click', (event) => {
            const clickedEl = event.target;
            if (clickedEl !== selectedAnnoteEl) {
              unselectCodeLines();
              const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
              if (activeEl) {
                activeEl.classList.remove('code-annotation-active');
              }
              selectCodeLines(clickedEl);
              clickedEl.classList.add('code-annotation-active');
            } else {
              // Unselect the line
              unselectCodeLines();
              clickedEl.classList.remove('code-annotation-active');
            }
          });
        }
    const findCites = (el) => {
      const parentEl = el.parentElement;
      if (parentEl) {
        const cites = parentEl.dataset.cites;
        if (cites) {
          return {
            el,
            cites: cites.split(' ')
          };
        } else {
          return findCites(el.parentElement)
        }
      } else {
        return undefined;
      }
    };
    var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
    for (var i=0; i<bibliorefs.length; i++) {
      const ref = bibliorefs[i];
      const citeInfo = findCites(ref);
      if (citeInfo) {
        tippyHover(citeInfo.el, function() {
          var popup = window.document.createElement('div');
          citeInfo.cites.forEach(function(cite) {
            var citeDiv = window.document.createElement('div');
            citeDiv.classList.add('hanging-indent');
            citeDiv.classList.add('csl-entry');
            var biblioDiv = window.document.getElementById('ref-' + cite);
            if (biblioDiv) {
              citeDiv.innerHTML = biblioDiv.innerHTML;
            }
            popup.appendChild(citeDiv);
          });
          return popup.innerHTML;
        });
      }
    }
  });
  </script>
</div> <!-- /content -->
<footer class="footer">
  <div class="nav-footer">
    <div class="nav-footer-left">
<p>EnrichGT</p>
</div>   
    <div class="nav-footer-center">
      &nbsp;
    <div class="toc-actions d-sm-block d-md-none"><ul><li><a href="https://github.com/ZhimingYe/EnrichGT_website/blob/main/pkgfuns.qmd" class="toc-action"><i class="bi bi-github"></i>View source</a></li><li><a href="https://github.com/ZhimingYe/EnrichGT_website/issues/new" class="toc-action"><i class="bi empty"></i>Report an issue</a></li></ul></div></div>
    <div class="nav-footer-right">
<p>Version 2.2</p>
</div>
  </div>
</footer>




</body></html>