<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.7.32">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">


<title>Second enrichment of enriched Results – EnrichGT</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
/* CSS for syntax highlighting */
html { -webkit-text-size-adjust: 100%; }
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
  }
pre.numberSource { margin-left: 3em;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
</style>


<script src="site_libs/quarto-nav/quarto-nav.js"></script>
<script src="site_libs/quarto-nav/headroom.min.js"></script>
<script src="site_libs/clipboard/clipboard.min.js"></script>
<script src="site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="site_libs/quarto-search/fuse.min.js"></script>
<script src="site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="./">
<script src="site_libs/quarto-html/quarto.js" type="module"></script>
<script src="site_libs/quarto-html/tabsets/tabsets.js" type="module"></script>
<script src="site_libs/quarto-html/popper.min.js"></script>
<script src="site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="site_libs/quarto-html/anchor.min.js"></script>
<link href="site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="site_libs/quarto-html/quarto-syntax-highlighting-37eea08aefeeee20ff55810ff984fec1.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="site_libs/bootstrap/bootstrap.min.js"></script>
<link href="site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="site_libs/bootstrap/bootstrap-06d22339cf1bae60befde528b0c36036.min.css" rel="stylesheet" append-hash="true" id="quarto-bootstrap" data-mode="light">
<script id="quarto-search-options" type="application/json">{
  "location": "navbar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "end",
  "type": "overlay",
  "limit": 50,
  "keyboard-shortcut": [
    "f",
    "/",
    "s"
  ],
  "show-item-context": false,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-text-placeholder": "",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit",
    "search-label": "Search"
  }
}</script>


<link rel="stylesheet" href="styles.css">
</head>

<body class="nav-sidebar docked nav-fixed quarto-light">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top">
    <nav class="navbar navbar-expand-lg " data-bs-theme="dark">
      <div class="navbar-container container-fluid">
      <div class="navbar-brand-container mx-auto">
    <a href="./index.html" class="navbar-brand navbar-brand-logo">
    <img src="./EnrichGTlogo.png" alt="" class="navbar-logo">
    </a>
    <a class="navbar-brand" href="./index.html">
    <span class="navbar-title">EnrichGT</span>
    </a>
  </div>
            <div id="quarto-search" class="" title="Search"></div>
          <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbarCollapse" aria-controls="navbarCollapse" role="menu" aria-expanded="false" aria-label="Toggle navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
  <span class="navbar-toggler-icon"></span>
</button>
          <div class="collapse navbar-collapse" id="navbarCollapse">
            <ul class="navbar-nav navbar-nav-scroll me-auto">
  <li class="nav-item">
    <a class="nav-link active" href="./index.html" aria-current="page"> 
<span class="menu-text">Home</span></a>
  </li>  
  <li class="nav-item">
    <a class="nav-link" href="./pkgfuns.html"> 
<span class="menu-text">Package Functions</span></a>
  </li>  
  <li class="nav-item">
    <a class="nav-link" href="https://github.com/ZhimingYe/EnrichGT/"> 
<span class="menu-text">Github</span></a>
  </li>  
  <li class="nav-item">
    <a class="nav-link" href="./acknowledgement.html"> 
<span class="menu-text">Acknowledgement</span></a>
  </li>  
</ul>
          </div> <!-- /navcollapse -->
            <div class="quarto-navbar-tools">
</div>
      </div> <!-- /container-fluid -->
    </nav>
  <nav class="quarto-secondary-nav">
    <div class="container-fluid d-flex">
      <button type="button" class="quarto-btn-toggle btn" data-bs-toggle="collapse" role="button" data-bs-target=".quarto-sidebar-collapse-item" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
        <i class="bi bi-layout-text-sidebar-reverse"></i>
      </button>
        <nav class="quarto-page-breadcrumbs" aria-label="breadcrumb"><ol class="breadcrumb"><li class="breadcrumb-item"><a href="./re_enrichment.html">Second enrichment of enriched Results</a></li></ol></nav>
        <a class="flex-grow-1" role="navigation" data-bs-toggle="collapse" data-bs-target=".quarto-sidebar-collapse-item" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">      
        </a>
    </div>
  </nav>
</header>
<!-- content -->
<div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-article page-navbar">
<!-- sidebar -->
  <nav id="quarto-sidebar" class="sidebar collapse collapse-horizontal quarto-sidebar-collapse-item sidebar-navigation docked overflow-auto">
    <div class="sidebar-menu-container"> 
    <ul class="list-unstyled mt-1">
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./index.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">EnrichGT: all-in-one enrichment analysis</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./basic.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Basic Workflow</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./interfaceshiny.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Web Interface</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./enrichment_core.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Core Enrichment Functions</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./re_enrichment.html" class="sidebar-item-text sidebar-link active">
 <span class="menu-text">Second enrichment of enriched Results</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./fusing.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Functions to FUSING results</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./visualize.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Visualize and Create Reports</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./llm.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Large language models integration</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./compare.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Compare Multiple Enrichment Results</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./database.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Select your databases for enrichment or annotation</span></a>
  </div>
</li>
    </ul>
    </div>
</nav>
<div id="quarto-sidebar-glass" class="quarto-sidebar-collapse-item" data-bs-toggle="collapse" data-bs-target=".quarto-sidebar-collapse-item"></div>
<!-- margin-sidebar -->
    <div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
        <nav id="TOC" role="doc-toc" class="toc-active" data-toc-expanded="99">
    <h2 id="toc-title">On this page</h2>
   
  <ul>
  <li><a href="#enrichment-of-enriched-results" id="toc-enrichment-of-enriched-results" class="nav-link active" data-scroll-target="#enrichment-of-enriched-results">Enrichment of Enriched Results</a>
  <ul class="collapse">
  <li><a href="#access-re-enriched-data-table" id="toc-access-re-enriched-data-table" class="nav-link" data-scroll-target="#access-re-enriched-data-table">Access re-enriched data table</a></li>
  <li><a href="#access-re-enriched-html-report" id="toc-access-re-enriched-html-report" class="nav-link" data-scroll-target="#access-re-enriched-html-report">Access re-enriched HTML report</a></li>
  <li><a href="#glance-details-of-each-module" id="toc-glance-details-of-each-module" class="nav-link" data-scroll-target="#glance-details-of-each-module">Glance details of each module</a>
  <ul class="collapse">
  <li><a href="#through-word-cloud" id="toc-through-word-cloud" class="nav-link" data-scroll-target="#through-word-cloud">Through Word Cloud</a></li>
  <li><a href="#through-morden-ides" id="toc-through-morden-ides" class="nav-link" data-scroll-target="#through-morden-ides">Through Morden IDEs</a></li>
  <li><a href="#through-r-console" id="toc-through-r-console" class="nav-link" data-scroll-target="#through-r-console">Through R Console</a></li>
  </ul></li>
  </ul></li>
  <li><a href="#mask-unnecessary-results" id="toc-mask-unnecessary-results" class="nav-link" data-scroll-target="#mask-unnecessary-results">Mask unnecessary results</a></li>
  <li><a href="#manual-visualize-the-relationship-of-specific-terms-intra-database" id="toc-manual-visualize-the-relationship-of-specific-terms-intra-database" class="nav-link" data-scroll-target="#manual-visualize-the-relationship-of-specific-terms-intra-database">Manual visualize the relationship of specific terms intra database</a></li>
  <li><a href="#infering-tfs-or-pathway-activity-and-more-based-on-meta-gene-modules" id="toc-infering-tfs-or-pathway-activity-and-more-based-on-meta-gene-modules" class="nav-link" data-scroll-target="#infering-tfs-or-pathway-activity-and-more-based-on-meta-gene-modules">Infering TFs or pathway activity and more based on meta-gene modules</a></li>
  </ul>
<div class="toc-actions"><ul><li><a href="https://github.com/ZhimingYe/EnrichGT_website/blob/main/re_enrichment.qmd" class="toc-action"><i class="bi bi-github"></i>View source</a></li><li><a href="https://github.com/ZhimingYe/EnrichGT_website/issues/new" class="toc-action"><i class="bi empty"></i>Report an issue</a></li></ul></div></nav>
    </div>
<!-- main -->
<main class="content" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default">
<div class="quarto-title">
<h1 class="title">Second enrichment of enriched Results</h1>
</div>



<div class="quarto-title-meta">

    
  
    
  </div>
  


</header>


<section id="enrichment-of-enriched-results" class="level1">
<h1>Enrichment of Enriched Results</h1>
<p>The enriched result is too messy? Clean up it!</p>
<p>EnrichGT generates insightful results by simply constructing a term frequency matrix of genes enriched in pathways and performing clustering. While the results may not be statistically optimal, they offer significant interpretive insights.</p>
<p>Print <code>?egt_recluster_analysis</code> for further help. But of note, you can adjust <code>ClusterNum</code> (Cluster the enrichment into N clusters) and <code>nTop</code> (Show how many top items in GT table) for a better result (the default is not all the best for your data).</p>
<div class="callout callout-style-default callout-note callout-titled">
<div class="callout-header d-flex align-content-center" data-bs-toggle="collapse" data-bs-target=".callout-1-contents" aria-controls="callout-1" aria-expanded="false" aria-label="Toggle callout">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-title-container flex-fill">
Why the <code>re-enrichment</code> is necessary?
</div>
<div class="callout-btn-toggle d-inline-block border-0 py-1 ps-1 pe-0 float-end"><i class="callout-toggle"></i></div>
</div>
<div id="callout-1" class="callout-1-contents callout-collapse collapse">
<section id="challenges-in-biological-gene-enrichment-analysis" class="level5 callout-body-container callout-body">
<h5 class="anchored" data-anchor-id="challenges-in-biological-gene-enrichment-analysis">Challenges in Biological Gene Enrichment Analysis</h5>
<p>Gene enrichment analysis can often be misleading due to the redundancy within gene set databases and the limitations of most enrichment tools. Many tools, by default, only display a few top results and fail to filter out redundancy. This can result in both biological misinterpretation and valuable information being overlooked.</p>
<p>For instance, high expression of certain immune genes can cause many immune-related gene sets to appear overrepresented. However, a closer look often reveals that these gene sets are derived from the same group of genes, which might represent only a small fraction (less than 10%) of the differential expressed genes (DEGs). What about the other 90%? Do they hold no biological significance?</p>
<p>The main purpose of developing this package is to provide a lightweight and practical solution to the problems mentioned above. Specifically, this package can cluster enrichment results based on hit genes or core enrichment from GSEA using term frequency analysis (from the output of the powerful <code>clusterProfiler</code>). This provides a clearer view of biological relevance by focusing on the genes that matter most.</p>
</section>
</div>
</div>
<div class="sourceCode" id="cb1"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a><span class="co"># From results generated before</span></span>
<span id="cb1-2"><a href="#cb1-2" aria-hidden="true" tabindex="-1"></a>res <span class="ot">&lt;-</span> <span class="fu">egt_enrichment_analysis</span>(<span class="at">genes =</span> DEGtable<span class="sc">$</span>Genes,</span>
<span id="cb1-3"><a href="#cb1-3" aria-hidden="true" tabindex="-1"></a><span class="at">database =</span> <span class="fu">database_GO_BP</span>(Org.Hs.eg.db))</span>
<span id="cb1-4"><a href="#cb1-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-5"><a href="#cb1-5" aria-hidden="true" tabindex="-1"></a>re_enrich <span class="ot">&lt;-</span> <span class="fu">egt_recluster_analysis</span>(</span>
<span id="cb1-6"><a href="#cb1-6" aria-hidden="true" tabindex="-1"></a>  res,</span>
<span id="cb1-7"><a href="#cb1-7" aria-hidden="true" tabindex="-1"></a>  <span class="at">ClusterNum =</span> <span class="dv">10</span>,</span>
<span id="cb1-8"><a href="#cb1-8" aria-hidden="true" tabindex="-1"></a>  <span class="at">P.adj =</span> <span class="fl">0.05</span>,</span>
<span id="cb1-9"><a href="#cb1-9" aria-hidden="true" tabindex="-1"></a>  <span class="at">force =</span> F,</span>
<span id="cb1-10"><a href="#cb1-10" aria-hidden="true" tabindex="-1"></a>  <span class="at">nTop =</span> <span class="dv">10</span>,</span>
<span id="cb1-11"><a href="#cb1-11" aria-hidden="true" tabindex="-1"></a>  <span class="at">method =</span> <span class="st">"ward.D2"</span></span>
<span id="cb1-12"><a href="#cb1-12" aria-hidden="true" tabindex="-1"></a>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<p>You can see the structure of <code>re_enrich</code> object above. The <code>re_enrich</code> object is an S4 <code>EnrichGT_obj</code> object. The first slot is the result table (a <code>data.frame</code>), and the second slot contains gt table.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb2"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1" aria-hidden="true" tabindex="-1"></a><span class="fu">str</span>(re_enrich,<span class="at">max.level =</span> <span class="dv">2</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>Formal class 'EnrichGT_obj' [package "EnrichGT"] with 12 slots
  ..@ enriched_result     : tibble [65 × 7] (S3: tbl_df/tbl/data.frame)
  ..@ gt_object           :List of 17
  .. ..- attr(*, "class")= chr [1:2] "gt_tbl" "list"
  ..@ gt_object_noHTML    :List of 17
  .. ..- attr(*, "class")= chr [1:2] "gt_tbl" "list"
  ..@ gene_modules        :List of 9
  ..@ pathway_clusters    :List of 9
  ..@ document_term_matrix:Formal class 'dgCMatrix' [package "Matrix"] with 6 slots
  ..@ clustering_tree     :List of 7
  .. ..- attr(*, "class")= chr "hclust"
  ..@ raw_enriched_result : tibble [3,003 × 9] (S3: tbl_df/tbl/data.frame)
  .. ..- attr(*, "Package")= chr "EnrichGT"
  .. ..- attr(*, "Input")= chr [1:465] "MT-ND4" "VAMP8" "PLAAT4" "LGALS1" ...
  .. ..- attr(*, "Database")='data.frame':  1284869 obs. of  2 variables:
  .. ..- attr(*, "Other_Params")=List of 5
  .. ..- attr(*, "Time")= POSIXct[1:1], format: "2025-11-23 02:56:33"
  ..@ fused               : logi FALSE
  ..@ param               :List of 6
  ..@ LLM_Annotation      :Formal class 'egt_llm' [package "EnrichGT"] with 3 slots
  ..@ LLM_Comparison      :Formal class 'egt_llm_comparison' [package "EnrichGT"] with 3 slots</code></pre>
</div>
</div>
<section id="access-re-enriched-data-table" class="level2">
<h2 class="anchored" data-anchor-id="access-re-enriched-data-table">Access re-enriched data table</h2>
<p>You can simple <code>View(re_enrich@enriched_result)</code> for the first slot.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb4"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb4-1"><a href="#cb4-1" aria-hidden="true" tabindex="-1"></a>re_enrich<span class="sc">@</span>enriched_result <span class="co"># Get the re-enrichment result table</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code># A tibble: 65 × 7
   Description                          ID    Count Cluster   PCT    Padj geneID
   &lt;chr&gt;                                &lt;chr&gt; &lt;int&gt; &lt;chr&gt;   &lt;dbl&gt;   &lt;dbl&gt; &lt;chr&gt; 
 1 synaptic transmission, glutamatergic GO:0…    18 Cluste…   3.9 1   e-6 ATP1A…
 2 regulation of synaptic transmission… GO:0…    15 Cluste…   3.2 1.20e-6 ATP1A…
 3 glutamate receptor signaling pathway GO:0…    12 Cluste…   2.6 1.10e-5 GRIA4…
 4 regulation of neuronal synaptic pla… GO:0…    12 Cluste…   2.6 1.20e-5 APOE,…
 5 ionotropic glutamate receptor signa… GO:0…     8 Cluste…   1.7 4.1 e-5 GRIA4…
 6 ligand-gated ion channel signaling … GO:1…     9 Cluste…   1.9 9.8 e-5 GRIA4…
 7 regulation of postsynaptic membrane… GO:0…    16 Cluste…   3.4 1.6 e-4 GABRD…
 8 regulation of membrane potential     GO:0…    30 Cluste…   6.5 2.5 e-4 AGT, …
 9 locomotory behavior                  GO:0…    19 Cluste…   4.1 2.5 e-4 ALK, …
10 transmission of nerve impulse        GO:0…    11 Cluste…   2.4 4.3 e-4 ATP1A…
# ℹ 55 more rows</code></pre>
</div>
</div>
</section>
<section id="access-re-enriched-html-report" class="level2">
<h2 class="anchored" data-anchor-id="access-re-enriched-html-report">Access re-enriched HTML report</h2>
<p>EnrichGT offers <strong>more than</strong> data frames. Please see <a href="./visualize.html">HTML reports (gt table table)</a> for further visualization.</p>
</section>
<section id="glance-details-of-each-module" class="level2">
<h2 class="anchored" data-anchor-id="glance-details-of-each-module">Glance details of each module</h2>
<section id="through-word-cloud" class="level3">
<h3 class="anchored" data-anchor-id="through-word-cloud">Through Word Cloud</h3>
<p><code>egt_fetch_biological_theme</code> function can do this. Just execute:</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb6"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb6-1"><a href="#cb6-1" aria-hidden="true" tabindex="-1"></a><span class="fu">egt_fetch_biological_theme</span>(re_enrich, <span class="dv">1</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="re_enrichment_files/figure-html/unnamed-chunk-6-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
</section>
<section id="through-morden-ides" class="level3">
<h3 class="anchored" data-anchor-id="through-morden-ides">Through Morden IDEs</h3>
<p>EnrichGT works well with IDEs <code>RStudio</code> and <code>Positron</code>. You can use <code>egt_summary()</code> function to see the LLM summaries.</p>
<p>For example, if you want to view <code>cluster 1</code>, you can use one of below:</p>
<div class="sourceCode" id="cb7"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb7-1"><a href="#cb7-1" aria-hidden="true" tabindex="-1"></a><span class="fu">egt_summary</span>(deepseekAnno, <span class="st">"1"</span>)</span>
<span id="cb7-2"><a href="#cb7-2" aria-hidden="true" tabindex="-1"></a><span class="fu">egt_summary</span>(deepseekAnno, <span class="dv">1</span>)</span>
<span id="cb7-3"><a href="#cb7-3" aria-hidden="true" tabindex="-1"></a><span class="fu">egt_summary</span>(deepseekAnno, <span class="st">"Cluster_1"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<p><img src="images/clipboard-657879404.png" class="img-fluid" width="800"></p>
</section>
<section id="through-r-console" class="level3">
<h3 class="anchored" data-anchor-id="through-r-console">Through R Console</h3>
<p>The re-enriched object from EnrichGT supports the <code>$</code> subset operator. You can use it to glance details inside each cluster. In morden IDE like <code>Positron</code>, type <code>$</code>, and then press the <code>Tab</code> key for auto-completion, as shown in the figure.</p>
<p><img src="images/paste-1.png" class="img-fluid" width="700"></p>
<p>But when you are still using <code>RStudio</code> (which auto-complete function is poor than <code>ARK</code> LSP), you can use <code>names</code> to get the cluster names.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb8"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb8-1"><a href="#cb8-1" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(re_enrich)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] "Cluster_1" "Cluster_2" "Cluster_3" "Cluster_4" "Cluster_5" "Cluster_6"
[7] "Cluster_7" "Cluster_8" "Cluster_9"</code></pre>
</div>
</div>
<p>In this example, all results haven’t got any extra annotation. But <code>EnrichGT</code> supports Large language models (LLMs) based enrichment result annotations, you can refer to <a href="./llm.html">large language models integration of EnrichGT</a> page for more details. After performed LLM annotations, this step will display more information and insights about enrichment results. If you feel typing full names is bored, you can use <code>c1</code> or <code>C1</code> and even <code>"1"</code> to access it.</p>
<p>For example:</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb10"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb10-1"><a href="#cb10-1" aria-hidden="true" tabindex="-1"></a>re_enrich<span class="sc">$</span>Cluster_1</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code></code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>── Enrichment Result of Cluster_1 (Local Summary) ──────────────────────────────</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>• This cluster contains synaptic transmission, glutamatergic, regulation of
synaptic transmission, glutamatergic, glutamate receptor signaling pathway,
regulation of neuronal synaptic plasticity, ionotropic glutamate receptor
signaling pathway ...</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>• Candidate genes includes ACHE, ADGRF1, AGT, ALK, ANKFN1, ANKH, APLP1, APOE,
ASPM, ATP1A2, ATP8B3, AVPR1A, BCAN, C4BPB, CA2, CACNG5, CAMK1D, CAMK2B, CDC20,
CELSR3, CIART, CNTN2, CNTNAP2, CPNE4, CPNE7, CTNNA2, DAAM2, DLX5, DNAJB1, DNER,
DPP4, DPYSL5, DSCAM, E2F1, EMX2, EPHA10, FOLR2, FOXD1, FSTL4, GABRD, GAD1,
GAP43, GFAP, GIPR, GPR37, GRIA4, GRID2, GRIK2, GRIK3, GRIN1, GRIN2A, GRIN2B,
GRIN2D, GRM1, GRM5, GRM8, HCN1, HCN2, HMOX1, HRAS, HTR3A, IGSF9, IL1RAPL2,
IMPA2, INA, KCNC3, KCNE4, KCNK2, KIAA1755, KIF1A, KIF5C, LHFPL4, LOXL2, LRFN2,
MAG, MAP1B, MAPK8IP2, MEGF10, MSX1, MT-CYB, MYO3B, NKX6-1, NLGN1, NMU, NQO1,
NRCAM, NRXN1, NTRK2, NUDT1, OPCML, PPP1R1B, PTPRH, PYCR1, RAC3, RASD2, RIMS3,
RNF207, RYR1, SCN8A, SDC1, SDC2, SEMA5B, SEZ6L2, SFRP2, SHISA9, SIX1, SLC30A10,
SLC30A3, SLC6A1, SLC6A3, SPOCK1, SRD5A1, STRA6, SYNDIG1, TBX18, TF, TFAP2A,
TREM2, TRPM2, TRPV4, TWIST1, UCHL1, UNC13A, UNC13C, VGF, WDR62, WNT4, WNT5A,
ZAN, ZIC1 (We will print all genes. Please stroll to top to read)</code></pre>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] "cli-33595-15"</code></pre>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb16"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb16-1"><a href="#cb16-1" aria-hidden="true" tabindex="-1"></a>re_enrich<span class="sc">$</span>c3</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code></code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>── Enrichment Result of Cluster_3 (Local Summary) ──────────────────────────────</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>• This cluster contains regulation of neuron differentiation, positive
regulation of nervous system development, regulation of nervous system
development, regulation of developmental growth, negative regulation of neuron
differentiation ...</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>• Candidate genes includes AGR2, ALK, APOE, ASPM, CAMK2B, CNTN2, DAAM2, DPYSL5,
DSCAM, FOXC2, FOXS1, FSTL4, GFAP, GPAM, GRID2, GRM5, HEY2, IL34, JAG1, KCNK2,
MAG, MAP1B, NKX6-1, NLGN1, NRCAM, NRXN1, NTRK2, RAC3, SFRP2, SIX1, SLC6A3,
SOX8, SYNDIG1, TP73, TREM2, UNC13A, WDR62, WNT5A, ZNF536 (We will print all
genes. Please stroll to top to read)</code></pre>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] "cli-33595-20"</code></pre>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb22"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb22-1"><a href="#cb22-1" aria-hidden="true" tabindex="-1"></a>re_enrich<span class="sc">$</span><span class="st">"5"</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code></code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>── Enrichment Result of Cluster_5 (Local Summary) ──────────────────────────────</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>• This cluster contains ATP synthesis coupled electron transport, mitochondrial
ATP synthesis coupled electron transport, aerobic electron transport chain,
aerobic respiration, respiratory electron transport chain ...</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>• Candidate genes includes ADGRF1, AK5, ALDH1L1, ANGPTL4, ATP1A2, ATP6V1C2,
AVPR1A, CA9, CCNB1, CD24, COL1A1, CRYAB, DPP4, ENO1, ENPP1, GIPR, HSPA1A, IDH1,
KCNK2, LOXL2, MB, MT-ATP6, MT-CO2, MT-CO3, MT-CYB, MT-ND1, MT-ND2, MT-ND3,
MT-ND4, NDUFB2, NDUFS6, OGDHL, PGF, RYR1, SLC15A1, TREM2, TRPV4, TWIST1, UCHL1,
UQCR10, VGF (We will print all genes. Please stroll to top to read)</code></pre>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] "cli-33595-25"</code></pre>
</div>
</div>
<p>Further more, you can can use <code>@</code> to get objects in S4. Like <code>result@gene_modules</code> returns genes in cluster.</p>
<div class="callout callout-style-default callout-tip callout-titled">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-title-container flex-fill">
How to get objects inside the S4 object?
</div>
</div>
<div class="callout-body-container callout-body">
<p>You can use <code>@</code>, for example, <code>x &lt;- re_enrich@enriched_result</code> returns a result table and <code>x &lt;- re_enrich@gt_object</code> returns a <code>gt</code> object.</p>
</div>
</div>
</section>
</section>
</section>
<section id="mask-unnecessary-results" class="level1">
<h1>Mask unnecessary results</h1>
<p>In transcriptomic sequencing, we often encounter a particular scenario. For example, T cell-related pathways can still be enriched in certain tumors from NOD-SCID mice, which are thymus-deficient. However, such immune infiltration patterns should not be predominant in thymus-deficient mice. Why does this happen? It’s likely due to shared immune response processes—such as interleukin and cytokine-related biological events—that are common across various immune cell types. What you’re seeing might be the surface reflection of a complex and chaotic underlying process.</p>
<p>Clearly, neither ORA nor GSEA is capable of correcting for such biases—let alone account for rare cell types or unique tissue microenvironments. In such cases, filtering out certain pathways isn’t falsification; it’s a way to minimize potential misunderstandings for readers or collaborators.</p>
<p>That said, I must emphasize: although this is a useful feature, please don’t use it to blindly dismiss pathways—existence implies relevance. It’s an art of trade-off. Nevertheless, every pathway with an FDR less than 0.05 deserves careful consideration.</p>
<p>In <code>EnrichGT</code>, you can simply use <code>%-delete-&gt;%</code> operator to achieve this:</p>
<div class="sourceCode" id="cb28"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb28-1"><a href="#cb28-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Filter out "ribosome" related terms in re-enriched object</span></span>
<span id="cb28-2"><a href="#cb28-2" aria-hidden="true" tabindex="-1"></a>filtered_results <span class="ot">&lt;-</span> reenrichment_obj <span class="sc">%-delete-&gt;%</span> <span class="st">"ribosome"</span></span>
<span id="cb28-3"><a href="#cb28-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-4"><a href="#cb28-4" aria-hidden="true" tabindex="-1"></a><span class="co"># Filter data.frame directly from ORA/GSEA result is also OK</span></span>
<span id="cb28-5"><a href="#cb28-5" aria-hidden="true" tabindex="-1"></a>filtered_df <span class="ot">&lt;-</span> df <span class="sc">%-delete-&gt;%</span> <span class="st">"metabolism"</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<p>It uses regular expression to help you remove them. Regular expression have many high-level ways to use, you can ask for Google for more details.</p>
</section>
<section id="manual-visualize-the-relationship-of-specific-terms-intra-database" class="level1">
<h1>Manual visualize the relationship of specific terms intra database</h1>
<p><code>egt_fetch_termwise_relationship</code> is designed for this. This directly call the cluster function internal and show you the result. The purpose of designing this function is to encourage you to explore more freely.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb29"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb29-1"><a href="#cb29-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(org.Hs.eg.db)</span>
<span id="cb29-2"><a href="#cb29-2" aria-hidden="true" tabindex="-1"></a><span class="fu">egt_fetch_termwise_relationship</span>(ora_result<span class="sc">$</span>Description[<span class="dv">1</span><span class="sc">:</span><span class="dv">100</span>],<span class="at">database =</span> <span class="fu">database_GO_BP</span>(org.Hs.eg.db),<span class="at">ClusterNum =</span> <span class="dv">5</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>✔ success loaded database, time used : 10.1783771514893 sec.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Joining with `by = join_by(label)`</code></pre>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="re_enrichment_files/figure-html/unnamed-chunk-11-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
</section>
<section id="infering-tfs-or-pathway-activity-and-more-based-on-meta-gene-modules" class="level1">
<h1>Infering TFs or pathway activity and more based on meta-gene modules</h1>
<p>Based on re-enriched result, the S4 object return from re-enrichment contains <code>gene_modules</code> slot and <code>pathway_clusters</code> slot. In <code>gene_modules</code> slot you can find a group of meta-genes take part in specific pathway cluster (in <code>pathway_clusters</code> slot).</p>
<p>EnrichGT supports inferring Pathway or Transcript Factors activity from re-enriched meta-gene modules. This is accomplished by two amazing database:</p>
<ul>
<li><p><a href="https://saezlab.github.io/progeny/">PROGENy</a> is a comprehensive resource containing a curated collection of pathways and their target genes, with weights for each interaction.</p></li>
<li><p><a href="https://github.com/saezlab/CollecTRI">CollecTRI</a> is a comprehensive resource containing a curated collection of TFs and their transcriptional targets compiled from 12 different resources. This collection provides an increased coverage of transcription factors and a superior performance in identifying perturbed TFs compared to our previous.</p></li>
</ul>
<p>Now let’s see this example:</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb32"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb32-1"><a href="#cb32-1" aria-hidden="true" tabindex="-1"></a>TF_Act <span class="ot">&lt;-</span> <span class="fu">egt_infer_act</span>(re_enrich,<span class="at">DB =</span> <span class="st">"collectri"</span>, <span class="at">species =</span> <span class="st">"human"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>! If when doing re-enrichment, you select a high number of clusters, that may cause low gene number in each meta-gene module, and then can't be infered sucessfully. So if result is empty, please increase the cut off of pvalue.  </code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>✔ success loaded self-contained database</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>✔ Done ORA in 0.0166840553283691 sec.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>✔ Done ORA in 0.0125110149383545 sec.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>✔ Done ORA in 0.0122601985931396 sec.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>✔ Done ORA in 0.016308069229126 sec.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>✔ Done ORA in 0.0119631290435791 sec.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>✔ Done ORA in 0.013624906539917 sec.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>✔ Done ORA in 0.0245451927185059 sec.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>✔ Done ORA in 0.0154321193695068 sec.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>✔ Done ORA in 0.0112709999084473 sec.</code></pre>
</div>
<div class="sourceCode cell-code" id="cb44"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb44-1"><a href="#cb44-1" aria-hidden="true" tabindex="-1"></a><span class="fu">egt_plot_results</span>(TF_Act<span class="sc">$</span>Cluster_1,<span class="at">P.adj =</span> <span class="fl">0.1</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>! You are drawing origin results, for better result you can re-cluster it by egt_recluster_analysis()</code></pre>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="re_enrichment_files/figure-html/unnamed-chunk-12-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
<div class="callout callout-style-default callout-tip callout-titled">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-title-container flex-fill">
Wants to interpret the regulator of whole inputted genes?
</div>
</div>
<div class="callout-body-container callout-body">
<p><code>PROGENy</code> and <code>CollecTRI</code> can be used just like other database in ORA or GSEA enrichment, for example, the <code>database_GO_BP()</code>. See [Progeny Database] and [CollecTRI Database] page for detail.</p>
<p>Example:</p>
<div class="sourceCode" id="cb46"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb46-1"><a href="#cb46-1" aria-hidden="true" tabindex="-1"></a>TFActivity <span class="ot">&lt;-</span> <span class="fu">egt_enrichment_analysis</span>(<span class="at">genes =</span> DEGtable<span class="sc">$</span>Genes,</span>
<span id="cb46-2"><a href="#cb46-2" aria-hidden="true" tabindex="-1"></a><span class="at">database =</span> <span class="fu">database_CollecTRI_human</span>())</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</div>
<div class="callout callout-style-default callout-important callout-titled">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-title-container flex-fill">
Why many inferred results are empty?
</div>
</div>
<div class="callout-body-container callout-body">
<p>If when doing re-enrichment with a high number of clusters, that may cause low gene number in each meta-gene module (splitting into too many clusters make gene in each cluster is not enough to enrich), and then can’t be inferred successfully. So if result is empty, please increase the number of re-clustering when doing it.</p>
</div>
</div>


</section>

<a onclick="window.scrollTo(0, 0); return false;" role="button" id="quarto-back-to-top"><i class="bi bi-arrow-up"></i> Back to top</a></main> <!-- /main -->
<script id="quarto-html-after-body" type="application/javascript">
  window.document.addEventListener("DOMContentLoaded", function (event) {
    const icon = "";
    const anchorJS = new window.AnchorJS();
    anchorJS.options = {
      placement: 'right',
      icon: icon
    };
    anchorJS.add('.anchored');
    const isCodeAnnotation = (el) => {
      for (const clz of el.classList) {
        if (clz.startsWith('code-annotation-')) {                     
          return true;
        }
      }
      return false;
    }
    const onCopySuccess = function(e) {
      // button target
      const button = e.trigger;
      // don't keep focus
      button.blur();
      // flash "checked"
      button.classList.add('code-copy-button-checked');
      var currentTitle = button.getAttribute("title");
      button.setAttribute("title", "Copied!");
      let tooltip;
      if (window.bootstrap) {
        button.setAttribute("data-bs-toggle", "tooltip");
        button.setAttribute("data-bs-placement", "left");
        button.setAttribute("data-bs-title", "Copied!");
        tooltip = new bootstrap.Tooltip(button, 
          { trigger: "manual", 
            customClass: "code-copy-button-tooltip",
            offset: [0, -8]});
        tooltip.show();    
      }
      setTimeout(function() {
        if (tooltip) {
          tooltip.hide();
          button.removeAttribute("data-bs-title");
          button.removeAttribute("data-bs-toggle");
          button.removeAttribute("data-bs-placement");
        }
        button.setAttribute("title", currentTitle);
        button.classList.remove('code-copy-button-checked');
      }, 1000);
      // clear code selection
      e.clearSelection();
    }
    const getTextToCopy = function(trigger) {
        const codeEl = trigger.previousElementSibling.cloneNode(true);
        for (const childEl of codeEl.children) {
          if (isCodeAnnotation(childEl)) {
            childEl.remove();
          }
        }
        return codeEl.innerText;
    }
    const clipboard = new window.ClipboardJS('.code-copy-button:not([data-in-quarto-modal])', {
      text: getTextToCopy
    });
    clipboard.on('success', onCopySuccess);
    if (window.document.getElementById('quarto-embedded-source-code-modal')) {
      const clipboardModal = new window.ClipboardJS('.code-copy-button[data-in-quarto-modal]', {
        text: getTextToCopy,
        container: window.document.getElementById('quarto-embedded-source-code-modal')
      });
      clipboardModal.on('success', onCopySuccess);
    }
      var localhostRegex = new RegExp(/^(?:http|https):\/\/localhost\:?[0-9]*\//);
      var mailtoRegex = new RegExp(/^mailto:/);
        var filterRegex = new RegExp('/' + window.location.host + '/');
      var isInternal = (href) => {
          return filterRegex.test(href) || localhostRegex.test(href) || mailtoRegex.test(href);
      }
      // Inspect non-navigation links and adorn them if external
     var links = window.document.querySelectorAll('a[href]:not(.nav-link):not(.navbar-brand):not(.toc-action):not(.sidebar-link):not(.sidebar-item-toggle):not(.pagination-link):not(.no-external):not([aria-hidden]):not(.dropdown-item):not(.quarto-navigation-tool):not(.about-link)');
      for (var i=0; i<links.length; i++) {
        const link = links[i];
        if (!isInternal(link.href)) {
          // undo the damage that might have been done by quarto-nav.js in the case of
          // links that we want to consider external
          if (link.dataset.originalHref !== undefined) {
            link.href = link.dataset.originalHref;
          }
        }
      }
    function tippyHover(el, contentFn, onTriggerFn, onUntriggerFn) {
      const config = {
        allowHTML: true,
        maxWidth: 500,
        delay: 100,
        arrow: false,
        appendTo: function(el) {
            return el.parentElement;
        },
        interactive: true,
        interactiveBorder: 10,
        theme: 'quarto',
        placement: 'bottom-start',
      };
      if (contentFn) {
        config.content = contentFn;
      }
      if (onTriggerFn) {
        config.onTrigger = onTriggerFn;
      }
      if (onUntriggerFn) {
        config.onUntrigger = onUntriggerFn;
      }
      window.tippy(el, config); 
    }
    const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
    for (var i=0; i<noterefs.length; i++) {
      const ref = noterefs[i];
      tippyHover(ref, function() {
        // use id or data attribute instead here
        let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
        try { href = new URL(href).hash; } catch {}
        const id = href.replace(/^#\/?/, "");
        const note = window.document.getElementById(id);
        if (note) {
          return note.innerHTML;
        } else {
          return "";
        }
      });
    }
    const xrefs = window.document.querySelectorAll('a.quarto-xref');
    const processXRef = (id, note) => {
      // Strip column container classes
      const stripColumnClz = (el) => {
        el.classList.remove("page-full", "page-columns");
        if (el.children) {
          for (const child of el.children) {
            stripColumnClz(child);
          }
        }
      }
      stripColumnClz(note)
      if (id === null || id.startsWith('sec-')) {
        // Special case sections, only their first couple elements
        const container = document.createElement("div");
        if (note.children && note.children.length > 2) {
          container.appendChild(note.children[0].cloneNode(true));
          for (let i = 1; i < note.children.length; i++) {
            const child = note.children[i];
            if (child.tagName === "P" && child.innerText === "") {
              continue;
            } else {
              container.appendChild(child.cloneNode(true));
              break;
            }
          }
          if (window.Quarto?.typesetMath) {
            window.Quarto.typesetMath(container);
          }
          return container.innerHTML
        } else {
          if (window.Quarto?.typesetMath) {
            window.Quarto.typesetMath(note);
          }
          return note.innerHTML;
        }
      } else {
        // Remove any anchor links if they are present
        const anchorLink = note.querySelector('a.anchorjs-link');
        if (anchorLink) {
          anchorLink.remove();
        }
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(note);
        }
        if (note.classList.contains("callout")) {
          return note.outerHTML;
        } else {
          return note.innerHTML;
        }
      }
    }
    for (var i=0; i<xrefs.length; i++) {
      const xref = xrefs[i];
      tippyHover(xref, undefined, function(instance) {
        instance.disable();
        let url = xref.getAttribute('href');
        let hash = undefined; 
        if (url.startsWith('#')) {
          hash = url;
        } else {
          try { hash = new URL(url).hash; } catch {}
        }
        if (hash) {
          const id = hash.replace(/^#\/?/, "");
          const note = window.document.getElementById(id);
          if (note !== null) {
            try {
              const html = processXRef(id, note.cloneNode(true));
              instance.setContent(html);
            } finally {
              instance.enable();
              instance.show();
            }
          } else {
            // See if we can fetch this
            fetch(url.split('#')[0])
            .then(res => res.text())
            .then(html => {
              const parser = new DOMParser();
              const htmlDoc = parser.parseFromString(html, "text/html");
              const note = htmlDoc.getElementById(id);
              if (note !== null) {
                const html = processXRef(id, note);
                instance.setContent(html);
              } 
            }).finally(() => {
              instance.enable();
              instance.show();
            });
          }
        } else {
          // See if we can fetch a full url (with no hash to target)
          // This is a special case and we should probably do some content thinning / targeting
          fetch(url)
          .then(res => res.text())
          .then(html => {
            const parser = new DOMParser();
            const htmlDoc = parser.parseFromString(html, "text/html");
            const note = htmlDoc.querySelector('main.content');
            if (note !== null) {
              // This should only happen for chapter cross references
              // (since there is no id in the URL)
              // remove the first header
              if (note.children.length > 0 && note.children[0].tagName === "HEADER") {
                note.children[0].remove();
              }
              const html = processXRef(null, note);
              instance.setContent(html);
            } 
          }).finally(() => {
            instance.enable();
            instance.show();
          });
        }
      }, function(instance) {
      });
    }
        let selectedAnnoteEl;
        const selectorForAnnotation = ( cell, annotation) => {
          let cellAttr = 'data-code-cell="' + cell + '"';
          let lineAttr = 'data-code-annotation="' +  annotation + '"';
          const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
          return selector;
        }
        const selectCodeLines = (annoteEl) => {
          const doc = window.document;
          const targetCell = annoteEl.getAttribute("data-target-cell");
          const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
          const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
          const lines = annoteSpan.getAttribute("data-code-lines").split(",");
          const lineIds = lines.map((line) => {
            return targetCell + "-" + line;
          })
          let top = null;
          let height = null;
          let parent = null;
          if (lineIds.length > 0) {
              //compute the position of the single el (top and bottom and make a div)
              const el = window.document.getElementById(lineIds[0]);
              top = el.offsetTop;
              height = el.offsetHeight;
              parent = el.parentElement.parentElement;
            if (lineIds.length > 1) {
              const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
              const bottom = lastEl.offsetTop + lastEl.offsetHeight;
              height = bottom - top;
            }
            if (top !== null && height !== null && parent !== null) {
              // cook up a div (if necessary) and position it 
              let div = window.document.getElementById("code-annotation-line-highlight");
              if (div === null) {
                div = window.document.createElement("div");
                div.setAttribute("id", "code-annotation-line-highlight");
                div.style.position = 'absolute';
                parent.appendChild(div);
              }
              div.style.top = top - 2 + "px";
              div.style.height = height + 4 + "px";
              div.style.left = 0;
              let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
              if (gutterDiv === null) {
                gutterDiv = window.document.createElement("div");
                gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
                gutterDiv.style.position = 'absolute';
                const codeCell = window.document.getElementById(targetCell);
                const gutter = codeCell.querySelector('.code-annotation-gutter');
                gutter.appendChild(gutterDiv);
              }
              gutterDiv.style.top = top - 2 + "px";
              gutterDiv.style.height = height + 4 + "px";
            }
            selectedAnnoteEl = annoteEl;
          }
        };
        const unselectCodeLines = () => {
          const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
          elementsIds.forEach((elId) => {
            const div = window.document.getElementById(elId);
            if (div) {
              div.remove();
            }
          });
          selectedAnnoteEl = undefined;
        };
          // Handle positioning of the toggle
      window.addEventListener(
        "resize",
        throttle(() => {
          elRect = undefined;
          if (selectedAnnoteEl) {
            selectCodeLines(selectedAnnoteEl);
          }
        }, 10)
      );
      function throttle(fn, ms) {
      let throttle = false;
      let timer;
        return (...args) => {
          if(!throttle) { // first call gets through
              fn.apply(this, args);
              throttle = true;
          } else { // all the others get throttled
              if(timer) clearTimeout(timer); // cancel #2
              timer = setTimeout(() => {
                fn.apply(this, args);
                timer = throttle = false;
              }, ms);
          }
        };
      }
        // Attach click handler to the DT
        const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
        for (const annoteDlNode of annoteDls) {
          annoteDlNode.addEventListener('click', (event) => {
            const clickedEl = event.target;
            if (clickedEl !== selectedAnnoteEl) {
              unselectCodeLines();
              const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
              if (activeEl) {
                activeEl.classList.remove('code-annotation-active');
              }
              selectCodeLines(clickedEl);
              clickedEl.classList.add('code-annotation-active');
            } else {
              // Unselect the line
              unselectCodeLines();
              clickedEl.classList.remove('code-annotation-active');
            }
          });
        }
    const findCites = (el) => {
      const parentEl = el.parentElement;
      if (parentEl) {
        const cites = parentEl.dataset.cites;
        if (cites) {
          return {
            el,
            cites: cites.split(' ')
          };
        } else {
          return findCites(el.parentElement)
        }
      } else {
        return undefined;
      }
    };
    var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
    for (var i=0; i<bibliorefs.length; i++) {
      const ref = bibliorefs[i];
      const citeInfo = findCites(ref);
      if (citeInfo) {
        tippyHover(citeInfo.el, function() {
          var popup = window.document.createElement('div');
          citeInfo.cites.forEach(function(cite) {
            var citeDiv = window.document.createElement('div');
            citeDiv.classList.add('hanging-indent');
            citeDiv.classList.add('csl-entry');
            var biblioDiv = window.document.getElementById('ref-' + cite);
            if (biblioDiv) {
              citeDiv.innerHTML = biblioDiv.innerHTML;
            }
            popup.appendChild(citeDiv);
          });
          return popup.innerHTML;
        });
      }
    }
  });
  </script>
</div> <!-- /content -->
<footer class="footer">
  <div class="nav-footer">
    <div class="nav-footer-left">
<p>EnrichGT</p>
</div>   
    <div class="nav-footer-center">
      &nbsp;
    <div class="toc-actions d-sm-block d-md-none"><ul><li><a href="https://github.com/ZhimingYe/EnrichGT_website/blob/main/re_enrichment.qmd" class="toc-action"><i class="bi bi-github"></i>View source</a></li><li><a href="https://github.com/ZhimingYe/EnrichGT_website/issues/new" class="toc-action"><i class="bi empty"></i>Report an issue</a></li></ul></div></div>
    <div class="nav-footer-right">
<p>Version 2.2</p>
</div>
  </div>
</footer>




</body></html>