<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.7.32">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">


<title>Visualize and Create Reports â€“ EnrichGT</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
/* CSS for syntax highlighting */
html { -webkit-text-size-adjust: 100%; }
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
  }
pre.numberSource { margin-left: 3em;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
</style>


<script src="site_libs/quarto-nav/quarto-nav.js"></script>
<script src="site_libs/quarto-nav/headroom.min.js"></script>
<script src="site_libs/clipboard/clipboard.min.js"></script>
<script src="site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="site_libs/quarto-search/fuse.min.js"></script>
<script src="site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="./">
<script src="site_libs/quarto-html/quarto.js" type="module"></script>
<script src="site_libs/quarto-html/tabsets/tabsets.js" type="module"></script>
<script src="site_libs/quarto-html/popper.min.js"></script>
<script src="site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="site_libs/quarto-html/anchor.min.js"></script>
<link href="site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="site_libs/quarto-html/quarto-syntax-highlighting-37eea08aefeeee20ff55810ff984fec1.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="site_libs/bootstrap/bootstrap.min.js"></script>
<link href="site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="site_libs/bootstrap/bootstrap-06d22339cf1bae60befde528b0c36036.min.css" rel="stylesheet" append-hash="true" id="quarto-bootstrap" data-mode="light">
<script id="quarto-search-options" type="application/json">{
  "location": "navbar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "end",
  "type": "overlay",
  "limit": 50,
  "keyboard-shortcut": [
    "f",
    "/",
    "s"
  ],
  "show-item-context": false,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-text-placeholder": "",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit",
    "search-label": "Search"
  }
}</script>


<link rel="stylesheet" href="styles.css">
</head>

<body class="nav-sidebar docked nav-fixed quarto-light">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top">
    <nav class="navbar navbar-expand-lg " data-bs-theme="dark">
      <div class="navbar-container container-fluid">
      <div class="navbar-brand-container mx-auto">
    <a href="./index.html" class="navbar-brand navbar-brand-logo">
    <img src="./EnrichGTlogo.png" alt="" class="navbar-logo">
    </a>
    <a class="navbar-brand" href="./index.html">
    <span class="navbar-title">EnrichGT</span>
    </a>
  </div>
            <div id="quarto-search" class="" title="Search"></div>
          <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbarCollapse" aria-controls="navbarCollapse" role="menu" aria-expanded="false" aria-label="Toggle navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
  <span class="navbar-toggler-icon"></span>
</button>
          <div class="collapse navbar-collapse" id="navbarCollapse">
            <ul class="navbar-nav navbar-nav-scroll me-auto">
  <li class="nav-item">
    <a class="nav-link active" href="./index.html" aria-current="page"> 
<span class="menu-text">Home</span></a>
  </li>  
  <li class="nav-item">
    <a class="nav-link" href="./pkgfuns.html"> 
<span class="menu-text">Package Functions</span></a>
  </li>  
  <li class="nav-item">
    <a class="nav-link" href="https://github.com/ZhimingYe/EnrichGT/"> 
<span class="menu-text">Github</span></a>
  </li>  
  <li class="nav-item">
    <a class="nav-link" href="./acknowledgement.html"> 
<span class="menu-text">Acknowledgement</span></a>
  </li>  
</ul>
          </div> <!-- /navcollapse -->
            <div class="quarto-navbar-tools">
</div>
      </div> <!-- /container-fluid -->
    </nav>
  <nav class="quarto-secondary-nav">
    <div class="container-fluid d-flex">
      <button type="button" class="quarto-btn-toggle btn" data-bs-toggle="collapse" role="button" data-bs-target=".quarto-sidebar-collapse-item" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
        <i class="bi bi-layout-text-sidebar-reverse"></i>
      </button>
        <nav class="quarto-page-breadcrumbs" aria-label="breadcrumb"><ol class="breadcrumb"><li class="breadcrumb-item"><a href="./visualize.html">Visualize and Create Reports</a></li></ol></nav>
        <a class="flex-grow-1" role="navigation" data-bs-toggle="collapse" data-bs-target=".quarto-sidebar-collapse-item" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">      
        </a>
    </div>
  </nav>
</header>
<!-- content -->
<div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-article page-navbar">
<!-- sidebar -->
  <nav id="quarto-sidebar" class="sidebar collapse collapse-horizontal quarto-sidebar-collapse-item sidebar-navigation docked overflow-auto">
    <div class="sidebar-menu-container"> 
    <ul class="list-unstyled mt-1">
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./index.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">EnrichGT: all-in-one enrichment analysis</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./basic.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Basic Workflow</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./interfaceshiny.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Web Interface</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./enrichment_core.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Core Enrichment Functions</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./re_enrichment.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Second enrichment of enriched Results</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./fusing.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Functions to FUSING results</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./visualize.html" class="sidebar-item-text sidebar-link active">
 <span class="menu-text">Visualize and Create Reports</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./llm.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Large language models integration</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./compare.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Compare Multiple Enrichment Results</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./database.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Select your databases for enrichment or annotation</span></a>
  </div>
</li>
    </ul>
    </div>
</nav>
<div id="quarto-sidebar-glass" class="quarto-sidebar-collapse-item" data-bs-toggle="collapse" data-bs-target=".quarto-sidebar-collapse-item"></div>
<!-- margin-sidebar -->
    <div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
        <nav id="TOC" role="doc-toc" class="toc-active" data-toc-expanded="99">
    <h2 id="toc-title">On this page</h2>
   
  <ul>
  <li><a href="#html-reports-gt-table" id="toc-html-reports-gt-table" class="nav-link active" data-scroll-target="#html-reports-gt-table">Dynamic document reports in HTML or Latex</a>
  <ul class="collapse">
  <li><a href="#full-result-inline-or-in-dynamic-documents" id="toc-full-result-inline-or-in-dynamic-documents" class="nav-link" data-scroll-target="#full-result-inline-or-in-dynamic-documents">Full Result: Inline or in dynamic documents</a></li>
  <li><a href="#each-cluster-html-widgets" id="toc-each-cluster-html-widgets" class="nav-link" data-scroll-target="#each-cluster-html-widgets">Each Cluster: HTML Widgets</a></li>
  <li><a href="#quarto-reports" id="toc-quarto-reports" class="nav-link" data-scroll-target="#quarto-reports">Quarto Reports</a></li>
  </ul></li>
  <li><a href="#ploting-functions" id="toc-ploting-functions" class="nav-link" data-scroll-target="#ploting-functions">Ploting functions</a>
  <ul class="collapse">
  <li><a href="#basic-result-plot" id="toc-basic-result-plot" class="nav-link" data-scroll-target="#basic-result-plot">Basic result plot</a></li>
  <li><a href="#additional-gsea-plots" id="toc-additional-gsea-plots" class="nav-link" data-scroll-target="#additional-gsea-plots">Additional GSEA plots</a>
  <ul class="collapse">
  <li><a href="#single-ranking-plot" id="toc-single-ranking-plot" class="nav-link" data-scroll-target="#single-ranking-plot">Single ranking plot</a></li>
  <li><a href="#table-like-visualization-gsea-result" id="toc-table-like-visualization-gsea-result" class="nav-link" data-scroll-target="#table-like-visualization-gsea-result">Table-like visualization GSEA result</a></li>
  </ul></li>
  <li><a href="#umap-plot-for-re-enrichment-no-longer-supported" id="toc-umap-plot-for-re-enrichment-no-longer-supported" class="nav-link" data-scroll-target="#umap-plot-for-re-enrichment-no-longer-supported">UMAP plot for re-enrichment (no longer supported)</a></li>
  </ul></li>
  </ul>
<div class="toc-actions"><ul><li><a href="https://github.com/ZhimingYe/EnrichGT_website/blob/main/visualize.qmd" class="toc-action"><i class="bi bi-github"></i>View source</a></li><li><a href="https://github.com/ZhimingYe/EnrichGT_website/issues/new" class="toc-action"><i class="bi empty"></i>Report an issue</a></li></ul></div></nav>
    </div>
<!-- main -->
<main class="content" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default">
<div class="quarto-title">
<h1 class="title">Visualize and Create Reports</h1>
</div>



<div class="quarto-title-meta">

    
  
    
  </div>
  


</header>


<section id="html-reports-gt-table" class="level1">
<h1>Dynamic document reports in HTML or Latex</h1>
<section id="full-result-inline-or-in-dynamic-documents" class="level2">
<h2 class="anchored" data-anchor-id="full-result-inline-or-in-dynamic-documents">Full Result: Inline or in dynamic documents</h2>
<p>Because of the messy result table is hardly to read, <code>EnrichGT</code> help you convert it into pretty <code>gt table</code> . This only supports the re-enriched results.</p>
<p>You can simple print by input the object name of re-enriched object or fused object to R console, to show the table. In IDE like <code>RStudio</code> and <code>Positron</code>, you can view the table in <code>View</code> window. You can also integrate to dynamic document publish systems like <code>rmarkdown</code> and <code>Quarto</code> which support HTML widget. Latex output is also supported.</p>
<p>To add the result to your dynamic document, simply call the second slot in your <code>rmarkdown</code> or <code>Quarto</code> document using <code>@gt_object</code>, like this :</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb1"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a>re_enrich<span class="sc">@</span>gt_object</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div id="eeenhhbogw" style="padding-left:0px;padding-right:0px;padding-top:10px;padding-bottom:10px;overflow-x:auto;overflow-y:auto;width:auto;height:auto;">
<style>#eeenhhbogw table {
  font-family: system-ui, 'Segoe UI', Roboto, Helvetica, Arial, sans-serif, 'Apple Color Emoji', 'Segoe UI Emoji', 'Segoe UI Symbol', 'Noto Color Emoji';
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
}

#eeenhhbogw thead, #eeenhhbogw tbody, #eeenhhbogw tfoot, #eeenhhbogw tr, #eeenhhbogw td, #eeenhhbogw th {
  border-style: none;
}

#eeenhhbogw p {
  margin: 0;
  padding: 0;
}

#eeenhhbogw .gt_table {
  display: table;
  border-collapse: collapse;
  line-height: normal;
  margin-left: auto;
  margin-right: auto;
  color: #333333;
  font-size: 16px;
  font-weight: normal;
  font-style: normal;
  background-color: #FFFFFF;
  width: auto;
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #A8A8A8;
  border-right-style: none;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #A8A8A8;
  border-left-style: none;
  border-left-width: 2px;
  border-left-color: #D3D3D3;
}

#eeenhhbogw .gt_caption {
  padding-top: 4px;
  padding-bottom: 4px;
}

#eeenhhbogw .gt_title {
  color: #333333;
  font-size: 125%;
  font-weight: initial;
  padding-top: 4px;
  padding-bottom: 4px;
  padding-left: 5px;
  padding-right: 5px;
  border-bottom-color: #FFFFFF;
  border-bottom-width: 0;
}

#eeenhhbogw .gt_subtitle {
  color: #333333;
  font-size: 85%;
  font-weight: initial;
  padding-top: 3px;
  padding-bottom: 5px;
  padding-left: 5px;
  padding-right: 5px;
  border-top-color: #FFFFFF;
  border-top-width: 0;
}

#eeenhhbogw .gt_heading {
  background-color: #FFFFFF;
  text-align: center;
  border-bottom-color: #FFFFFF;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
}

#eeenhhbogw .gt_bottom_border {
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
}

#eeenhhbogw .gt_col_headings {
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
}

#eeenhhbogw .gt_col_heading {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: normal;
  text-transform: inherit;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
  vertical-align: bottom;
  padding-top: 5px;
  padding-bottom: 6px;
  padding-left: 5px;
  padding-right: 5px;
  overflow-x: hidden;
}

#eeenhhbogw .gt_column_spanner_outer {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: normal;
  text-transform: inherit;
  padding-top: 0;
  padding-bottom: 0;
  padding-left: 4px;
  padding-right: 4px;
}

#eeenhhbogw .gt_column_spanner_outer:first-child {
  padding-left: 0;
}

#eeenhhbogw .gt_column_spanner_outer:last-child {
  padding-right: 0;
}

#eeenhhbogw .gt_column_spanner {
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  vertical-align: bottom;
  padding-top: 5px;
  padding-bottom: 5px;
  overflow-x: hidden;
  display: inline-block;
  width: 100%;
}

#eeenhhbogw .gt_spanner_row {
  border-bottom-style: hidden;
}

#eeenhhbogw .gt_group_heading {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  text-transform: inherit;
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
  vertical-align: middle;
  text-align: left;
}

#eeenhhbogw .gt_empty_group_heading {
  padding: 0.5px;
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  vertical-align: middle;
}

#eeenhhbogw .gt_from_md > :first-child {
  margin-top: 0;
}

#eeenhhbogw .gt_from_md > :last-child {
  margin-bottom: 0;
}

#eeenhhbogw .gt_row {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  margin: 10px;
  border-top-style: solid;
  border-top-width: 1px;
  border-top-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
  vertical-align: middle;
  overflow-x: hidden;
}

#eeenhhbogw .gt_stub {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  text-transform: inherit;
  border-right-style: solid;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
  padding-left: 5px;
  padding-right: 5px;
}

#eeenhhbogw .gt_stub_row_group {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  text-transform: inherit;
  border-right-style: solid;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
  padding-left: 5px;
  padding-right: 5px;
  vertical-align: top;
}

#eeenhhbogw .gt_row_group_first td {
  border-top-width: 2px;
}

#eeenhhbogw .gt_row_group_first th {
  border-top-width: 2px;
}

#eeenhhbogw .gt_summary_row {
  color: #333333;
  background-color: #FFFFFF;
  text-transform: inherit;
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
}

#eeenhhbogw .gt_first_summary_row {
  border-top-style: solid;
  border-top-color: #D3D3D3;
}

#eeenhhbogw .gt_first_summary_row.thick {
  border-top-width: 2px;
}

#eeenhhbogw .gt_last_summary_row {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
}

#eeenhhbogw .gt_grand_summary_row {
  color: #333333;
  background-color: #FFFFFF;
  text-transform: inherit;
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
}

#eeenhhbogw .gt_first_grand_summary_row {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  border-top-style: double;
  border-top-width: 6px;
  border-top-color: #D3D3D3;
}

#eeenhhbogw .gt_last_grand_summary_row_top {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  border-bottom-style: double;
  border-bottom-width: 6px;
  border-bottom-color: #D3D3D3;
}

#eeenhhbogw .gt_striped {
  background-color: rgba(128, 128, 128, 0.05);
}

#eeenhhbogw .gt_table_body {
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
}

#eeenhhbogw .gt_footnotes {
  color: #333333;
  background-color: #FFFFFF;
  border-bottom-style: none;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 2px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
}

#eeenhhbogw .gt_footnote {
  margin: 0px;
  font-size: 90%;
  padding-top: 4px;
  padding-bottom: 4px;
  padding-left: 5px;
  padding-right: 5px;
}

#eeenhhbogw .gt_sourcenotes {
  color: #333333;
  background-color: #FFFFFF;
  border-bottom-style: none;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 2px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
}

#eeenhhbogw .gt_sourcenote {
  font-size: 90%;
  padding-top: 4px;
  padding-bottom: 4px;
  padding-left: 5px;
  padding-right: 5px;
}

#eeenhhbogw .gt_left {
  text-align: left;
}

#eeenhhbogw .gt_center {
  text-align: center;
}

#eeenhhbogw .gt_right {
  text-align: right;
  font-variant-numeric: tabular-nums;
}

#eeenhhbogw .gt_font_normal {
  font-weight: normal;
}

#eeenhhbogw .gt_font_bold {
  font-weight: bold;
}

#eeenhhbogw .gt_font_italic {
  font-style: italic;
}

#eeenhhbogw .gt_super {
  font-size: 65%;
}

#eeenhhbogw .gt_footnote_marks {
  font-size: 75%;
  vertical-align: 0.4em;
  position: initial;
}

#eeenhhbogw .gt_asterisk {
  font-size: 100%;
  vertical-align: 0;
}

#eeenhhbogw .gt_indent_1 {
  text-indent: 5px;
}

#eeenhhbogw .gt_indent_2 {
  text-indent: 10px;
}

#eeenhhbogw .gt_indent_3 {
  text-indent: 15px;
}

#eeenhhbogw .gt_indent_4 {
  text-indent: 20px;
}

#eeenhhbogw .gt_indent_5 {
  text-indent: 25px;
}

#eeenhhbogw .katex-display {
  display: inline-flex !important;
  margin-bottom: 0.75em !important;
}

#eeenhhbogw div.Reactable > div.rt-table > div.rt-thead > div.rt-tr.rt-tr-group-header > div.rt-th-group:after {
  height: 0px !important;
}
</style>

<table class="gt_table caption-top table table-sm table-striped small" data-quarto-postprocess="true" data-quarto-disable-processing="false" data-quarto-bootstrap="false">
<colgroup>
<col style="width: 20%">
<col style="width: 20%">
<col style="width: 20%">
<col style="width: 20%">
<col style="width: 20%">
</colgroup>
<thead>
<tr class="gt_heading header">
<th colspan="5" class="gt_heading gt_title gt_font_normal">Parse form: ora_result</th>
</tr>
<tr class="gt_heading even">
<th colspan="5" class="gt_heading gt_subtitle gt_font_normal gt_bottom_border">Split into 10 Clusters. Generated by EnrichGT</th>
</tr>
<tr class="gt_col_headings header">
<th id="Description" class="gt_col_heading gt_columns_bottom_border gt_left" data-quarto-table-cell-role="th" scope="col">Description</th>
<th id="Count" class="gt_col_heading gt_columns_bottom_border gt_right" data-quarto-table-cell-role="th" scope="col">Count</th>
<th id="PCT" class="gt_col_heading gt_columns_bottom_border gt_right" data-quarto-table-cell-role="th" scope="col">PCT</th>
<th id="Padj" class="gt_col_heading gt_columns_bottom_border gt_right" data-quarto-table-cell-role="th" scope="col">Padj</th>
<th id="geneID" class="gt_col_heading gt_columns_bottom_border gt_left" data-quarto-table-cell-role="th" scope="col">geneID</th>
</tr>
</thead>
<tbody class="gt_table_body">
<tr class="gt_group_heading_row odd">
<td colspan="5" id="Cluster_1" class="gt_group_heading" data-quarto-table-cell-role="th" scope="colgroup">Cluster_1</td>
</tr>
<tr class="gt_row_group_first even">
<td class="gt_row gt_left" headers="Cluster_1  Description" style="font-size: 13px"><div style="line-height:14px">
<span class="smallcaps">synaptic transmission, glutamatergic</span>
</div>
<div style="line-height:10px">
<span style="font-weight:bold;color:#BEBEBE;font-size:10px">GO:0035249</span>
</div></td>
<td class="gt_row gt_right" headers="Cluster_1  Count" style="background-color: #91B6D7; color: #000000; font-size: 13px">18</td>
<td class="gt_row gt_right" headers="Cluster_1  PCT" style="background-color: #F5E8E4; color: #000000; font-size: 13px">3.9</td>
<td class="gt_row gt_right" headers="Cluster_1  Padj" style="background-color: #D53E4F; color: #FFFFFF; font-size: 13px">1.0e-06</td>
<td class="gt_row gt_left" headers="Cluster_1  geneID" style="font-size: 13px">ATP1A2, GRIA4, GRID2, GRIK2, GRIK3, GRIN1, GRIN2A, GRIN2B, GRIN2D, GRM1, GRM5, GRM8, NRXN1, NLGN1, UNC13A, MAPK8IP2, CACNG5, UNC13C</td>
</tr>
<tr class="odd">
<td class="gt_row gt_left" headers="Cluster_1  Description" style="font-size: 13px"><div style="line-height:14px">
<span class="smallcaps">regulation of synaptic transmission, glutamatergic</span>
</div>
<div style="line-height:10px">
<span style="font-weight:bold;color:#BEBEBE;font-size:10px">GO:0051966</span>
</div></td>
<td class="gt_row gt_right" headers="Cluster_1  Count" style="background-color: #5EA4CC; color: #FFFFFF; font-size: 13px">15</td>
<td class="gt_row gt_right" headers="Cluster_1  PCT" style="background-color: #DAF0BB; color: #000000; font-size: 13px">3.2</td>
<td class="gt_row gt_right" headers="Cluster_1  Padj" style="background-color: #D53E4F; color: #FFFFFF; font-size: 13px">1.2e-06</td>
<td class="gt_row gt_left" headers="Cluster_1  geneID" style="font-size: 13px">ATP1A2, GRIK2, GRIK3, GRIN1, GRIN2A, GRIN2B, GRIN2D, GRM1, GRM5, GRM8, NRXN1, NLGN1, UNC13A, MAPK8IP2, CACNG5</td>
</tr>
<tr class="even">
<td class="gt_row gt_left" headers="Cluster_1  Description" style="font-size: 13px"><div style="line-height:14px">
<span class="smallcaps">glutamate receptor signaling pathway</span>
</div>
<div style="line-height:10px">
<span style="font-weight:bold;color:#BEBEBE;font-size:10px">GO:0007215</span>
</div></td>
<td class="gt_row gt_right" headers="Cluster_1  Count" style="background-color: #358FBE; color: #FFFFFF; font-size: 13px">12</td>
<td class="gt_row gt_right" headers="Cluster_1  PCT" style="background-color: #B5DF82; color: #000000; font-size: 13px">2.6</td>
<td class="gt_row gt_right" headers="Cluster_1  Padj" style="background-color: #D53E4F; color: #FFFFFF; font-size: 13px">1.1e-05</td>
<td class="gt_row gt_left" headers="Cluster_1  geneID" style="font-size: 13px">GRIA4, GRID2, GRIK2, GRIK3, GRIN1, GRIN2A, GRIN2B, GRIN2D, GRM1, GRM5, GRM8, NRXN1</td>
</tr>
<tr class="odd">
<td class="gt_row gt_left" headers="Cluster_1  Description" style="font-size: 13px"><div style="line-height:14px">
<span class="smallcaps">regulation of neuronal synaptic plasticity</span>
</div>
<div style="line-height:10px">
<span style="font-weight:bold;color:#BEBEBE;font-size:10px">GO:0048168</span>
</div></td>
<td class="gt_row gt_right" headers="Cluster_1  Count" style="background-color: #358FBE; color: #FFFFFF; font-size: 13px">12</td>
<td class="gt_row gt_right" headers="Cluster_1  PCT" style="background-color: #B5DF82; color: #000000; font-size: 13px">2.6</td>
<td class="gt_row gt_right" headers="Cluster_1  Padj" style="background-color: #D53E4F; color: #FFFFFF; font-size: 13px">1.2e-05</td>
<td class="gt_row gt_left" headers="Cluster_1  geneID" style="font-size: 13px">APOE, CAMK2B, GRIK2, GRIN1, GRIN2A, GRIN2B, GRIN2D, GRM5, HRAS, CNTN2, VGF, SHISA9</td>
</tr>
<tr class="even">
<td class="gt_row gt_left" headers="Cluster_1  Description" style="font-size: 13px"><div style="line-height:14px">
<span class="smallcaps">ionotropic glutamate receptor signaling pathway</span>
</div>
<div style="line-height:10px">
<span style="font-weight:bold;color:#BEBEBE;font-size:10px">GO:0035235</span>
</div></td>
<td class="gt_row gt_right" headers="Cluster_1  Count" style="background-color: #047C80; color: #FFFFFF; font-size: 13px">8</td>
<td class="gt_row gt_right" headers="Cluster_1  PCT" style="background-color: #74B33A; color: #FFFFFF; font-size: 13px">1.7</td>
<td class="gt_row gt_right" headers="Cluster_1  Padj" style="background-color: #D53E4F; color: #FFFFFF; font-size: 13px">4.1e-05</td>
<td class="gt_row gt_left" headers="Cluster_1  geneID" style="font-size: 13px">GRIA4, GRIK2, GRIK3, GRIN1, GRIN2A, GRIN2B, GRIN2D, NRXN1</td>
</tr>
<tr class="odd">
<td class="gt_row gt_left" headers="Cluster_1  Description" style="font-size: 13px"><div style="line-height:14px">
<span class="smallcaps">ligand-gated ion channel signaling pathway</span>
</div>
<div style="line-height:10px">
<span style="font-weight:bold;color:#BEBEBE;font-size:10px">GO:1990806</span>
</div></td>
<td class="gt_row gt_right" headers="Cluster_1  Count" style="background-color: #0E8390; color: #FFFFFF; font-size: 13px">9</td>
<td class="gt_row gt_right" headers="Cluster_1  PCT" style="background-color: #81BD44; color: #000000; font-size: 13px">1.9</td>
<td class="gt_row gt_right" headers="Cluster_1  Padj" style="background-color: #D53F4F; color: #FFFFFF; font-size: 13px">9.8e-05</td>
<td class="gt_row gt_left" headers="Cluster_1  geneID" style="font-size: 13px">GRIA4, GRIK2, GRIK3, GRIN1, GRIN2A, GRIN2B, GRIN2D, HTR3A, NRXN1</td>
</tr>
<tr class="even">
<td class="gt_row gt_left" headers="Cluster_1  Description" style="font-size: 13px"><div style="line-height:14px">
<span class="smallcaps">regulation of postsynaptic membrane potential</span>
</div>
<div style="line-height:10px">
<span style="font-weight:bold;color:#BEBEBE;font-size:10px">GO:0060078</span>
</div></td>
<td class="gt_row gt_right" headers="Cluster_1  Count" style="background-color: #6DABD0; color: #000000; font-size: 13px">16</td>
<td class="gt_row gt_right" headers="Cluster_1  PCT" style="background-color: #E5F5CF; color: #000000; font-size: 13px">3.4</td>
<td class="gt_row gt_right" headers="Cluster_1  Padj" style="background-color: #D63F4F; color: #FFFFFF; font-size: 13px">1.6e-04</td>
<td class="gt_row gt_left" headers="Cluster_1  geneID" style="font-size: 13px">GABRD, GRIA4, GRID2, GRIK2, GRIK3, GRIN1, GRIN2A, GRIN2B, GRIN2D, GRM1, GRM5, HTR3A, NRXN1, NLGN1, MAPK8IP2, HCN1</td>
</tr>
<tr class="odd">
<td class="gt_row gt_left" headers="Cluster_1  Description" style="font-size: 13px"><div style="line-height:14px">
<span class="smallcaps">regulation of membrane potential</span>
</div>
<div style="line-height:10px">
<span style="font-weight:bold;color:#BEBEBE;font-size:10px">GO:0042391</span>
</div></td>
<td class="gt_row gt_right" headers="Cluster_1  Count" style="background-color: #FFF7FB; color: #000000; font-size: 13px">30</td>
<td class="gt_row gt_right" headers="Cluster_1  PCT" style="background-color: #C51B7D; color: #FFFFFF; font-size: 13px">6.5</td>
<td class="gt_row gt_right" headers="Cluster_1  Padj" style="background-color: #D6404F; color: #FFFFFF; font-size: 13px">2.5e-04</td>
<td class="gt_row gt_left" headers="Cluster_1  geneID" style="font-size: 13px">AGT, ATP1A2, HCN2, GABRD, GRIA4, GRID2, GRIK2, GRIK3, GRIN1, GRIN2A, GRIN2B, GRIN2D, GRM1, GRM5, HTR3A, KCNC3, KCNK2, NRCAM, NTRK2, PYCR1, SCN8A, TBX18, NRXN1, RIMS3, NLGN1, MAPK8IP2, KCNE4, TREM2, HCN1, RNF207</td>
</tr>
<tr class="even">
<td class="gt_row gt_left" headers="Cluster_1  Description" style="font-size: 13px"><div style="line-height:14px">
<span class="smallcaps">locomotory behavior</span>
</div>
<div style="line-height:10px">
<span style="font-weight:bold;color:#BEBEBE;font-size:10px">GO:0007626</span>
</div></td>
<td class="gt_row gt_right" headers="Cluster_1  Count" style="background-color: #A1BBDA; color: #000000; font-size: 13px">19</td>
<td class="gt_row gt_right" headers="Cluster_1  PCT" style="background-color: #FBE2EC; color: #000000; font-size: 13px">4.1</td>
<td class="gt_row gt_right" headers="Cluster_1  Padj" style="background-color: #D6404F; color: #FFFFFF; font-size: 13px">2.5e-04</td>
<td class="gt_row gt_left" headers="Cluster_1  geneID" style="font-size: 13px">ALK, APOE, ATP1A2, DPP4, DSCAM, GAD1, GPR37, GRIN2D, GRM1, GRM5, SLC6A3, CNTN2, UCHL1, ZIC1, RASD2, ANKH, PPP1R1B, CIART, ANKFN1</td>
</tr>
<tr class="odd">
<td class="gt_row gt_left" headers="Cluster_1  Description" style="font-size: 13px"><div style="line-height:14px">
<span class="smallcaps">transmission of nerve impulse</span>
</div>
<div style="line-height:10px">
<span style="font-weight:bold;color:#BEBEBE;font-size:10px">GO:0019226</span>
</div></td>
<td class="gt_row gt_right" headers="Cluster_1  Count" style="background-color: #2B8BAE; color: #FFFFFF; font-size: 13px">11</td>
<td class="gt_row gt_right" headers="Cluster_1  PCT" style="background-color: #A6D571; color: #000000; font-size: 13px">2.4</td>
<td class="gt_row gt_right" headers="Cluster_1  Padj" style="background-color: #D7414E; color: #FFFFFF; font-size: 13px">4.3e-04</td>
<td class="gt_row gt_left" headers="Cluster_1  geneID" style="font-size: 13px">ATP1A2, AVPR1A, GRIK2, KCNK2, MAG, NRCAM, NTRK2, SCN8A, CNTNAP2, CACNG5, HCN1</td>
</tr>
<tr class="gt_group_heading_row even">
<td colspan="5" id="Cluster_2" class="gt_group_heading" data-quarto-table-cell-role="th" scope="colgroup">Cluster_2</td>
</tr>
<tr class="gt_row_group_first odd">
<td class="gt_row gt_left" headers="Cluster_2  Description" style="font-size: 13px"><div style="line-height:14px">
<span class="smallcaps">mitotic spindle assembly checkpoint signaling</span>
</div>
<div style="line-height:10px">
<span style="font-weight:bold;color:#BEBEBE;font-size:10px">GO:0007094</span>
</div></td>
<td class="gt_row gt_right" headers="Cluster_2  Count" style="background-color: #2B8BAE; color: #FFFFFF; font-size: 13px">11</td>
<td class="gt_row gt_right" headers="Cluster_2  PCT" style="background-color: #A6D571; color: #000000; font-size: 13px">2.4</td>
<td class="gt_row gt_right" headers="Cluster_2  Padj" style="background-color: #D53E4F; color: #FFFFFF; font-size: 13px">1.2e-05</td>
<td class="gt_row gt_left" headers="Cluster_2  geneID" style="font-size: 13px">BIRC5, BUB1, CCNB1, CDC20, CENPF, PLK1, NDC80, ZWINT, CDCA8, NUF2, SPC24</td>
</tr>
<tr class="even">
<td class="gt_row gt_left" headers="Cluster_2  Description" style="font-size: 13px"><div style="line-height:14px">
<span class="smallcaps">mitotic spindle checkpoint signaling</span>
</div>
<div style="line-height:10px">
<span style="font-weight:bold;color:#BEBEBE;font-size:10px">GO:0071174</span>
</div></td>
<td class="gt_row gt_right" headers="Cluster_2  Count" style="background-color: #2B8BAE; color: #FFFFFF; font-size: 13px">11</td>
<td class="gt_row gt_right" headers="Cluster_2  PCT" style="background-color: #A6D571; color: #000000; font-size: 13px">2.4</td>
<td class="gt_row gt_right" headers="Cluster_2  Padj" style="background-color: #D53E4F; color: #FFFFFF; font-size: 13px">1.2e-05</td>
<td class="gt_row gt_left" headers="Cluster_2  geneID" style="font-size: 13px">BIRC5, BUB1, CCNB1, CDC20, CENPF, PLK1, NDC80, ZWINT, CDCA8, NUF2, SPC24</td>
</tr>
<tr class="odd">
<td class="gt_row gt_left" headers="Cluster_2  Description" style="font-size: 13px"><div style="line-height:14px">
<span class="smallcaps">spindle assembly checkpoint signaling</span>
</div>
<div style="line-height:10px">
<span style="font-weight:bold;color:#BEBEBE;font-size:10px">GO:0071173</span>
</div></td>
<td class="gt_row gt_right" headers="Cluster_2  Count" style="background-color: #2B8BAE; color: #FFFFFF; font-size: 13px">11</td>
<td class="gt_row gt_right" headers="Cluster_2  PCT" style="background-color: #A6D571; color: #000000; font-size: 13px">2.4</td>
<td class="gt_row gt_right" headers="Cluster_2  Padj" style="background-color: #D53E4F; color: #FFFFFF; font-size: 13px">1.2e-05</td>
<td class="gt_row gt_left" headers="Cluster_2  geneID" style="font-size: 13px">BIRC5, BUB1, CCNB1, CDC20, CENPF, PLK1, NDC80, ZWINT, CDCA8, NUF2, SPC24</td>
</tr>
<tr class="even">
<td class="gt_row gt_left" headers="Cluster_2  Description" style="font-size: 13px"><div style="line-height:14px">
<span class="smallcaps">spindle checkpoint signaling</span>
</div>
<div style="line-height:10px">
<span style="font-weight:bold;color:#BEBEBE;font-size:10px">GO:0031577</span>
</div></td>
<td class="gt_row gt_right" headers="Cluster_2  Count" style="background-color: #2B8BAE; color: #FFFFFF; font-size: 13px">11</td>
<td class="gt_row gt_right" headers="Cluster_2  PCT" style="background-color: #A6D571; color: #000000; font-size: 13px">2.4</td>
<td class="gt_row gt_right" headers="Cluster_2  Padj" style="background-color: #D53E4F; color: #FFFFFF; font-size: 13px">1.2e-05</td>
<td class="gt_row gt_left" headers="Cluster_2  geneID" style="font-size: 13px">BIRC5, BUB1, CCNB1, CDC20, CENPF, PLK1, NDC80, ZWINT, CDCA8, NUF2, SPC24</td>
</tr>
<tr class="odd">
<td class="gt_row gt_left" headers="Cluster_2  Description" style="font-size: 13px"><div style="line-height:14px">
<span class="smallcaps">regulation of mitotic sister chromatid separation</span>
</div>
<div style="line-height:10px">
<span style="font-weight:bold;color:#BEBEBE;font-size:10px">GO:0010965</span>
</div></td>
<td class="gt_row gt_right" headers="Cluster_2  Count" style="background-color: #358FBE; color: #FFFFFF; font-size: 13px">12</td>
<td class="gt_row gt_right" headers="Cluster_2  PCT" style="background-color: #B5DF82; color: #000000; font-size: 13px">2.6</td>
<td class="gt_row gt_right" headers="Cluster_2  Padj" style="background-color: #D53E4F; color: #FFFFFF; font-size: 13px">1.2e-05</td>
<td class="gt_row gt_left" headers="Cluster_2  geneID" style="font-size: 13px">BIRC5, BUB1, CCNB1, CDC20, CENPF, PLK1, NDC80, UBE2C, ZWINT, CDCA8, NUF2, SPC24</td>
</tr>
<tr class="even">
<td class="gt_row gt_left" headers="Cluster_2  Description" style="font-size: 13px"><div style="line-height:14px">
<span class="smallcaps">negative regulation of mitotic metaphase/anaphase transition</span>
</div>
<div style="line-height:10px">
<span style="font-weight:bold;color:#BEBEBE;font-size:10px">GO:0045841</span>
</div></td>
<td class="gt_row gt_right" headers="Cluster_2  Count" style="background-color: #2B8BAE; color: #FFFFFF; font-size: 13px">11</td>
<td class="gt_row gt_right" headers="Cluster_2  PCT" style="background-color: #A6D571; color: #000000; font-size: 13px">2.4</td>
<td class="gt_row gt_right" headers="Cluster_2  Padj" style="background-color: #D53E4F; color: #FFFFFF; font-size: 13px">1.2e-05</td>
<td class="gt_row gt_left" headers="Cluster_2  geneID" style="font-size: 13px">BIRC5, BUB1, CCNB1, CDC20, CENPF, PLK1, NDC80, ZWINT, CDCA8, NUF2, SPC24</td>
</tr>
<tr class="odd">
<td class="gt_row gt_left" headers="Cluster_2  Description" style="font-size: 13px"><div style="line-height:14px">
<span class="smallcaps">negative regulation of mitotic sister chromatid segregation</span>
</div>
<div style="line-height:10px">
<span style="font-weight:bold;color:#BEBEBE;font-size:10px">GO:0033048</span>
</div></td>
<td class="gt_row gt_right" headers="Cluster_2  Count" style="background-color: #2B8BAE; color: #FFFFFF; font-size: 13px">11</td>
<td class="gt_row gt_right" headers="Cluster_2  PCT" style="background-color: #A6D571; color: #000000; font-size: 13px">2.4</td>
<td class="gt_row gt_right" headers="Cluster_2  Padj" style="background-color: #D53E4F; color: #FFFFFF; font-size: 13px">1.2e-05</td>
<td class="gt_row gt_left" headers="Cluster_2  geneID" style="font-size: 13px">BIRC5, BUB1, CCNB1, CDC20, CENPF, PLK1, NDC80, ZWINT, CDCA8, NUF2, SPC24</td>
</tr>
<tr class="even">
<td class="gt_row gt_left" headers="Cluster_2  Description" style="font-size: 13px"><div style="line-height:14px">
<span class="smallcaps">negative regulation of mitotic sister chromatid separation</span>
</div>
<div style="line-height:10px">
<span style="font-weight:bold;color:#BEBEBE;font-size:10px">GO:2000816</span>
</div></td>
<td class="gt_row gt_right" headers="Cluster_2  Count" style="background-color: #2B8BAE; color: #FFFFFF; font-size: 13px">11</td>
<td class="gt_row gt_right" headers="Cluster_2  PCT" style="background-color: #A6D571; color: #000000; font-size: 13px">2.4</td>
<td class="gt_row gt_right" headers="Cluster_2  Padj" style="background-color: #D53E4F; color: #FFFFFF; font-size: 13px">1.2e-05</td>
<td class="gt_row gt_left" headers="Cluster_2  geneID" style="font-size: 13px">BIRC5, BUB1, CCNB1, CDC20, CENPF, PLK1, NDC80, ZWINT, CDCA8, NUF2, SPC24</td>
</tr>
<tr class="odd">
<td class="gt_row gt_left" headers="Cluster_2  Description" style="font-size: 13px"><div style="line-height:14px">
<span class="smallcaps">negative regulation of sister chromatid segregation</span>
</div>
<div style="line-height:10px">
<span style="font-weight:bold;color:#BEBEBE;font-size:10px">GO:0033046</span>
</div></td>
<td class="gt_row gt_right" headers="Cluster_2  Count" style="background-color: #2B8BAE; color: #FFFFFF; font-size: 13px">11</td>
<td class="gt_row gt_right" headers="Cluster_2  PCT" style="background-color: #A6D571; color: #000000; font-size: 13px">2.4</td>
<td class="gt_row gt_right" headers="Cluster_2  Padj" style="background-color: #D53E4F; color: #FFFFFF; font-size: 13px">1.2e-05</td>
<td class="gt_row gt_left" headers="Cluster_2  geneID" style="font-size: 13px">BIRC5, BUB1, CCNB1, CDC20, CENPF, PLK1, NDC80, ZWINT, CDCA8, NUF2, SPC24</td>
</tr>
<tr class="even">
<td class="gt_row gt_left" headers="Cluster_2  Description" style="font-size: 13px"><div style="line-height:14px">
<span class="smallcaps">mitotic sister chromatid separation</span>
</div>
<div style="line-height:10px">
<span style="font-weight:bold;color:#BEBEBE;font-size:10px">GO:0051306</span>
</div></td>
<td class="gt_row gt_right" headers="Cluster_2  Count" style="background-color: #358FBE; color: #FFFFFF; font-size: 13px">12</td>
<td class="gt_row gt_right" headers="Cluster_2  PCT" style="background-color: #B5DF82; color: #000000; font-size: 13px">2.6</td>
<td class="gt_row gt_right" headers="Cluster_2  Padj" style="background-color: #D53E4F; color: #FFFFFF; font-size: 13px">1.5e-05</td>
<td class="gt_row gt_left" headers="Cluster_2  geneID" style="font-size: 13px">BIRC5, BUB1, CCNB1, CDC20, CENPF, PLK1, NDC80, UBE2C, ZWINT, CDCA8, NUF2, SPC24</td>
</tr>
<tr class="gt_group_heading_row odd">
<td colspan="5" id="Cluster_3" class="gt_group_heading" data-quarto-table-cell-role="th" scope="colgroup">Cluster_3</td>
</tr>
<tr class="gt_row_group_first even">
<td class="gt_row gt_left" headers="Cluster_3  Description" style="font-size: 13px"><div style="line-height:14px">
<span class="smallcaps">regulation of neuron differentiation</span>
</div>
<div style="line-height:10px">
<span style="font-weight:bold;color:#BEBEBE;font-size:10px">GO:0045664</span>
</div></td>
<td class="gt_row gt_right" headers="Cluster_3  Count" style="background-color: #80B0D3; color: #000000; font-size: 13px">17</td>
<td class="gt_row gt_right" headers="Cluster_3  PCT" style="background-color: #EFEDDC; color: #000000; font-size: 13px">3.7</td>
<td class="gt_row gt_right" headers="Cluster_3  Padj" style="background-color: #D9444E; color: #FFFFFF; font-size: 13px">8.4e-04</td>
<td class="gt_row gt_left" headers="Cluster_3  geneID" style="font-size: 13px">JAG1, ALK, MAG, MAP1B, NKX6-1, NRCAM, RAC3, SFRP2, SIX1, CNTN2, TP73, ZNF536, NLGN1, HEY2, SOX8, ASPM, WDR62</td>
</tr>
<tr class="odd">
<td class="gt_row gt_left" headers="Cluster_3  Description" style="font-size: 13px"><div style="line-height:14px">
<span class="smallcaps">positive regulation of nervous system development</span>
</div>
<div style="line-height:10px">
<span style="font-weight:bold;color:#BEBEBE;font-size:10px">GO:0051962</span>
</div></td>
<td class="gt_row gt_right" headers="Cluster_3  Count" style="background-color: #80B0D3; color: #000000; font-size: 13px">17</td>
<td class="gt_row gt_right" headers="Cluster_3  PCT" style="background-color: #EFEDDC; color: #000000; font-size: 13px">3.7</td>
<td class="gt_row gt_right" headers="Cluster_3  Padj" style="background-color: #EFEE93; color: #000000; font-size: 13px">2.5e-02</td>
<td class="gt_row gt_left" headers="Cluster_3  geneID" style="font-size: 13px">CAMK2B, DSCAM, GFAP, GRID2, GRM5, MAG, MAP1B, NKX6-1, NTRK2, TP73, NRXN1, NLGN1, SOX8, SYNDIG1, IL34, ASPM, WDR62</td>
</tr>
<tr class="even">
<td class="gt_row gt_left" headers="Cluster_3  Description" style="font-size: 13px"><div style="line-height:14px">
<span class="smallcaps">regulation of nervous system development</span>
</div>
<div style="line-height:10px">
<span style="font-weight:bold;color:#BEBEBE;font-size:10px">GO:0051960</span>
</div></td>
<td class="gt_row gt_right" headers="Cluster_3  Count" style="background-color: #D1D2E6; color: #000000; font-size: 13px">23</td>
<td class="gt_row gt_right" headers="Cluster_3  PCT" style="background-color: #F2B9DC; color: #000000; font-size: 13px">4.9</td>
<td class="gt_row gt_right" headers="Cluster_3  Padj" style="background-color: #D9EF9B; color: #000000; font-size: 13px">2.9e-02</td>
<td class="gt_row gt_left" headers="Cluster_3  geneID" style="font-size: 13px">CAMK2B, DSCAM, GFAP, GRID2, GRM5, MAG, MAP1B, NKX6-1, NTRK2, TP73, WNT5A, NRXN1, NLGN1, FSTL4, HEY2, DAAM2, SOX8, TREM2, DPYSL5, SYNDIG1, IL34, ASPM, WDR62</td>
</tr>
<tr class="odd">
<td class="gt_row gt_left" headers="Cluster_3  Description" style="font-size: 13px"><div style="line-height:14px">
<span class="smallcaps">regulation of developmental growth</span>
</div>
<div style="line-height:10px">
<span style="font-weight:bold;color:#BEBEBE;font-size:10px">GO:0048638</span>
</div></td>
<td class="gt_row gt_right" headers="Cluster_3  Count" style="background-color: #91B6D7; color: #000000; font-size: 13px">18</td>
<td class="gt_row gt_right" headers="Cluster_3  PCT" style="background-color: #F5E8E4; color: #000000; font-size: 13px">3.9</td>
<td class="gt_row gt_right" headers="Cluster_3  Padj" style="background-color: #9BD6A4; color: #000000; font-size: 13px">3.6e-02</td>
<td class="gt_row gt_left" headers="Cluster_3  geneID" style="font-size: 13px">APOE, DSCAM, FOXC2, FOXS1, KCNK2, MAG, MAP1B, NKX6-1, NRCAM, SIX1, SLC6A3, TP73, WNT5A, AGR2, UNC13A, FSTL4, HEY2, GPAM</td>
</tr>
<tr class="even">
<td class="gt_row gt_left" headers="Cluster_3  Description" style="font-size: 13px"><div style="line-height:14px">
<span class="smallcaps">negative regulation of neuron differentiation</span>
</div>
<div style="line-height:10px">
<span style="font-weight:bold;color:#BEBEBE;font-size:10px">GO:0045665</span>
</div></td>
<td class="gt_row gt_right" headers="Cluster_3  Count" style="background-color: #057470; color: #FFFFFF; font-size: 13px">7</td>
<td class="gt_row gt_right" headers="Cluster_3  PCT" style="background-color: #67A832; color: #FFFFFF; font-size: 13px">1.5</td>
<td class="gt_row gt_right" headers="Cluster_3  Padj" style="background-color: #50A1B3; color: #FFFFFF; font-size: 13px">4.5e-02</td>
<td class="gt_row gt_left" headers="Cluster_3  geneID" style="font-size: 13px">JAG1, MAG, CNTN2, TP73, ZNF536, SOX8, ASPM</td>
</tr>
<tr class="odd">
<td class="gt_row gt_left" headers="Cluster_3  Description" style="font-size: 13px"><div style="line-height:14px">
<span class="smallcaps">regulation of neurogenesis</span>
</div>
<div style="line-height:10px">
<span style="font-weight:bold;color:#BEBEBE;font-size:10px">GO:0050767</span>
</div></td>
<td class="gt_row gt_right" headers="Cluster_3  Count" style="background-color: #A1BBDA; color: #000000; font-size: 13px">19</td>
<td class="gt_row gt_right" headers="Cluster_3  PCT" style="background-color: #FBE2EC; color: #000000; font-size: 13px">4.1</td>
<td class="gt_row gt_right" headers="Cluster_3  Padj" style="background-color: #3288BD; color: #FFFFFF; font-size: 13px">4.8e-02</td>
<td class="gt_row gt_left" headers="Cluster_3  geneID" style="font-size: 13px">CAMK2B, DSCAM, GFAP, GRM5, MAG, MAP1B, NKX6-1, NTRK2, TP73, WNT5A, FSTL4, HEY2, DAAM2, SOX8, TREM2, DPYSL5, IL34, ASPM, WDR62</td>
</tr>
<tr class="gt_group_heading_row even">
<td colspan="5" id="Cluster_4" class="gt_group_heading" data-quarto-table-cell-role="th" scope="colgroup">Cluster_4</td>
</tr>
<tr class="gt_row_group_first odd">
<td class="gt_row gt_left" headers="Cluster_4  Description" style="font-size: 13px"><div style="line-height:14px">
<span class="smallcaps">behavioral fear response</span>
</div>
<div style="line-height:10px">
<span style="font-weight:bold;color:#BEBEBE;font-size:10px">GO:0001662</span>
</div></td>
<td class="gt_row gt_right" headers="Cluster_4  Count" style="background-color: #047C80; color: #FFFFFF; font-size: 13px">8</td>
<td class="gt_row gt_right" headers="Cluster_4  PCT" style="background-color: #74B33A; color: #FFFFFF; font-size: 13px">1.7</td>
<td class="gt_row gt_right" headers="Cluster_4  Padj" style="background-color: #D9454E; color: #FFFFFF; font-size: 13px">9.7e-04</td>
<td class="gt_row gt_left" headers="Cluster_4  geneID" style="font-size: 13px">APOE, ATP1A2, DPP4, GRIK2, NPAS2, MAPK8IP2, LYPD1, ANKFN1</td>
</tr>
<tr class="even">
<td class="gt_row gt_left" headers="Cluster_4  Description" style="font-size: 13px"><div style="line-height:14px">
<span class="smallcaps">behavioral defense response</span>
</div>
<div style="line-height:10px">
<span style="font-weight:bold;color:#BEBEBE;font-size:10px">GO:0002209</span>
</div></td>
<td class="gt_row gt_right" headers="Cluster_4  Count" style="background-color: #047C80; color: #FFFFFF; font-size: 13px">8</td>
<td class="gt_row gt_right" headers="Cluster_4  PCT" style="background-color: #74B33A; color: #FFFFFF; font-size: 13px">1.7</td>
<td class="gt_row gt_right" headers="Cluster_4  Padj" style="background-color: #DA464D; color: #FFFFFF; font-size: 13px">1.1e-03</td>
<td class="gt_row gt_left" headers="Cluster_4  geneID" style="font-size: 13px">APOE, ATP1A2, DPP4, GRIK2, NPAS2, MAPK8IP2, LYPD1, ANKFN1</td>
</tr>
<tr class="odd">
<td class="gt_row gt_left" headers="Cluster_4  Description" style="font-size: 13px"><div style="line-height:14px">
<span class="smallcaps">fear response</span>
</div>
<div style="line-height:10px">
<span style="font-weight:bold;color:#BEBEBE;font-size:10px">GO:0042596</span>
</div></td>
<td class="gt_row gt_right" headers="Cluster_4  Count" style="background-color: #047C80; color: #FFFFFF; font-size: 13px">8</td>
<td class="gt_row gt_right" headers="Cluster_4  PCT" style="background-color: #74B33A; color: #FFFFFF; font-size: 13px">1.7</td>
<td class="gt_row gt_right" headers="Cluster_4  Padj" style="background-color: #DF4E4C; color: #FFFFFF; font-size: 13px">2.2e-03</td>
<td class="gt_row gt_left" headers="Cluster_4  geneID" style="font-size: 13px">APOE, ATP1A2, DPP4, GRIK2, NPAS2, MAPK8IP2, LYPD1, ANKFN1</td>
</tr>
<tr class="even">
<td class="gt_row gt_left" headers="Cluster_4  Description" style="font-size: 13px"><div style="line-height:14px">
<span class="smallcaps">startle response</span>
</div>
<div style="line-height:10px">
<span style="font-weight:bold;color:#BEBEBE;font-size:10px">GO:0001964</span>
</div></td>
<td class="gt_row gt_right" headers="Cluster_4  Count" style="background-color: #046C60; color: #FFFFFF; font-size: 13px">6</td>
<td class="gt_row gt_right" headers="Cluster_4  PCT" style="background-color: #5A9D29; color: #FFFFFF; font-size: 13px">1.3</td>
<td class="gt_row gt_right" headers="Cluster_4  Padj" style="background-color: #E75A49; color: #FFFFFF; font-size: 13px">4.0e-03</td>
<td class="gt_row gt_left" headers="Cluster_4  geneID" style="font-size: 13px">CTNNA2, GRID2, GRIN2A, GRIN2D, SLC6A3, CNTNAP2</td>
</tr>
<tr class="odd">
<td class="gt_row gt_left" headers="Cluster_4  Description" style="font-size: 13px"><div style="line-height:14px">
<span class="smallcaps">protein localization to synapse</span>
</div>
<div style="line-height:10px">
<span style="font-weight:bold;color:#BEBEBE;font-size:10px">GO:0035418</span>
</div></td>
<td class="gt_row gt_right" headers="Cluster_4  Count" style="background-color: #1F879F; color: #FFFFFF; font-size: 13px">10</td>
<td class="gt_row gt_right" headers="Cluster_4  PCT" style="background-color: #98CC5F; color: #000000; font-size: 13px">2.2</td>
<td class="gt_row gt_right" headers="Cluster_4  Padj" style="background-color: #EA5E48; color: #FFFFFF; font-size: 13px">4.6e-03</td>
<td class="gt_row gt_left" headers="Cluster_4  geneID" style="font-size: 13px">GRIN2A, HRAS, HSPB1, KIF5C, WNT5A, NRXN1, NLGN1, GRIP1, GRIP2, LHFPL4</td>
</tr>
<tr class="even">
<td class="gt_row gt_left" headers="Cluster_4  Description" style="font-size: 13px"><div style="line-height:14px">
<span class="smallcaps">neuromuscular process</span>
</div>
<div style="line-height:10px">
<span style="font-weight:bold;color:#BEBEBE;font-size:10px">GO:0050905</span>
</div></td>
<td class="gt_row gt_right" headers="Cluster_4  Count" style="background-color: #529DC8; color: #FFFFFF; font-size: 13px">14</td>
<td class="gt_row gt_right" headers="Cluster_4  PCT" style="background-color: #CEEAA8; color: #000000; font-size: 13px">3.0</td>
<td class="gt_row gt_right" headers="Cluster_4  Padj" style="background-color: #F57245; color: #FFFFFF; font-size: 13px">7.3e-03</td>
<td class="gt_row gt_left" headers="Cluster_4  geneID" style="font-size: 13px">CTNNA2, FOXS1, GRID2, GRIN2A, GRIN2D, RAC3, SLC6A3, TNNT1, UCHL1, NRXN1, CNTNAP2, STRA6, ANKFN1, STAC2</td>
</tr>
<tr class="odd">
<td class="gt_row gt_left" headers="Cluster_4  Description" style="font-size: 13px"><div style="line-height:14px">
<span class="smallcaps">multicellular organismal response to stress</span>
</div>
<div style="line-height:10px">
<span style="font-weight:bold;color:#BEBEBE;font-size:10px">GO:0033555</span>
</div></td>
<td class="gt_row gt_right" headers="Cluster_4  Count" style="background-color: #1F879F; color: #FFFFFF; font-size: 13px">10</td>
<td class="gt_row gt_right" headers="Cluster_4  PCT" style="background-color: #98CC5F; color: #000000; font-size: 13px">2.2</td>
<td class="gt_row gt_right" headers="Cluster_4  Padj" style="background-color: #F67847; color: #FFFFFF; font-size: 13px">7.9e-03</td>
<td class="gt_row gt_left" headers="Cluster_4  geneID" style="font-size: 13px">APOE, ATP1A2, DPP4, GRIK2, NPAS2, THBS4, MAPK8IP2, LYPD1, ANKFN1, HCN1</td>
</tr>
<tr class="even">
<td class="gt_row gt_left" headers="Cluster_4  Description" style="font-size: 13px"><div style="line-height:14px">
<span class="smallcaps">gas transport</span>
</div>
<div style="line-height:10px">
<span style="font-weight:bold;color:#BEBEBE;font-size:10px">GO:0015669</span>
</div></td>
<td class="gt_row gt_right" headers="Cluster_4  Count" style="background-color: #016450; color: #FFFFFF; font-size: 13px">5</td>
<td class="gt_row gt_right" headers="Cluster_4  PCT" style="background-color: #4D9221; color: #FFFFFF; font-size: 13px">1.1</td>
<td class="gt_row gt_right" headers="Cluster_4  Padj" style="background-color: #F67B49; color: #FFFFFF; font-size: 13px">8.2e-03</td>
<td class="gt_row gt_left" headers="Cluster_4  geneID" style="font-size: 13px">CA2, HBA1, HBA2, HBB, MB</td>
</tr>
<tr class="odd">
<td class="gt_row gt_left" headers="Cluster_4  Description" style="font-size: 13px"><div style="line-height:14px">
<span class="smallcaps">cellular oxidant detoxification</span>
</div>
<div style="line-height:10px">
<span style="font-weight:bold;color:#BEBEBE;font-size:10px">GO:0098869</span>
</div></td>
<td class="gt_row gt_right" headers="Cluster_4  Count" style="background-color: #1F879F; color: #FFFFFF; font-size: 13px">10</td>
<td class="gt_row gt_right" headers="Cluster_4  PCT" style="background-color: #98CC5F; color: #000000; font-size: 13px">2.2</td>
<td class="gt_row gt_right" headers="Cluster_4  Padj" style="background-color: #F67B49; color: #FFFFFF; font-size: 13px">8.2e-03</td>
<td class="gt_row gt_left" headers="Cluster_4  geneID" style="font-size: 13px">APOE, CP, NQO1, GPX1, HBA1, HBA2, HBB, MB, MGST1, TXNDC17</td>
</tr>
<tr class="even">
<td class="gt_row gt_left" headers="Cluster_4  Description" style="font-size: 13px"><div style="line-height:14px">
<span class="smallcaps">response to hydrogen peroxide</span>
</div>
<div style="line-height:10px">
<span style="font-weight:bold;color:#BEBEBE;font-size:10px">GO:0042542</span>
</div></td>
<td class="gt_row gt_right" headers="Cluster_4  Count" style="background-color: #1F879F; color: #FFFFFF; font-size: 13px">10</td>
<td class="gt_row gt_right" headers="Cluster_4  PCT" style="background-color: #98CC5F; color: #000000; font-size: 13px">2.2</td>
<td class="gt_row gt_right" headers="Cluster_4  Padj" style="background-color: #F67B49; color: #FFFFFF; font-size: 13px">8.2e-03</td>
<td class="gt_row gt_left" headers="Cluster_4  geneID" style="font-size: 13px">COL1A1, CRYAB, NQO1, GPX1, HBA1, HBA2, HBB, HMOX1, SDC1, TRPM2</td>
</tr>
<tr class="gt_group_heading_row odd">
<td colspan="5" id="Cluster_5" class="gt_group_heading" data-quarto-table-cell-role="th" scope="colgroup">Cluster_5</td>
</tr>
<tr class="gt_row_group_first even">
<td class="gt_row gt_left" headers="Cluster_5  Description" style="font-size: 13px"><div style="line-height:14px">
<span class="smallcaps">ATP synthesis coupled electron transport</span>
</div>
<div style="line-height:10px">
<span style="font-weight:bold;color:#BEBEBE;font-size:10px">GO:0042773</span>
</div></td>
<td class="gt_row gt_right" headers="Cluster_5  Count" style="background-color: #2B8BAE; color: #FFFFFF; font-size: 13px">11</td>
<td class="gt_row gt_right" headers="Cluster_5  PCT" style="background-color: #A6D571; color: #000000; font-size: 13px">2.4</td>
<td class="gt_row gt_right" headers="Cluster_5  Padj" style="background-color: #E4564A; color: #FFFFFF; font-size: 13px">3.3e-03</td>
<td class="gt_row gt_left" headers="Cluster_5  geneID" style="font-size: 13px">CCNB1, MT-CO2, MT-CO3, MT-CYB, MT-ND1, MT-ND2, MT-ND3, MT-ND4, NDUFB2, NDUFS6, UQCR10</td>
</tr>
<tr class="odd">
<td class="gt_row gt_left" headers="Cluster_5  Description" style="font-size: 13px"><div style="line-height:14px">
<span class="smallcaps">mitochondrial ATP synthesis coupled electron transport</span>
</div>
<div style="line-height:10px">
<span style="font-weight:bold;color:#BEBEBE;font-size:10px">GO:0042775</span>
</div></td>
<td class="gt_row gt_right" headers="Cluster_5  Count" style="background-color: #2B8BAE; color: #FFFFFF; font-size: 13px">11</td>
<td class="gt_row gt_right" headers="Cluster_5  PCT" style="background-color: #A6D571; color: #000000; font-size: 13px">2.4</td>
<td class="gt_row gt_right" headers="Cluster_5  Padj" style="background-color: #E4564A; color: #FFFFFF; font-size: 13px">3.3e-03</td>
<td class="gt_row gt_left" headers="Cluster_5  geneID" style="font-size: 13px">CCNB1, MT-CO2, MT-CO3, MT-CYB, MT-ND1, MT-ND2, MT-ND3, MT-ND4, NDUFB2, NDUFS6, UQCR10</td>
</tr>
<tr class="even">
<td class="gt_row gt_left" headers="Cluster_5  Description" style="font-size: 13px"><div style="line-height:14px">
<span class="smallcaps">aerobic electron transport chain</span>
</div>
<div style="line-height:10px">
<span style="font-weight:bold;color:#BEBEBE;font-size:10px">GO:0019646</span>
</div></td>
<td class="gt_row gt_right" headers="Cluster_5  Count" style="background-color: #1F879F; color: #FFFFFF; font-size: 13px">10</td>
<td class="gt_row gt_right" headers="Cluster_5  PCT" style="background-color: #98CC5F; color: #000000; font-size: 13px">2.2</td>
<td class="gt_row gt_right" headers="Cluster_5  Padj" style="background-color: #F26944; color: #FFFFFF; font-size: 13px">6.3e-03</td>
<td class="gt_row gt_left" headers="Cluster_5  geneID" style="font-size: 13px">MT-CO2, MT-CO3, MT-CYB, MT-ND1, MT-ND2, MT-ND3, MT-ND4, NDUFB2, NDUFS6, UQCR10</td>
</tr>
<tr class="odd">
<td class="gt_row gt_left" headers="Cluster_5  Description" style="font-size: 13px"><div style="line-height:14px">
<span class="smallcaps">aerobic respiration</span>
</div>
<div style="line-height:10px">
<span style="font-weight:bold;color:#BEBEBE;font-size:10px">GO:0009060</span>
</div></td>
<td class="gt_row gt_right" headers="Cluster_5  Count" style="background-color: #5EA4CC; color: #FFFFFF; font-size: 13px">15</td>
<td class="gt_row gt_right" headers="Cluster_5  PCT" style="background-color: #DAF0BB; color: #000000; font-size: 13px">3.2</td>
<td class="gt_row gt_right" headers="Cluster_5  Padj" style="background-color: #F67A48; color: #FFFFFF; font-size: 13px">8.1e-03</td>
<td class="gt_row gt_left" headers="Cluster_5  geneID" style="font-size: 13px">CCNB1, IDH1, MT-ATP6, MT-CO2, MT-CO3, MT-CYB, MT-ND1, MT-ND2, MT-ND3, MT-ND4, NDUFB2, NDUFS6, UQCR10, OGDHL, TRPV4</td>
</tr>
<tr class="even">
<td class="gt_row gt_left" headers="Cluster_5  Description" style="font-size: 13px"><div style="line-height:14px">
<span class="smallcaps">respiratory electron transport chain</span>
</div>
<div style="line-height:10px">
<span style="font-weight:bold;color:#BEBEBE;font-size:10px">GO:0022904</span>
</div></td>
<td class="gt_row gt_right" headers="Cluster_5  Count" style="background-color: #2B8BAE; color: #FFFFFF; font-size: 13px">11</td>
<td class="gt_row gt_right" headers="Cluster_5  PCT" style="background-color: #A6D571; color: #000000; font-size: 13px">2.4</td>
<td class="gt_row gt_right" headers="Cluster_5  Padj" style="background-color: #F67A48; color: #FFFFFF; font-size: 13px">8.1e-03</td>
<td class="gt_row gt_left" headers="Cluster_5  geneID" style="font-size: 13px">CCNB1, MT-CO2, MT-CO3, MT-CYB, MT-ND1, MT-ND2, MT-ND3, MT-ND4, NDUFB2, NDUFS6, UQCR10</td>
</tr>
<tr class="odd">
<td class="gt_row gt_left" headers="Cluster_5  Description" style="font-size: 13px"><div style="line-height:14px">
<span class="smallcaps">response to oxygen levels</span>
</div>
<div style="line-height:10px">
<span style="font-weight:bold;color:#BEBEBE;font-size:10px">GO:0070482</span>
</div></td>
<td class="gt_row gt_right" headers="Cluster_5  Count" style="background-color: #BAC7E0; color: #000000; font-size: 13px">21</td>
<td class="gt_row gt_right" headers="Cluster_5  PCT" style="background-color: #F8CFE6; color: #000000; font-size: 13px">4.5</td>
<td class="gt_row gt_right" headers="Cluster_5  Padj" style="background-color: #F98C50; color: #000000; font-size: 13px">1.0e-02</td>
<td class="gt_row gt_left" headers="Cluster_5  geneID" style="font-size: 13px">CA9, COL1A1, CRYAB, DPP4, ENO1, KCNK2, LOXL2, MB, MT-ATP6, MT-CO2, MT-CYB, MT-ND1, MT-ND2, MT-ND4, PGF, RYR1, TWIST1, ANGPTL4, TREM2, TRPV4, CD24</td>
</tr>
<tr class="even">
<td class="gt_row gt_left" headers="Cluster_5  Description" style="font-size: 13px"><div style="line-height:14px">
<span class="smallcaps">proton transmembrane transport</span>
</div>
<div style="line-height:10px">
<span style="font-weight:bold;color:#BEBEBE;font-size:10px">GO:1902600</span>
</div></td>
<td class="gt_row gt_right" headers="Cluster_5  Count" style="background-color: #529DC8; color: #FFFFFF; font-size: 13px">14</td>
<td class="gt_row gt_right" headers="Cluster_5  PCT" style="background-color: #CEEAA8; color: #000000; font-size: 13px">3.0</td>
<td class="gt_row gt_right" headers="Cluster_5  Padj" style="background-color: #FB9F59; color: #000000; font-size: 13px">1.2e-02</td>
<td class="gt_row gt_left" headers="Cluster_5  geneID" style="font-size: 13px">ATP1A2, MT-ATP6, MT-CO2, MT-CO3, MT-CYB, MT-ND1, MT-ND2, MT-ND3, MT-ND4, NDUFB2, NDUFS6, SLC15A1, UQCR10, ATP6V1C2</td>
</tr>
<tr class="odd">
<td class="gt_row gt_left" headers="Cluster_5  Description" style="font-size: 13px"><div style="line-height:14px">
<span class="smallcaps">electron transport chain</span>
</div>
<div style="line-height:10px">
<span style="font-weight:bold;color:#BEBEBE;font-size:10px">GO:0022900</span>
</div></td>
<td class="gt_row gt_right" headers="Cluster_5  Count" style="background-color: #2B8BAE; color: #FFFFFF; font-size: 13px">11</td>
<td class="gt_row gt_right" headers="Cluster_5  PCT" style="background-color: #A6D571; color: #000000; font-size: 13px">2.4</td>
<td class="gt_row gt_right" headers="Cluster_5  Padj" style="background-color: #FCA85E; color: #000000; font-size: 13px">1.3e-02</td>
<td class="gt_row gt_left" headers="Cluster_5  geneID" style="font-size: 13px">CCNB1, MT-CO2, MT-CO3, MT-CYB, MT-ND1, MT-ND2, MT-ND3, MT-ND4, NDUFB2, NDUFS6, UQCR10</td>
</tr>
<tr class="even">
<td class="gt_row gt_left" headers="Cluster_5  Description" style="font-size: 13px"><div style="line-height:14px">
<span class="smallcaps">response to hypoxia</span>
</div>
<div style="line-height:10px">
<span style="font-weight:bold;color:#BEBEBE;font-size:10px">GO:0001666</span>
</div></td>
<td class="gt_row gt_right" headers="Cluster_5  Count" style="background-color: #A1BBDA; color: #000000; font-size: 13px">19</td>
<td class="gt_row gt_right" headers="Cluster_5  PCT" style="background-color: #FBE2EC; color: #000000; font-size: 13px">4.1</td>
<td class="gt_row gt_right" headers="Cluster_5  Padj" style="background-color: #FDB063; color: #000000; font-size: 13px">1.4e-02</td>
<td class="gt_row gt_left" headers="Cluster_5  geneID" style="font-size: 13px">CA9, CRYAB, DPP4, ENO1, KCNK2, LOXL2, MB, MT-CO2, MT-CYB, MT-ND1, MT-ND2, MT-ND4, PGF, RYR1, TWIST1, ANGPTL4, TREM2, TRPV4, CD24</td>
</tr>
<tr class="odd">
<td class="gt_row gt_left" headers="Cluster_5  Description" style="font-size: 13px"><div style="line-height:14px">
<span class="smallcaps">oxidative phosphorylation</span>
</div>
<div style="line-height:10px">
<span style="font-weight:bold;color:#BEBEBE;font-size:10px">GO:0006119</span>
</div></td>
<td class="gt_row gt_right" headers="Cluster_5  Count" style="background-color: #358FBE; color: #FFFFFF; font-size: 13px">12</td>
<td class="gt_row gt_right" headers="Cluster_5  PCT" style="background-color: #B5DF82; color: #000000; font-size: 13px">2.6</td>
<td class="gt_row gt_right" headers="Cluster_5  Padj" style="background-color: #FEB769; color: #000000; font-size: 13px">1.5e-02</td>
<td class="gt_row gt_left" headers="Cluster_5  geneID" style="font-size: 13px">CCNB1, MT-ATP6, MT-CO2, MT-CO3, MT-CYB, MT-ND1, MT-ND2, MT-ND3, MT-ND4, NDUFB2, NDUFS6, UQCR10</td>
</tr>
<tr class="gt_group_heading_row even">
<td colspan="5" id="Cluster_6" class="gt_group_heading" data-quarto-table-cell-role="th" scope="colgroup">Cluster_6</td>
</tr>
<tr class="gt_row_group_first odd">
<td class="gt_row gt_left" headers="Cluster_6  Description" style="font-size: 13px"><div style="line-height:14px">
<span class="smallcaps">ear development</span>
</div>
<div style="line-height:10px">
<span style="font-weight:bold;color:#BEBEBE;font-size:10px">GO:0043583</span>
</div></td>
<td class="gt_row gt_right" headers="Cluster_6  Count" style="background-color: #80B0D3; color: #000000; font-size: 13px">17</td>
<td class="gt_row gt_right" headers="Cluster_6  PCT" style="background-color: #EFEDDC; color: #000000; font-size: 13px">3.7</td>
<td class="gt_row gt_right" headers="Cluster_6  Padj" style="background-color: #EB6047; color: #FFFFFF; font-size: 13px">4.9e-03</td>
<td class="gt_row gt_left" headers="Cluster_6  geneID" style="font-size: 13px">JAG1, COL11A1, GSDME, DLX5, KCNK2, MSX1, SIX1, TFAP2A, TWIST1, WNT5A, ZIC1, TBX18, SIX2, HEY2, STRA6, CTHRC1, MYO3B</td>
</tr>
<tr class="even">
<td class="gt_row gt_left" headers="Cluster_6  Description" style="font-size: 13px"><div style="line-height:14px">
<span class="smallcaps">sensory organ morphogenesis</span>
</div>
<div style="line-height:10px">
<span style="font-weight:bold;color:#BEBEBE;font-size:10px">GO:0090596</span>
</div></td>
<td class="gt_row gt_right" headers="Cluster_6  Count" style="background-color: #A1BBDA; color: #000000; font-size: 13px">19</td>
<td class="gt_row gt_right" headers="Cluster_6  PCT" style="background-color: #FBE2EC; color: #000000; font-size: 13px">4.1</td>
<td class="gt_row gt_right" headers="Cluster_6  Padj" style="background-color: #F26944; color: #FFFFFF; font-size: 13px">6.3e-03</td>
<td class="gt_row gt_left" headers="Cluster_6  geneID" style="font-size: 13px">JAG1, COL11A1, DLX5, DSCAM, MSX1, NTRK2, SIX1, TFAP2A, TWIST1, WNT5A, ZIC1, TBX18, SIX2, FJX1, SOX8, STRA6, CTHRC1, MYO3B, HCN1</td>
</tr>
<tr class="odd">
<td class="gt_row gt_left" headers="Cluster_6  Description" style="font-size: 13px"><div style="line-height:14px">
<span class="smallcaps">ear morphogenesis</span>
</div>
<div style="line-height:10px">
<span style="font-weight:bold;color:#BEBEBE;font-size:10px">GO:0042471</span>
</div></td>
<td class="gt_row gt_right" headers="Cluster_6  Count" style="background-color: #358FBE; color: #FFFFFF; font-size: 13px">12</td>
<td class="gt_row gt_right" headers="Cluster_6  PCT" style="background-color: #B5DF82; color: #000000; font-size: 13px">2.6</td>
<td class="gt_row gt_right" headers="Cluster_6  Padj" style="background-color: #F26944; color: #FFFFFF; font-size: 13px">6.3e-03</td>
<td class="gt_row gt_left" headers="Cluster_6  geneID" style="font-size: 13px">COL11A1, DLX5, MSX1, SIX1, TFAP2A, TWIST1, WNT5A, ZIC1, TBX18, SIX2, CTHRC1, MYO3B</td>
</tr>
<tr class="even">
<td class="gt_row gt_left" headers="Cluster_6  Description" style="font-size: 13px"><div style="line-height:14px">
<span class="smallcaps">regulation of animal organ morphogenesis</span>
</div>
<div style="line-height:10px">
<span style="font-weight:bold;color:#BEBEBE;font-size:10px">GO:2000027</span>
</div></td>
<td class="gt_row gt_right" headers="Cluster_6  Count" style="background-color: #0E8390; color: #FFFFFF; font-size: 13px">9</td>
<td class="gt_row gt_right" headers="Cluster_6  PCT" style="background-color: #81BD44; color: #000000; font-size: 13px">1.9</td>
<td class="gt_row gt_right" headers="Cluster_6  Padj" style="background-color: #F8874E; color: #000000; font-size: 13px">9.5e-03</td>
<td class="gt_row gt_left" headers="Cluster_6  geneID" style="font-size: 13px">AGT, MSX1, SFRP2, SIX1, WNT5A, SIX2, SOX8, WNT4, SAPCD2</td>
</tr>
<tr class="odd">
<td class="gt_row gt_left" headers="Cluster_6  Description" style="font-size: 13px"><div style="line-height:14px">
<span class="smallcaps">cochlea development</span>
</div>
<div style="line-height:10px">
<span style="font-weight:bold;color:#BEBEBE;font-size:10px">GO:0090102</span>
</div></td>
<td class="gt_row gt_right" headers="Cluster_6  Count" style="background-color: #057470; color: #FFFFFF; font-size: 13px">7</td>
<td class="gt_row gt_right" headers="Cluster_6  PCT" style="background-color: #67A832; color: #FFFFFF; font-size: 13px">1.5</td>
<td class="gt_row gt_right" headers="Cluster_6  Padj" style="background-color: #FCA85E; color: #000000; font-size: 13px">1.3e-02</td>
<td class="gt_row gt_left" headers="Cluster_6  geneID" style="font-size: 13px">KCNK2, SIX1, WNT5A, TBX18, HEY2, CTHRC1, MYO3B</td>
</tr>
<tr class="even">
<td class="gt_row gt_left" headers="Cluster_6  Description" style="font-size: 13px"><div style="line-height:14px">
<span class="smallcaps">cochlea morphogenesis</span>
</div>
<div style="line-height:10px">
<span style="font-weight:bold;color:#BEBEBE;font-size:10px">GO:0090103</span>
</div></td>
<td class="gt_row gt_right" headers="Cluster_6  Count" style="background-color: #016450; color: #FFFFFF; font-size: 13px">5</td>
<td class="gt_row gt_right" headers="Cluster_6  PCT" style="background-color: #4D9221; color: #FFFFFF; font-size: 13px">1.1</td>
<td class="gt_row gt_right" headers="Cluster_6  Padj" style="background-color: #FEB769; color: #000000; font-size: 13px">1.5e-02</td>
<td class="gt_row gt_left" headers="Cluster_6  geneID" style="font-size: 13px">SIX1, WNT5A, TBX18, CTHRC1, MYO3B</td>
</tr>
<tr class="odd">
<td class="gt_row gt_left" headers="Cluster_6  Description" style="font-size: 13px"><div style="line-height:14px">
<span class="smallcaps">muscle cell development</span>
</div>
<div style="line-height:10px">
<span style="font-weight:bold;color:#BEBEBE;font-size:10px">GO:0055001</span>
</div></td>
<td class="gt_row gt_right" headers="Cluster_6  Count" style="background-color: #529DC8; color: #FFFFFF; font-size: 13px">14</td>
<td class="gt_row gt_right" headers="Cluster_6  PCT" style="background-color: #CEEAA8; color: #000000; font-size: 13px">3.0</td>
<td class="gt_row gt_right" headers="Cluster_6  Padj" style="background-color: #FECD7B; color: #000000; font-size: 13px">1.8e-02</td>
<td class="gt_row gt_left" headers="Cluster_6  geneID" style="font-size: 13px">GPX1, RYR1, SDC1, SIX1, TNNT1, UCHL1, TBX18, HEY2, MEGF10, MYO18B, DNER, ALPK2, WFIKKN1, SGCZ</td>
</tr>
<tr class="even">
<td class="gt_row gt_left" headers="Cluster_6  Description" style="font-size: 13px"><div style="line-height:14px">
<span class="smallcaps">inner ear development</span>
</div>
<div style="line-height:10px">
<span style="font-weight:bold;color:#BEBEBE;font-size:10px">GO:0048839</span>
</div></td>
<td class="gt_row gt_right" headers="Cluster_6  Count" style="background-color: #529DC8; color: #FFFFFF; font-size: 13px">14</td>
<td class="gt_row gt_right" headers="Cluster_6  PCT" style="background-color: #CEEAA8; color: #000000; font-size: 13px">3.0</td>
<td class="gt_row gt_right" headers="Cluster_6  Padj" style="background-color: #FDE18C; color: #000000; font-size: 13px">2.1e-02</td>
<td class="gt_row gt_left" headers="Cluster_6  geneID" style="font-size: 13px">JAG1, COL11A1, GSDME, DLX5, KCNK2, MSX1, SIX1, TFAP2A, WNT5A, ZIC1, TBX18, HEY2, CTHRC1, MYO3B</td>
</tr>
<tr class="odd">
<td class="gt_row gt_left" headers="Cluster_6  Description" style="font-size: 13px"><div style="line-height:14px">
<span class="smallcaps">muscle tissue development</span>
</div>
<div style="line-height:10px">
<span style="font-weight:bold;color:#BEBEBE;font-size:10px">GO:0060537</span>
</div></td>
<td class="gt_row gt_right" headers="Cluster_6  Count" style="background-color: #D1D2E6; color: #000000; font-size: 13px">23</td>
<td class="gt_row gt_right" headers="Cluster_6  PCT" style="background-color: #F2B9DC; color: #000000; font-size: 13px">4.9</td>
<td class="gt_row gt_right" headers="Cluster_6  Padj" style="background-color: #EFEE93; color: #000000; font-size: 13px">2.5e-02</td>
<td class="gt_row gt_left" headers="Cluster_6  geneID" style="font-size: 13px">CENPF, COL11A1, EYA2, FOXC2, GPX1, KCNK2, MSX1, RYR1, SIX1, TP73, TWIST1, BARX2, TBX18, HEY2, SOX8, NOX4, STRA6, MEGF10, MYO18B, MYLK2, DNER, ALPK2, SGCZ</td>
</tr>
<tr class="even">
<td class="gt_row gt_left" headers="Cluster_6  Description" style="font-size: 13px"><div style="line-height:14px">
<span class="smallcaps">embryonic cranial skeleton morphogenesis</span>
</div>
<div style="line-height:10px">
<span style="font-weight:bold;color:#BEBEBE;font-size:10px">GO:0048701</span>
</div></td>
<td class="gt_row gt_right" headers="Cluster_6  Count" style="background-color: #046C60; color: #FFFFFF; font-size: 13px">6</td>
<td class="gt_row gt_right" headers="Cluster_6  PCT" style="background-color: #5A9D29; color: #FFFFFF; font-size: 13px">1.3</td>
<td class="gt_row gt_right" headers="Cluster_6  Padj" style="background-color: #EFEE93; color: #000000; font-size: 13px">2.5e-02</td>
<td class="gt_row gt_left" headers="Cluster_6  geneID" style="font-size: 13px">FOXC2, SIX1, TBX15, TFAP2A, TWIST1, SIX2</td>
</tr>
<tr class="gt_group_heading_row odd">
<td colspan="5" id="Cluster_7" class="gt_group_heading" data-quarto-table-cell-role="th" scope="colgroup">Cluster_7</td>
</tr>
<tr class="gt_row_group_first even">
<td class="gt_row gt_left" headers="Cluster_7  Description" style="font-size: 13px"><div style="line-height:14px">
<span class="smallcaps">positive regulation of chemotaxis</span>
</div>
<div style="line-height:10px">
<span style="font-weight:bold;color:#BEBEBE;font-size:10px">GO:0050921</span>
</div></td>
<td class="gt_row gt_right" headers="Cluster_7  Count" style="background-color: #4496C4; color: #FFFFFF; font-size: 13px">13</td>
<td class="gt_row gt_right" headers="Cluster_7  PCT" style="background-color: #C2E595; color: #000000; font-size: 13px">2.8</td>
<td class="gt_row gt_right" headers="Cluster_7  Padj" style="background-color: #EB6047; color: #FFFFFF; font-size: 13px">4.9e-03</td>
<td class="gt_row gt_left" headers="Cluster_7  geneID" style="font-size: 13px">C3AR1, DSCAM, HSPB1, PGF, THBS4, WNT5A, SCG2, TREM2, CAMK1D, S100A14, TRPV4, SMOC2, IL34</td>
</tr>
<tr class="odd">
<td class="gt_row gt_left" headers="Cluster_7  Description" style="font-size: 13px"><div style="line-height:14px">
<span class="smallcaps">regulation of granulocyte chemotaxis</span>
</div>
<div style="line-height:10px">
<span style="font-weight:bold;color:#BEBEBE;font-size:10px">GO:0071622</span>
</div></td>
<td class="gt_row gt_right" headers="Cluster_7  Count" style="background-color: #057470; color: #FFFFFF; font-size: 13px">7</td>
<td class="gt_row gt_right" headers="Cluster_7  PCT" style="background-color: #67A832; color: #FFFFFF; font-size: 13px">1.5</td>
<td class="gt_row gt_right" headers="Cluster_7  Padj" style="background-color: #FDB063; color: #000000; font-size: 13px">1.4e-02</td>
<td class="gt_row gt_left" headers="Cluster_7  geneID" style="font-size: 13px">C3AR1, DPP4, THBS4, CAMK1D, S100A14, TRPV4, IL34</td>
</tr>
<tr class="even">
<td class="gt_row gt_left" headers="Cluster_7  Description" style="font-size: 13px"><div style="line-height:14px">
<span class="smallcaps">regulation of chemotaxis</span>
</div>
<div style="line-height:10px">
<span style="font-weight:bold;color:#BEBEBE;font-size:10px">GO:0050920</span>
</div></td>
<td class="gt_row gt_right" headers="Cluster_7  Count" style="background-color: #529DC8; color: #FFFFFF; font-size: 13px">14</td>
<td class="gt_row gt_right" headers="Cluster_7  PCT" style="background-color: #CEEAA8; color: #000000; font-size: 13px">3.0</td>
<td class="gt_row gt_right" headers="Cluster_7  Padj" style="background-color: #D1EC9D; color: #000000; font-size: 13px">3.0e-02</td>
<td class="gt_row gt_left" headers="Cluster_7  geneID" style="font-size: 13px">C3AR1, DPP4, DSCAM, HSPB1, PGF, THBS4, WNT5A, SCG2, TREM2, CAMK1D, S100A14, TRPV4, SMOC2, IL34</td>
</tr>
<tr class="gt_group_heading_row odd">
<td colspan="5" id="Cluster_8" class="gt_group_heading" data-quarto-table-cell-role="th" scope="colgroup">Cluster_8</td>
</tr>
<tr class="gt_row_group_first even">
<td class="gt_row gt_left" headers="Cluster_8  Description" style="font-size: 13px"><div style="line-height:14px">
<span class="smallcaps">renal system development</span>
</div>
<div style="line-height:10px">
<span style="font-weight:bold;color:#BEBEBE;font-size:10px">GO:0072001</span>
</div></td>
<td class="gt_row gt_right" headers="Cluster_8  Count" style="background-color: #BAC7E0; color: #000000; font-size: 13px">21</td>
<td class="gt_row gt_right" headers="Cluster_8  PCT" style="background-color: #F8CFE6; color: #000000; font-size: 13px">4.5</td>
<td class="gt_row gt_right" headers="Cluster_8  Padj" style="background-color: #F26944; color: #FFFFFF; font-size: 13px">6.3e-03</td>
<td class="gt_row gt_left" headers="Cluster_8  geneID" style="font-size: 13px">JAG1, AGT, AKR1B1, CENPF, EMX2, ENPEP, FOXD1, FOXC2, MMP17, SDC1, SIX1, TFAP2A, TP73, WNT5A, TBX18, GCNT3, SIX2, SOX8, WNT4, STRA6, CD24</td>
</tr>
<tr class="odd">
<td class="gt_row gt_left" headers="Cluster_8  Description" style="font-size: 13px"><div style="line-height:14px">
<span class="smallcaps">regulation of kidney development</span>
</div>
<div style="line-height:10px">
<span style="font-weight:bold;color:#BEBEBE;font-size:10px">GO:0090183</span>
</div></td>
<td class="gt_row gt_right" headers="Cluster_8  Count" style="background-color: #046C60; color: #FFFFFF; font-size: 13px">6</td>
<td class="gt_row gt_right" headers="Cluster_8  PCT" style="background-color: #5A9D29; color: #FFFFFF; font-size: 13px">1.3</td>
<td class="gt_row gt_right" headers="Cluster_8  Padj" style="background-color: #F98C50; color: #000000; font-size: 13px">1.0e-02</td>
<td class="gt_row gt_left" headers="Cluster_8  geneID" style="font-size: 13px">AGT, FOXD1, SIX1, SIX2, SOX8, WNT4</td>
</tr>
<tr class="even">
<td class="gt_row gt_left" headers="Cluster_8  Description" style="font-size: 13px"><div style="line-height:14px">
<span class="smallcaps">kidney development</span>
</div>
<div style="line-height:10px">
<span style="font-weight:bold;color:#BEBEBE;font-size:10px">GO:0001822</span>
</div></td>
<td class="gt_row gt_right" headers="Cluster_8  Count" style="background-color: #A1BBDA; color: #000000; font-size: 13px">19</td>
<td class="gt_row gt_right" headers="Cluster_8  PCT" style="background-color: #FBE2EC; color: #000000; font-size: 13px">4.1</td>
<td class="gt_row gt_right" headers="Cluster_8  Padj" style="background-color: #FECD7B; color: #000000; font-size: 13px">1.8e-02</td>
<td class="gt_row gt_left" headers="Cluster_8  geneID" style="font-size: 13px">JAG1, AGT, AKR1B1, CENPF, ENPEP, FOXD1, FOXC2, MMP17, SDC1, SIX1, TFAP2A, TP73, WNT5A, GCNT3, SIX2, SOX8, WNT4, STRA6, CD24</td>
</tr>
<tr class="odd">
<td class="gt_row gt_left" headers="Cluster_8  Description" style="font-size: 13px"><div style="line-height:14px">
<span class="smallcaps">regulation of morphogenesis of an epithelium</span>
</div>
<div style="line-height:10px">
<span style="font-weight:bold;color:#BEBEBE;font-size:10px">GO:1905330</span>
</div></td>
<td class="gt_row gt_right" headers="Cluster_8  Count" style="background-color: #057470; color: #FFFFFF; font-size: 13px">7</td>
<td class="gt_row gt_right" headers="Cluster_8  PCT" style="background-color: #67A832; color: #FFFFFF; font-size: 13px">1.5</td>
<td class="gt_row gt_right" headers="Cluster_8  Padj" style="background-color: #BFE5A0; color: #000000; font-size: 13px">3.2e-02</td>
<td class="gt_row gt_left" headers="Cluster_8  geneID" style="font-size: 13px">AGT, SIX1, WNT5A, SIX2, SOX8, WNT4, RNF207</td>
</tr>
<tr class="gt_group_heading_row even">
<td colspan="5" id="Cluster_9" class="gt_group_heading" data-quarto-table-cell-role="th" scope="colgroup">Cluster_9</td>
</tr>
<tr class="gt_row_group_first odd">
<td class="gt_row gt_left" headers="Cluster_9  Description" style="font-size: 13px"><div style="line-height:14px">
<span class="smallcaps">olefinic compound metabolic process</span>
</div>
<div style="line-height:10px">
<span style="font-weight:bold;color:#BEBEBE;font-size:10px">GO:0120254</span>
</div></td>
<td class="gt_row gt_right" headers="Cluster_9  Count" style="background-color: #4496C4; color: #FFFFFF; font-size: 13px">13</td>
<td class="gt_row gt_right" headers="Cluster_9  PCT" style="background-color: #C2E595; color: #000000; font-size: 13px">2.8</td>
<td class="gt_row gt_right" headers="Cluster_9  Padj" style="background-color: #F98C50; color: #000000; font-size: 13px">1.0e-02</td>
<td class="gt_row gt_left" headers="Cluster_9  geneID" style="font-size: 13px">ABCA4, AKR1B1, ALOX15B, AKR1C1, AKR1C2, GPX1, GRIN1, SRD5A1, PLA2G4C, RDH16, DHRS3, WNT4, CTHRC1</td>
</tr>
<tr class="even">
<td class="gt_row gt_left" headers="Cluster_9  Description" style="font-size: 13px"><div style="line-height:14px">
<span class="smallcaps">C21-steroid hormone metabolic process</span>
</div>
<div style="line-height:10px">
<span style="font-weight:bold;color:#BEBEBE;font-size:10px">GO:0008207</span>
</div></td>
<td class="gt_row gt_right" headers="Cluster_9  Count" style="background-color: #016450; color: #FFFFFF; font-size: 13px">5</td>
<td class="gt_row gt_right" headers="Cluster_9  PCT" style="background-color: #4D9221; color: #FFFFFF; font-size: 13px">1.1</td>
<td class="gt_row gt_right" headers="Cluster_9  Padj" style="background-color: #3E90BA; color: #FFFFFF; font-size: 13px">4.7e-02</td>
<td class="gt_row gt_left" headers="Cluster_9  geneID" style="font-size: 13px">AKR1B1, AKR1C1, AKR1C2, SRD5A1, WNT4</td>
</tr>
</tbody>
</table>

</div>
</div>
</div>
<p>This just shows the second slot inside the <code>EnrichGT_obj</code> object. The second slot - <code>gt_object</code> is a pure object of <code>gt table</code> package, you can use any <a href="https://vincentarelbundock.github.io/gt%20table/vignettes/gt%20table.html">function</a> on it.</p>
<p>For further usage of <code>gt table</code> package, please refer to <a href="https://gt.rstudio.com/reference/index.html#export-and-extraction-functions">gt package website</a>.</p>
<p>See re-enrichment example for further demo.</p>
</section>
<section id="each-cluster-html-widgets" class="level2">
<h2 class="anchored" data-anchor-id="each-cluster-html-widgets">Each Cluster: HTML Widgets</h2>
<p>If you want to show <code>cluster 1</code> in your documents(the publish system or IDE should support HTML widgets, like RStudio), you can use one of below:</p>
<div class="sourceCode" id="cb2"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1" aria-hidden="true" tabindex="-1"></a><span class="fu">egt_summary</span>(deepseekAnno, <span class="st">"1"</span>)</span>
<span id="cb2-2"><a href="#cb2-2" aria-hidden="true" tabindex="-1"></a><span class="fu">egt_summary</span>(deepseekAnno, <span class="dv">1</span>)</span>
<span id="cb2-3"><a href="#cb2-3" aria-hidden="true" tabindex="-1"></a><span class="fu">egt_summary</span>(deepseekAnno, <span class="st">"Cluster_1"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<p>In <code>Quarto</code> or <code>rmarkdown</code>, you can use <code>#| output: asis</code> header in code chunk and print it to show HTML widgets.</p>
<p><img src="images/clipboard-1084314783.png" class="img-fluid"></p>
<div class="sourceCode cell-code" id="cb3"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1" aria-hidden="true" tabindex="-1"></a>HTMLres <span class="ot">&lt;-</span> <span class="fu">egt_summary</span>(re_enrich, <span class="st">"1"</span>)</span>
<span id="cb3-2"><a href="#cb3-2" aria-hidden="true" tabindex="-1"></a>HTMLres</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div style="font-family: -apple-system, BlinkMacSystemFont, 'Segoe UI', Roboto, sans-serif; margin: 20px;">
<h2 style="color: #495057; border-bottom: 2px solid #007bff; padding-bottom: 10px; margin-bottom: 20px;" class="anchored">Enrichment Result of Cluster_1 (Local Summary)</h2>
<div style="background-color: #f8f9fa; padding: 20px; border-radius: 8px; margin-bottom: 15px;">
<p style="color: #495057; line-height: 1.6; margin-bottom: 10px;"><strong>This cluster contains:</strong> synaptic transmission, glutamatergic, regulation of synaptic transmission, glutamatergic, glutamate receptor signaling pathway, regulation of neuronal synaptic plasticity, ionotropic glutamate receptor signaling pathway ...</p>
<p style="color: #495057; line-height: 1.6;"><strong>Candidate genes include:</strong> ACHE, ADGRF1, AGT, ALK, ANKFN1, ANKH, APLP1, APOE, ASPM, ATP1A2, ATP8B3, AVPR1A, BCAN, C4BPB, CA2, CACNG5, CAMK1D, CAMK2B, CDC20, CELSR3, CIART, CNTN2, CNTNAP2, CPNE4, CPNE7, CTNNA2, DAAM2, DLX5, DNAJB1, DNER, DPP4, DPYSL5, DSCAM, E2F1, EMX2, EPHA10, FOLR2, FOXD1, FSTL4, GABRD, GAD1, GAP43, GFAP, GIPR, GPR37, GRIA4, GRID2, GRIK2, GRIK3, GRIN1, GRIN2A, GRIN2B, GRIN2D, GRM1, GRM5, GRM8, HCN1, HCN2, HMOX1, HRAS, HTR3A, IGSF9, IL1RAPL2, IMPA2, INA, KCNC3, KCNE4, KCNK2, KIAA1755, KIF1A, KIF5C, LHFPL4, LOXL2, LRFN2, MAG, MAP1B, MAPK8IP2, MEGF10, MSX1, MT-CYB, MYO3B, NKX6-1, NLGN1, NMU, NQO1, NRCAM, NRXN1, NTRK2, NUDT1, OPCML, PPP1R1B, PTPRH, PYCR1, RAC3, RASD2, RIMS3, RNF207, RYR1, SCN8A, SDC1, SDC2, SEMA5B, SEZ6L2, SFRP2, SHISA9, SIX1, SLC30A10, SLC30A3, SLC6A1, SLC6A3, SPOCK1, SRD5A1, STRA6, SYNDIG1, TBX18, TF, TFAP2A, TREM2, TRPM2, TRPV4, TWIST1, UCHL1, UNC13A, UNC13C, VGF, WDR62, WNT4, WNT5A, ZAN, ZIC1</p>
</div>
<div style="color: #6c757d; font-size: 0.9em; font-style: italic;">We can't find LLM annotations. Please using egt_llm_* functions for further analysis. </div>
</div>
</section>
<section id="quarto-reports" class="level2">
<h2 class="anchored" data-anchor-id="quarto-reports">Quarto Reports</h2>
<p>Function <code>egt_generate_quarto_report()</code> can help you convert re-enrichment object to an all-in-one Quarto document containing all enrichment results. You can use <code>RStudio</code> or <code>Positron</code> to knit them (preview) into HTML document.</p>
<p><img src="images/clipboard-1284614460.png" class="img-fluid" width="1000"></p>
</section>
</section>
<section id="ploting-functions" class="level1">
<h1>Ploting functions</h1>
<p>HTML <code>gt table</code> table satisfied most of things, but for others. Though we donâ€™t want this package become complex (i.e., you can simple draw your figure using ggplot2 for enriched tables by yourself.) But we still provide limited figure ploting functions.</p>
<section id="basic-result-plot" class="level2">
<h2 class="anchored" data-anchor-id="basic-result-plot">Basic result plot</h2>
<p>The Dot Plot provide basic viewing of results. So it supports both simple enrichment result <code>data.frame</code> and re-enriched <code>egt_object</code>.</p>
<p>You can adjust this figure by these params:</p>
<ul>
<li><code>ntop</code>: Show top N in each cluster. In default, for origin enriched result, showing top 15; for re-clustered object, showing top 5 in each cluster.</li>
<li><code>showIDs</code>: bool, show pathway IDs or not. Default is FALSE</li>
<li><code>max_len_descript</code>: the label format length, default as 40.</li>
<li><code>P.adj</code>: If pass an origin data.frame from original enriched result, you can specify the P-adjust value cut off. If is null, default is 0.05. When passing <code>EnrichGT_obj</code>, this filter is previously done by <code>egt_recluster_analysis</code>.</li>
<li><code>low.col</code>: the color for the lowest</li>
<li><code>hi.col</code>: the color for the highest</li>
</ul>
<section id="ploting-ora-result" class="level4">
<h4 class="anchored" data-anchor-id="ploting-ora-result">Ploting ORA result</h4>
<div class="cell">
<div class="sourceCode cell-code" id="cb4"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb4-1"><a href="#cb4-1" aria-hidden="true" tabindex="-1"></a><span class="fu">egt_plot_results</span>(ora_result)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>â„¹ Use Default P-adjust cut-off 0.05. You can pass `P.adj=xxx` arugument to filter. </code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>! You are drawing origin results, for better result you can re-cluster it by egt_recluster_analysis()</code></pre>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="visualize_files/figure-html/unnamed-chunk-4-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
</section>
<section id="ploting-gsea-result" class="level4">
<h4 class="anchored" data-anchor-id="ploting-gsea-result">Ploting GSEA result</h4>
<div class="cell">
<div class="sourceCode cell-code" id="cb7"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb7-1"><a href="#cb7-1" aria-hidden="true" tabindex="-1"></a><span class="fu">egt_plot_results</span>(resGSEA)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>â„¹ Use Default P-adjust cut-off 0.05. You can pass `P.adj=xxx` arugument to filter. </code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>! You are drawing origin results, for better result you can re-cluster it by egt_recluster_analysis()</code></pre>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="visualize_files/figure-html/unnamed-chunk-5-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
</section>
<section id="ploting-re-enrichment-result" class="level4">
<h4 class="anchored" data-anchor-id="ploting-re-enrichment-result">Ploting re-enrichment result</h4>
<div class="cell">
<div class="sourceCode cell-code" id="cb10"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb10-1"><a href="#cb10-1" aria-hidden="true" tabindex="-1"></a><span class="fu">egt_plot_results</span>(re_enrich)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="visualize_files/figure-html/unnamed-chunk-6-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
</section>
</section>
<section id="additional-gsea-plots" class="level2">
<h2 class="anchored" data-anchor-id="additional-gsea-plots">Additional GSEA plots</h2>
<p>Since version 0.8.6, EnrichGT provides a function called <code>egt_plot_gsea()</code> to help users to get typical ranking plots. If you pass a pathway name to the first param of this function, <code>egt_plot_gsea()</code> will return a ranking plot of single pathway. For multiple pathway, you need to subset the <code>data.frame</code> result of <code>egt_gsea_analysis()</code>.</p>
<p>The param contains:</p>
<ul>
<li><code>x</code>: A GSEA result object. Can be either:
<ul>
<li>A data frame containing GSEA results (requires columns: pvalue, p.adjust, Description)</li>
<li>A character string specifying a single pathway name</li>
</ul></li>
<li><code>genes</code> and <code>database</code> should be the same as you doing <code>egt_gsea_analysis()</code></li>
</ul>
<div class="callout callout-style-default callout-note callout-titled">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-title-container flex-fill">
Why multiple results plotting function needs a <code>data.frame</code>
</div>
</div>
<div class="callout-body-container callout-body">
<p>You can see the result figure, we need the <code>NES</code> and <code>p-values</code> for display.</p>
</div>
</div>
<section id="single-ranking-plot" class="level3">
<h3 class="anchored" data-anchor-id="single-ranking-plot">Single ranking plot</h3>
<p>You need to prepare your favourite pathwayâ€™s name:</p>
<div class="cell">
<div class="sourceCode cell-code" id="annotated-cell-6"><pre class="sourceCode r code-annotation-code code-with-copy code-annotated"><code class="sourceCode r"><a class="code-annotation-anchor" data-target-cell="annotated-cell-6" data-target-annotation="1" onclick="event.preventDefault();">1</a><span id="annotated-cell-6-1" class="code-annotation-target"><a href="#annotated-cell-6-1" aria-hidden="true" tabindex="-1"></a><span class="fu">class</span>(resGSEA<span class="sc">$</span>Description[<span class="dv">1</span>])</span>
<a class="code-annotation-anchor" data-target-cell="annotated-cell-6" data-target-annotation="2" onclick="event.preventDefault();">2</a><span id="annotated-cell-6-2" class="code-annotation-target"><a href="#annotated-cell-6-2" aria-hidden="true" tabindex="-1"></a>resGSEA<span class="sc">$</span>Description[<span class="dv">1</span>]</span><div class="code-annotation-gutter-bg"></div><div class="code-annotation-gutter"></div></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-annotation">
<dl class="code-annotation-container-grid">
<dt data-target-cell="annotated-cell-6" data-target-annotation="1">1</dt>
<dd>
<span data-code-cell="annotated-cell-6" data-code-lines="1" data-code-annotation="1">This is a vector.</span>
</dd>
<dt data-target-cell="annotated-cell-6" data-target-annotation="2">2</dt>
<dd>
<span data-code-cell="annotated-cell-6" data-code-lines="2" data-code-annotation="2">If you need the single-plot, all you pass to <code>egt_plot_gsea()</code> is the name of this pathway, subset from <code>resGSEA</code>.</span>
</dd>
</dl>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] "character"
[1] "meiotic chromosome segregation"</code></pre>
</div>
</div>
<p>And then using <code>egt_plot_gsea()</code> for drawing. The other params should be the <strong>same</strong> as you provided in origin GSEA analysis.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb12"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb12-1"><a href="#cb12-1" aria-hidden="true" tabindex="-1"></a><span class="fu">egt_plot_gsea</span>(resGSEA<span class="sc">$</span>Description[<span class="dv">1</span>],</span>
<span id="cb12-2"><a href="#cb12-2" aria-hidden="true" tabindex="-1"></a>              <span class="at">genes =</span> <span class="fu">genes_with_weights</span>(<span class="at">genes =</span> DEGexample<span class="sc">$</span>...<span class="dv">1</span>, </span>
<span id="cb12-3"><a href="#cb12-3" aria-hidden="true" tabindex="-1"></a>                                              <span class="at">weights =</span> DEGexample<span class="sc">$</span>log2FoldChange),</span>
<span id="cb12-4"><a href="#cb12-4" aria-hidden="true" tabindex="-1"></a>              <span class="at">database =</span> <span class="fu">database_GO_BP</span>(org.Hs.eg.db))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>âœ” success loaded database, time used : 15.935772895813 sec.</code></pre>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="visualize_files/figure-html/unnamed-chunk-8-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
</section>
<section id="table-like-visualization-gsea-result" class="level3">
<h3 class="anchored" data-anchor-id="table-like-visualization-gsea-result">Table-like visualization GSEA result</h3>
<div class="callout callout-style-default callout-warning callout-titled">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-title-container flex-fill">
Warning
</div>
</div>
<div class="callout-body-container callout-body">
<p>If you want to plot this plot, remember to filter the GSEA results to ~10 to ~20, you can base on <code>NES</code> or <code>p-values</code>. This is to avoid too many loading to this plotting.</p>
</div>
</div>
<p>You need to subset the origin GSEA result <code>data.frame</code> according to <code>NES</code> or <code>p-val</code> to gain your targets. In this demo, we choose row 1 to row 8.</p>
<div class="cell">
<div class="sourceCode cell-code" id="annotated-cell-8"><pre class="sourceCode r code-annotation-code code-with-copy code-annotated"><code class="sourceCode r"><a class="code-annotation-anchor" data-target-cell="annotated-cell-8" data-target-annotation="1" onclick="event.preventDefault();">1</a><span id="annotated-cell-8-1" class="code-annotation-target"><a href="#annotated-cell-8-1" aria-hidden="true" tabindex="-1"></a><span class="fu">class</span>(resGSEA[<span class="dv">1</span><span class="sc">:</span><span class="dv">8</span>,])</span>
<a class="code-annotation-anchor" data-target-cell="annotated-cell-8" data-target-annotation="2" onclick="event.preventDefault();">2</a><span id="annotated-cell-8-2" class="code-annotation-target"><a href="#annotated-cell-8-2" aria-hidden="true" tabindex="-1"></a>resGSEA[<span class="dv">1</span><span class="sc">:</span><span class="dv">8</span>,]</span><div class="code-annotation-gutter-bg"></div><div class="code-annotation-gutter"></div></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-annotation">
<dl class="code-annotation-container-grid">
<dt data-target-cell="annotated-cell-8" data-target-annotation="1">1</dt>
<dd>
<span data-code-cell="annotated-cell-8" data-code-lines="1" data-code-annotation="1">This is a <code>data.frame</code>. Please subset it to avoid too many results and the waste of time.</span>
</dd>
<dt data-target-cell="annotated-cell-8" data-target-annotation="2">2</dt>
<dd>
<span data-code-cell="annotated-cell-8" data-code-lines="2" data-code-annotation="2">You need to subset data frame according to NES (suggests: <code>abs(NES)&gt;1</code>) and p-value, too avoiding too many outputs and wasting.</span>
</dd>
</dl>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] "tbl_df"     "tbl"        "data.frame"
# A tibble: 8 Ã— 12
  ID        Description    ES   NES  pvalue p.adjust core_enrichment  rank tags 
  &lt;chr&gt;     &lt;chr&gt;       &lt;dbl&gt; &lt;dbl&gt;   &lt;dbl&gt;    &lt;dbl&gt; &lt;chr&gt;           &lt;dbl&gt; &lt;chr&gt;
1 GO:00451â€¦ meiotic châ€¦ 0.612  1.96 3.73e-5  0.00117 SPATA22/KASH5/â€¦  2679 43%  
2 GO:00092â€¦ purine ribâ€¦ 0.575  1.92 4.15e-5  0.00126 ADCY10/NME9/NDâ€¦  4558 55%  
3 GO:00509â€¦ detection â€¦ 0.767  1.92 3.41e-4  0.00611 OR51E1/CST1/ORâ€¦  2023 63%  
4 GO:00091â€¦ purine nucâ€¦ 0.570  1.91 3.21e-5  0.00104 ADCY10/NME9/NDâ€¦  4558 54%  
5 GO:00067â€¦ ATP biosynâ€¦ 0.582  1.90 6.38e-5  0.00179 ADCY10/NDUFS6/â€¦  4403 56%  
6 GO:00352â€¦ synaptic tâ€¦ 0.557  1.88 3.23e-5  0.00104 GRIK3/UNC13C/Gâ€¦  1330 31%  
7 GO:00076â€¦ adult walkâ€¦ 0.714  1.88 6.84e-4  0.0103  ZIC1/CNTN2/CACâ€¦   866 25%  
8 GO:00315â€¦ spindle châ€¦ 0.637  1.88 3.12e-4  0.00574 SPC24/BIRC5/PLâ€¦  1435 39%  
# â„¹ 3 more variables: list &lt;chr&gt;, signal &lt;chr&gt;, leading_edge &lt;chr&gt;</code></pre>
</div>
</div>
<p>And then using <code>egt_plot_gsea()</code> for drawing. The other params should be the <strong>same</strong> as you provided in origin GSEA analysis.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb15"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb15-1"><a href="#cb15-1" aria-hidden="true" tabindex="-1"></a><span class="fu">egt_plot_gsea</span>(resGSEA[<span class="dv">1</span><span class="sc">:</span><span class="dv">8</span>,], </span>
<span id="cb15-2"><a href="#cb15-2" aria-hidden="true" tabindex="-1"></a>              <span class="at">genes =</span> <span class="fu">genes_with_weights</span>(<span class="at">genes =</span> DEGexample<span class="sc">$</span>...<span class="dv">1</span>, </span>
<span id="cb15-3"><a href="#cb15-3" aria-hidden="true" tabindex="-1"></a>                                              <span class="at">weights =</span> DEGexample<span class="sc">$</span>log2FoldChange),</span>
<span id="cb15-4"><a href="#cb15-4" aria-hidden="true" tabindex="-1"></a>              <span class="at">database =</span> <span class="fu">database_GO_BP</span>(org.Hs.eg.db))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>âœ” Use cached database: GO_BP_org.Hs.eg.db</code></pre>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="visualize_files/figure-html/unnamed-chunk-10-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
</section>
</section>
<section id="umap-plot-for-re-enrichment-no-longer-supported" class="level2">
<h2 class="anchored" data-anchor-id="umap-plot-for-re-enrichment-no-longer-supported">UMAP plot for re-enrichment (no longer supported)</h2>
<p>Before version 0.8.6, UMAP plot is provided in re-enriched <code>egt_object</code>, to show the dimensionality reduction view of enriched results.</p>
<p>A word frequency matrix represents the frequency of words or tokens across different documents or text samples. Each row corresponds to a document, and each column represents a word or token, with the cell values indicating the frequency of the respective word in that document.However, high-dimensional data like word frequency matrices can be challenging to interpret directly. To make such data easier to analyze, we can reduce its dimensionality and visualize the patterns or clusters in a 2D or 3D space. UMAP (Uniform Manifold Approximation and Projection) is a powerful, non-linear dimensionality reduction technique widely used for this purpose.</p>
<p>But since 0.8.6, because <code>umap</code> and <code>ggrepel</code> have too many dependencies, and this function is not nesessary in most of cases. We now donâ€™t support it. If you still needs this figure, you can execute following code.</p>
<div class="sourceCode" id="cb17"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb17-1"><a href="#cb17-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(umap)</span>
<span id="cb17-2"><a href="#cb17-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ggrepel)</span>
<span id="cb17-3"><a href="#cb17-3" aria-hidden="true" tabindex="-1"></a>mat<span class="ot">&lt;-</span>x<span class="sc">@</span>document_term_matrix</span>
<span id="cb17-4"><a href="#cb17-4" aria-hidden="true" tabindex="-1"></a>umap_result <span class="ot">&lt;-</span> umap<span class="sc">::</span><span class="fu">umap</span>(mat)</span>
<span id="cb17-5"><a href="#cb17-5" aria-hidden="true" tabindex="-1"></a>umap_df <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">ID=</span><span class="fu">rownames</span>(umap_result[[<span class="st">"layout"</span>]]),</span>
<span id="cb17-6"><a href="#cb17-6" aria-hidden="true" tabindex="-1"></a>                      <span class="at">UMAP1 =</span> umap_result<span class="sc">$</span>layout[, <span class="dv">1</span>],</span>
<span id="cb17-7"><a href="#cb17-7" aria-hidden="true" tabindex="-1"></a>                      <span class="at">UMAP2 =</span> umap_result<span class="sc">$</span>layout[, <span class="dv">2</span>])</span>
<span id="cb17-8"><a href="#cb17-8" aria-hidden="true" tabindex="-1"></a>udf<span class="ot">&lt;-</span>x<span class="sc">@</span>enriched_result <span class="sc">|&gt;</span> <span class="fu">left_join</span>(umap_df,<span class="at">by=</span><span class="st">"ID"</span>)</span>
<span id="cb17-9"><a href="#cb17-9" aria-hidden="true" tabindex="-1"></a>fig<span class="ot">&lt;-</span><span class="fu">ggplot</span>(udf, <span class="fu">aes</span>(<span class="at">x =</span> UMAP1, <span class="at">y =</span> UMAP2, <span class="at">color =</span> Cluster)) <span class="sc">+</span></span>
<span id="cb17-10"><a href="#cb17-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="at">size =</span> <span class="dv">2</span>) <span class="sc">+</span></span>
<span id="cb17-11"><a href="#cb17-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_text_repel</span>(<span class="fu">aes</span>(<span class="at">label =</span> Description),</span>
<span id="cb17-12"><a href="#cb17-12" aria-hidden="true" tabindex="-1"></a>                  <span class="at">size =</span> <span class="dv">3</span>,</span>
<span id="cb17-13"><a href="#cb17-13" aria-hidden="true" tabindex="-1"></a>                  <span class="at">max.overlaps =</span> <span class="dv">20</span>,</span>
<span id="cb17-14"><a href="#cb17-14" aria-hidden="true" tabindex="-1"></a>                  <span class="at">box.padding =</span> <span class="fl">0.3</span>,</span>
<span id="cb17-15"><a href="#cb17-15" aria-hidden="true" tabindex="-1"></a>                  <span class="at">point.padding =</span> <span class="fl">0.2</span>) <span class="sc">+</span></span>
<span id="cb17-16"><a href="#cb17-16" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">title =</span> <span class="st">"Enrichment Results"</span>,</span>
<span id="cb17-17"><a href="#cb17-17" aria-hidden="true" tabindex="-1"></a>        <span class="at">x =</span> <span class="st">"UMAP1"</span>, <span class="at">y =</span> <span class="st">"UMAP2"</span>) <span class="sc">+</span></span>
<span id="cb17-18"><a href="#cb17-18" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_classic</span>()</span>
<span id="cb17-19"><a href="#cb17-19" aria-hidden="true" tabindex="-1"></a>fig</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>


</section>
</section>

<a onclick="window.scrollTo(0, 0); return false;" role="button" id="quarto-back-to-top"><i class="bi bi-arrow-up"></i> Back to top</a></main> <!-- /main -->
<script id="quarto-html-after-body" type="application/javascript">
  window.document.addEventListener("DOMContentLoaded", function (event) {
    const icon = "î§‹";
    const anchorJS = new window.AnchorJS();
    anchorJS.options = {
      placement: 'right',
      icon: icon
    };
    anchorJS.add('.anchored');
    const isCodeAnnotation = (el) => {
      for (const clz of el.classList) {
        if (clz.startsWith('code-annotation-')) {                     
          return true;
        }
      }
      return false;
    }
    const onCopySuccess = function(e) {
      // button target
      const button = e.trigger;
      // don't keep focus
      button.blur();
      // flash "checked"
      button.classList.add('code-copy-button-checked');
      var currentTitle = button.getAttribute("title");
      button.setAttribute("title", "Copied!");
      let tooltip;
      if (window.bootstrap) {
        button.setAttribute("data-bs-toggle", "tooltip");
        button.setAttribute("data-bs-placement", "left");
        button.setAttribute("data-bs-title", "Copied!");
        tooltip = new bootstrap.Tooltip(button, 
          { trigger: "manual", 
            customClass: "code-copy-button-tooltip",
            offset: [0, -8]});
        tooltip.show();    
      }
      setTimeout(function() {
        if (tooltip) {
          tooltip.hide();
          button.removeAttribute("data-bs-title");
          button.removeAttribute("data-bs-toggle");
          button.removeAttribute("data-bs-placement");
        }
        button.setAttribute("title", currentTitle);
        button.classList.remove('code-copy-button-checked');
      }, 1000);
      // clear code selection
      e.clearSelection();
    }
    const getTextToCopy = function(trigger) {
        const codeEl = trigger.previousElementSibling.cloneNode(true);
        for (const childEl of codeEl.children) {
          if (isCodeAnnotation(childEl)) {
            childEl.remove();
          }
        }
        return codeEl.innerText;
    }
    const clipboard = new window.ClipboardJS('.code-copy-button:not([data-in-quarto-modal])', {
      text: getTextToCopy
    });
    clipboard.on('success', onCopySuccess);
    if (window.document.getElementById('quarto-embedded-source-code-modal')) {
      const clipboardModal = new window.ClipboardJS('.code-copy-button[data-in-quarto-modal]', {
        text: getTextToCopy,
        container: window.document.getElementById('quarto-embedded-source-code-modal')
      });
      clipboardModal.on('success', onCopySuccess);
    }
      var localhostRegex = new RegExp(/^(?:http|https):\/\/localhost\:?[0-9]*\//);
      var mailtoRegex = new RegExp(/^mailto:/);
        var filterRegex = new RegExp('/' + window.location.host + '/');
      var isInternal = (href) => {
          return filterRegex.test(href) || localhostRegex.test(href) || mailtoRegex.test(href);
      }
      // Inspect non-navigation links and adorn them if external
     var links = window.document.querySelectorAll('a[href]:not(.nav-link):not(.navbar-brand):not(.toc-action):not(.sidebar-link):not(.sidebar-item-toggle):not(.pagination-link):not(.no-external):not([aria-hidden]):not(.dropdown-item):not(.quarto-navigation-tool):not(.about-link)');
      for (var i=0; i<links.length; i++) {
        const link = links[i];
        if (!isInternal(link.href)) {
          // undo the damage that might have been done by quarto-nav.js in the case of
          // links that we want to consider external
          if (link.dataset.originalHref !== undefined) {
            link.href = link.dataset.originalHref;
          }
        }
      }
    function tippyHover(el, contentFn, onTriggerFn, onUntriggerFn) {
      const config = {
        allowHTML: true,
        maxWidth: 500,
        delay: 100,
        arrow: false,
        appendTo: function(el) {
            return el.parentElement;
        },
        interactive: true,
        interactiveBorder: 10,
        theme: 'quarto',
        placement: 'bottom-start',
      };
      if (contentFn) {
        config.content = contentFn;
      }
      if (onTriggerFn) {
        config.onTrigger = onTriggerFn;
      }
      if (onUntriggerFn) {
        config.onUntrigger = onUntriggerFn;
      }
      window.tippy(el, config); 
    }
    const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
    for (var i=0; i<noterefs.length; i++) {
      const ref = noterefs[i];
      tippyHover(ref, function() {
        // use id or data attribute instead here
        let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
        try { href = new URL(href).hash; } catch {}
        const id = href.replace(/^#\/?/, "");
        const note = window.document.getElementById(id);
        if (note) {
          return note.innerHTML;
        } else {
          return "";
        }
      });
    }
    const xrefs = window.document.querySelectorAll('a.quarto-xref');
    const processXRef = (id, note) => {
      // Strip column container classes
      const stripColumnClz = (el) => {
        el.classList.remove("page-full", "page-columns");
        if (el.children) {
          for (const child of el.children) {
            stripColumnClz(child);
          }
        }
      }
      stripColumnClz(note)
      if (id === null || id.startsWith('sec-')) {
        // Special case sections, only their first couple elements
        const container = document.createElement("div");
        if (note.children && note.children.length > 2) {
          container.appendChild(note.children[0].cloneNode(true));
          for (let i = 1; i < note.children.length; i++) {
            const child = note.children[i];
            if (child.tagName === "P" && child.innerText === "") {
              continue;
            } else {
              container.appendChild(child.cloneNode(true));
              break;
            }
          }
          if (window.Quarto?.typesetMath) {
            window.Quarto.typesetMath(container);
          }
          return container.innerHTML
        } else {
          if (window.Quarto?.typesetMath) {
            window.Quarto.typesetMath(note);
          }
          return note.innerHTML;
        }
      } else {
        // Remove any anchor links if they are present
        const anchorLink = note.querySelector('a.anchorjs-link');
        if (anchorLink) {
          anchorLink.remove();
        }
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(note);
        }
        if (note.classList.contains("callout")) {
          return note.outerHTML;
        } else {
          return note.innerHTML;
        }
      }
    }
    for (var i=0; i<xrefs.length; i++) {
      const xref = xrefs[i];
      tippyHover(xref, undefined, function(instance) {
        instance.disable();
        let url = xref.getAttribute('href');
        let hash = undefined; 
        if (url.startsWith('#')) {
          hash = url;
        } else {
          try { hash = new URL(url).hash; } catch {}
        }
        if (hash) {
          const id = hash.replace(/^#\/?/, "");
          const note = window.document.getElementById(id);
          if (note !== null) {
            try {
              const html = processXRef(id, note.cloneNode(true));
              instance.setContent(html);
            } finally {
              instance.enable();
              instance.show();
            }
          } else {
            // See if we can fetch this
            fetch(url.split('#')[0])
            .then(res => res.text())
            .then(html => {
              const parser = new DOMParser();
              const htmlDoc = parser.parseFromString(html, "text/html");
              const note = htmlDoc.getElementById(id);
              if (note !== null) {
                const html = processXRef(id, note);
                instance.setContent(html);
              } 
            }).finally(() => {
              instance.enable();
              instance.show();
            });
          }
        } else {
          // See if we can fetch a full url (with no hash to target)
          // This is a special case and we should probably do some content thinning / targeting
          fetch(url)
          .then(res => res.text())
          .then(html => {
            const parser = new DOMParser();
            const htmlDoc = parser.parseFromString(html, "text/html");
            const note = htmlDoc.querySelector('main.content');
            if (note !== null) {
              // This should only happen for chapter cross references
              // (since there is no id in the URL)
              // remove the first header
              if (note.children.length > 0 && note.children[0].tagName === "HEADER") {
                note.children[0].remove();
              }
              const html = processXRef(null, note);
              instance.setContent(html);
            } 
          }).finally(() => {
            instance.enable();
            instance.show();
          });
        }
      }, function(instance) {
      });
    }
        let selectedAnnoteEl;
        const selectorForAnnotation = ( cell, annotation) => {
          let cellAttr = 'data-code-cell="' + cell + '"';
          let lineAttr = 'data-code-annotation="' +  annotation + '"';
          const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
          return selector;
        }
        const selectCodeLines = (annoteEl) => {
          const doc = window.document;
          const targetCell = annoteEl.getAttribute("data-target-cell");
          const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
          const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
          const lines = annoteSpan.getAttribute("data-code-lines").split(",");
          const lineIds = lines.map((line) => {
            return targetCell + "-" + line;
          })
          let top = null;
          let height = null;
          let parent = null;
          if (lineIds.length > 0) {
              //compute the position of the single el (top and bottom and make a div)
              const el = window.document.getElementById(lineIds[0]);
              top = el.offsetTop;
              height = el.offsetHeight;
              parent = el.parentElement.parentElement;
            if (lineIds.length > 1) {
              const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
              const bottom = lastEl.offsetTop + lastEl.offsetHeight;
              height = bottom - top;
            }
            if (top !== null && height !== null && parent !== null) {
              // cook up a div (if necessary) and position it 
              let div = window.document.getElementById("code-annotation-line-highlight");
              if (div === null) {
                div = window.document.createElement("div");
                div.setAttribute("id", "code-annotation-line-highlight");
                div.style.position = 'absolute';
                parent.appendChild(div);
              }
              div.style.top = top - 2 + "px";
              div.style.height = height + 4 + "px";
              div.style.left = 0;
              let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
              if (gutterDiv === null) {
                gutterDiv = window.document.createElement("div");
                gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
                gutterDiv.style.position = 'absolute';
                const codeCell = window.document.getElementById(targetCell);
                const gutter = codeCell.querySelector('.code-annotation-gutter');
                gutter.appendChild(gutterDiv);
              }
              gutterDiv.style.top = top - 2 + "px";
              gutterDiv.style.height = height + 4 + "px";
            }
            selectedAnnoteEl = annoteEl;
          }
        };
        const unselectCodeLines = () => {
          const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
          elementsIds.forEach((elId) => {
            const div = window.document.getElementById(elId);
            if (div) {
              div.remove();
            }
          });
          selectedAnnoteEl = undefined;
        };
          // Handle positioning of the toggle
      window.addEventListener(
        "resize",
        throttle(() => {
          elRect = undefined;
          if (selectedAnnoteEl) {
            selectCodeLines(selectedAnnoteEl);
          }
        }, 10)
      );
      function throttle(fn, ms) {
      let throttle = false;
      let timer;
        return (...args) => {
          if(!throttle) { // first call gets through
              fn.apply(this, args);
              throttle = true;
          } else { // all the others get throttled
              if(timer) clearTimeout(timer); // cancel #2
              timer = setTimeout(() => {
                fn.apply(this, args);
                timer = throttle = false;
              }, ms);
          }
        };
      }
        // Attach click handler to the DT
        const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
        for (const annoteDlNode of annoteDls) {
          annoteDlNode.addEventListener('click', (event) => {
            const clickedEl = event.target;
            if (clickedEl !== selectedAnnoteEl) {
              unselectCodeLines();
              const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
              if (activeEl) {
                activeEl.classList.remove('code-annotation-active');
              }
              selectCodeLines(clickedEl);
              clickedEl.classList.add('code-annotation-active');
            } else {
              // Unselect the line
              unselectCodeLines();
              clickedEl.classList.remove('code-annotation-active');
            }
          });
        }
    const findCites = (el) => {
      const parentEl = el.parentElement;
      if (parentEl) {
        const cites = parentEl.dataset.cites;
        if (cites) {
          return {
            el,
            cites: cites.split(' ')
          };
        } else {
          return findCites(el.parentElement)
        }
      } else {
        return undefined;
      }
    };
    var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
    for (var i=0; i<bibliorefs.length; i++) {
      const ref = bibliorefs[i];
      const citeInfo = findCites(ref);
      if (citeInfo) {
        tippyHover(citeInfo.el, function() {
          var popup = window.document.createElement('div');
          citeInfo.cites.forEach(function(cite) {
            var citeDiv = window.document.createElement('div');
            citeDiv.classList.add('hanging-indent');
            citeDiv.classList.add('csl-entry');
            var biblioDiv = window.document.getElementById('ref-' + cite);
            if (biblioDiv) {
              citeDiv.innerHTML = biblioDiv.innerHTML;
            }
            popup.appendChild(citeDiv);
          });
          return popup.innerHTML;
        });
      }
    }
  });
  </script>
</div> <!-- /content -->
<footer class="footer">
  <div class="nav-footer">
    <div class="nav-footer-left">
<p>EnrichGT</p>
</div>   
    <div class="nav-footer-center">
      &nbsp;
    <div class="toc-actions d-sm-block d-md-none"><ul><li><a href="https://github.com/ZhimingYe/EnrichGT_website/blob/main/visualize.qmd" class="toc-action"><i class="bi bi-github"></i>View source</a></li><li><a href="https://github.com/ZhimingYe/EnrichGT_website/issues/new" class="toc-action"><i class="bi empty"></i>Report an issue</a></li></ul></div></div>
    <div class="nav-footer-right">
<p>Version 2.2</p>
</div>
  </div>
</footer>




</body></html>