<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<meta name="description" content="EnrichGT">
<title>Clustering and viewing ORA results • EnrichGT</title>
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- bootstrap-toc --><script src="https://cdn.jsdelivr.net/gh/afeld/bootstrap-toc@v1.0.1/dist/bootstrap-toc.min.js" integrity="sha256-4veVQbu7//Lk5TSmc7YV48MxtMy98e26cf5MrgZYnwo=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Clustering and viewing ORA results">
<meta property="og:description" content="EnrichGT">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-light navbar-expand-lg bg-light"><div class="container">
    
    <a class="navbar-brand me-2" href="../index.html">EnrichGT</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.2.3</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item">
  <a class="nav-link" href="../reference/index.html">Reference</a>
</li>
<li class="active nav-item dropdown">
  <a href="#" class="nav-link dropdown-toggle" data-bs-toggle="dropdown" role="button" aria-expanded="false" aria-haspopup="true" id="dropdown-articles">Articles</a>
  <div class="dropdown-menu" aria-labelledby="dropdown-articles">
    <a class="dropdown-item" href="../articles/c_ora.html">Clustering and viewing ORA results</a>
    <a class="dropdown-item" href="../articles/c_gsea.html">Clustering and viewing GSEA results</a>
    <a class="dropdown-item" href="../articles/c_multi_grp.html">Clustering and viewing compareClusterResult</a>
  </div>
</li>
      </ul>
<form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="../search.json" id="search-input" placeholder="Search for" autocomplete="off">
</form>

      <ul class="navbar-nav"></ul>
</div>

    
  </div>
</nav><div class="container template-article">




<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="" class="logo" alt=""><h1>Clustering and viewing ORA results</h1>
            
      
      
      <div class="d-none name"><code>c_ora.Rmd</code></div>
    </div>

    
    
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://zhimingye.github.io/EnrichGT/">EnrichGT</a></span><span class="op">)</span></span>
<span><span class="co">#&gt; Loading required package: dplyr</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Attaching package: 'dplyr'</span></span>
<span><span class="co">#&gt; The following objects are masked from 'package:stats':</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;     filter, lag</span></span>
<span><span class="co">#&gt; The following objects are masked from 'package:base':</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;     intersect, setdiff, setequal, union</span></span>
<span><span class="co">#&gt; Loading required package: tibble</span></span>
<span><span class="co">#&gt; Loading required package: gt</span></span>
<span><span class="co">#&gt; View your enrichment result by entring `EnrichGT(result)`</span></span>
<span><span class="co">#&gt; by Zhiming Ye, https://github.com/ZhimingYe/EnrichGT</span></span></code></pre></div>
<div class="section level2">
<h2 id="summary">Summary<a class="anchor" aria-label="anchor" href="#summary"></a>
</h2>
<p>For overexpression enrichment analysis (What is overexpression
enrichment analysis? See webpage: <a href="https://yulab-smu.top/biomedical-knowledge-mining-book/enrichment-overview.html" class="external-link uri">https://yulab-smu.top/biomedical-knowledge-mining-book/enrichment-overview.html</a>),
EnrichGT can read results from <code>enrichResult</code> generated by
<code>clusterProfiler</code> or an exported table from clusterProfiler
enrichment (including at least the columns “ID”, “Description”,
“GeneRatio”, “pvalue”, “p.adjust”, “geneID” and “Count”).
<code>EnrichGT</code> performs term frequency statistics and automatic
clustering based on genes enriched in hits, returns an S4 object, and
outputs a beautifully formatted result wrapped by gt.</p>
</div>
<div class="section level2">
<h2 id="generating-enriched-result">Generating enriched Result<a class="anchor" aria-label="anchor" href="#generating-enriched-result"></a>
</h2>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/message.html" class="external-link">suppressMessages</a></span><span class="op">(</span><span class="op">{</span></span>
<span>  <span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://tidyverse.tidyverse.org" class="external-link">tidyverse</a></span><span class="op">)</span></span>
<span>  <span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://gt.rstudio.com" class="external-link">gt</a></span><span class="op">)</span></span>
<span>  <span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://yulab-smu.top/biomedical-knowledge-mining-book/" class="external-link">clusterProfiler</a></span><span class="op">)</span></span>
<span>  <span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">org.Hs.eg.db</span><span class="op">)</span></span>
<span>  <span class="op">}</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">geneList</span>, package<span class="op">=</span><span class="st">"DOSE"</span><span class="op">)</span></span>
<span><span class="va">gene</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">geneList</span><span class="op">)</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/MathFun.html" class="external-link">abs</a></span><span class="op">(</span><span class="va">geneList</span><span class="op">)</span> <span class="op">&gt;</span> <span class="fl">2</span><span class="op">]</span></span>
<span><span class="va">ego</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/clusterProfiler/man/enrichGO.html" class="external-link">enrichGO</a></span><span class="op">(</span>gene          <span class="op">=</span> <span class="va">gene</span>,</span>
<span>                  universe      <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">geneList</span><span class="op">)</span>,</span>
<span>                  OrgDb         <span class="op">=</span> <span class="va">org.Hs.eg.db</span>,</span>
<span>                  ont           <span class="op">=</span> <span class="st">"BP"</span>,</span>
<span>                  pAdjustMethod <span class="op">=</span> <span class="st">"BH"</span>,</span>
<span>                  pvalueCutoff  <span class="op">=</span> <span class="fl">0.5</span>,</span>
<span>                  qvalueCutoff  <span class="op">=</span> <span class="fl">0.5</span>,</span>
<span>                  readable      <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="go-enrichgt">go EnrichGT!<a class="anchor" aria-label="anchor" href="#go-enrichgt"></a>
</h2>
<div class="section level3">
<h3 id="from-enrichresult-object">From <code>enrichResult</code> object<a class="anchor" aria-label="anchor" href="#from-enrichresult-object"></a>
</h3>
<div class="section level4">
<h4 id="do-enriching">Do enriching<a class="anchor" aria-label="anchor" href="#do-enriching"></a>
</h4>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">ego</span></span>
<span><span class="co">#&gt; #</span></span>
<span><span class="co">#&gt; # over-representation test</span></span>
<span><span class="co">#&gt; #</span></span>
<span><span class="co">#&gt; #...@organism     Homo sapiens </span></span>
<span><span class="co">#&gt; #...@ontology     BP </span></span>
<span><span class="co">#&gt; #...@keytype      ENTREZID </span></span>
<span><span class="co">#&gt; #...@gene     chr [1:207] "4312" "8318" "10874" "55143" "55388" "991" "6280" "2305" ...</span></span>
<span><span class="co">#&gt; #...pvalues adjusted by 'BH' with cutoff &lt;0.5 </span></span>
<span><span class="co">#&gt; #...470 enriched terms found</span></span>
<span><span class="co">#&gt; 'data.frame':    470 obs. of  9 variables:</span></span>
<span><span class="co">#&gt;  $ ID         : chr  "GO:0098813" "GO:0007059" "GO:0000070" "GO:0000819" ...</span></span>
<span><span class="co">#&gt;  $ Description: chr  "nuclear chromosome segregation" "chromosome segregation" "mitotic sister chromatid segregation" "sister chromatid segregation" ...</span></span>
<span><span class="co">#&gt;  $ GeneRatio  : chr  "33/196" "37/196" "27/196" "29/196" ...</span></span>
<span><span class="co">#&gt;  $ BgRatio    : chr  "236/11580" "316/11580" "151/11580" "184/11580" ...</span></span>
<span><span class="co">#&gt;  $ pvalue     : num  2.78e-21 4.45e-21 2.10e-20 2.92e-20 6.84e-20 ...</span></span>
<span><span class="co">#&gt;  $ p.adjust   : num  6.69e-18 6.69e-18 2.10e-17 2.20e-17 4.12e-17 ...</span></span>
<span><span class="co">#&gt;  $ qvalue     : num  6.16e-18 6.16e-18 1.94e-17 2.02e-17 3.79e-17 ...</span></span>
<span><span class="co">#&gt;  $ geneID     : chr  "CDCA8/CDC20/KIF23/CENPE/MYBL2/CCNB2/NDC80/TOP2A/NCAPH/ASPM/DLGAP5/UBE2C/NUSAP1/TPX2/TACC3/NEK2/CDK1/MAD2L1/KIF1"| __truncated__ "CDCA8/CDC20/KIF23/CENPE/MYBL2/CCNB2/NDC80/TOP2A/NCAPH/ASPM/DLGAP5/UBE2C/HJURP/SKA1/NUSAP1/TPX2/TACC3/NEK2/CENPM"| __truncated__ "CDCA8/CDC20/KIF23/CENPE/MYBL2/NDC80/NCAPH/DLGAP5/UBE2C/NUSAP1/TPX2/NEK2/CDK1/MAD2L1/KIF18A/CDT1/BIRC5/KIF11/TTK"| __truncated__ "CDCA8/CDC20/KIF23/CENPE/MYBL2/NDC80/TOP2A/NCAPH/DLGAP5/UBE2C/NUSAP1/TPX2/TACC3/NEK2/CDK1/MAD2L1/KIF18A/CDT1/BIR"| __truncated__ ...</span></span>
<span><span class="co">#&gt;  $ Count      : int  33 37 27 29 31 36 37 23 38 37 ...</span></span>
<span><span class="co">#&gt; #...Citation</span></span>
<span><span class="co">#&gt;  T Wu, E Hu, S Xu, M Chen, P Guo, Z Dai, T Feng, L Zhou, W Tang, L Zhan, X Fu, S Liu, X Bo, and G Yu.</span></span>
<span><span class="co">#&gt;  clusterProfiler 4.0: A universal enrichment tool for interpreting omics data.</span></span>
<span><span class="co">#&gt;  The Innovation. 2021, 2(3):100141</span></span></code></pre></div>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">obj</span> <span class="op">&lt;-</span> <span class="va">ego</span> <span class="op">|&gt;</span> <span class="fu"><a href="../reference/EnrichGT.html">EnrichGT</a></span><span class="op">(</span>P.adj <span class="op">=</span> <span class="fl">0.2</span>,ClusterNum <span class="op">=</span> <span class="fl">10</span>,nTop <span class="op">=</span> <span class="fl">10</span><span class="op">)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; =====[SUGGESTION]=====</span></span>
<span><span class="co">#&gt; You are passing an object from GO Enrichment.</span></span>
<span><span class="co">#&gt; Please ensure that `obj |&gt; clusterProfiler::simplify()` is executed, to pre-simplify result,</span></span>
<span><span class="co">#&gt; For better enriched result.</span></span>
<span><span class="co">#&gt; Loading required package: proxy</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Attaching package: 'proxy'</span></span>
<span><span class="co">#&gt; The following object is masked from 'package:IRanges':</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;     as.matrix</span></span>
<span><span class="co">#&gt; The following object is masked from 'package:S4Vectors':</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;     as.matrix</span></span>
<span><span class="co">#&gt; The following objects are masked from 'package:stats':</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;     as.dist, dist</span></span>
<span><span class="co">#&gt; The following object is masked from 'package:base':</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;     as.matrix</span></span>
<span><span class="co">#&gt; Loading required package: text2vec</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Attaching package: 'text2vec'</span></span>
<span><span class="co">#&gt; The following object is masked from 'package:BiocGenerics':</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;     normalize</span></span>
<span><span class="co">#&gt; Joining with `by = join_by(ID)`</span></span></code></pre></div>
<p><img src="c_ora_files/figure-html/unnamed-chunk-4-1.png" width="672"></p>
<p>You can define how many clusters and how many top terms would be
shown. And you can adjust the p-adj cut-off.</p>
</div>
<div class="section level4">
<h4 id="understanding-enrichgt_obj">Understanding <code>EnrichGT_obj</code><a class="anchor" aria-label="anchor" href="#understanding-enrichgt_obj"></a>
</h4>
<p>The created <code>obj</code> contains several slots:</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/str.html" class="external-link">str</a></span><span class="op">(</span><span class="va">obj</span>,max.level<span class="op">=</span><span class="fl">2</span><span class="op">)</span></span>
<span><span class="co">#&gt; Formal class 'EnrichGT_obj' [package "EnrichGT"] with 4 slots</span></span>
<span><span class="co">#&gt;   ..@ enriched_result : tibble [34 × 7] (S3: tbl_df/tbl/data.frame)</span></span>
<span><span class="co">#&gt;   ..@ gt_object       :List of 17</span></span>
<span><span class="co">#&gt;   .. ..- attr(*, "class")= chr [1:2] "gt_tbl" "list"</span></span>
<span><span class="co">#&gt;   ..@ gene_modules    :List of 7</span></span>
<span><span class="co">#&gt;   ..@ pathway_clusters:List of 7</span></span></code></pre></div>
<p>Simply enter <code>obj</code> in console also can returns a rendered
gt view in <code>viewer</code> plane in RStudio.</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">obj</span></span></code></pre></div>
<p><code>gt</code> object is storaged in <code>gt_object</code>. You can
use all <code>gt</code> functions on it like gt::gtsave() by calling
<code>obj@gt_object</code>.</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">obj</span><span class="op">@</span><span class="va">gt_object</span></span></code></pre></div>
<div id="ivzpyyeanu" style="padding-left:0px;padding-right:0px;padding-top:10px;padding-bottom:10px;overflow-x:auto;overflow-y:auto;width:auto;height:auto;">
<style>#ivzpyyeanu table {
  font-family: system-ui, 'Segoe UI', Roboto, Helvetica, Arial, sans-serif, 'Apple Color Emoji', 'Segoe UI Emoji', 'Segoe UI Symbol', 'Noto Color Emoji';
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
}

#ivzpyyeanu thead, #ivzpyyeanu tbody, #ivzpyyeanu tfoot, #ivzpyyeanu tr, #ivzpyyeanu td, #ivzpyyeanu th {
  border-style: none;
}

#ivzpyyeanu p {
  margin: 0;
  padding: 0;
}

#ivzpyyeanu .gt_table {
  display: table;
  border-collapse: collapse;
  line-height: normal;
  margin-left: auto;
  margin-right: auto;
  color: #333333;
  font-size: 16px;
  font-weight: normal;
  font-style: normal;
  background-color: #FFFFFF;
  width: auto;
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #A8A8A8;
  border-right-style: none;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #A8A8A8;
  border-left-style: none;
  border-left-width: 2px;
  border-left-color: #D3D3D3;
}

#ivzpyyeanu .gt_caption {
  padding-top: 4px;
  padding-bottom: 4px;
}

#ivzpyyeanu .gt_title {
  color: #333333;
  font-size: 125%;
  font-weight: initial;
  padding-top: 4px;
  padding-bottom: 4px;
  padding-left: 5px;
  padding-right: 5px;
  border-bottom-color: #FFFFFF;
  border-bottom-width: 0;
}

#ivzpyyeanu .gt_subtitle {
  color: #333333;
  font-size: 85%;
  font-weight: initial;
  padding-top: 3px;
  padding-bottom: 5px;
  padding-left: 5px;
  padding-right: 5px;
  border-top-color: #FFFFFF;
  border-top-width: 0;
}

#ivzpyyeanu .gt_heading {
  background-color: #FFFFFF;
  text-align: center;
  border-bottom-color: #FFFFFF;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
}

#ivzpyyeanu .gt_bottom_border {
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
}

#ivzpyyeanu .gt_col_headings {
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
}

#ivzpyyeanu .gt_col_heading {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: normal;
  text-transform: inherit;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
  vertical-align: bottom;
  padding-top: 5px;
  padding-bottom: 6px;
  padding-left: 5px;
  padding-right: 5px;
  overflow-x: hidden;
}

#ivzpyyeanu .gt_column_spanner_outer {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: normal;
  text-transform: inherit;
  padding-top: 0;
  padding-bottom: 0;
  padding-left: 4px;
  padding-right: 4px;
}

#ivzpyyeanu .gt_column_spanner_outer:first-child {
  padding-left: 0;
}

#ivzpyyeanu .gt_column_spanner_outer:last-child {
  padding-right: 0;
}

#ivzpyyeanu .gt_column_spanner {
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  vertical-align: bottom;
  padding-top: 5px;
  padding-bottom: 5px;
  overflow-x: hidden;
  display: inline-block;
  width: 100%;
}

#ivzpyyeanu .gt_spanner_row {
  border-bottom-style: hidden;
}

#ivzpyyeanu .gt_group_heading {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  text-transform: inherit;
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
  vertical-align: middle;
  text-align: left;
}

#ivzpyyeanu .gt_empty_group_heading {
  padding: 0.5px;
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  vertical-align: middle;
}

#ivzpyyeanu .gt_from_md > :first-child {
  margin-top: 0;
}

#ivzpyyeanu .gt_from_md > :last-child {
  margin-bottom: 0;
}

#ivzpyyeanu .gt_row {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  margin: 10px;
  border-top-style: solid;
  border-top-width: 1px;
  border-top-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
  vertical-align: middle;
  overflow-x: hidden;
}

#ivzpyyeanu .gt_stub {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  text-transform: inherit;
  border-right-style: solid;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
  padding-left: 5px;
  padding-right: 5px;
}

#ivzpyyeanu .gt_stub_row_group {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  text-transform: inherit;
  border-right-style: solid;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
  padding-left: 5px;
  padding-right: 5px;
  vertical-align: top;
}

#ivzpyyeanu .gt_row_group_first td {
  border-top-width: 2px;
}

#ivzpyyeanu .gt_row_group_first th {
  border-top-width: 2px;
}

#ivzpyyeanu .gt_summary_row {
  color: #333333;
  background-color: #FFFFFF;
  text-transform: inherit;
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
}

#ivzpyyeanu .gt_first_summary_row {
  border-top-style: solid;
  border-top-color: #D3D3D3;
}

#ivzpyyeanu .gt_first_summary_row.thick {
  border-top-width: 2px;
}

#ivzpyyeanu .gt_last_summary_row {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
}

#ivzpyyeanu .gt_grand_summary_row {
  color: #333333;
  background-color: #FFFFFF;
  text-transform: inherit;
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
}

#ivzpyyeanu .gt_first_grand_summary_row {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  border-top-style: double;
  border-top-width: 6px;
  border-top-color: #D3D3D3;
}

#ivzpyyeanu .gt_last_grand_summary_row_top {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  border-bottom-style: double;
  border-bottom-width: 6px;
  border-bottom-color: #D3D3D3;
}

#ivzpyyeanu .gt_striped {
  background-color: rgba(128, 128, 128, 0.05);
}

#ivzpyyeanu .gt_table_body {
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
}

#ivzpyyeanu .gt_footnotes {
  color: #333333;
  background-color: #FFFFFF;
  border-bottom-style: none;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 2px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
}

#ivzpyyeanu .gt_footnote {
  margin: 0px;
  font-size: 90%;
  padding-top: 4px;
  padding-bottom: 4px;
  padding-left: 5px;
  padding-right: 5px;
}

#ivzpyyeanu .gt_sourcenotes {
  color: #333333;
  background-color: #FFFFFF;
  border-bottom-style: none;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 2px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
}

#ivzpyyeanu .gt_sourcenote {
  font-size: 90%;
  padding-top: 4px;
  padding-bottom: 4px;
  padding-left: 5px;
  padding-right: 5px;
}

#ivzpyyeanu .gt_left {
  text-align: left;
}

#ivzpyyeanu .gt_center {
  text-align: center;
}

#ivzpyyeanu .gt_right {
  text-align: right;
  font-variant-numeric: tabular-nums;
}

#ivzpyyeanu .gt_font_normal {
  font-weight: normal;
}

#ivzpyyeanu .gt_font_bold {
  font-weight: bold;
}

#ivzpyyeanu .gt_font_italic {
  font-style: italic;
}

#ivzpyyeanu .gt_super {
  font-size: 65%;
}

#ivzpyyeanu .gt_footnote_marks {
  font-size: 75%;
  vertical-align: 0.4em;
  position: initial;
}

#ivzpyyeanu .gt_asterisk {
  font-size: 100%;
  vertical-align: 0;
}

#ivzpyyeanu .gt_indent_1 {
  text-indent: 5px;
}

#ivzpyyeanu .gt_indent_2 {
  text-indent: 10px;
}

#ivzpyyeanu .gt_indent_3 {
  text-indent: 15px;
}

#ivzpyyeanu .gt_indent_4 {
  text-indent: 20px;
}

#ivzpyyeanu .gt_indent_5 {
  text-indent: 25px;
}

#ivzpyyeanu .katex-display {
  display: inline-flex !important;
  margin-bottom: 0.75em !important;
}

#ivzpyyeanu div.Reactable > div.rt-table > div.rt-thead > div.rt-tr.rt-tr-group-header > div.rt-th-group:after {
  height: 0px !important;
}
</style>
<table class="table gt_table" data-quarto-disable-processing="false" data-quarto-bootstrap="false">
<thead>
<tr class="gt_heading">
<td colspan="5" class="gt_heading gt_title gt_font_normal" style>Parse form: ego</td>
    </tr>
<tr class="gt_heading">
<td colspan="5" class="gt_heading gt_subtitle gt_font_normal gt_bottom_border" style>Split into 10 Clusters. Generated by github@zhimingye/EnrichGT</td>
    </tr>
<tr class="gt_col_headings">
<th class="gt_col_heading gt_columns_bottom_border gt_left" rowspan="1" colspan="1" scope="col" id="Description">Description</th>
      <th class="gt_col_heading gt_columns_bottom_border gt_right" rowspan="1" colspan="1" scope="col" id="Count">Count</th>
      <th class="gt_col_heading gt_columns_bottom_border gt_right" rowspan="1" colspan="1" scope="col" id="PCT">PCT</th>
      <th class="gt_col_heading gt_columns_bottom_border gt_right" rowspan="1" colspan="1" scope="col" id="Padj">Padj</th>
      <th class="gt_col_heading gt_columns_bottom_border gt_left" rowspan="1" colspan="1" scope="col" id="geneID">geneID</th>
    </tr>
</thead>
<tbody class="gt_table_body">
<tr class="gt_group_heading_row">
<th colspan="5" class="gt_group_heading" scope="colgroup" id="Cluster_1">Cluster_1</th>
    </tr>
<tr class="gt_row_group_first">
<td headers="Cluster_1  Description" class="gt_row gt_left" style="font-size: 13px;">
<div style="line-height:14px"><span style="font-weight:bold;font-variant:small-caps;color:#000000;font-size:14px">nuclear chromosome segregation</span></div>
<div style="line-height:10px"><span style="font-weight:bold;color:#BEBEBE;font-size:10px">GO:0098813</span></div>
</td>
<td headers="Cluster_1  Count" class="gt_row gt_right" style="background-color: #EAE1EF; color: #000000; font-size: 13px;">33</td>
<td headers="Cluster_1  PCT" class="gt_row gt_right" style="background-color: #DB6CA7; color: #FFFFFF; font-size: 13px;">17.0</td>
<td headers="Cluster_1  Padj" class="gt_row gt_right" style="background-color: #D53E4F; color: #FFFFFF; font-size: 13px;">6.7e-18</td>
<td headers="Cluster_1  geneID" class="gt_row gt_left" style="font-size: 13px;">CDCA8, CDC20, KIF23, CENPE, MYBL2, CCNB2, NDC80, TOP2A, NCAPH, ASPM, DLGAP5, UBE2C, NUSAP1, TPX2, TACC3, NEK2, CDK1, MAD2L1, KIF18A, CDT1, BIRC5, KIF11, TTK, NCAPG, AURKB, TRIP13, PRC1, KIFC1, KIF18B, AURKA, CCNB1, KIF4A, PTTG1</td>
</tr>
<tr>
<td headers="Cluster_1  Description" class="gt_row gt_left" style="font-size: 13px;">
<div style="line-height:14px"><span style="font-weight:bold;font-variant:small-caps;color:#000000;font-size:14px">chromosome segregation</span></div>
<div style="line-height:10px"><span style="font-weight:bold;color:#BEBEBE;font-size:10px">GO:0007059</span></div>
</td>
<td headers="Cluster_1  Count" class="gt_row gt_right" style="background-color: #FBF3F9; color: #000000; font-size: 13px;">37</td>
<td headers="Cluster_1  PCT" class="gt_row gt_right" style="background-color: #C51B7D; color: #FFFFFF; font-size: 13px;">19.0</td>
<td headers="Cluster_1  Padj" class="gt_row gt_right" style="background-color: #D53E4F; color: #FFFFFF; font-size: 13px;">6.7e-18</td>
<td headers="Cluster_1  geneID" class="gt_row gt_left" style="font-size: 13px;">CDCA8, CDC20, KIF23, CENPE, MYBL2, CCNB2, NDC80, TOP2A, NCAPH, ASPM, DLGAP5, UBE2C, HJURP, SKA1, NUSAP1, TPX2, TACC3, NEK2, CENPM, CENPN, CDK1, MAD2L1, KIF18A, CDT1, BIRC5, KIF11, TTK, NCAPG, AURKB, TRIP13, PRC1, KIFC1, KIF18B, AURKA, CCNB1, KIF4A, PTTG1</td>
</tr>
<tr>
<td headers="Cluster_1  Description" class="gt_row gt_left" style="font-size: 13px;">
<div style="line-height:14px"><span style="font-weight:bold;font-variant:small-caps;color:#000000;font-size:14px">mitotic sister chromatid segregation</span></div>
<div style="line-height:10px"><span style="font-weight:bold;color:#BEBEBE;font-size:10px">GO:0000070</span></div>
</td>
<td headers="Cluster_1  Count" class="gt_row gt_right" style="background-color: #C2CAE2; color: #000000; font-size: 13px;">27</td>
<td headers="Cluster_1  PCT" class="gt_row gt_right" style="background-color: #F3BCDD; color: #000000; font-size: 13px;">14.0</td>
<td headers="Cluster_1  Padj" class="gt_row gt_right" style="background-color: #D53E4F; color: #FFFFFF; font-size: 13px;">2.1e-17</td>
<td headers="Cluster_1  geneID" class="gt_row gt_left" style="font-size: 13px;">CDCA8, CDC20, KIF23, CENPE, MYBL2, NDC80, NCAPH, DLGAP5, UBE2C, NUSAP1, TPX2, NEK2, CDK1, MAD2L1, KIF18A, CDT1, BIRC5, KIF11, TTK, NCAPG, AURKB, TRIP13, PRC1, KIFC1, KIF18B, CCNB1, KIF4A</td>
</tr>
<tr>
<td headers="Cluster_1  Description" class="gt_row gt_left" style="font-size: 13px;">
<div style="line-height:14px"><span style="font-weight:bold;font-variant:small-caps;color:#000000;font-size:14px">sister chromatid segregation</span></div>
<div style="line-height:10px"><span style="font-weight:bold;color:#BEBEBE;font-size:10px">GO:0000819</span></div>
</td>
<td headers="Cluster_1  Count" class="gt_row gt_right" style="background-color: #D3D3E7; color: #000000; font-size: 13px;">29</td>
<td headers="Cluster_1  PCT" class="gt_row gt_right" style="background-color: #ECA4CD; color: #000000; font-size: 13px;">15.0</td>
<td headers="Cluster_1  Padj" class="gt_row gt_right" style="background-color: #D53E4F; color: #FFFFFF; font-size: 13px;">2.2e-17</td>
<td headers="Cluster_1  geneID" class="gt_row gt_left" style="font-size: 13px;">CDCA8, CDC20, KIF23, CENPE, MYBL2, NDC80, TOP2A, NCAPH, DLGAP5, UBE2C, NUSAP1, TPX2, TACC3, NEK2, CDK1, MAD2L1, KIF18A, CDT1, BIRC5, KIF11, TTK, NCAPG, AURKB, TRIP13, PRC1, KIFC1, KIF18B, CCNB1, KIF4A</td>
</tr>
<tr>
<td headers="Cluster_1  Description" class="gt_row gt_left" style="font-size: 13px;">
<div style="line-height:14px"><span style="font-weight:bold;font-variant:small-caps;color:#000000;font-size:14px">mitotic nuclear division</span></div>
<div style="line-height:10px"><span style="font-weight:bold;color:#BEBEBE;font-size:10px">GO:0140014</span></div>
</td>
<td headers="Cluster_1  Count" class="gt_row gt_right" style="background-color: #DEDAEB; color: #000000; font-size: 13px;">31</td>
<td headers="Cluster_1  PCT" class="gt_row gt_right" style="background-color: #E489BA; color: #000000; font-size: 13px;">16.0</td>
<td headers="Cluster_1  Padj" class="gt_row gt_right" style="background-color: #D53E4F; color: #FFFFFF; font-size: 13px;">4.1e-17</td>
<td headers="Cluster_1  geneID" class="gt_row gt_left" style="font-size: 13px;">CDCA8, CDC20, KIF23, CENPE, MYBL2, NDC80, NCAPH, DLGAP5, UBE2C, NUSAP1, TPX2, NEK2, UBE2S, CDK1, MAD2L1, KIF18A, CDT1, BIRC5, KIF11, TTK, NCAPG, AURKB, CHEK1, TRIP13, PRC1, KIFC1, KIF18B, AURKA, CCNB1, KIF4A, BMP4</td>
</tr>
<tr>
<td headers="Cluster_1  Description" class="gt_row gt_left" style="font-size: 13px;">
<div style="line-height:14px"><span style="font-weight:bold;font-variant:small-caps;color:#000000;font-size:14px">nuclear division</span></div>
<div style="line-height:10px"><span style="font-weight:bold;color:#BEBEBE;font-size:10px">GO:0000280</span></div>
</td>
<td headers="Cluster_1  Count" class="gt_row gt_right" style="background-color: #F7EEF6; color: #000000; font-size: 13px;">36</td>
<td headers="Cluster_1  PCT" class="gt_row gt_right" style="background-color: #D04B92; color: #FFFFFF; font-size: 13px;">18.0</td>
<td headers="Cluster_1  Padj" class="gt_row gt_right" style="background-color: #D53E4F; color: #FFFFFF; font-size: 13px;">5.3e-17</td>
<td headers="Cluster_1  geneID" class="gt_row gt_left" style="font-size: 13px;">CDCA8, CDC20, KIF23, CENPE, MYBL2, CCNB2, NDC80, TOP2A, NCAPH, ASPM, DLGAP5, UBE2C, NUSAP1, TPX2, NEK2, RAD51AP1, UBE2S, CDK1, MAD2L1, KIF18A, CDT1, BIRC5, KIF11, TTK, NCAPG, AURKB, CHEK1, TRIP13, PRC1, KIFC1, KIF18B, AURKA, CCNB1, KIF4A, PTTG1, BMP4</td>
</tr>
<tr>
<td headers="Cluster_1  Description" class="gt_row gt_left" style="font-size: 13px;">
<div style="line-height:14px"><span style="font-weight:bold;font-variant:small-caps;color:#000000;font-size:14px">organelle fission</span></div>
<div style="line-height:10px"><span style="font-weight:bold;color:#BEBEBE;font-size:10px">GO:0048285</span></div>
</td>
<td headers="Cluster_1  Count" class="gt_row gt_right" style="background-color: #FBF3F9; color: #000000; font-size: 13px;">37</td>
<td headers="Cluster_1  PCT" class="gt_row gt_right" style="background-color: #C51B7D; color: #FFFFFF; font-size: 13px;">19.0</td>
<td headers="Cluster_1  Padj" class="gt_row gt_right" style="background-color: #D53E4F; color: #FFFFFF; font-size: 13px;">1.7e-16</td>
<td headers="Cluster_1  geneID" class="gt_row gt_left" style="font-size: 13px;">CDCA8, CDC20, KIF23, CENPE, MYBL2, CCNB2, NDC80, TOP2A, NCAPH, ASPM, DLGAP5, UBE2C, NUSAP1, TPX2, NEK2, RAD51AP1, UBE2S, CDK1, MAD2L1, KIF18A, CDT1, BIRC5, KIF11, TTK, NCAPG, AURKB, CHEK1, TRIP13, PRC1, KIFC1, KIF18B, AURKA, CCNB1, KIF4A, PTTG1, BMP4, MAPT</td>
</tr>
<tr>
<td headers="Cluster_1  Description" class="gt_row gt_left" style="font-size: 13px;">
<div style="line-height:14px"><span style="font-weight:bold;font-variant:small-caps;color:#000000;font-size:14px">microtubule cytoskeleton organization involved in mitosis</span></div>
<div style="line-height:10px"><span style="font-weight:bold;color:#BEBEBE;font-size:10px">GO:1902850</span></div>
</td>
<td headers="Cluster_1  Count" class="gt_row gt_right" style="background-color: #9BB9D9; color: #000000; font-size: 13px;">23</td>
<td headers="Cluster_1  PCT" class="gt_row gt_right" style="background-color: #FDDFEF; color: #000000; font-size: 13px;">12.0</td>
<td headers="Cluster_1  Padj" class="gt_row gt_right" style="background-color: #D53E4F; color: #FFFFFF; font-size: 13px;">1.9e-14</td>
<td headers="Cluster_1  geneID" class="gt_row gt_left" style="font-size: 13px;">CDCA8, CDC20, KIF23, CENPE, MYBL2, NDC80, DLGAP5, NUSAP1, TPX2, TACC3, NEK2, CDK1, MAD2L1, PAX6, BIRC5, KIF11, TTK, AURKB, PRC1, KIFC1, AURKA, CCNB1, KIF4A</td>
</tr>
<tr>
<td headers="Cluster_1  Description" class="gt_row gt_left" style="font-size: 13px;">
<div style="line-height:14px"><span style="font-weight:bold;font-variant:small-caps;color:#000000;font-size:14px">chromosome organization</span></div>
<div style="line-height:10px"><span style="font-weight:bold;color:#BEBEBE;font-size:10px">GO:0051276</span></div>
</td>
<td headers="Cluster_1  Count" class="gt_row gt_right" style="background-color: #FFF7FB; color: #000000; font-size: 13px;">38</td>
<td headers="Cluster_1  PCT" class="gt_row gt_right" style="background-color: #C51B7D; color: #FFFFFF; font-size: 13px;">19.0</td>
<td headers="Cluster_1  Padj" class="gt_row gt_right" style="background-color: #D53E4F; color: #FFFFFF; font-size: 13px;">1.4e-13</td>
<td headers="Cluster_1  geneID" class="gt_row gt_left" style="font-size: 13px;">CDC45, CDCA8, CDC20, KIF23, CENPE, MYBL2, NDC80, TOP2A, NCAPH, DLGAP5, UBE2C, HJURP, NUSAP1, TPX2, TACC3, NEK2, CENPN, CDK1, MAD2L1, GINS1, KIF18A, CDT1, BIRC5, KIF11, EZH2, TTK, NCAPG, AURKB, GINS2, TRIP13, PRC1, KIFC1, KIF18B, CCNB1, KIF4A, MCM5, PTTG1, MAPT</td>
</tr>
<tr>
<td headers="Cluster_1  Description" class="gt_row gt_left" style="font-size: 13px;">
<div style="line-height:14px"><span style="font-weight:bold;font-variant:small-caps;color:#000000;font-size:14px">cell division</span></div>
<div style="line-height:10px"><span style="font-weight:bold;color:#BEBEBE;font-size:10px">GO:0051301</span></div>
</td>
<td headers="Cluster_1  Count" class="gt_row gt_right" style="background-color: #FBF3F9; color: #000000; font-size: 13px;">37</td>
<td headers="Cluster_1  PCT" class="gt_row gt_right" style="background-color: #C51B7D; color: #FFFFFF; font-size: 13px;">19.0</td>
<td headers="Cluster_1  Padj" class="gt_row gt_right" style="background-color: #D53E4F; color: #FFFFFF; font-size: 13px;">2.5e-12</td>
<td headers="Cluster_1  geneID" class="gt_row gt_left" style="font-size: 13px;">CDCA8, CDC20, KIF23, CENPE, CCNB2, NDC80, TOP2A, NCAPH, E2F8, ASPM, CEP55, UBE2C, SKA1, NUSAP1, CDCA3, TPX2, TACC3, NEK2, UBE2S, CCNA2, CDK1, ERCC6L, MAD2L1, PAX6, CDT1, BIRC5, KIF11, NCAPG, AURKB, PRC1, KIFC1, KIF18B, KIF20A, AURKA, CCNB1, KIF4A, PTTG1</td>
</tr>
<tr class="gt_group_heading_row">
<th colspan="5" class="gt_group_heading" scope="colgroup" id="Cluster_2">Cluster_2</th>
    </tr>
<tr class="gt_row_group_first">
<td headers="Cluster_2  Description" class="gt_row gt_left" style="font-size: 13px;">
<div style="line-height:14px"><span style="font-weight:bold;font-variant:small-caps;color:#000000;font-size:14px">female meiotic nuclear division</span></div>
<div style="line-height:10px"><span style="font-weight:bold;color:#BEBEBE;font-size:10px">GO:0007143</span></div>
</td>
<td headers="Cluster_2  Count" class="gt_row gt_right" style="background-color: #057068; color: #FFFFFF; font-size: 13px;">7</td>
<td headers="Cluster_2  PCT" class="gt_row gt_right" style="background-color: #62A42F; color: #FFFFFF; font-size: 13px;">3.6</td>
<td headers="Cluster_2  Padj" class="gt_row gt_right" style="background-color: #D53E4F; color: #FFFFFF; font-size: 13px;">1.0e-05</td>
<td headers="Cluster_2  geneID" class="gt_row gt_left" style="font-size: 13px;">CCNB2, NDC80, TOP2A, NCAPH, TTK, TRIP13, AURKA</td>
</tr>
<tr>
<td headers="Cluster_2  Description" class="gt_row gt_left" style="font-size: 13px;">
<div style="line-height:14px"><span style="font-weight:bold;font-variant:small-caps;color:#000000;font-size:14px">meiotic nuclear division</span></div>
<div style="line-height:10px"><span style="font-weight:bold;color:#BEBEBE;font-size:10px">GO:0140013</span></div>
</td>
<td headers="Cluster_2  Count" class="gt_row gt_right" style="background-color: #2388A4; color: #FFFFFF; font-size: 13px;">12</td>
<td headers="Cluster_2  PCT" class="gt_row gt_right" style="background-color: #9BCE64; color: #000000; font-size: 13px;">6.1</td>
<td headers="Cluster_2  Padj" class="gt_row gt_right" style="background-color: #D53E4F; color: #FFFFFF; font-size: 13px;">5.4e-05</td>
<td headers="Cluster_2  geneID" class="gt_row gt_left" style="font-size: 13px;">CDC20, CCNB2, NDC80, TOP2A, NCAPH, ASPM, RAD51AP1, KIF18A, TTK, TRIP13, AURKA, PTTG1</td>
</tr>
<tr>
<td headers="Cluster_2  Description" class="gt_row gt_left" style="font-size: 13px;">
<div style="line-height:14px"><span style="font-weight:bold;font-variant:small-caps;color:#000000;font-size:14px">meiotic cell cycle process</span></div>
<div style="line-height:10px"><span style="font-weight:bold;color:#BEBEBE;font-size:10px">GO:1903046</span></div>
</td>
<td headers="Cluster_2  Count" class="gt_row gt_right" style="background-color: #2388A4; color: #FFFFFF; font-size: 13px;">12</td>
<td headers="Cluster_2  PCT" class="gt_row gt_right" style="background-color: #9BCE64; color: #000000; font-size: 13px;">6.1</td>
<td headers="Cluster_2  Padj" class="gt_row gt_right" style="background-color: #D53E4F; color: #FFFFFF; font-size: 13px;">1.8e-04</td>
<td headers="Cluster_2  geneID" class="gt_row gt_left" style="font-size: 13px;">CDC20, CCNB2, NDC80, TOP2A, NCAPH, ASPM, RAD51AP1, KIF18A, TTK, TRIP13, AURKA, PTTG1</td>
</tr>
<tr>
<td headers="Cluster_2  Description" class="gt_row gt_left" style="font-size: 13px;">
<div style="line-height:14px"><span style="font-weight:bold;font-variant:small-caps;color:#000000;font-size:14px">meiotic chromosome segregation</span></div>
<div style="line-height:10px"><span style="font-weight:bold;color:#BEBEBE;font-size:10px">GO:0045132</span></div>
</td>
<td headers="Cluster_2  Count" class="gt_row gt_right" style="background-color: #057674; color: #FFFFFF; font-size: 13px;">8</td>
<td headers="Cluster_2  PCT" class="gt_row gt_right" style="background-color: #6DAD36; color: #FFFFFF; font-size: 13px;">4.1</td>
<td headers="Cluster_2  Padj" class="gt_row gt_right" style="background-color: #D53F4F; color: #FFFFFF; font-size: 13px;">2.3e-04</td>
<td headers="Cluster_2  geneID" class="gt_row gt_left" style="font-size: 13px;">CCNB2, NDC80, NCAPH, ASPM, TTK, TRIP13, AURKA, PTTG1</td>
</tr>
<tr>
<td headers="Cluster_2  Description" class="gt_row gt_left" style="font-size: 13px;">
<div style="line-height:14px"><span style="font-weight:bold;font-variant:small-caps;color:#000000;font-size:14px">cell cycle G2/M phase transition</span></div>
<div style="line-height:10px"><span style="font-weight:bold;color:#BEBEBE;font-size:10px">GO:0044839</span></div>
</td>
<td headers="Cluster_2  Count" class="gt_row gt_right" style="background-color: #188598; color: #FFFFFF; font-size: 13px;">11</td>
<td headers="Cluster_2  PCT" class="gt_row gt_right" style="background-color: #8FC655; color: #000000; font-size: 13px;">5.6</td>
<td headers="Cluster_2  Padj" class="gt_row gt_right" style="background-color: #D63F4F; color: #FFFFFF; font-size: 13px;">3.6e-04</td>
<td headers="Cluster_2  geneID" class="gt_row gt_left" style="font-size: 13px;">FOXM1, MELK, CCNB2, NDC80, CCNA2, CDK1, AURKB, CHEK1, DTL, AURKA, CCNB1</td>
</tr>
<tr>
<td headers="Cluster_2  Description" class="gt_row gt_left" style="font-size: 13px;">
<div style="line-height:14px"><span style="font-weight:bold;font-variant:small-caps;color:#000000;font-size:14px">meiotic cell cycle</span></div>
<div style="line-height:10px"><span style="font-weight:bold;color:#BEBEBE;font-size:10px">GO:0051321</span></div>
</td>
<td headers="Cluster_2  Count" class="gt_row gt_right" style="background-color: #2C8CAF; color: #FFFFFF; font-size: 13px;">13</td>
<td headers="Cluster_2  PCT" class="gt_row gt_right" style="background-color: #A8D672; color: #000000; font-size: 13px;">6.6</td>
<td headers="Cluster_2  Padj" class="gt_row gt_right" style="background-color: #D63F4F; color: #FFFFFF; font-size: 13px;">5.8e-04</td>
<td headers="Cluster_2  geneID" class="gt_row gt_left" style="font-size: 13px;">CDC20, CCNB2, NDC80, TOP2A, NCAPH, ASPM, NEK2, RAD51AP1, KIF18A, TTK, TRIP13, AURKA, PTTG1</td>
</tr>
<tr>
<td headers="Cluster_2  Description" class="gt_row gt_left" style="font-size: 13px;">
<div style="line-height:14px"><span style="font-weight:bold;font-variant:small-caps;color:#000000;font-size:14px">DNA replication</span></div>
<div style="line-height:10px"><span style="font-weight:bold;color:#BEBEBE;font-size:10px">GO:0006260</span></div>
</td>
<td headers="Cluster_2  Count" class="gt_row gt_right" style="background-color: #338FBB; color: #FFFFFF; font-size: 13px;">14</td>
<td headers="Cluster_2  PCT" class="gt_row gt_right" style="background-color: #B4DE81; color: #000000; font-size: 13px;">7.1</td>
<td headers="Cluster_2  Padj" class="gt_row gt_right" style="background-color: #D6404F; color: #FFFFFF; font-size: 13px;">8.8e-04</td>
<td headers="Cluster_2  geneID" class="gt_row gt_left" style="font-size: 13px;">CDC45, MCM10, E2F8, RRM2, CCNA2, CDK1, GINS1, CDT1, GINS2, CHAF1B, CHEK1, DTL, MCM5, DACH1</td>
</tr>
<tr>
<td headers="Cluster_2  Description" class="gt_row gt_left" style="font-size: 13px;">
<div style="line-height:14px"><span style="font-weight:bold;font-variant:small-caps;color:#000000;font-size:14px">meiosis I cell cycle process</span></div>
<div style="line-height:10px"><span style="font-weight:bold;color:#BEBEBE;font-size:10px">GO:0061982</span></div>
</td>
<td headers="Cluster_2  Count" class="gt_row gt_right" style="background-color: #057674; color: #FFFFFF; font-size: 13px;">8</td>
<td headers="Cluster_2  PCT" class="gt_row gt_right" style="background-color: #6DAD36; color: #FFFFFF; font-size: 13px;">4.1</td>
<td headers="Cluster_2  Padj" class="gt_row gt_right" style="background-color: #DA464E; color: #FFFFFF; font-size: 13px;">3.2e-03</td>
<td headers="Cluster_2  geneID" class="gt_row gt_left" style="font-size: 13px;">CDC20, CCNB2, NDC80, TOP2A, RAD51AP1, TRIP13, AURKA, PTTG1</td>
</tr>
<tr>
<td headers="Cluster_2  Description" class="gt_row gt_left" style="font-size: 13px;">
<div style="line-height:14px"><span style="font-weight:bold;font-variant:small-caps;color:#000000;font-size:14px">female gamete generation</span></div>
<div style="line-height:10px"><span style="font-weight:bold;color:#BEBEBE;font-size:10px">GO:0007292</span></div>
</td>
<td headers="Cluster_2  Count" class="gt_row gt_right" style="background-color: #047D81; color: #FFFFFF; font-size: 13px;">9</td>
<td headers="Cluster_2  PCT" class="gt_row gt_right" style="background-color: #78B63C; color: #000000; font-size: 13px;">4.6</td>
<td headers="Cluster_2  Padj" class="gt_row gt_right" style="background-color: #DA464E; color: #FFFFFF; font-size: 13px;">3.3e-03</td>
<td headers="Cluster_2  geneID" class="gt_row gt_left" style="font-size: 13px;">CCNB2, NDC80, TOP2A, NCAPH, ASPM, TTK, TRIP13, AURKA, PGR</td>
</tr>
<tr>
<td headers="Cluster_2  Description" class="gt_row gt_left" style="font-size: 13px;">
<div style="line-height:14px"><span style="font-weight:bold;font-variant:small-caps;color:#000000;font-size:14px">G2/M transition of mitotic cell cycle</span></div>
<div style="line-height:10px"><span style="font-weight:bold;color:#BEBEBE;font-size:10px">GO:0000086</span></div>
</td>
<td headers="Cluster_2  Count" class="gt_row gt_right" style="background-color: #047D81; color: #FFFFFF; font-size: 13px;">9</td>
<td headers="Cluster_2  PCT" class="gt_row gt_right" style="background-color: #78B63C; color: #000000; font-size: 13px;">4.6</td>
<td headers="Cluster_2  Padj" class="gt_row gt_right" style="background-color: #DB474D; color: #FFFFFF; font-size: 13px;">3.9e-03</td>
<td headers="Cluster_2  geneID" class="gt_row gt_left" style="font-size: 13px;">FOXM1, MELK, CCNA2, CDK1, AURKB, CHEK1, DTL, AURKA, CCNB1</td>
</tr>
<tr class="gt_group_heading_row">
<th colspan="5" class="gt_group_heading" scope="colgroup" id="Cluster_3">Cluster_3</th>
    </tr>
<tr class="gt_row_group_first">
<td headers="Cluster_3  Description" class="gt_row gt_left" style="font-size: 13px;">
<div style="line-height:14px"><span style="font-weight:bold;font-variant:small-caps;color:#000000;font-size:14px">antimicrobial humoral immune response mediated by antimicrobial peptide</span></div>
<div style="line-height:10px"><span style="font-weight:bold;color:#BEBEBE;font-size:10px">GO:0061844</span></div>
</td>
<td headers="Cluster_3  Count" class="gt_row gt_right" style="background-color: #047D81; color: #FFFFFF; font-size: 13px;">9</td>
<td headers="Cluster_3  PCT" class="gt_row gt_right" style="background-color: #78B63C; color: #000000; font-size: 13px;">4.6</td>
<td headers="Cluster_3  Padj" class="gt_row gt_right" style="background-color: #D53E4F; color: #FFFFFF; font-size: 13px;">4.4e-05</td>
<td headers="Cluster_3  geneID" class="gt_row gt_left" style="font-size: 13px;">S100A9, S100A7, CXCL10, CXCL13, CXCL11, CAMP, CXCL9, GNLY, CXCL14</td>
</tr>
<tr>
<td headers="Cluster_3  Description" class="gt_row gt_left" style="font-size: 13px;">
<div style="line-height:14px"><span style="font-weight:bold;font-variant:small-caps;color:#000000;font-size:14px">antimicrobial humoral response</span></div>
<div style="line-height:10px"><span style="font-weight:bold;color:#BEBEBE;font-size:10px">GO:0019730</span></div>
</td>
<td headers="Cluster_3  Count" class="gt_row gt_right" style="background-color: #08828D; color: #FFFFFF; font-size: 13px;">10</td>
<td headers="Cluster_3  PCT" class="gt_row gt_right" style="background-color: #83BE46; color: #000000; font-size: 13px;">5.1</td>
<td headers="Cluster_3  Padj" class="gt_row gt_right" style="background-color: #D53E4F; color: #FFFFFF; font-size: 13px;">1.2e-04</td>
<td headers="Cluster_3  geneID" class="gt_row gt_left" style="font-size: 13px;">S100A9, S100A7, CXCL10, CXCL13, CXCL11, CAMP, CXCL9, GNLY, RNASE4, CXCL14</td>
</tr>
<tr>
<td headers="Cluster_3  Description" class="gt_row gt_left" style="font-size: 13px;">
<div style="line-height:14px"><span style="font-weight:bold;font-variant:small-caps;color:#000000;font-size:14px">chemokine-mediated signaling pathway</span></div>
<div style="line-height:10px"><span style="font-weight:bold;color:#BEBEBE;font-size:10px">GO:0070098</span></div>
</td>
<td headers="Cluster_3  Count" class="gt_row gt_right" style="background-color: #057674; color: #FFFFFF; font-size: 13px;">8</td>
<td headers="Cluster_3  PCT" class="gt_row gt_right" style="background-color: #6DAD36; color: #FFFFFF; font-size: 13px;">4.1</td>
<td headers="Cluster_3  Padj" class="gt_row gt_right" style="background-color: #D8444E; color: #FFFFFF; font-size: 13px;">2.3e-03</td>
<td headers="Cluster_3  geneID" class="gt_row gt_left" style="font-size: 13px;">CXCL10, CXCL13, CXCL11, CXCL9, CCL18, CCL8, ACKR1, CX3CR1</td>
</tr>
<tr>
<td headers="Cluster_3  Description" class="gt_row gt_left" style="font-size: 13px;">
<div style="line-height:14px"><span style="font-weight:bold;font-variant:small-caps;color:#000000;font-size:14px">lymphocyte chemotaxis</span></div>
<div style="line-height:10px"><span style="font-weight:bold;color:#BEBEBE;font-size:10px">GO:0048247</span></div>
</td>
<td headers="Cluster_3  Count" class="gt_row gt_right" style="background-color: #036A5C; color: #FFFFFF; font-size: 13px;">6</td>
<td headers="Cluster_3  PCT" class="gt_row gt_right" style="background-color: #589B28; color: #FFFFFF; font-size: 13px;">3.1</td>
<td headers="Cluster_3  Padj" class="gt_row gt_right" style="background-color: #DA464E; color: #FFFFFF; font-size: 13px;">3.3e-03</td>
<td headers="Cluster_3  geneID" class="gt_row gt_left" style="font-size: 13px;">S100A7, CXCL10, CXCL13, CXCL11, CCL18, CCL8</td>
</tr>
<tr>
<td headers="Cluster_3  Description" class="gt_row gt_left" style="font-size: 13px;">
<div style="line-height:14px"><span style="font-weight:bold;font-variant:small-caps;color:#000000;font-size:14px">granulocyte chemotaxis</span></div>
<div style="line-height:10px"><span style="font-weight:bold;color:#BEBEBE;font-size:10px">GO:0071621</span></div>
</td>
<td headers="Cluster_3  Count" class="gt_row gt_right" style="background-color: #047D81; color: #FFFFFF; font-size: 13px;">9</td>
<td headers="Cluster_3  PCT" class="gt_row gt_right" style="background-color: #78B63C; color: #000000; font-size: 13px;">4.6</td>
<td headers="Cluster_3  Padj" class="gt_row gt_right" style="background-color: #DA464D; color: #FFFFFF; font-size: 13px;">3.5e-03</td>
<td headers="Cluster_3  geneID" class="gt_row gt_left" style="font-size: 13px;">S100A9, S100A8, S100A7, CXCL10, CXCL13, CXCL11, CXCL9, CCL18, CCL8</td>
</tr>
<tr>
<td headers="Cluster_3  Description" class="gt_row gt_left" style="font-size: 13px;">
<div style="line-height:14px"><span style="font-weight:bold;font-variant:small-caps;color:#000000;font-size:14px">response to chemokine</span></div>
<div style="line-height:10px"><span style="font-weight:bold;color:#BEBEBE;font-size:10px">GO:1990868</span></div>
</td>
<td headers="Cluster_3  Count" class="gt_row gt_right" style="background-color: #057674; color: #FFFFFF; font-size: 13px;">8</td>
<td headers="Cluster_3  PCT" class="gt_row gt_right" style="background-color: #6DAD36; color: #FFFFFF; font-size: 13px;">4.1</td>
<td headers="Cluster_3  Padj" class="gt_row gt_right" style="background-color: #DB474D; color: #FFFFFF; font-size: 13px;">3.8e-03</td>
<td headers="Cluster_3  geneID" class="gt_row gt_left" style="font-size: 13px;">CXCL10, CXCL13, CXCL11, CXCL9, CCL18, CCL8, ACKR1, CX3CR1</td>
</tr>
<tr>
<td headers="Cluster_3  Description" class="gt_row gt_left" style="font-size: 13px;">
<div style="line-height:14px"><span style="font-weight:bold;font-variant:small-caps;color:#000000;font-size:14px">cellular response to chemokine</span></div>
<div style="line-height:10px"><span style="font-weight:bold;color:#BEBEBE;font-size:10px">GO:1990869</span></div>
</td>
<td headers="Cluster_3  Count" class="gt_row gt_right" style="background-color: #057674; color: #FFFFFF; font-size: 13px;">8</td>
<td headers="Cluster_3  PCT" class="gt_row gt_right" style="background-color: #6DAD36; color: #FFFFFF; font-size: 13px;">4.1</td>
<td headers="Cluster_3  Padj" class="gt_row gt_right" style="background-color: #DB474D; color: #FFFFFF; font-size: 13px;">3.8e-03</td>
<td headers="Cluster_3  geneID" class="gt_row gt_left" style="font-size: 13px;">CXCL10, CXCL13, CXCL11, CXCL9, CCL18, CCL8, ACKR1, CX3CR1</td>
</tr>
<tr>
<td headers="Cluster_3  Description" class="gt_row gt_left" style="font-size: 13px;">
<div style="line-height:14px"><span style="font-weight:bold;font-variant:small-caps;color:#000000;font-size:14px">humoral immune response</span></div>
<div style="line-height:10px"><span style="font-weight:bold;color:#BEBEBE;font-size:10px">GO:0006959</span></div>
</td>
<td headers="Cluster_3  Count" class="gt_row gt_right" style="background-color: #2388A4; color: #FFFFFF; font-size: 13px;">12</td>
<td headers="Cluster_3  PCT" class="gt_row gt_right" style="background-color: #9BCE64; color: #000000; font-size: 13px;">6.1</td>
<td headers="Cluster_3  Padj" class="gt_row gt_right" style="background-color: #DB484D; color: #FFFFFF; font-size: 13px;">4.0e-03</td>
<td headers="Cluster_3  geneID" class="gt_row gt_left" style="font-size: 13px;">S100A9, S100A7, CXCL10, CXCL13, CXCL11, CAMP, CXCL9, GNLY, RNASE4, C7, CXCL14, GATA3</td>
</tr>
<tr>
<td headers="Cluster_3  Description" class="gt_row gt_left" style="font-size: 13px;">
<div style="line-height:14px"><span style="font-weight:bold;font-variant:small-caps;color:#000000;font-size:14px">neutrophil chemotaxis</span></div>
<div style="line-height:10px"><span style="font-weight:bold;color:#BEBEBE;font-size:10px">GO:0030593</span></div>
</td>
<td headers="Cluster_3  Count" class="gt_row gt_right" style="background-color: #057674; color: #FFFFFF; font-size: 13px;">8</td>
<td headers="Cluster_3  PCT" class="gt_row gt_right" style="background-color: #6DAD36; color: #FFFFFF; font-size: 13px;">4.1</td>
<td headers="Cluster_3  Padj" class="gt_row gt_right" style="background-color: #DB484D; color: #FFFFFF; font-size: 13px;">4.2e-03</td>
<td headers="Cluster_3  geneID" class="gt_row gt_left" style="font-size: 13px;">S100A9, S100A8, CXCL10, CXCL13, CXCL11, CXCL9, CCL18, CCL8</td>
</tr>
<tr>
<td headers="Cluster_3  Description" class="gt_row gt_left" style="font-size: 13px;">
<div style="line-height:14px"><span style="font-weight:bold;font-variant:small-caps;color:#000000;font-size:14px">response to molecule of bacterial origin</span></div>
<div style="line-height:10px"><span style="font-weight:bold;color:#BEBEBE;font-size:10px">GO:0002237</span></div>
</td>
<td headers="Cluster_3  Count" class="gt_row gt_right" style="background-color: #3D93C2; color: #FFFFFF; font-size: 13px;">15</td>
<td headers="Cluster_3  PCT" class="gt_row gt_right" style="background-color: #C0E593; color: #000000; font-size: 13px;">7.7</td>
<td headers="Cluster_3  Padj" class="gt_row gt_right" style="background-color: #DC494D; color: #FFFFFF; font-size: 13px;">4.6e-03</td>
<td headers="Cluster_3  geneID" class="gt_row gt_left" style="font-size: 13px;">S100A9, S100A8, S100A7, CXCL10, CXCL13, CXCL11, SLC7A5, CAMP, CXCL9, INAVA, IDO1, MAOB, PCK1, HMGCS2, CX3CR1</td>
</tr>
<tr class="gt_group_heading_row">
<th colspan="5" class="gt_group_heading" scope="colgroup" id="Cluster_4">Cluster_4</th>
    </tr>
<tr class="gt_row_group_first">
<td headers="Cluster_4  Description" class="gt_row gt_left" style="font-size: 13px;">
<div style="line-height:14px"><span style="font-weight:bold;font-variant:small-caps;color:#000000;font-size:14px">lens development in camera-type eye</span></div>
<div style="line-height:10px"><span style="font-weight:bold;color:#BEBEBE;font-size:10px">GO:0002088</span></div>
</td>
<td headers="Cluster_4  Count" class="gt_row gt_right" style="background-color: #016450; color: #FFFFFF; font-size: 13px;">5</td>
<td headers="Cluster_4  PCT" class="gt_row gt_right" style="background-color: #4D9221; color: #FFFFFF; font-size: 13px;">2.6</td>
<td headers="Cluster_4  Padj" class="gt_row gt_right" style="background-color: #3288BD; color: #FFFFFF; font-size: 13px;">1.5e-01</td>
<td headers="Cluster_4  geneID" class="gt_row gt_left" style="font-size: 13px;">PAX6, PLAAT1, BMP4, NDP, GATA3</td>
</tr>
<tr class="gt_group_heading_row">
<th colspan="5" class="gt_group_heading" scope="colgroup" id="Cluster_6">Cluster_6</th>
    </tr>
<tr class="gt_row_group_first">
<td headers="Cluster_6  Description" class="gt_row gt_left" style="font-size: 13px;">
<div style="line-height:14px"><span style="font-weight:bold;font-variant:small-caps;color:#000000;font-size:14px">microtubule-based movement</span></div>
<div style="line-height:10px"><span style="font-weight:bold;color:#BEBEBE;font-size:10px">GO:0007018</span></div>
</td>
<td headers="Cluster_6  Count" class="gt_row gt_right" style="background-color: #2388A4; color: #FFFFFF; font-size: 13px;">12</td>
<td headers="Cluster_6  PCT" class="gt_row gt_right" style="background-color: #9BCE64; color: #000000; font-size: 13px;">6.1</td>
<td headers="Cluster_6  Padj" class="gt_row gt_right" style="background-color: #F98B50; color: #000000; font-size: 13px;">3.1e-02</td>
<td headers="Cluster_6  geneID" class="gt_row gt_left" style="font-size: 13px;">KIF23, CENPE, DLGAP5, KIF18A, KIF11, KIFC1, KIF18B, KIF20A, KIF4A, MAPT, TRIM58, CFAP69</td>
</tr>
<tr class="gt_group_heading_row">
<th colspan="5" class="gt_group_heading" scope="colgroup" id="Cluster_7">Cluster_7</th>
    </tr>
<tr class="gt_row_group_first">
<td headers="Cluster_7  Description" class="gt_row gt_left" style="font-size: 13px;">
<div style="line-height:14px"><span style="font-weight:bold;font-variant:small-caps;color:#000000;font-size:14px">regulation of DNA biosynthetic process</span></div>
<div style="line-height:10px"><span style="font-weight:bold;color:#BEBEBE;font-size:10px">GO:2000278</span></div>
</td>
<td headers="Cluster_7  Count" class="gt_row gt_right" style="background-color: #036A5C; color: #FFFFFF; font-size: 13px;">6</td>
<td headers="Cluster_7  PCT" class="gt_row gt_right" style="background-color: #589B28; color: #FFFFFF; font-size: 13px;">3.1</td>
<td headers="Cluster_7  Padj" class="gt_row gt_right" style="background-color: #84CDA5; color: #000000; font-size: 13px;">1.2e-01</td>
<td headers="Cluster_7  geneID" class="gt_row gt_left" style="font-size: 13px;">NEK2, CCNA2, AURKB, CHEK1, ADIPOQ, DACH1</td>
</tr>
<tr class="gt_group_heading_row">
<th colspan="5" class="gt_group_heading" scope="colgroup" id="Cluster_9">Cluster_9</th>
    </tr>
<tr class="gt_row_group_first">
<td headers="Cluster_9  Description" class="gt_row gt_left" style="font-size: 13px;">
<div style="line-height:14px"><span style="font-weight:bold;font-variant:small-caps;color:#000000;font-size:14px">response to activity</span></div>
<div style="line-height:10px"><span style="font-weight:bold;color:#BEBEBE;font-size:10px">GO:0014823</span></div>
</td>
<td headers="Cluster_9  Count" class="gt_row gt_right" style="background-color: #016450; color: #FFFFFF; font-size: 13px;">5</td>
<td headers="Cluster_9  PCT" class="gt_row gt_right" style="background-color: #4D9221; color: #FFFFFF; font-size: 13px;">2.6</td>
<td headers="Cluster_9  Padj" class="gt_row gt_right" style="background-color: #FED884; color: #000000; font-size: 13px;">6.1e-02</td>
<td headers="Cluster_9  geneID" class="gt_row gt_left" style="font-size: 13px;">SLC7A5, CDK1, ADIPOQ, PCK1, CRY2</td>
</tr>
</tbody>
</table>
</div>
<p>Full enriched enriched (enrich^2) result is in
<code>obj@enriched_result</code>.</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">df1</span><span class="op">&lt;-</span><span class="va">obj</span><span class="op">@</span><span class="va">enriched_result</span> <span class="op">|&gt;</span> <span class="fu"><a href="https://tibble.tidyverse.org/reference/as_tibble.html" class="external-link">as_tibble</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="va">df1</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 34 × 7</span></span></span>
<span><span class="co">#&gt;    Description                         ID    Count Cluster   PCT     Padj geneID</span></span>
<span><span class="co">#&gt;    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>                               <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> <span style="color: #949494; font-style: italic;">&lt;int&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>    <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 1</span> nuclear chromosome segregation      GO:0…    33 Cluste…    17 6.70<span style="color: #949494;">e</span><span style="color: #BB0000;">-18</span> CDCA8…</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 2</span> chromosome segregation              GO:0…    37 Cluste…    19 6.70<span style="color: #949494;">e</span><span style="color: #BB0000;">-18</span> CDCA8…</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 3</span> mitotic sister chromatid segregati… GO:0…    27 Cluste…    14 2.10<span style="color: #949494;">e</span><span style="color: #BB0000;">-17</span> CDCA8…</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 4</span> sister chromatid segregation        GO:0…    29 Cluste…    15 2.20<span style="color: #949494;">e</span><span style="color: #BB0000;">-17</span> CDCA8…</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 5</span> mitotic nuclear division            GO:0…    31 Cluste…    16 4.10<span style="color: #949494;">e</span><span style="color: #BB0000;">-17</span> CDCA8…</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 6</span> nuclear division                    GO:0…    36 Cluste…    18 5.30<span style="color: #949494;">e</span><span style="color: #BB0000;">-17</span> CDCA8…</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 7</span> organelle fission                   GO:0…    37 Cluste…    19 1.7 <span style="color: #949494;">e</span><span style="color: #BB0000;">-16</span> CDCA8…</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 8</span> microtubule cytoskeleton organizat… GO:1…    23 Cluste…    12 1.9 <span style="color: #949494;">e</span><span style="color: #BB0000;">-14</span> CDCA8…</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 9</span> chromosome organization             GO:0…    38 Cluste…    19 1.4 <span style="color: #949494;">e</span><span style="color: #BB0000;">-13</span> CDC45…</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">10</span> cell division                       GO:0…    37 Cluste…    19 2.5 <span style="color: #949494;">e</span><span style="color: #BB0000;">-12</span> CDCA8…</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># ℹ 24 more rows</span></span></span></code></pre></div>
<p>We also generated gene modules and pathway clusters based on
clustering results. You can assess them by <code>obj@gene_modules</code>
and <code>obj@pathway_clusters</code>.</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">obj</span><span class="op">@</span><span class="va">gene_modules</span><span class="op">)</span></span>
<span><span class="co">#&gt; $Cluster_1</span></span>
<span><span class="co">#&gt;  [1] "ASPM"     "AURKA"    "AURKB"    "BIRC5"    "BMP4"     "CCNA2"   </span></span>
<span><span class="co">#&gt;  [7] "CCNB1"    "CCNB2"    "CDC20"    "CDC45"    "CDCA3"    "CDCA8"   </span></span>
<span><span class="co">#&gt; [13] "CDK1"     "CDT1"     "CENPE"    "CENPM"    "CENPN"    "CEP55"   </span></span>
<span><span class="co">#&gt; [19] "CHEK1"    "DLGAP5"   "E2F8"     "ERCC6L"   "EZH2"     "GINS1"   </span></span>
<span><span class="co">#&gt; [25] "GINS2"    "HJURP"    "KIF11"    "KIF18A"   "KIF18B"   "KIF20A"  </span></span>
<span><span class="co">#&gt; [31] "KIF23"    "KIF4A"    "KIFC1"    "MAD2L1"   "MAPT"     "MCM5"    </span></span>
<span><span class="co">#&gt; [37] "MYBL2"    "NCAPG"    "NCAPH"    "NDC80"    "NEK2"     "NUSAP1"  </span></span>
<span><span class="co">#&gt; [43] "PAX6"     "PRC1"     "PTTG1"    "RAD51AP1" "SKA1"     "TACC3"   </span></span>
<span><span class="co">#&gt; [49] "TOP2A"    "TPX2"     "TRIP13"   "TTK"      "UBE2C"    "UBE2S"   </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; $Cluster_2</span></span>
<span><span class="co">#&gt;  [1] "ASPM"     "AURKA"    "AURKB"    "CCNA2"    "CCNB1"    "CCNB2"   </span></span>
<span><span class="co">#&gt;  [7] "CDC20"    "CDC45"    "CDK1"     "CDT1"     "CHAF1B"   "CHEK1"   </span></span>
<span><span class="co">#&gt; [13] "DACH1"    "DTL"      "E2F8"     "FOXM1"    "GINS1"    "GINS2"   </span></span>
<span><span class="co">#&gt; [19] "KIF18A"   "MCM10"    "MCM5"     "MELK"     "NCAPH"    "NDC80"   </span></span>
<span><span class="co">#&gt; [25] "NEK2"     "PGR"      "PTTG1"    "RAD51AP1" "RRM2"     "TOP2A"   </span></span>
<span><span class="co">#&gt; [31] "TRIP13"   "TTK"     </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; $Cluster_3</span></span>
<span><span class="co">#&gt;  [1] "ACKR1"  "C7"     "CAMP"   "CCL18"  "CCL8"   "CX3CR1" "CXCL10" "CXCL11"</span></span>
<span><span class="co">#&gt;  [9] "CXCL13" "CXCL14" "CXCL9"  "GATA3"  "GNLY"   "HMGCS2" "IDO1"   "INAVA" </span></span>
<span><span class="co">#&gt; [17] "MAOB"   "PCK1"   "RNASE4" "S100A7" "S100A8" "S100A9" "SLC7A5"</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; $Cluster_4</span></span>
<span><span class="co">#&gt; [1] "BMP4"   "GATA3"  "NDP"    "PAX6"   "PLAAT1"</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; $Cluster_6</span></span>
<span><span class="co">#&gt;  [1] "CENPE"  "CFAP69" "DLGAP5" "KIF11"  "KIF18A" "KIF18B" "KIF20A" "KIF23" </span></span>
<span><span class="co">#&gt;  [9] "KIF4A"  "KIFC1"  "MAPT"   "TRIM58"</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; $Cluster_7</span></span>
<span><span class="co">#&gt; [1] "ADIPOQ" "AURKB"  "CCNA2"  "CHEK1"  "DACH1"  "NEK2"</span></span></code></pre></div>
</div>
</div>
<div class="section level3">
<h3 id="from-saved-data-frame">From saved data frame<a class="anchor" aria-label="anchor" href="#from-saved-data-frame"></a>
</h3>
<p>We can parse any data frame saved from clusterProfiler on disk, for
re-producing results. Like this:</p>
<div class="section level4">
<h4 id="save-data-frame">Save data frame<a class="anchor" aria-label="anchor" href="#save-data-frame"></a>
</h4>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/write.table.html" class="external-link">write.csv</a></span><span class="op">(</span><span class="va">ego</span>,<span class="st">"ego_enriched.csv"</span><span class="op">)</span></span></code></pre></div>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">res0</span><span class="op">&lt;-</span><span class="fu"><a href="https://readr.tidyverse.org/reference/read_delim.html" class="external-link">read_csv</a></span><span class="op">(</span><span class="st">"ego_enriched.csv"</span><span class="op">)</span></span>
<span><span class="co">#&gt; New names:</span></span>
<span><span class="co">#&gt; <span style="font-weight: bold;">Rows: </span><span style="color: #0000BB;">470</span> <span style="font-weight: bold;">Columns: </span><span style="color: #0000BB;">10</span></span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">──</span> <span style="font-weight: bold;">Column specification</span></span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">────────────────────────────────────────────────────────</span> <span style="font-weight: bold;">Delimiter:</span> "," <span style="color: #BB0000;">chr</span></span></span>
<span><span class="co">#&gt; (6): ...1, ID, Description, GeneRatio, BgRatio, geneID <span style="color: #00BB00;">dbl</span> (4): pvalue,</span></span>
<span><span class="co">#&gt; p.adjust, qvalue, Count</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> Use `spec()` to retrieve the full column specification for this data. <span style="color: #00BBBB;">ℹ</span></span></span>
<span><span class="co">#&gt; Specify the column types or set `show_col_types = FALSE` to quiet this message.</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">•</span> `` -&gt; `...1`</span></span>
<span><span class="va">res0</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 470 × 10</span></span></span>
<span><span class="co">#&gt;    ...1    ID    Description GeneRatio BgRatio   pvalue p.adjust   qvalue geneID</span></span>
<span><span class="co">#&gt;    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>       <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>     <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>      <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>    <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>    <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 1</span> GO:009… GO:0… nuclear ch… 33/196    236/11… 2.78<span style="color: #949494;">e</span><span style="color: #BB0000;">-21</span> 6.69<span style="color: #949494;">e</span><span style="color: #BB0000;">-18</span> 6.16<span style="color: #949494;">e</span><span style="color: #BB0000;">-18</span> CDCA8…</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 2</span> GO:000… GO:0… chromosome… 37/196    316/11… 4.45<span style="color: #949494;">e</span><span style="color: #BB0000;">-21</span> 6.69<span style="color: #949494;">e</span><span style="color: #BB0000;">-18</span> 6.16<span style="color: #949494;">e</span><span style="color: #BB0000;">-18</span> CDCA8…</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 3</span> GO:000… GO:0… mitotic si… 27/196    151/11… 2.10<span style="color: #949494;">e</span><span style="color: #BB0000;">-20</span> 2.10<span style="color: #949494;">e</span><span style="color: #BB0000;">-17</span> 1.94<span style="color: #949494;">e</span><span style="color: #BB0000;">-17</span> CDCA8…</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 4</span> GO:000… GO:0… sister chr… 29/196    184/11… 2.92<span style="color: #949494;">e</span><span style="color: #BB0000;">-20</span> 2.20<span style="color: #949494;">e</span><span style="color: #BB0000;">-17</span> 2.02<span style="color: #949494;">e</span><span style="color: #BB0000;">-17</span> CDCA8…</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 5</span> GO:014… GO:0… mitotic nu… 31/196    224/11… 6.84<span style="color: #949494;">e</span><span style="color: #BB0000;">-20</span> 4.12<span style="color: #949494;">e</span><span style="color: #BB0000;">-17</span> 3.79<span style="color: #949494;">e</span><span style="color: #BB0000;">-17</span> CDCA8…</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 6</span> GO:000… GO:0… nuclear di… 36/196    325/11… 1.05<span style="color: #949494;">e</span><span style="color: #BB0000;">-19</span> 5.29<span style="color: #949494;">e</span><span style="color: #BB0000;">-17</span> 4.87<span style="color: #949494;">e</span><span style="color: #BB0000;">-17</span> CDCA8…</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 7</span> GO:004… GO:0… organelle … 37/196    360/11… 3.98<span style="color: #949494;">e</span><span style="color: #BB0000;">-19</span> 1.71<span style="color: #949494;">e</span><span style="color: #BB0000;">-16</span> 1.57<span style="color: #949494;">e</span><span style="color: #BB0000;">-16</span> CDCA8…</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 8</span> GO:190… GO:1… microtubul… 23/196    135/11… 5.01<span style="color: #949494;">e</span><span style="color: #BB0000;">-17</span> 1.88<span style="color: #949494;">e</span><span style="color: #BB0000;">-14</span> 1.73<span style="color: #949494;">e</span><span style="color: #BB0000;">-14</span> CDCA8…</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 9</span> GO:005… GO:0… chromosome… 38/196    470/11… 4.33<span style="color: #949494;">e</span><span style="color: #BB0000;">-16</span> 1.45<span style="color: #949494;">e</span><span style="color: #BB0000;">-13</span> 1.33<span style="color: #949494;">e</span><span style="color: #BB0000;">-13</span> CDC45…</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">10</span> GO:005… GO:0… cell divis… 37/196    487/11… 8.15<span style="color: #949494;">e</span><span style="color: #BB0000;">-15</span> 2.45<span style="color: #949494;">e</span><span style="color: #BB0000;">-12</span> 2.26<span style="color: #949494;">e</span><span style="color: #BB0000;">-12</span> CDCA8…</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># ℹ 460 more rows</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># ℹ 1 more variable: Count &lt;dbl&gt;</span></span></span></code></pre></div>
</div>
<div class="section level4">
<h4 id="do-enrichgt">do EnrichGT<a class="anchor" aria-label="anchor" href="#do-enrichgt"></a>
</h4>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">obj1</span><span class="op">&lt;-</span><span class="va">res0</span><span class="op">|&gt;</span> <span class="fu"><a href="../reference/EnrichGT.html">EnrichGT</a></span><span class="op">(</span>P.adj <span class="op">=</span> <span class="fl">0.2</span>,ClusterNum <span class="op">=</span> <span class="fl">10</span>,nTop <span class="op">=</span> <span class="fl">10</span><span class="op">)</span></span>
<span><span class="co">#&gt; Joining with `by = join_by(ID)`</span></span></code></pre></div>
<p><img src="c_ora_files/figure-html/unnamed-chunk-11-1.png" width="672"></p>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">obj</span><span class="op">@</span><span class="va">gt_object</span></span></code></pre></div>
<div id="oqxtuqgjix" style="padding-left:0px;padding-right:0px;padding-top:10px;padding-bottom:10px;overflow-x:auto;overflow-y:auto;width:auto;height:auto;">
<style>#oqxtuqgjix table {
  font-family: system-ui, 'Segoe UI', Roboto, Helvetica, Arial, sans-serif, 'Apple Color Emoji', 'Segoe UI Emoji', 'Segoe UI Symbol', 'Noto Color Emoji';
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
}

#oqxtuqgjix thead, #oqxtuqgjix tbody, #oqxtuqgjix tfoot, #oqxtuqgjix tr, #oqxtuqgjix td, #oqxtuqgjix th {
  border-style: none;
}

#oqxtuqgjix p {
  margin: 0;
  padding: 0;
}

#oqxtuqgjix .gt_table {
  display: table;
  border-collapse: collapse;
  line-height: normal;
  margin-left: auto;
  margin-right: auto;
  color: #333333;
  font-size: 16px;
  font-weight: normal;
  font-style: normal;
  background-color: #FFFFFF;
  width: auto;
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #A8A8A8;
  border-right-style: none;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #A8A8A8;
  border-left-style: none;
  border-left-width: 2px;
  border-left-color: #D3D3D3;
}

#oqxtuqgjix .gt_caption {
  padding-top: 4px;
  padding-bottom: 4px;
}

#oqxtuqgjix .gt_title {
  color: #333333;
  font-size: 125%;
  font-weight: initial;
  padding-top: 4px;
  padding-bottom: 4px;
  padding-left: 5px;
  padding-right: 5px;
  border-bottom-color: #FFFFFF;
  border-bottom-width: 0;
}

#oqxtuqgjix .gt_subtitle {
  color: #333333;
  font-size: 85%;
  font-weight: initial;
  padding-top: 3px;
  padding-bottom: 5px;
  padding-left: 5px;
  padding-right: 5px;
  border-top-color: #FFFFFF;
  border-top-width: 0;
}

#oqxtuqgjix .gt_heading {
  background-color: #FFFFFF;
  text-align: center;
  border-bottom-color: #FFFFFF;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
}

#oqxtuqgjix .gt_bottom_border {
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
}

#oqxtuqgjix .gt_col_headings {
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
}

#oqxtuqgjix .gt_col_heading {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: normal;
  text-transform: inherit;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
  vertical-align: bottom;
  padding-top: 5px;
  padding-bottom: 6px;
  padding-left: 5px;
  padding-right: 5px;
  overflow-x: hidden;
}

#oqxtuqgjix .gt_column_spanner_outer {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: normal;
  text-transform: inherit;
  padding-top: 0;
  padding-bottom: 0;
  padding-left: 4px;
  padding-right: 4px;
}

#oqxtuqgjix .gt_column_spanner_outer:first-child {
  padding-left: 0;
}

#oqxtuqgjix .gt_column_spanner_outer:last-child {
  padding-right: 0;
}

#oqxtuqgjix .gt_column_spanner {
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  vertical-align: bottom;
  padding-top: 5px;
  padding-bottom: 5px;
  overflow-x: hidden;
  display: inline-block;
  width: 100%;
}

#oqxtuqgjix .gt_spanner_row {
  border-bottom-style: hidden;
}

#oqxtuqgjix .gt_group_heading {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  text-transform: inherit;
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
  vertical-align: middle;
  text-align: left;
}

#oqxtuqgjix .gt_empty_group_heading {
  padding: 0.5px;
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  vertical-align: middle;
}

#oqxtuqgjix .gt_from_md > :first-child {
  margin-top: 0;
}

#oqxtuqgjix .gt_from_md > :last-child {
  margin-bottom: 0;
}

#oqxtuqgjix .gt_row {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  margin: 10px;
  border-top-style: solid;
  border-top-width: 1px;
  border-top-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
  vertical-align: middle;
  overflow-x: hidden;
}

#oqxtuqgjix .gt_stub {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  text-transform: inherit;
  border-right-style: solid;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
  padding-left: 5px;
  padding-right: 5px;
}

#oqxtuqgjix .gt_stub_row_group {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  text-transform: inherit;
  border-right-style: solid;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
  padding-left: 5px;
  padding-right: 5px;
  vertical-align: top;
}

#oqxtuqgjix .gt_row_group_first td {
  border-top-width: 2px;
}

#oqxtuqgjix .gt_row_group_first th {
  border-top-width: 2px;
}

#oqxtuqgjix .gt_summary_row {
  color: #333333;
  background-color: #FFFFFF;
  text-transform: inherit;
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
}

#oqxtuqgjix .gt_first_summary_row {
  border-top-style: solid;
  border-top-color: #D3D3D3;
}

#oqxtuqgjix .gt_first_summary_row.thick {
  border-top-width: 2px;
}

#oqxtuqgjix .gt_last_summary_row {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
}

#oqxtuqgjix .gt_grand_summary_row {
  color: #333333;
  background-color: #FFFFFF;
  text-transform: inherit;
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
}

#oqxtuqgjix .gt_first_grand_summary_row {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  border-top-style: double;
  border-top-width: 6px;
  border-top-color: #D3D3D3;
}

#oqxtuqgjix .gt_last_grand_summary_row_top {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  border-bottom-style: double;
  border-bottom-width: 6px;
  border-bottom-color: #D3D3D3;
}

#oqxtuqgjix .gt_striped {
  background-color: rgba(128, 128, 128, 0.05);
}

#oqxtuqgjix .gt_table_body {
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
}

#oqxtuqgjix .gt_footnotes {
  color: #333333;
  background-color: #FFFFFF;
  border-bottom-style: none;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 2px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
}

#oqxtuqgjix .gt_footnote {
  margin: 0px;
  font-size: 90%;
  padding-top: 4px;
  padding-bottom: 4px;
  padding-left: 5px;
  padding-right: 5px;
}

#oqxtuqgjix .gt_sourcenotes {
  color: #333333;
  background-color: #FFFFFF;
  border-bottom-style: none;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 2px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
}

#oqxtuqgjix .gt_sourcenote {
  font-size: 90%;
  padding-top: 4px;
  padding-bottom: 4px;
  padding-left: 5px;
  padding-right: 5px;
}

#oqxtuqgjix .gt_left {
  text-align: left;
}

#oqxtuqgjix .gt_center {
  text-align: center;
}

#oqxtuqgjix .gt_right {
  text-align: right;
  font-variant-numeric: tabular-nums;
}

#oqxtuqgjix .gt_font_normal {
  font-weight: normal;
}

#oqxtuqgjix .gt_font_bold {
  font-weight: bold;
}

#oqxtuqgjix .gt_font_italic {
  font-style: italic;
}

#oqxtuqgjix .gt_super {
  font-size: 65%;
}

#oqxtuqgjix .gt_footnote_marks {
  font-size: 75%;
  vertical-align: 0.4em;
  position: initial;
}

#oqxtuqgjix .gt_asterisk {
  font-size: 100%;
  vertical-align: 0;
}

#oqxtuqgjix .gt_indent_1 {
  text-indent: 5px;
}

#oqxtuqgjix .gt_indent_2 {
  text-indent: 10px;
}

#oqxtuqgjix .gt_indent_3 {
  text-indent: 15px;
}

#oqxtuqgjix .gt_indent_4 {
  text-indent: 20px;
}

#oqxtuqgjix .gt_indent_5 {
  text-indent: 25px;
}

#oqxtuqgjix .katex-display {
  display: inline-flex !important;
  margin-bottom: 0.75em !important;
}

#oqxtuqgjix div.Reactable > div.rt-table > div.rt-thead > div.rt-tr.rt-tr-group-header > div.rt-th-group:after {
  height: 0px !important;
}
</style>
<table class="table gt_table" data-quarto-disable-processing="false" data-quarto-bootstrap="false">
<thead>
<tr class="gt_heading">
<td colspan="5" class="gt_heading gt_title gt_font_normal" style>Parse form: ego</td>
    </tr>
<tr class="gt_heading">
<td colspan="5" class="gt_heading gt_subtitle gt_font_normal gt_bottom_border" style>Split into 10 Clusters. Generated by github@zhimingye/EnrichGT</td>
    </tr>
<tr class="gt_col_headings">
<th class="gt_col_heading gt_columns_bottom_border gt_left" rowspan="1" colspan="1" scope="col" id="Description">Description</th>
      <th class="gt_col_heading gt_columns_bottom_border gt_right" rowspan="1" colspan="1" scope="col" id="Count">Count</th>
      <th class="gt_col_heading gt_columns_bottom_border gt_right" rowspan="1" colspan="1" scope="col" id="PCT">PCT</th>
      <th class="gt_col_heading gt_columns_bottom_border gt_right" rowspan="1" colspan="1" scope="col" id="Padj">Padj</th>
      <th class="gt_col_heading gt_columns_bottom_border gt_left" rowspan="1" colspan="1" scope="col" id="geneID">geneID</th>
    </tr>
</thead>
<tbody class="gt_table_body">
<tr class="gt_group_heading_row">
<th colspan="5" class="gt_group_heading" scope="colgroup" id="Cluster_1">Cluster_1</th>
    </tr>
<tr class="gt_row_group_first">
<td headers="Cluster_1  Description" class="gt_row gt_left" style="font-size: 13px;">
<div style="line-height:14px"><span style="font-weight:bold;font-variant:small-caps;color:#000000;font-size:14px">nuclear chromosome segregation</span></div>
<div style="line-height:10px"><span style="font-weight:bold;color:#BEBEBE;font-size:10px">GO:0098813</span></div>
</td>
<td headers="Cluster_1  Count" class="gt_row gt_right" style="background-color: #EAE1EF; color: #000000; font-size: 13px;">33</td>
<td headers="Cluster_1  PCT" class="gt_row gt_right" style="background-color: #DB6CA7; color: #FFFFFF; font-size: 13px;">17.0</td>
<td headers="Cluster_1  Padj" class="gt_row gt_right" style="background-color: #D53E4F; color: #FFFFFF; font-size: 13px;">6.7e-18</td>
<td headers="Cluster_1  geneID" class="gt_row gt_left" style="font-size: 13px;">CDCA8, CDC20, KIF23, CENPE, MYBL2, CCNB2, NDC80, TOP2A, NCAPH, ASPM, DLGAP5, UBE2C, NUSAP1, TPX2, TACC3, NEK2, CDK1, MAD2L1, KIF18A, CDT1, BIRC5, KIF11, TTK, NCAPG, AURKB, TRIP13, PRC1, KIFC1, KIF18B, AURKA, CCNB1, KIF4A, PTTG1</td>
</tr>
<tr>
<td headers="Cluster_1  Description" class="gt_row gt_left" style="font-size: 13px;">
<div style="line-height:14px"><span style="font-weight:bold;font-variant:small-caps;color:#000000;font-size:14px">chromosome segregation</span></div>
<div style="line-height:10px"><span style="font-weight:bold;color:#BEBEBE;font-size:10px">GO:0007059</span></div>
</td>
<td headers="Cluster_1  Count" class="gt_row gt_right" style="background-color: #FBF3F9; color: #000000; font-size: 13px;">37</td>
<td headers="Cluster_1  PCT" class="gt_row gt_right" style="background-color: #C51B7D; color: #FFFFFF; font-size: 13px;">19.0</td>
<td headers="Cluster_1  Padj" class="gt_row gt_right" style="background-color: #D53E4F; color: #FFFFFF; font-size: 13px;">6.7e-18</td>
<td headers="Cluster_1  geneID" class="gt_row gt_left" style="font-size: 13px;">CDCA8, CDC20, KIF23, CENPE, MYBL2, CCNB2, NDC80, TOP2A, NCAPH, ASPM, DLGAP5, UBE2C, HJURP, SKA1, NUSAP1, TPX2, TACC3, NEK2, CENPM, CENPN, CDK1, MAD2L1, KIF18A, CDT1, BIRC5, KIF11, TTK, NCAPG, AURKB, TRIP13, PRC1, KIFC1, KIF18B, AURKA, CCNB1, KIF4A, PTTG1</td>
</tr>
<tr>
<td headers="Cluster_1  Description" class="gt_row gt_left" style="font-size: 13px;">
<div style="line-height:14px"><span style="font-weight:bold;font-variant:small-caps;color:#000000;font-size:14px">mitotic sister chromatid segregation</span></div>
<div style="line-height:10px"><span style="font-weight:bold;color:#BEBEBE;font-size:10px">GO:0000070</span></div>
</td>
<td headers="Cluster_1  Count" class="gt_row gt_right" style="background-color: #C2CAE2; color: #000000; font-size: 13px;">27</td>
<td headers="Cluster_1  PCT" class="gt_row gt_right" style="background-color: #F3BCDD; color: #000000; font-size: 13px;">14.0</td>
<td headers="Cluster_1  Padj" class="gt_row gt_right" style="background-color: #D53E4F; color: #FFFFFF; font-size: 13px;">2.1e-17</td>
<td headers="Cluster_1  geneID" class="gt_row gt_left" style="font-size: 13px;">CDCA8, CDC20, KIF23, CENPE, MYBL2, NDC80, NCAPH, DLGAP5, UBE2C, NUSAP1, TPX2, NEK2, CDK1, MAD2L1, KIF18A, CDT1, BIRC5, KIF11, TTK, NCAPG, AURKB, TRIP13, PRC1, KIFC1, KIF18B, CCNB1, KIF4A</td>
</tr>
<tr>
<td headers="Cluster_1  Description" class="gt_row gt_left" style="font-size: 13px;">
<div style="line-height:14px"><span style="font-weight:bold;font-variant:small-caps;color:#000000;font-size:14px">sister chromatid segregation</span></div>
<div style="line-height:10px"><span style="font-weight:bold;color:#BEBEBE;font-size:10px">GO:0000819</span></div>
</td>
<td headers="Cluster_1  Count" class="gt_row gt_right" style="background-color: #D3D3E7; color: #000000; font-size: 13px;">29</td>
<td headers="Cluster_1  PCT" class="gt_row gt_right" style="background-color: #ECA4CD; color: #000000; font-size: 13px;">15.0</td>
<td headers="Cluster_1  Padj" class="gt_row gt_right" style="background-color: #D53E4F; color: #FFFFFF; font-size: 13px;">2.2e-17</td>
<td headers="Cluster_1  geneID" class="gt_row gt_left" style="font-size: 13px;">CDCA8, CDC20, KIF23, CENPE, MYBL2, NDC80, TOP2A, NCAPH, DLGAP5, UBE2C, NUSAP1, TPX2, TACC3, NEK2, CDK1, MAD2L1, KIF18A, CDT1, BIRC5, KIF11, TTK, NCAPG, AURKB, TRIP13, PRC1, KIFC1, KIF18B, CCNB1, KIF4A</td>
</tr>
<tr>
<td headers="Cluster_1  Description" class="gt_row gt_left" style="font-size: 13px;">
<div style="line-height:14px"><span style="font-weight:bold;font-variant:small-caps;color:#000000;font-size:14px">mitotic nuclear division</span></div>
<div style="line-height:10px"><span style="font-weight:bold;color:#BEBEBE;font-size:10px">GO:0140014</span></div>
</td>
<td headers="Cluster_1  Count" class="gt_row gt_right" style="background-color: #DEDAEB; color: #000000; font-size: 13px;">31</td>
<td headers="Cluster_1  PCT" class="gt_row gt_right" style="background-color: #E489BA; color: #000000; font-size: 13px;">16.0</td>
<td headers="Cluster_1  Padj" class="gt_row gt_right" style="background-color: #D53E4F; color: #FFFFFF; font-size: 13px;">4.1e-17</td>
<td headers="Cluster_1  geneID" class="gt_row gt_left" style="font-size: 13px;">CDCA8, CDC20, KIF23, CENPE, MYBL2, NDC80, NCAPH, DLGAP5, UBE2C, NUSAP1, TPX2, NEK2, UBE2S, CDK1, MAD2L1, KIF18A, CDT1, BIRC5, KIF11, TTK, NCAPG, AURKB, CHEK1, TRIP13, PRC1, KIFC1, KIF18B, AURKA, CCNB1, KIF4A, BMP4</td>
</tr>
<tr>
<td headers="Cluster_1  Description" class="gt_row gt_left" style="font-size: 13px;">
<div style="line-height:14px"><span style="font-weight:bold;font-variant:small-caps;color:#000000;font-size:14px">nuclear division</span></div>
<div style="line-height:10px"><span style="font-weight:bold;color:#BEBEBE;font-size:10px">GO:0000280</span></div>
</td>
<td headers="Cluster_1  Count" class="gt_row gt_right" style="background-color: #F7EEF6; color: #000000; font-size: 13px;">36</td>
<td headers="Cluster_1  PCT" class="gt_row gt_right" style="background-color: #D04B92; color: #FFFFFF; font-size: 13px;">18.0</td>
<td headers="Cluster_1  Padj" class="gt_row gt_right" style="background-color: #D53E4F; color: #FFFFFF; font-size: 13px;">5.3e-17</td>
<td headers="Cluster_1  geneID" class="gt_row gt_left" style="font-size: 13px;">CDCA8, CDC20, KIF23, CENPE, MYBL2, CCNB2, NDC80, TOP2A, NCAPH, ASPM, DLGAP5, UBE2C, NUSAP1, TPX2, NEK2, RAD51AP1, UBE2S, CDK1, MAD2L1, KIF18A, CDT1, BIRC5, KIF11, TTK, NCAPG, AURKB, CHEK1, TRIP13, PRC1, KIFC1, KIF18B, AURKA, CCNB1, KIF4A, PTTG1, BMP4</td>
</tr>
<tr>
<td headers="Cluster_1  Description" class="gt_row gt_left" style="font-size: 13px;">
<div style="line-height:14px"><span style="font-weight:bold;font-variant:small-caps;color:#000000;font-size:14px">organelle fission</span></div>
<div style="line-height:10px"><span style="font-weight:bold;color:#BEBEBE;font-size:10px">GO:0048285</span></div>
</td>
<td headers="Cluster_1  Count" class="gt_row gt_right" style="background-color: #FBF3F9; color: #000000; font-size: 13px;">37</td>
<td headers="Cluster_1  PCT" class="gt_row gt_right" style="background-color: #C51B7D; color: #FFFFFF; font-size: 13px;">19.0</td>
<td headers="Cluster_1  Padj" class="gt_row gt_right" style="background-color: #D53E4F; color: #FFFFFF; font-size: 13px;">1.7e-16</td>
<td headers="Cluster_1  geneID" class="gt_row gt_left" style="font-size: 13px;">CDCA8, CDC20, KIF23, CENPE, MYBL2, CCNB2, NDC80, TOP2A, NCAPH, ASPM, DLGAP5, UBE2C, NUSAP1, TPX2, NEK2, RAD51AP1, UBE2S, CDK1, MAD2L1, KIF18A, CDT1, BIRC5, KIF11, TTK, NCAPG, AURKB, CHEK1, TRIP13, PRC1, KIFC1, KIF18B, AURKA, CCNB1, KIF4A, PTTG1, BMP4, MAPT</td>
</tr>
<tr>
<td headers="Cluster_1  Description" class="gt_row gt_left" style="font-size: 13px;">
<div style="line-height:14px"><span style="font-weight:bold;font-variant:small-caps;color:#000000;font-size:14px">microtubule cytoskeleton organization involved in mitosis</span></div>
<div style="line-height:10px"><span style="font-weight:bold;color:#BEBEBE;font-size:10px">GO:1902850</span></div>
</td>
<td headers="Cluster_1  Count" class="gt_row gt_right" style="background-color: #9BB9D9; color: #000000; font-size: 13px;">23</td>
<td headers="Cluster_1  PCT" class="gt_row gt_right" style="background-color: #FDDFEF; color: #000000; font-size: 13px;">12.0</td>
<td headers="Cluster_1  Padj" class="gt_row gt_right" style="background-color: #D53E4F; color: #FFFFFF; font-size: 13px;">1.9e-14</td>
<td headers="Cluster_1  geneID" class="gt_row gt_left" style="font-size: 13px;">CDCA8, CDC20, KIF23, CENPE, MYBL2, NDC80, DLGAP5, NUSAP1, TPX2, TACC3, NEK2, CDK1, MAD2L1, PAX6, BIRC5, KIF11, TTK, AURKB, PRC1, KIFC1, AURKA, CCNB1, KIF4A</td>
</tr>
<tr>
<td headers="Cluster_1  Description" class="gt_row gt_left" style="font-size: 13px;">
<div style="line-height:14px"><span style="font-weight:bold;font-variant:small-caps;color:#000000;font-size:14px">chromosome organization</span></div>
<div style="line-height:10px"><span style="font-weight:bold;color:#BEBEBE;font-size:10px">GO:0051276</span></div>
</td>
<td headers="Cluster_1  Count" class="gt_row gt_right" style="background-color: #FFF7FB; color: #000000; font-size: 13px;">38</td>
<td headers="Cluster_1  PCT" class="gt_row gt_right" style="background-color: #C51B7D; color: #FFFFFF; font-size: 13px;">19.0</td>
<td headers="Cluster_1  Padj" class="gt_row gt_right" style="background-color: #D53E4F; color: #FFFFFF; font-size: 13px;">1.4e-13</td>
<td headers="Cluster_1  geneID" class="gt_row gt_left" style="font-size: 13px;">CDC45, CDCA8, CDC20, KIF23, CENPE, MYBL2, NDC80, TOP2A, NCAPH, DLGAP5, UBE2C, HJURP, NUSAP1, TPX2, TACC3, NEK2, CENPN, CDK1, MAD2L1, GINS1, KIF18A, CDT1, BIRC5, KIF11, EZH2, TTK, NCAPG, AURKB, GINS2, TRIP13, PRC1, KIFC1, KIF18B, CCNB1, KIF4A, MCM5, PTTG1, MAPT</td>
</tr>
<tr>
<td headers="Cluster_1  Description" class="gt_row gt_left" style="font-size: 13px;">
<div style="line-height:14px"><span style="font-weight:bold;font-variant:small-caps;color:#000000;font-size:14px">cell division</span></div>
<div style="line-height:10px"><span style="font-weight:bold;color:#BEBEBE;font-size:10px">GO:0051301</span></div>
</td>
<td headers="Cluster_1  Count" class="gt_row gt_right" style="background-color: #FBF3F9; color: #000000; font-size: 13px;">37</td>
<td headers="Cluster_1  PCT" class="gt_row gt_right" style="background-color: #C51B7D; color: #FFFFFF; font-size: 13px;">19.0</td>
<td headers="Cluster_1  Padj" class="gt_row gt_right" style="background-color: #D53E4F; color: #FFFFFF; font-size: 13px;">2.5e-12</td>
<td headers="Cluster_1  geneID" class="gt_row gt_left" style="font-size: 13px;">CDCA8, CDC20, KIF23, CENPE, CCNB2, NDC80, TOP2A, NCAPH, E2F8, ASPM, CEP55, UBE2C, SKA1, NUSAP1, CDCA3, TPX2, TACC3, NEK2, UBE2S, CCNA2, CDK1, ERCC6L, MAD2L1, PAX6, CDT1, BIRC5, KIF11, NCAPG, AURKB, PRC1, KIFC1, KIF18B, KIF20A, AURKA, CCNB1, KIF4A, PTTG1</td>
</tr>
<tr class="gt_group_heading_row">
<th colspan="5" class="gt_group_heading" scope="colgroup" id="Cluster_2">Cluster_2</th>
    </tr>
<tr class="gt_row_group_first">
<td headers="Cluster_2  Description" class="gt_row gt_left" style="font-size: 13px;">
<div style="line-height:14px"><span style="font-weight:bold;font-variant:small-caps;color:#000000;font-size:14px">female meiotic nuclear division</span></div>
<div style="line-height:10px"><span style="font-weight:bold;color:#BEBEBE;font-size:10px">GO:0007143</span></div>
</td>
<td headers="Cluster_2  Count" class="gt_row gt_right" style="background-color: #057068; color: #FFFFFF; font-size: 13px;">7</td>
<td headers="Cluster_2  PCT" class="gt_row gt_right" style="background-color: #62A42F; color: #FFFFFF; font-size: 13px;">3.6</td>
<td headers="Cluster_2  Padj" class="gt_row gt_right" style="background-color: #D53E4F; color: #FFFFFF; font-size: 13px;">1.0e-05</td>
<td headers="Cluster_2  geneID" class="gt_row gt_left" style="font-size: 13px;">CCNB2, NDC80, TOP2A, NCAPH, TTK, TRIP13, AURKA</td>
</tr>
<tr>
<td headers="Cluster_2  Description" class="gt_row gt_left" style="font-size: 13px;">
<div style="line-height:14px"><span style="font-weight:bold;font-variant:small-caps;color:#000000;font-size:14px">meiotic nuclear division</span></div>
<div style="line-height:10px"><span style="font-weight:bold;color:#BEBEBE;font-size:10px">GO:0140013</span></div>
</td>
<td headers="Cluster_2  Count" class="gt_row gt_right" style="background-color: #2388A4; color: #FFFFFF; font-size: 13px;">12</td>
<td headers="Cluster_2  PCT" class="gt_row gt_right" style="background-color: #9BCE64; color: #000000; font-size: 13px;">6.1</td>
<td headers="Cluster_2  Padj" class="gt_row gt_right" style="background-color: #D53E4F; color: #FFFFFF; font-size: 13px;">5.4e-05</td>
<td headers="Cluster_2  geneID" class="gt_row gt_left" style="font-size: 13px;">CDC20, CCNB2, NDC80, TOP2A, NCAPH, ASPM, RAD51AP1, KIF18A, TTK, TRIP13, AURKA, PTTG1</td>
</tr>
<tr>
<td headers="Cluster_2  Description" class="gt_row gt_left" style="font-size: 13px;">
<div style="line-height:14px"><span style="font-weight:bold;font-variant:small-caps;color:#000000;font-size:14px">meiotic cell cycle process</span></div>
<div style="line-height:10px"><span style="font-weight:bold;color:#BEBEBE;font-size:10px">GO:1903046</span></div>
</td>
<td headers="Cluster_2  Count" class="gt_row gt_right" style="background-color: #2388A4; color: #FFFFFF; font-size: 13px;">12</td>
<td headers="Cluster_2  PCT" class="gt_row gt_right" style="background-color: #9BCE64; color: #000000; font-size: 13px;">6.1</td>
<td headers="Cluster_2  Padj" class="gt_row gt_right" style="background-color: #D53E4F; color: #FFFFFF; font-size: 13px;">1.8e-04</td>
<td headers="Cluster_2  geneID" class="gt_row gt_left" style="font-size: 13px;">CDC20, CCNB2, NDC80, TOP2A, NCAPH, ASPM, RAD51AP1, KIF18A, TTK, TRIP13, AURKA, PTTG1</td>
</tr>
<tr>
<td headers="Cluster_2  Description" class="gt_row gt_left" style="font-size: 13px;">
<div style="line-height:14px"><span style="font-weight:bold;font-variant:small-caps;color:#000000;font-size:14px">meiotic chromosome segregation</span></div>
<div style="line-height:10px"><span style="font-weight:bold;color:#BEBEBE;font-size:10px">GO:0045132</span></div>
</td>
<td headers="Cluster_2  Count" class="gt_row gt_right" style="background-color: #057674; color: #FFFFFF; font-size: 13px;">8</td>
<td headers="Cluster_2  PCT" class="gt_row gt_right" style="background-color: #6DAD36; color: #FFFFFF; font-size: 13px;">4.1</td>
<td headers="Cluster_2  Padj" class="gt_row gt_right" style="background-color: #D53F4F; color: #FFFFFF; font-size: 13px;">2.3e-04</td>
<td headers="Cluster_2  geneID" class="gt_row gt_left" style="font-size: 13px;">CCNB2, NDC80, NCAPH, ASPM, TTK, TRIP13, AURKA, PTTG1</td>
</tr>
<tr>
<td headers="Cluster_2  Description" class="gt_row gt_left" style="font-size: 13px;">
<div style="line-height:14px"><span style="font-weight:bold;font-variant:small-caps;color:#000000;font-size:14px">cell cycle G2/M phase transition</span></div>
<div style="line-height:10px"><span style="font-weight:bold;color:#BEBEBE;font-size:10px">GO:0044839</span></div>
</td>
<td headers="Cluster_2  Count" class="gt_row gt_right" style="background-color: #188598; color: #FFFFFF; font-size: 13px;">11</td>
<td headers="Cluster_2  PCT" class="gt_row gt_right" style="background-color: #8FC655; color: #000000; font-size: 13px;">5.6</td>
<td headers="Cluster_2  Padj" class="gt_row gt_right" style="background-color: #D63F4F; color: #FFFFFF; font-size: 13px;">3.6e-04</td>
<td headers="Cluster_2  geneID" class="gt_row gt_left" style="font-size: 13px;">FOXM1, MELK, CCNB2, NDC80, CCNA2, CDK1, AURKB, CHEK1, DTL, AURKA, CCNB1</td>
</tr>
<tr>
<td headers="Cluster_2  Description" class="gt_row gt_left" style="font-size: 13px;">
<div style="line-height:14px"><span style="font-weight:bold;font-variant:small-caps;color:#000000;font-size:14px">meiotic cell cycle</span></div>
<div style="line-height:10px"><span style="font-weight:bold;color:#BEBEBE;font-size:10px">GO:0051321</span></div>
</td>
<td headers="Cluster_2  Count" class="gt_row gt_right" style="background-color: #2C8CAF; color: #FFFFFF; font-size: 13px;">13</td>
<td headers="Cluster_2  PCT" class="gt_row gt_right" style="background-color: #A8D672; color: #000000; font-size: 13px;">6.6</td>
<td headers="Cluster_2  Padj" class="gt_row gt_right" style="background-color: #D63F4F; color: #FFFFFF; font-size: 13px;">5.8e-04</td>
<td headers="Cluster_2  geneID" class="gt_row gt_left" style="font-size: 13px;">CDC20, CCNB2, NDC80, TOP2A, NCAPH, ASPM, NEK2, RAD51AP1, KIF18A, TTK, TRIP13, AURKA, PTTG1</td>
</tr>
<tr>
<td headers="Cluster_2  Description" class="gt_row gt_left" style="font-size: 13px;">
<div style="line-height:14px"><span style="font-weight:bold;font-variant:small-caps;color:#000000;font-size:14px">DNA replication</span></div>
<div style="line-height:10px"><span style="font-weight:bold;color:#BEBEBE;font-size:10px">GO:0006260</span></div>
</td>
<td headers="Cluster_2  Count" class="gt_row gt_right" style="background-color: #338FBB; color: #FFFFFF; font-size: 13px;">14</td>
<td headers="Cluster_2  PCT" class="gt_row gt_right" style="background-color: #B4DE81; color: #000000; font-size: 13px;">7.1</td>
<td headers="Cluster_2  Padj" class="gt_row gt_right" style="background-color: #D6404F; color: #FFFFFF; font-size: 13px;">8.8e-04</td>
<td headers="Cluster_2  geneID" class="gt_row gt_left" style="font-size: 13px;">CDC45, MCM10, E2F8, RRM2, CCNA2, CDK1, GINS1, CDT1, GINS2, CHAF1B, CHEK1, DTL, MCM5, DACH1</td>
</tr>
<tr>
<td headers="Cluster_2  Description" class="gt_row gt_left" style="font-size: 13px;">
<div style="line-height:14px"><span style="font-weight:bold;font-variant:small-caps;color:#000000;font-size:14px">meiosis I cell cycle process</span></div>
<div style="line-height:10px"><span style="font-weight:bold;color:#BEBEBE;font-size:10px">GO:0061982</span></div>
</td>
<td headers="Cluster_2  Count" class="gt_row gt_right" style="background-color: #057674; color: #FFFFFF; font-size: 13px;">8</td>
<td headers="Cluster_2  PCT" class="gt_row gt_right" style="background-color: #6DAD36; color: #FFFFFF; font-size: 13px;">4.1</td>
<td headers="Cluster_2  Padj" class="gt_row gt_right" style="background-color: #DA464E; color: #FFFFFF; font-size: 13px;">3.2e-03</td>
<td headers="Cluster_2  geneID" class="gt_row gt_left" style="font-size: 13px;">CDC20, CCNB2, NDC80, TOP2A, RAD51AP1, TRIP13, AURKA, PTTG1</td>
</tr>
<tr>
<td headers="Cluster_2  Description" class="gt_row gt_left" style="font-size: 13px;">
<div style="line-height:14px"><span style="font-weight:bold;font-variant:small-caps;color:#000000;font-size:14px">female gamete generation</span></div>
<div style="line-height:10px"><span style="font-weight:bold;color:#BEBEBE;font-size:10px">GO:0007292</span></div>
</td>
<td headers="Cluster_2  Count" class="gt_row gt_right" style="background-color: #047D81; color: #FFFFFF; font-size: 13px;">9</td>
<td headers="Cluster_2  PCT" class="gt_row gt_right" style="background-color: #78B63C; color: #000000; font-size: 13px;">4.6</td>
<td headers="Cluster_2  Padj" class="gt_row gt_right" style="background-color: #DA464E; color: #FFFFFF; font-size: 13px;">3.3e-03</td>
<td headers="Cluster_2  geneID" class="gt_row gt_left" style="font-size: 13px;">CCNB2, NDC80, TOP2A, NCAPH, ASPM, TTK, TRIP13, AURKA, PGR</td>
</tr>
<tr>
<td headers="Cluster_2  Description" class="gt_row gt_left" style="font-size: 13px;">
<div style="line-height:14px"><span style="font-weight:bold;font-variant:small-caps;color:#000000;font-size:14px">G2/M transition of mitotic cell cycle</span></div>
<div style="line-height:10px"><span style="font-weight:bold;color:#BEBEBE;font-size:10px">GO:0000086</span></div>
</td>
<td headers="Cluster_2  Count" class="gt_row gt_right" style="background-color: #047D81; color: #FFFFFF; font-size: 13px;">9</td>
<td headers="Cluster_2  PCT" class="gt_row gt_right" style="background-color: #78B63C; color: #000000; font-size: 13px;">4.6</td>
<td headers="Cluster_2  Padj" class="gt_row gt_right" style="background-color: #DB474D; color: #FFFFFF; font-size: 13px;">3.9e-03</td>
<td headers="Cluster_2  geneID" class="gt_row gt_left" style="font-size: 13px;">FOXM1, MELK, CCNA2, CDK1, AURKB, CHEK1, DTL, AURKA, CCNB1</td>
</tr>
<tr class="gt_group_heading_row">
<th colspan="5" class="gt_group_heading" scope="colgroup" id="Cluster_3">Cluster_3</th>
    </tr>
<tr class="gt_row_group_first">
<td headers="Cluster_3  Description" class="gt_row gt_left" style="font-size: 13px;">
<div style="line-height:14px"><span style="font-weight:bold;font-variant:small-caps;color:#000000;font-size:14px">antimicrobial humoral immune response mediated by antimicrobial peptide</span></div>
<div style="line-height:10px"><span style="font-weight:bold;color:#BEBEBE;font-size:10px">GO:0061844</span></div>
</td>
<td headers="Cluster_3  Count" class="gt_row gt_right" style="background-color: #047D81; color: #FFFFFF; font-size: 13px;">9</td>
<td headers="Cluster_3  PCT" class="gt_row gt_right" style="background-color: #78B63C; color: #000000; font-size: 13px;">4.6</td>
<td headers="Cluster_3  Padj" class="gt_row gt_right" style="background-color: #D53E4F; color: #FFFFFF; font-size: 13px;">4.4e-05</td>
<td headers="Cluster_3  geneID" class="gt_row gt_left" style="font-size: 13px;">S100A9, S100A7, CXCL10, CXCL13, CXCL11, CAMP, CXCL9, GNLY, CXCL14</td>
</tr>
<tr>
<td headers="Cluster_3  Description" class="gt_row gt_left" style="font-size: 13px;">
<div style="line-height:14px"><span style="font-weight:bold;font-variant:small-caps;color:#000000;font-size:14px">antimicrobial humoral response</span></div>
<div style="line-height:10px"><span style="font-weight:bold;color:#BEBEBE;font-size:10px">GO:0019730</span></div>
</td>
<td headers="Cluster_3  Count" class="gt_row gt_right" style="background-color: #08828D; color: #FFFFFF; font-size: 13px;">10</td>
<td headers="Cluster_3  PCT" class="gt_row gt_right" style="background-color: #83BE46; color: #000000; font-size: 13px;">5.1</td>
<td headers="Cluster_3  Padj" class="gt_row gt_right" style="background-color: #D53E4F; color: #FFFFFF; font-size: 13px;">1.2e-04</td>
<td headers="Cluster_3  geneID" class="gt_row gt_left" style="font-size: 13px;">S100A9, S100A7, CXCL10, CXCL13, CXCL11, CAMP, CXCL9, GNLY, RNASE4, CXCL14</td>
</tr>
<tr>
<td headers="Cluster_3  Description" class="gt_row gt_left" style="font-size: 13px;">
<div style="line-height:14px"><span style="font-weight:bold;font-variant:small-caps;color:#000000;font-size:14px">chemokine-mediated signaling pathway</span></div>
<div style="line-height:10px"><span style="font-weight:bold;color:#BEBEBE;font-size:10px">GO:0070098</span></div>
</td>
<td headers="Cluster_3  Count" class="gt_row gt_right" style="background-color: #057674; color: #FFFFFF; font-size: 13px;">8</td>
<td headers="Cluster_3  PCT" class="gt_row gt_right" style="background-color: #6DAD36; color: #FFFFFF; font-size: 13px;">4.1</td>
<td headers="Cluster_3  Padj" class="gt_row gt_right" style="background-color: #D8444E; color: #FFFFFF; font-size: 13px;">2.3e-03</td>
<td headers="Cluster_3  geneID" class="gt_row gt_left" style="font-size: 13px;">CXCL10, CXCL13, CXCL11, CXCL9, CCL18, CCL8, ACKR1, CX3CR1</td>
</tr>
<tr>
<td headers="Cluster_3  Description" class="gt_row gt_left" style="font-size: 13px;">
<div style="line-height:14px"><span style="font-weight:bold;font-variant:small-caps;color:#000000;font-size:14px">lymphocyte chemotaxis</span></div>
<div style="line-height:10px"><span style="font-weight:bold;color:#BEBEBE;font-size:10px">GO:0048247</span></div>
</td>
<td headers="Cluster_3  Count" class="gt_row gt_right" style="background-color: #036A5C; color: #FFFFFF; font-size: 13px;">6</td>
<td headers="Cluster_3  PCT" class="gt_row gt_right" style="background-color: #589B28; color: #FFFFFF; font-size: 13px;">3.1</td>
<td headers="Cluster_3  Padj" class="gt_row gt_right" style="background-color: #DA464E; color: #FFFFFF; font-size: 13px;">3.3e-03</td>
<td headers="Cluster_3  geneID" class="gt_row gt_left" style="font-size: 13px;">S100A7, CXCL10, CXCL13, CXCL11, CCL18, CCL8</td>
</tr>
<tr>
<td headers="Cluster_3  Description" class="gt_row gt_left" style="font-size: 13px;">
<div style="line-height:14px"><span style="font-weight:bold;font-variant:small-caps;color:#000000;font-size:14px">granulocyte chemotaxis</span></div>
<div style="line-height:10px"><span style="font-weight:bold;color:#BEBEBE;font-size:10px">GO:0071621</span></div>
</td>
<td headers="Cluster_3  Count" class="gt_row gt_right" style="background-color: #047D81; color: #FFFFFF; font-size: 13px;">9</td>
<td headers="Cluster_3  PCT" class="gt_row gt_right" style="background-color: #78B63C; color: #000000; font-size: 13px;">4.6</td>
<td headers="Cluster_3  Padj" class="gt_row gt_right" style="background-color: #DA464D; color: #FFFFFF; font-size: 13px;">3.5e-03</td>
<td headers="Cluster_3  geneID" class="gt_row gt_left" style="font-size: 13px;">S100A9, S100A8, S100A7, CXCL10, CXCL13, CXCL11, CXCL9, CCL18, CCL8</td>
</tr>
<tr>
<td headers="Cluster_3  Description" class="gt_row gt_left" style="font-size: 13px;">
<div style="line-height:14px"><span style="font-weight:bold;font-variant:small-caps;color:#000000;font-size:14px">response to chemokine</span></div>
<div style="line-height:10px"><span style="font-weight:bold;color:#BEBEBE;font-size:10px">GO:1990868</span></div>
</td>
<td headers="Cluster_3  Count" class="gt_row gt_right" style="background-color: #057674; color: #FFFFFF; font-size: 13px;">8</td>
<td headers="Cluster_3  PCT" class="gt_row gt_right" style="background-color: #6DAD36; color: #FFFFFF; font-size: 13px;">4.1</td>
<td headers="Cluster_3  Padj" class="gt_row gt_right" style="background-color: #DB474D; color: #FFFFFF; font-size: 13px;">3.8e-03</td>
<td headers="Cluster_3  geneID" class="gt_row gt_left" style="font-size: 13px;">CXCL10, CXCL13, CXCL11, CXCL9, CCL18, CCL8, ACKR1, CX3CR1</td>
</tr>
<tr>
<td headers="Cluster_3  Description" class="gt_row gt_left" style="font-size: 13px;">
<div style="line-height:14px"><span style="font-weight:bold;font-variant:small-caps;color:#000000;font-size:14px">cellular response to chemokine</span></div>
<div style="line-height:10px"><span style="font-weight:bold;color:#BEBEBE;font-size:10px">GO:1990869</span></div>
</td>
<td headers="Cluster_3  Count" class="gt_row gt_right" style="background-color: #057674; color: #FFFFFF; font-size: 13px;">8</td>
<td headers="Cluster_3  PCT" class="gt_row gt_right" style="background-color: #6DAD36; color: #FFFFFF; font-size: 13px;">4.1</td>
<td headers="Cluster_3  Padj" class="gt_row gt_right" style="background-color: #DB474D; color: #FFFFFF; font-size: 13px;">3.8e-03</td>
<td headers="Cluster_3  geneID" class="gt_row gt_left" style="font-size: 13px;">CXCL10, CXCL13, CXCL11, CXCL9, CCL18, CCL8, ACKR1, CX3CR1</td>
</tr>
<tr>
<td headers="Cluster_3  Description" class="gt_row gt_left" style="font-size: 13px;">
<div style="line-height:14px"><span style="font-weight:bold;font-variant:small-caps;color:#000000;font-size:14px">humoral immune response</span></div>
<div style="line-height:10px"><span style="font-weight:bold;color:#BEBEBE;font-size:10px">GO:0006959</span></div>
</td>
<td headers="Cluster_3  Count" class="gt_row gt_right" style="background-color: #2388A4; color: #FFFFFF; font-size: 13px;">12</td>
<td headers="Cluster_3  PCT" class="gt_row gt_right" style="background-color: #9BCE64; color: #000000; font-size: 13px;">6.1</td>
<td headers="Cluster_3  Padj" class="gt_row gt_right" style="background-color: #DB484D; color: #FFFFFF; font-size: 13px;">4.0e-03</td>
<td headers="Cluster_3  geneID" class="gt_row gt_left" style="font-size: 13px;">S100A9, S100A7, CXCL10, CXCL13, CXCL11, CAMP, CXCL9, GNLY, RNASE4, C7, CXCL14, GATA3</td>
</tr>
<tr>
<td headers="Cluster_3  Description" class="gt_row gt_left" style="font-size: 13px;">
<div style="line-height:14px"><span style="font-weight:bold;font-variant:small-caps;color:#000000;font-size:14px">neutrophil chemotaxis</span></div>
<div style="line-height:10px"><span style="font-weight:bold;color:#BEBEBE;font-size:10px">GO:0030593</span></div>
</td>
<td headers="Cluster_3  Count" class="gt_row gt_right" style="background-color: #057674; color: #FFFFFF; font-size: 13px;">8</td>
<td headers="Cluster_3  PCT" class="gt_row gt_right" style="background-color: #6DAD36; color: #FFFFFF; font-size: 13px;">4.1</td>
<td headers="Cluster_3  Padj" class="gt_row gt_right" style="background-color: #DB484D; color: #FFFFFF; font-size: 13px;">4.2e-03</td>
<td headers="Cluster_3  geneID" class="gt_row gt_left" style="font-size: 13px;">S100A9, S100A8, CXCL10, CXCL13, CXCL11, CXCL9, CCL18, CCL8</td>
</tr>
<tr>
<td headers="Cluster_3  Description" class="gt_row gt_left" style="font-size: 13px;">
<div style="line-height:14px"><span style="font-weight:bold;font-variant:small-caps;color:#000000;font-size:14px">response to molecule of bacterial origin</span></div>
<div style="line-height:10px"><span style="font-weight:bold;color:#BEBEBE;font-size:10px">GO:0002237</span></div>
</td>
<td headers="Cluster_3  Count" class="gt_row gt_right" style="background-color: #3D93C2; color: #FFFFFF; font-size: 13px;">15</td>
<td headers="Cluster_3  PCT" class="gt_row gt_right" style="background-color: #C0E593; color: #000000; font-size: 13px;">7.7</td>
<td headers="Cluster_3  Padj" class="gt_row gt_right" style="background-color: #DC494D; color: #FFFFFF; font-size: 13px;">4.6e-03</td>
<td headers="Cluster_3  geneID" class="gt_row gt_left" style="font-size: 13px;">S100A9, S100A8, S100A7, CXCL10, CXCL13, CXCL11, SLC7A5, CAMP, CXCL9, INAVA, IDO1, MAOB, PCK1, HMGCS2, CX3CR1</td>
</tr>
<tr class="gt_group_heading_row">
<th colspan="5" class="gt_group_heading" scope="colgroup" id="Cluster_4">Cluster_4</th>
    </tr>
<tr class="gt_row_group_first">
<td headers="Cluster_4  Description" class="gt_row gt_left" style="font-size: 13px;">
<div style="line-height:14px"><span style="font-weight:bold;font-variant:small-caps;color:#000000;font-size:14px">lens development in camera-type eye</span></div>
<div style="line-height:10px"><span style="font-weight:bold;color:#BEBEBE;font-size:10px">GO:0002088</span></div>
</td>
<td headers="Cluster_4  Count" class="gt_row gt_right" style="background-color: #016450; color: #FFFFFF; font-size: 13px;">5</td>
<td headers="Cluster_4  PCT" class="gt_row gt_right" style="background-color: #4D9221; color: #FFFFFF; font-size: 13px;">2.6</td>
<td headers="Cluster_4  Padj" class="gt_row gt_right" style="background-color: #3288BD; color: #FFFFFF; font-size: 13px;">1.5e-01</td>
<td headers="Cluster_4  geneID" class="gt_row gt_left" style="font-size: 13px;">PAX6, PLAAT1, BMP4, NDP, GATA3</td>
</tr>
<tr class="gt_group_heading_row">
<th colspan="5" class="gt_group_heading" scope="colgroup" id="Cluster_6">Cluster_6</th>
    </tr>
<tr class="gt_row_group_first">
<td headers="Cluster_6  Description" class="gt_row gt_left" style="font-size: 13px;">
<div style="line-height:14px"><span style="font-weight:bold;font-variant:small-caps;color:#000000;font-size:14px">microtubule-based movement</span></div>
<div style="line-height:10px"><span style="font-weight:bold;color:#BEBEBE;font-size:10px">GO:0007018</span></div>
</td>
<td headers="Cluster_6  Count" class="gt_row gt_right" style="background-color: #2388A4; color: #FFFFFF; font-size: 13px;">12</td>
<td headers="Cluster_6  PCT" class="gt_row gt_right" style="background-color: #9BCE64; color: #000000; font-size: 13px;">6.1</td>
<td headers="Cluster_6  Padj" class="gt_row gt_right" style="background-color: #F98B50; color: #000000; font-size: 13px;">3.1e-02</td>
<td headers="Cluster_6  geneID" class="gt_row gt_left" style="font-size: 13px;">KIF23, CENPE, DLGAP5, KIF18A, KIF11, KIFC1, KIF18B, KIF20A, KIF4A, MAPT, TRIM58, CFAP69</td>
</tr>
<tr class="gt_group_heading_row">
<th colspan="5" class="gt_group_heading" scope="colgroup" id="Cluster_7">Cluster_7</th>
    </tr>
<tr class="gt_row_group_first">
<td headers="Cluster_7  Description" class="gt_row gt_left" style="font-size: 13px;">
<div style="line-height:14px"><span style="font-weight:bold;font-variant:small-caps;color:#000000;font-size:14px">regulation of DNA biosynthetic process</span></div>
<div style="line-height:10px"><span style="font-weight:bold;color:#BEBEBE;font-size:10px">GO:2000278</span></div>
</td>
<td headers="Cluster_7  Count" class="gt_row gt_right" style="background-color: #036A5C; color: #FFFFFF; font-size: 13px;">6</td>
<td headers="Cluster_7  PCT" class="gt_row gt_right" style="background-color: #589B28; color: #FFFFFF; font-size: 13px;">3.1</td>
<td headers="Cluster_7  Padj" class="gt_row gt_right" style="background-color: #84CDA5; color: #000000; font-size: 13px;">1.2e-01</td>
<td headers="Cluster_7  geneID" class="gt_row gt_left" style="font-size: 13px;">NEK2, CCNA2, AURKB, CHEK1, ADIPOQ, DACH1</td>
</tr>
<tr class="gt_group_heading_row">
<th colspan="5" class="gt_group_heading" scope="colgroup" id="Cluster_9">Cluster_9</th>
    </tr>
<tr class="gt_row_group_first">
<td headers="Cluster_9  Description" class="gt_row gt_left" style="font-size: 13px;">
<div style="line-height:14px"><span style="font-weight:bold;font-variant:small-caps;color:#000000;font-size:14px">response to activity</span></div>
<div style="line-height:10px"><span style="font-weight:bold;color:#BEBEBE;font-size:10px">GO:0014823</span></div>
</td>
<td headers="Cluster_9  Count" class="gt_row gt_right" style="background-color: #016450; color: #FFFFFF; font-size: 13px;">5</td>
<td headers="Cluster_9  PCT" class="gt_row gt_right" style="background-color: #4D9221; color: #FFFFFF; font-size: 13px;">2.6</td>
<td headers="Cluster_9  Padj" class="gt_row gt_right" style="background-color: #FED884; color: #000000; font-size: 13px;">6.1e-02</td>
<td headers="Cluster_9  geneID" class="gt_row gt_left" style="font-size: 13px;">SLC7A5, CDK1, ADIPOQ, PCK1, CRY2</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
</div>
<div class="section level2">
<h2 id="session-info">Session info<a class="anchor" aria-label="anchor" href="#session-info"></a>
</h2>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">sessioninfo</span><span class="fu">::</span><span class="fu"><a href="https://r-lib.github.io/sessioninfo/reference/session_info.html" class="external-link">session_info</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB; font-weight: bold;">─ Session info ───────────────────────────────────────────────────────────────</span></span></span>
<span><span class="co">#&gt;  <span style="color: #555555; font-style: italic;">setting </span> <span style="color: #555555; font-style: italic;">value</span></span></span>
<span><span class="co">#&gt;  version  R version 4.3.3 (2024-02-29)</span></span>
<span><span class="co">#&gt;  os       macOS 15.0</span></span>
<span><span class="co">#&gt;  system   aarch64, darwin20</span></span>
<span><span class="co">#&gt;  ui       X11</span></span>
<span><span class="co">#&gt;  language en</span></span>
<span><span class="co">#&gt;  collate  en_US.UTF-8</span></span>
<span><span class="co">#&gt;  ctype    en_US.UTF-8</span></span>
<span><span class="co">#&gt;  tz       Asia/Shanghai</span></span>
<span><span class="co">#&gt;  date     2024-10-08</span></span>
<span><span class="co">#&gt;  pandoc   3.1.11 @ /Applications/RStudio.app/Contents/Resources/app/quarto/bin/tools/aarch64/ (via rmarkdown)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB; font-weight: bold;">─ Packages ───────────────────────────────────────────────────────────────────</span></span></span>
<span><span class="co">#&gt;  <span style="color: #555555; font-style: italic;">package         </span> <span style="color: #555555; font-style: italic;">*</span> <span style="color: #555555; font-style: italic;">version   </span> <span style="color: #555555; font-style: italic;">date (UTC)</span> <span style="color: #555555; font-style: italic;">lib</span> <span style="color: #555555; font-style: italic;">source</span></span></span>
<span><span class="co">#&gt;  AnnotationDbi    * 1.64.1     <span style="color: #555555;">2023-11-02</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">#&gt;  ape                5.7-1      <span style="color: #555555;">2023-03-13</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.3.0)</span></span></span>
<span><span class="co">#&gt;  aplot              0.2.2      <span style="color: #555555;">2023-10-06</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.3.1)</span></span></span>
<span><span class="co">#&gt;  Biobase          * 2.62.0     <span style="color: #555555;">2023-10-26</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">#&gt;  BiocGenerics     * 0.48.1     <span style="color: #555555;">2023-11-02</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">#&gt;  BiocParallel       1.36.0     <span style="color: #555555;">2023-10-26</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">#&gt;  Biostrings         2.70.2     <span style="color: #555555;">2024-01-30</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor 3.18 (R 4.3.2)</span></span></span>
<span><span class="co">#&gt;  bit                4.0.5      <span style="color: #555555;">2022-11-15</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.3.0)</span></span></span>
<span><span class="co">#&gt;  bit64              4.0.5      <span style="color: #555555;">2020-08-30</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.3.0)</span></span></span>
<span><span class="co">#&gt;  bitops             1.0-7      <span style="color: #555555;">2021-04-24</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.3.0)</span></span></span>
<span><span class="co">#&gt;  blob               1.2.4      <span style="color: #555555;">2023-03-17</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.3.0)</span></span></span>
<span><span class="co">#&gt;  bslib              0.6.1      <span style="color: #555555;">2023-11-28</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.3.1)</span></span></span>
<span><span class="co">#&gt;  cachem             1.0.8      <span style="color: #555555;">2023-05-01</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.3.0)</span></span></span>
<span><span class="co">#&gt;  cli                3.6.3      <span style="color: #555555;">2024-06-21</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.3.3)</span></span></span>
<span><span class="co">#&gt;  clusterProfiler  * 4.10.1     <span style="color: #555555;">2024-03-09</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor 3.18 (R 4.3.3)</span></span></span>
<span><span class="co">#&gt;  codetools          0.2-19     <span style="color: #555555;">2023-02-01</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.3.3)</span></span></span>
<span><span class="co">#&gt;  colorspace         2.1-0      <span style="color: #555555;">2023-01-23</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.3.0)</span></span></span>
<span><span class="co">#&gt;  cowplot            1.1.3      <span style="color: #555555;">2024-01-22</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.3.1)</span></span></span>
<span><span class="co">#&gt;  crayon             1.5.2      <span style="color: #555555;">2022-09-29</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.3.0)</span></span></span>
<span><span class="co">#&gt;  data.table         1.15.4     <span style="color: #555555;">2024-03-30</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.3.1)</span></span></span>
<span><span class="co">#&gt;  DBI                1.2.2      <span style="color: #555555;">2024-02-16</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.3.1)</span></span></span>
<span><span class="co">#&gt;  desc               1.4.3      <span style="color: #555555;">2023-12-10</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.3.1)</span></span></span>
<span><span class="co">#&gt;  digest             0.6.35     <span style="color: #555555;">2024-03-11</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.3.1)</span></span></span>
<span><span class="co">#&gt;  DOSE               3.28.2     <span style="color: #555555;">2023-12-12</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor 3.18 (R 4.3.2)</span></span></span>
<span><span class="co">#&gt;  dplyr            * 1.1.4      <span style="color: #555555;">2023-11-17</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.3.1)</span></span></span>
<span><span class="co">#&gt;  EnrichGT         * 0.2.3      <span style="color: #555555;">2024-10-08</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">local</span></span></span>
<span><span class="co">#&gt;  enrichplot         1.22.0     <span style="color: #555555;">2023-11-06</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">#&gt;  evaluate           0.23       <span style="color: #555555;">2023-11-01</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.3.1)</span></span></span>
<span><span class="co">#&gt;  fansi              1.0.6      <span style="color: #555555;">2023-12-08</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.3.1)</span></span></span>
<span><span class="co">#&gt;  farver             2.1.1      <span style="color: #555555;">2022-07-06</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.3.0)</span></span></span>
<span><span class="co">#&gt;  fastmap            1.1.1      <span style="color: #555555;">2023-02-24</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.3.0)</span></span></span>
<span><span class="co">#&gt;  fastmatch          1.1-4      <span style="color: #555555;">2023-08-18</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.3.0)</span></span></span>
<span><span class="co">#&gt;  fgsea              1.28.0     <span style="color: #555555;">2023-10-26</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">#&gt;  float              0.3-2      <span style="color: #555555;">2023-12-10</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.3.1)</span></span></span>
<span><span class="co">#&gt;  forcats          * 1.0.0      <span style="color: #555555;">2023-01-29</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.3.0)</span></span></span>
<span><span class="co">#&gt;  fs                 1.6.3      <span style="color: #555555;">2023-07-20</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.3.0)</span></span></span>
<span><span class="co">#&gt;  generics           0.1.3      <span style="color: #555555;">2022-07-05</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.3.0)</span></span></span>
<span><span class="co">#&gt;  GenomeInfoDb       1.38.7     <span style="color: #555555;">2024-03-09</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor 3.18 (R 4.3.3)</span></span></span>
<span><span class="co">#&gt;  GenomeInfoDbData   1.2.11     <span style="color: #555555;">2024-03-18</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">#&gt;  ggforce            0.4.2      <span style="color: #555555;">2024-02-19</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.3.1)</span></span></span>
<span><span class="co">#&gt;  ggfun              0.1.4      <span style="color: #555555;">2024-01-19</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.3.1)</span></span></span>
<span><span class="co">#&gt;  ggplot2          * 3.5.0      <span style="color: #555555;">2024-02-23</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.3.1)</span></span></span>
<span><span class="co">#&gt;  ggplotify          0.1.2      <span style="color: #555555;">2023-08-09</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.3.0)</span></span></span>
<span><span class="co">#&gt;  ggraph             2.2.1      <span style="color: #555555;">2024-03-07</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.3.1)</span></span></span>
<span><span class="co">#&gt;  ggrepel            0.9.5      <span style="color: #555555;">2024-01-10</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.3.1)</span></span></span>
<span><span class="co">#&gt;  ggtree             3.10.1     <span style="color: #555555;">2024-02-27</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor 3.18 (R 4.3.2)</span></span></span>
<span><span class="co">#&gt;  glue               1.7.0      <span style="color: #555555;">2024-01-09</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.3.1)</span></span></span>
<span><span class="co">#&gt;  GO.db              3.18.0     <span style="color: #555555;">2024-03-18</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">#&gt;  GOSemSim           2.28.1     <span style="color: #555555;">2024-01-20</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor 3.18 (R 4.3.2)</span></span></span>
<span><span class="co">#&gt;  graphlayouts       1.1.1      <span style="color: #555555;">2024-03-09</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.3.1)</span></span></span>
<span><span class="co">#&gt;  gridExtra          2.3        <span style="color: #555555;">2017-09-09</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.3.0)</span></span></span>
<span><span class="co">#&gt;  gridGraphics       0.5-1      <span style="color: #555555;">2020-12-13</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.3.0)</span></span></span>
<span><span class="co">#&gt;  gson               0.1.0      <span style="color: #555555;">2023-03-07</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.3.0)</span></span></span>
<span><span class="co">#&gt;  gt               * 0.11.0     <span style="color: #555555;">2024-07-09</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.3.3)</span></span></span>
<span><span class="co">#&gt;  gtable             0.3.4      <span style="color: #555555;">2023-08-21</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.3.0)</span></span></span>
<span><span class="co">#&gt;  HDO.db             0.99.1     <span style="color: #555555;">2024-03-18</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">#&gt;  highr              0.10       <span style="color: #555555;">2022-12-22</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.3.0)</span></span></span>
<span><span class="co">#&gt;  hms                1.1.3      <span style="color: #555555;">2023-03-21</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.3.0)</span></span></span>
<span><span class="co">#&gt;  htmltools          0.5.7      <span style="color: #555555;">2023-11-03</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.3.1)</span></span></span>
<span><span class="co">#&gt;  httr               1.4.7      <span style="color: #555555;">2023-08-15</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.3.0)</span></span></span>
<span><span class="co">#&gt;  igraph             2.0.3      <span style="color: #555555;">2024-03-13</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.3.1)</span></span></span>
<span><span class="co">#&gt;  IRanges          * 2.36.0     <span style="color: #555555;">2023-10-26</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">#&gt;  jquerylib          0.1.4      <span style="color: #555555;">2021-04-26</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.3.0)</span></span></span>
<span><span class="co">#&gt;  jsonlite           1.8.8      <span style="color: #555555;">2023-12-04</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.3.1)</span></span></span>
<span><span class="co">#&gt;  KEGGREST           1.42.0     <span style="color: #555555;">2023-10-26</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">#&gt;  knitr              1.45       <span style="color: #555555;">2023-10-30</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.3.1)</span></span></span>
<span><span class="co">#&gt;  lattice            0.22-5     <span style="color: #555555;">2023-10-24</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.3.3)</span></span></span>
<span><span class="co">#&gt;  lazyeval           0.2.2      <span style="color: #555555;">2019-03-15</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.3.0)</span></span></span>
<span><span class="co">#&gt;  lgr                0.4.4      <span style="color: #555555;">2022-09-05</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.3.0)</span></span></span>
<span><span class="co">#&gt;  lifecycle          1.0.4      <span style="color: #555555;">2023-11-07</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.3.1)</span></span></span>
<span><span class="co">#&gt;  lubridate        * 1.9.3      <span style="color: #555555;">2023-09-27</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.3.1)</span></span></span>
<span><span class="co">#&gt;  magrittr           2.0.3      <span style="color: #555555;">2022-03-30</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.3.0)</span></span></span>
<span><span class="co">#&gt;  MASS               7.3-60.0.1 <span style="color: #555555;">2024-01-13</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.3.3)</span></span></span>
<span><span class="co">#&gt;  Matrix             1.6-5      <span style="color: #555555;">2024-01-11</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.3.3)</span></span></span>
<span><span class="co">#&gt;  memoise            2.0.1      <span style="color: #555555;">2021-11-26</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.3.0)</span></span></span>
<span><span class="co">#&gt;  mlapi              0.1.1      <span style="color: #555555;">2022-04-24</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.3.0)</span></span></span>
<span><span class="co">#&gt;  munsell            0.5.0      <span style="color: #555555;">2018-06-12</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.3.0)</span></span></span>
<span><span class="co">#&gt;  nlme               3.1-164    <span style="color: #555555;">2023-11-27</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.3.3)</span></span></span>
<span><span class="co">#&gt;  org.Hs.eg.db     * 3.18.0     <span style="color: #555555;">2024-03-18</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">#&gt;  patchwork          1.2.0      <span style="color: #555555;">2024-01-08</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.3.1)</span></span></span>
<span><span class="co">#&gt;  pillar             1.9.0      <span style="color: #555555;">2023-03-22</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.3.0)</span></span></span>
<span><span class="co">#&gt;  pkgconfig          2.0.3      <span style="color: #555555;">2019-09-22</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.3.0)</span></span></span>
<span><span class="co">#&gt;  pkgdown            2.0.7      <span style="color: #555555;">2022-12-14</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.3.0)</span></span></span>
<span><span class="co">#&gt;  plyr               1.8.9      <span style="color: #555555;">2023-10-02</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.3.1)</span></span></span>
<span><span class="co">#&gt;  png                0.1-8      <span style="color: #555555;">2022-11-29</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.3.0)</span></span></span>
<span><span class="co">#&gt;  polyclip           1.10-6     <span style="color: #555555;">2023-09-27</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.3.1)</span></span></span>
<span><span class="co">#&gt;  proxy            * 0.4-27     <span style="color: #555555;">2022-06-09</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.3.0)</span></span></span>
<span><span class="co">#&gt;  purrr            * 1.0.2      <span style="color: #555555;">2023-08-10</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.3.0)</span></span></span>
<span><span class="co">#&gt;  qvalue             2.34.0     <span style="color: #555555;">2023-10-26</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">#&gt;  R6                 2.5.1      <span style="color: #555555;">2021-08-19</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.3.0)</span></span></span>
<span><span class="co">#&gt;  ragg               1.3.0      <span style="color: #555555;">2024-03-13</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.3.1)</span></span></span>
<span><span class="co">#&gt;  RColorBrewer       1.1-3      <span style="color: #555555;">2022-04-03</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.3.0)</span></span></span>
<span><span class="co">#&gt;  Rcpp               1.0.12     <span style="color: #555555;">2024-01-09</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.3.1)</span></span></span>
<span><span class="co">#&gt;  RCurl              1.98-1.14  <span style="color: #555555;">2024-01-09</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.3.1)</span></span></span>
<span><span class="co">#&gt;  readr            * 2.1.5      <span style="color: #555555;">2024-01-10</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.3.1)</span></span></span>
<span><span class="co">#&gt;  reshape2           1.4.4      <span style="color: #555555;">2020-04-09</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.3.0)</span></span></span>
<span><span class="co">#&gt;  RhpcBLASctl        0.23-42    <span style="color: #555555;">2023-02-11</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.3.0)</span></span></span>
<span><span class="co">#&gt;  rlang              1.1.3      <span style="color: #555555;">2024-01-10</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.3.1)</span></span></span>
<span><span class="co">#&gt;  rmarkdown          2.26       <span style="color: #555555;">2024-03-05</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.3.1)</span></span></span>
<span><span class="co">#&gt;  rsparse            0.5.2      <span style="color: #555555;">2024-06-28</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.3.3)</span></span></span>
<span><span class="co">#&gt;  RSQLite            2.3.5      <span style="color: #555555;">2024-01-21</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.3.1)</span></span></span>
<span><span class="co">#&gt;  rstudioapi         0.15.0     <span style="color: #555555;">2023-07-07</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.3.0)</span></span></span>
<span><span class="co">#&gt;  S4Vectors        * 0.40.2     <span style="color: #555555;">2023-11-25</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor 3.18 (R 4.3.2)</span></span></span>
<span><span class="co">#&gt;  sass               0.4.9      <span style="color: #555555;">2024-03-15</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.3.1)</span></span></span>
<span><span class="co">#&gt;  scales             1.3.0      <span style="color: #555555;">2023-11-28</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.3.1)</span></span></span>
<span><span class="co">#&gt;  scatterpie         0.2.1      <span style="color: #555555;">2023-06-07</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.3.0)</span></span></span>
<span><span class="co">#&gt;  sessioninfo        1.2.2      <span style="color: #555555;">2021-12-06</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.3.0)</span></span></span>
<span><span class="co">#&gt;  shadowtext         0.1.3      <span style="color: #555555;">2024-01-19</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.3.1)</span></span></span>
<span><span class="co">#&gt;  stringi            1.8.3      <span style="color: #555555;">2023-12-11</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.3.1)</span></span></span>
<span><span class="co">#&gt;  stringr          * 1.5.1      <span style="color: #555555;">2023-11-14</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.3.1)</span></span></span>
<span><span class="co">#&gt;  systemfonts        1.1.0      <span style="color: #555555;">2024-05-15</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.3.3)</span></span></span>
<span><span class="co">#&gt;  text2vec         * 0.6.4      <span style="color: #555555;">2023-11-09</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.3.1)</span></span></span>
<span><span class="co">#&gt;  textshaping        0.3.7      <span style="color: #555555;">2023-10-09</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.3.1)</span></span></span>
<span><span class="co">#&gt;  tibble           * 3.2.1      <span style="color: #555555;">2023-03-20</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.3.0)</span></span></span>
<span><span class="co">#&gt;  tidygraph          1.3.1      <span style="color: #555555;">2024-01-30</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.3.1)</span></span></span>
<span><span class="co">#&gt;  tidyr            * 1.3.1      <span style="color: #555555;">2024-01-24</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.3.1)</span></span></span>
<span><span class="co">#&gt;  tidyselect         1.2.1      <span style="color: #555555;">2024-03-11</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.3.1)</span></span></span>
<span><span class="co">#&gt;  tidytree           0.4.6      <span style="color: #555555;">2023-12-12</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.3.1)</span></span></span>
<span><span class="co">#&gt;  tidyverse        * 2.0.0      <span style="color: #555555;">2023-02-22</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.3.0)</span></span></span>
<span><span class="co">#&gt;  timechange         0.3.0      <span style="color: #555555;">2024-01-18</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.3.1)</span></span></span>
<span><span class="co">#&gt;  treeio             1.26.0     <span style="color: #555555;">2023-11-06</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">#&gt;  tweenr             2.0.3      <span style="color: #555555;">2024-02-26</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.3.1)</span></span></span>
<span><span class="co">#&gt;  tzdb               0.4.0      <span style="color: #555555;">2023-05-12</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.3.0)</span></span></span>
<span><span class="co">#&gt;  utf8               1.2.4      <span style="color: #555555;">2023-10-22</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.3.1)</span></span></span>
<span><span class="co">#&gt;  vctrs              0.6.5      <span style="color: #555555;">2023-12-01</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.3.1)</span></span></span>
<span><span class="co">#&gt;  viridis            0.6.5      <span style="color: #555555;">2024-01-29</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.3.1)</span></span></span>
<span><span class="co">#&gt;  viridisLite        0.4.2      <span style="color: #555555;">2023-05-02</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.3.0)</span></span></span>
<span><span class="co">#&gt;  vroom              1.6.5      <span style="color: #555555;">2023-12-05</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.3.1)</span></span></span>
<span><span class="co">#&gt;  withr              3.0.0      <span style="color: #555555;">2024-01-16</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.3.1)</span></span></span>
<span><span class="co">#&gt;  xfun               0.42       <span style="color: #555555;">2024-02-08</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.3.1)</span></span></span>
<span><span class="co">#&gt;  xml2               1.3.6      <span style="color: #555555;">2023-12-04</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.3.1)</span></span></span>
<span><span class="co">#&gt;  XVector            0.42.0     <span style="color: #555555;">2023-10-26</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">#&gt;  yaml               2.3.8      <span style="color: #555555;">2023-12-11</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.3.1)</span></span></span>
<span><span class="co">#&gt;  yulab.utils        0.1.4      <span style="color: #555555;">2024-01-28</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.3.1)</span></span></span>
<span><span class="co">#&gt;  zlibbioc           1.48.0     <span style="color: #555555;">2023-10-26</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; <span style="color: #555555;"> [1] /Library/Frameworks/R.framework/Versions/4.3-arm64/Resources/library</span></span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB; font-weight: bold;">──────────────────────────────────────────────────────────────────────────────</span></span></span></code></pre></div>
</div>
  </main><aside class="col-md-3"><nav id="toc"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p></p>
<p>Developed by Zhiming Ye.</p>
</div>

<div class="pkgdown-footer-right">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

    </footer>
</div>

  

  

  </body>
</html>
